From ae733857bfcf9fff49760ccbd489a23b0cb18256 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Mon, 18 Mar 2019 03:53:37 +0200
Subject: [PATCH] [SW-2799]: move pci code into a dedicated file

This is making preboot.c more readable

Change-Id: I0c70b8e933e0a14267933d9f5b4435bcbba5a280
---

diff --git a/pci.c b/pci.c
new file mode 100644
index 0000000..670c2ab
--- /dev/null
+++ b/pci.c
@@ -0,0 +1,149 @@
+/*
+ * (C) Copyright 2019 HabanaLabs, Ltd.
+ * Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
+
+#include <common.h>
+#include <linux/io.h>
+#include <mach/hl_regs.h>
+#include <mach/hl_fw_if.h>
+#include "preboot.h"
+#include "uart.h"
+
+#ifdef DEV_IS_GOYA
+/* Use DumpSel[8] to skip PCIe initialization */
+#define SKIP_PCIE_INIT BIT(21)
+#else
+#define SKIP_PCIE_INIT BIT(16)
+#endif
+
+#if defined(DEV_IS_GOYA)
+static int preboot_wait_for_bit(unsigned long reg, unsigned long timeout_ns)
+{
+	unsigned int val;
+	unsigned long count = (timeout_ns * SYS_REF_CLOCK) / 1000000000UL;
+
+	while (count--) {
+		val = readl(reg);
+		if (val & 1)
+			return 0;
+	}
+
+	return 1;
+}
+
+static void preboot_wait_for_ext_ld_done(void)
+{
+	/*
+	 * Wait for HW FSM to complete 16KB copy to PHY
+	 * this is enough to set ext_ld_done.
+	 * FSM will set ext_ld_done after 32KB but why wait
+	 * if we can already move on
+	 */
+	while ((readl(mmPCIE_PHY_CR_PARA_CMD) & 0xFFFF) <= 0xD780)
+		;
+	writel(1, mmPCIE_PHY_PHYX_SRAM_EXT_LD_DONE_0);
+	writel(1, mmPCIE_PHY_PHYX_SRAM_EXT_LD_DONE_1);
+	writel(1, mmPCIE_PHY_PHYX_SRAM_EXT_LD_DONE_2);
+	writel(1, mmPCIE_PHY_PHYX_SRAM_EXT_LD_DONE_3);
+}
+
+static void preboot_wait_for_stable_pclk(void)
+{
+	int ret;
+
+	while (1) {
+		ret = 0;
+		/* Wait for PCLK stable - check MPLLA [no timeout] */
+		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLA_STATE_0, 100);
+		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLA_STATE_1, 100);
+		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLA_STATE_2, 100);
+		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLA_STATE_3, 100);
+		if (!ret)
+			break;
+
+		ret = 0;
+		/* Wait for PCLK stable - check MPLLB [no timeout] */
+		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLB_STATE_0, 100);
+		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLB_STATE_1, 100);
+		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLB_STATE_2, 100);
+		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLB_STATE_3, 100);
+		if (!ret)
+			break;
+	}
+}
+
+/* Handle PCIe errata: make sure reset_n de-asserted only after perst */
+static void preboot_pcie_wait_for_link(void)
+{
+	preboot_wait_for_ext_ld_done();
+
+	/* Wait for PRSTn */
+	while (!readl(mmPSOC_GLOBAL_CONF_PRSTN))
+		;
+
+	/* Enable FW FSM after PRSTn */
+	writel(1, mmPSOC_GLOBAL_CONF_PCI_FW_FSM);
+
+	preboot_wait_for_stable_pclk();
+
+	while (1) {
+		unsigned int prstn, ltssm, smlh, rdlh, hot_reset;
+
+		prstn = readl(mmPSOC_GLOBAL_CONF_PRSTN);
+		ltssm = readl(mmPCIE_AUX_LTSSM_EN);
+		smlh = readl(mmPCIE_AUX_SMLH_LINK_UP);
+		rdlh = readl(mmPCIE_AUX_RDLH_LINK_UP);
+		hot_reset = readl(mmPCIE_CORE_HOT_RST);
+
+		if ((hot_reset & BIT(0)) == BIT(0)) {
+			writel(0x10, mmPCIE_CORE_HOT_RST);
+			writel(1, mmPCIE_AUX_LTSSM_EN);
+		}
+
+		if (prstn && smlh && rdlh)
+			break;
+
+		/*
+		 * Some host assert prstn during link up while moving to GEN3
+		 * This cause BAR size to be reset
+		 * We stop link up process by disabling ltssm
+		 * till BAR is reconfigured
+		 */
+		if (!prstn && ltssm) {
+			writel(0, mmPCIE_AUX_LTSSM_EN);
+			continue;
+		}
+
+		if (prstn && !ltssm) {
+			writel(1, mmPCIE_AUX_LTSSM_EN);
+			continue;
+		}
+	}
+}
+#else
+static void preboot_pcie_wait_for_link(void)
+{
+	while (1) {
+		unsigned int smlh, rdlh;
+
+		smlh = readl(mmPCIE_AUX_SMLH_LINK_UP);
+		rdlh = readl(mmPCIE_AUX_RDLH_LINK_UP);
+		if (smlh && rdlh)
+			break;
+	}
+}
+#endif
+
+void preboot_pcie_init(void)
+{
+	if (readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS) & SKIP_PCIE_INIT) {
+		preboot_puts("PCIe initialization is skipped!\n");
+		return;
+	}
+
+	preboot_pcie_wait_for_link();
+	preboot_puts("PCIe link is UP!\n");
+}
diff --git a/pci.h b/pci.h
new file mode 100644
index 0000000..dfa1b7b
--- /dev/null
+++ b/pci.h
@@ -0,0 +1,14 @@
+/*
+ * Copyright (C) 2019 HabanaLabs Ltd.
+ *   Author: Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
+
+#ifndef _HL_PCI_H_
+#define _HL_PCI_H_
+
+
+void preboot_pcie_init(void);
+
+#endif /* _HL_PCI_H_ */
diff --git a/preboot.c b/preboot.c
index 3e7d202..63fb26d 100644
--- a/preboot.c
+++ b/preboot.c
@@ -12,13 +12,14 @@
 #include "auto_generated.h"
 #include <stddef.h>
 #include <stdbool.h>
+#include "preboot.h"
 #include "pll.h"
 #include "uart.h"
 #include "spi.h"
+#include "pci.h"
 
-#define SYS_REF_CLOCK			50000000
-#define BTL_COUNTER_MAX			BIT(22)
-#define VERSION_MAX_LEN	128
+#define BTL_COUNTER_MAX		BIT(22)
+#define VERSION_MAX_LEN		128
 
 #define PREBOOT_VER_STR	"Preboot version " PREBOOT_VER \
 	" (" PREBOOT_DATE " - " PREBOOT_TIME ")"
@@ -104,138 +105,9 @@
 	}
 }
 
-/* Use DumpSel[8] to skip PCIe initialization */
-#define SKIP_PCIE_INIT BIT(21)
 #else
 #define preboot_delay()
-#define SKIP_PCIE_INIT BIT(16)
 #endif
-
-#if defined (DEV_IS_GOYA)
-static int preboot_wait_for_bit(unsigned long reg, unsigned long timeout_ns)
-{
-	unsigned int val;
-	unsigned long count = (timeout_ns * SYS_REF_CLOCK) / 1000000000UL;
-
-	while (count--) {
-		val = readl(reg);
-		if (val & 1)
-			return 0;
-	}
-
-	return 1;
-}
-
-static void preboot_wait_for_ext_ld_done(void)
-{
-	/*
-	 * Wait for HW FSM to complete 16KB copy to PHY
-	 * this is enough to set ext_ld_done.
-	 * FSM will set ext_ld_done after 32KB but why wait
-	 * if we can already move on
-	 */
-	while ((readl(mmPCIE_PHY_CR_PARA_CMD) & 0xFFFF) <= 0xD780) ;
-	writel(1, mmPCIE_PHY_PHYX_SRAM_EXT_LD_DONE_0);
-	writel(1, mmPCIE_PHY_PHYX_SRAM_EXT_LD_DONE_1);
-	writel(1, mmPCIE_PHY_PHYX_SRAM_EXT_LD_DONE_2);
-	writel(1, mmPCIE_PHY_PHYX_SRAM_EXT_LD_DONE_3);
-}
-
-static void preboot_wait_for_stable_pclk(void)
-{
-	int ret;
-
-	while (1) {
-		ret = 0;
-		/* Wait for PCLK stable - check MPLLA [no timeout] */
-		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLA_STATE_0, 100);
-		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLA_STATE_1, 100);
-		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLA_STATE_2, 100);
-		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLA_STATE_3, 100);
-		if (!ret)
-			break;
-
-		ret = 0;
-		/* Wait for PCLK stable - check MPLLB [no timeout] */
-		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLB_STATE_0, 100);
-		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLB_STATE_1, 100);
-		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLB_STATE_2, 100);
-		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLB_STATE_3, 100);
-		if (!ret)
-			break;
-	}
-}
-
-/* Handle PCIe errata: make sure reset_n de-asserted only after perst */
-static void preboot_pcie_wait_for_link(void)
-{
-	preboot_wait_for_ext_ld_done();
-
-	/* Wait for PRSTn */
-	while (!readl(mmPSOC_GLOBAL_CONF_PRSTN)) ;
-
-	/* Enable FW FSM after PRSTn */
-	writel(1, mmPSOC_GLOBAL_CONF_PCI_FW_FSM);
-
-	preboot_wait_for_stable_pclk();
-
-	while (1) {
-		unsigned int prstn, ltssm, smlh, rdlh, hot_reset;
-
-		prstn = readl(mmPSOC_GLOBAL_CONF_PRSTN);
-		ltssm = readl(mmPCIE_AUX_LTSSM_EN);
-		smlh = readl(mmPCIE_AUX_SMLH_LINK_UP);
-		rdlh = readl(mmPCIE_AUX_RDLH_LINK_UP);
-		hot_reset = readl(mmPCIE_CORE_HOT_RST);
-
-		if ((hot_reset & BIT(0)) == BIT(0)) {
-			writel(0x10, mmPCIE_CORE_HOT_RST);
-			writel(1, mmPCIE_AUX_LTSSM_EN);
-		}
-
-		if (prstn && smlh && rdlh)
-			break;
-
-		/*
-		 * Some host assert prstn during link up while moving to GEN3
-		 * This cause BAR size to be reset
-		 * We stop link up process by disabling ltssm
-		 * till BAR is reconfigured
-		 */
-		if(!prstn && ltssm) {
-			writel(0, mmPCIE_AUX_LTSSM_EN);
-			continue;
-		}
-		if(prstn && !ltssm) {
-			writel(1, mmPCIE_AUX_LTSSM_EN);
-			continue;
-		}
-	}
-}
-#else
-static void preboot_pcie_wait_for_link(void)
-{
-	while (1) {
-		unsigned int smlh, rdlh;
-
-		smlh = readl(mmPCIE_AUX_SMLH_LINK_UP);
-		rdlh = readl(mmPCIE_AUX_RDLH_LINK_UP);
-		if (smlh && rdlh)
-			break;
-	}
-}
-#endif
-
-static void preboot_pcie_init(void)
-{
-	if (readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS) & SKIP_PCIE_INIT) {
-		preboot_puts("PCIe initialization is skipped!\n");
-		return;
-	}
-
-	preboot_pcie_wait_for_link();
-	preboot_puts("PCIe link is UP!\n");
-}
 
 #ifndef TARGET_IS_TESTER
 static const char *preboot_ver_str = PREBOOT_VER_STR;
diff --git a/preboot.h b/preboot.h
new file mode 100644
index 0000000..825c1d3
--- /dev/null
+++ b/preboot.h
@@ -0,0 +1,13 @@
+/*
+ * Copyright (C) 2019 HabanaLabs Ltd.
+ *   Author: Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
+
+#ifndef _HL_PREBOOT_H_
+#define _HL_PREBOOT_H_
+
+#define SYS_REF_CLOCK			50000000
+
+#endif /* _HL_PREBOOT_H_ */
