From b23dc94452717117ce1052320fc9cc9d9e2f0bec Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Sun, 07 Apr 2019 14:36:13 +0300
Subject: [PATCH] [SW-0] preboot: remove unused header inclusion

Change-Id: I77ac99ae01130d728cba638339528223ce0a69ef
---

diff --git a/common/preboot.c b/common/preboot.c
index 861afeb..f4ce164 100644
--- a/common/preboot.c
+++ b/common/preboot.c
@@ -14,7 +14,6 @@
 #include <stdbool.h>
 #include "uart.h"
 #include "spi.h"
-#include "pci.h"
 
 #define VERSION_MAX_LEN		128
 
