From 46753d93481f89738e3747410c07a6bbee6c6ea3 Mon Sep 17 00:00:00 2001
From: Moshe Makmel <mmakmel@habana.ai>
Date: Tue, 13 Jul 2021 14:16:54 +0300
Subject: [PATCH] [SW-34965] HLS: install script typo fix

Change-Id: I04bb2e3e8986850c23f821a2dece91b437271dc6
---

diff --git a/package/openbmc/install-hls.sh b/package/openbmc/install-hls.sh
index 0009e7f..f077fa2 100644
--- a/package/openbmc/install-hls.sh
+++ b/package/openbmc/install-hls.sh
@@ -223,7 +223,7 @@
 
 		echo You are about to downgrade fw version "(${curr_ver} -> ${new_ver})"
 		echo This means that the flash layout will change
-		read -p "Are you sure you wish to continue? [y/N] " -r
+		read -p "Are you sure you wish to continue? [Y/n] " -r
 		if [[ $REPLY =~ ^[Yy]$ ]]; then
 			export whole_chip="yes"
 			export BMC_IMAGE=${PWD}/obmc-phosphor-image-hls.static.mtd.all.tar
