From a988c9c02f6da50f9266b049c5cc28c373e85c76 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Thu, 15 Jul 2021 06:52:03 +0300
Subject: [PATCH] [SW-44497] preboot: typo fixes for ts

Typo fixes. Does not impact functionality.

Change-Id: I4f476328b81fbfd1a34a65f067c2736752848d23
Signed-off-by: Karthikeyan K <karthikk@habana.ai>
---

diff --git a/arch/greco/arc/tstd.h b/arch/greco/arc/tstd.h
index d0f4930..5ba1f76 100644
--- a/arch/greco/arc/tstd.h
+++ b/arch/greco/arc/tstd.h
@@ -18,7 +18,7 @@
 #define HL_TS_CMN_SDIF_PROG_SHIFT	DCON0_TSTD_TS_CMN_SDIF_SDIF_PROG_SHIFT
 #define HL_TS_CMN_SDIF_PROG_MASK	DCON0_TSTD_TS_CMN_SDIF_SDIF_PROG_MASK
 #define HL_TS_CMN_SDIF_STATUS_BUSY_MASK	\
-	DCON0_TSTD_VM_CMN_SDIF_STATUS_SDIF_BUSY_MASK
+	DCON0_TSTD_TS_CMN_SDIF_STATUS_SDIF_BUSY_MASK
 /*
  * sample_data register width is 16 bits, as per MR74137 spec resolution is
  * from 8 - 12 bits only.
diff --git a/arch/greco/arm/tstd.h b/arch/greco/arm/tstd.h
index d0f4930..5ba1f76 100644
--- a/arch/greco/arm/tstd.h
+++ b/arch/greco/arm/tstd.h
@@ -18,7 +18,7 @@
 #define HL_TS_CMN_SDIF_PROG_SHIFT	DCON0_TSTD_TS_CMN_SDIF_SDIF_PROG_SHIFT
 #define HL_TS_CMN_SDIF_PROG_MASK	DCON0_TSTD_TS_CMN_SDIF_SDIF_PROG_MASK
 #define HL_TS_CMN_SDIF_STATUS_BUSY_MASK	\
-	DCON0_TSTD_VM_CMN_SDIF_STATUS_SDIF_BUSY_MASK
+	DCON0_TSTD_TS_CMN_SDIF_STATUS_SDIF_BUSY_MASK
 /*
  * sample_data register width is 16 bits, as per MR74137 spec resolution is
  * from 8 - 12 bits only.
diff --git a/lib/preboot/thermal_v2.c b/lib/preboot/thermal_v2.c
index 091ae8a..41a9f5c 100644
--- a/lib/preboot/thermal_v2.c
+++ b/lib/preboot/thermal_v2.c
@@ -128,7 +128,7 @@
 	val |= (1 << HL_TS_CMN_CLK_SYNTH_ENA_SHIFT)
 		& HL_TS_CMN_CLK_SYNTH_ENA_MASK;
 
-	regmap_writel(val, addr + + HL_TS_CMN_CLK_SYNTH);
+	regmap_writel(val, addr + HL_TS_CMN_CLK_SYNTH);
 }
 
 /*
