From 18424c091dbbcd809f8b389150c6b6f30f67a829 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Tue, 23 Apr 2019 11:35:23 +0300
Subject: [PATCH] [SW-0] pll: fix macro spelling

Change-Id: I8dac47286316e6fc9f071f6f417d108083e3d18d
---

diff --git a/common/pll.c b/common/pll.c
index 8727c9d..a4f1dba 100644
--- a/common/pll.c
+++ b/common/pll.c
@@ -37,7 +37,7 @@
 	if (bval == HL_PLL_REF_CLK || bval == HL_PLL_DIV_REF)
 		cnt = BYPASS_CYCLES_PER_USEC;
 	else
-		cnt = CPU_CYCELS_PER_USEC;
+		cnt = CPU_CYCLES_PER_USEC;
 
 	cnt = (cnt * usec) / MIN_CYCLES_PER_ITER;
 
diff --git a/common/pll.h b/common/pll.h
index b8a9c5e..15c74c7 100644
--- a/common/pll.h
+++ b/common/pll.h
@@ -26,7 +26,7 @@
 #define HL_PLL_LOCK_TOUT	5000	/* 5 ms */
 
 #define USEC_PER_SEC		1000000
-#define CPU_CYCELS_PER_USEC	(HL_CPU_CYCELS_PER_USEC / USEC_PER_SEC)
+#define CPU_CYCLES_PER_USEC	(HL_CPU_CYCLES_PER_USEC / USEC_PER_SEC)
 #define BYPASS_CYCLES_PER_USEC	(50000000 / USEC_PER_SEC)
 #define MIN_CYCLES_PER_ITER	5
 
diff --git a/gaudi/gaudi.h b/gaudi/gaudi.h
index 8af8949..443573e 100644
--- a/gaudi/gaudi.h
+++ b/gaudi/gaudi.h
@@ -30,7 +30,7 @@
 #define mmNIC2_TPC_PLL_BASE		0xD7A000
 #define CPU_PLL_BASE			mmPSOC_CPU_PLL_BASE
 
-#define HL_CPU_CYCELS_PER_USEC		120000000
+#define HL_CPU_CYCLES_PER_USEC		120000000
 
 #define PCI_DIV2_CLOCK			(830000000 / 5)
 #define UART_BAUDDIV			(PCI_DIV2_CLOCK / 115200)
diff --git a/goya/goya.h b/goya/goya.h
index f6aeaa9..1478364 100644
--- a/goya/goya.h
+++ b/goya/goya.h
@@ -12,7 +12,7 @@
 #define mmIC_PLL_BASE		0x4A3000
 #define CPU_PLL_BASE		mmCPU_PLL_BASE
 
-#define HL_CPU_CYCELS_PER_USEC	133000000
+#define HL_CPU_CYCLES_PER_USEC	133000000
 
 #define NON_RST_FLOPS_3		mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3
 
