From 1ecb6c76b2afe315ab068fbea1b5a7f9975ef842 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Thu, 05 Aug 2021 17:19:15 +0300
Subject: [PATCH] [SW-0] gaudi: security: white space fix

No functional changes - only white space.

Change-Id: I805192692e476e0787d07f5c1f3df7763b32a61b
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index a92de3a..01c11f4 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -937,7 +937,7 @@
 			 PROT_BIT(mmMME0_ACC_WBC));
 		regmap_writel(mask, PRIV_PROT_REG(prot_base));
 
-		prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0x80 - 0xF80 */
+		prot_base += SNGL_REG_PROT_SIZE; /* offsets: 0x80 - 0xF80 */
 		set_prot_bits_range_all(prot_base,
 					PROT_SIZE_MAX - (prot_base - base));
 	}
@@ -960,7 +960,7 @@
 			 PROT_BIT(mmMME0_SBAB_ARUSER1));
 		regmap_writel(mask, PRIV_PROT_REG(prot_base));
 
-		prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0x80 - 0xF80 */
+		prot_base += SNGL_REG_PROT_SIZE; /* offsets: 0x80 - 0xF80 */
 		set_prot_bits_range_all(prot_base,
 					PROT_SIZE_MAX - (prot_base - base));
 	}
