From deca237fa0a4cca6c8d568f5aa507a4a8b32aef5 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Sat, 09 Oct 2021 17:56:33 +0300
Subject: [PATCH] test patch

Change-Id: I8061a95e80a7229655ceed5d688b6cbc6d611a43
---

diff --git a/board/habanalabs/common/agent-fw_loader.its b/board/habanalabs/common/agent-fw_loader.its
index 5e4a7aa..be63577 100644
--- a/board/habanalabs/common/agent-fw_loader.its
+++ b/board/habanalabs/common/agent-fw_loader.its
@@ -58,7 +58,7 @@
 		};
 
 		img-ptr {
-			description = "Habana image pointers";
+			description = "Habana image pointers test";
 			data = /incbin/("@HL_DEV@-images-pointers.bin.le");
 			compression = "none";
 			offset = <0x00060000>;
