From 05b97d30ba1cad828be66ec221f6a06c9c67c5db Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Sun, 24 Oct 2021 19:35:28 +0300
Subject: [PATCH] [SW-0] fit: fix typo

Fxing a typo mistake

Change-Id: Idbcc41c5a4d86f1395eb99b90c6988b9afd3d847
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/lib/preboot/fit.c b/lib/preboot/fit.c
index 15da8d3..f6d8187 100644
--- a/lib/preboot/fit.c
+++ b/lib/preboot/fit.c
@@ -710,7 +710,7 @@
 		return true;
 
 	if (len != sizeof(*cell)) {
-		preboot_puts("Currupted load type\n");
+		preboot_puts("Corrupted load type\n");
 		return false;
 	}
 
