From 187681df49ccbc2fd170654c4339bb9217a50b48 Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Thu, 25 Nov 2021 09:47:50 +0200
Subject: [PATCH] Revert "[SW-56233] security: gaudi2: fix prot bits loop"

This reverts commit 9bc2912988a2e775252571b718fa5cf47cd59e3a.

Change-Id: I13a08a8007492a51669306035a5072f2f2d823d4
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 7918a3f..5dd97c9 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -432,7 +432,6 @@
 				do {
 					init_prot_bits_block(cfg_idx, major_idx, minor_idx,
 							     sub_minor_idx);
-					sub_minor_idx++;
 				} while (sub_minor_idx < block_addr.sub_minor);
 				minor_idx++;
 			} while (minor_idx < block_addr.minor);
