From 255fc9acc3de4662c9a4eccff39a336e9c79da5f Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Sat, 18 Dec 2021 16:44:03 +0200
Subject: [PATCH] run to local_run distinguish from rmt_host_run

Change-Id: I290e6b4b77015ea18dc7d09c7426d1d5ff422945
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/scripts/fwi/fw_installer.sh b/scripts/fwi/fw_installer.sh
index f5dfb86..869223d 100755
--- a/scripts/fwi/fw_installer.sh
+++ b/scripts/fwi/fw_installer.sh
@@ -67,7 +67,7 @@
 	echo $*
 }
 
-function run()
+function local_run()
 {
 	local ret
 	[ ${is_dbg_mode:-0} -ne 0 ] && set -x
@@ -83,10 +83,10 @@
 	for ((i=0; i < 3; i++)); do
 
 		[ ${local_ctrl:-0} -eq 1 ] && {
-			${dry_run_echo} run $*
+			${dry_run_echo} local_run $*
 			ret=$?
 		} || {
-#			${dry_run_echo} run hlp_ssh ${ctrl_user} ${rmt_sudo} "$*"
+#			${dry_run_echo} local_run hlp_ssh ${ctrl_user} ${rmt_sudo} "$*"
 			ssh ${ctrl_user} ${rmt_sudo} "$*"
 			ret=$?
 		}
@@ -214,7 +214,7 @@
 function burn_flash()
 {
 	host_run sudo hl-fw-loader -y
-#run sudo hl-fw-loader -r <<END
+#host_run sudo hl-fw-loader -r <<END
 #y
 #
 #END
