From d757bbd5a4ae5ecc914a3f4689a61d22ef54c349 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Sun, 30 Jan 2022 21:07:40 +0200
Subject: [PATCH] sq read-eeprom

Change-Id: Ida2820b4b244ec4f68c90723d868444558689387
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/scripts/fwi/fw_installer.sh b/scripts/fwi/fw_installer.sh
index 8ebd0a2..8a1e01a 100755
--- a/scripts/fwi/fw_installer.sh
+++ b/scripts/fwi/fw_installer.sh
@@ -1524,7 +1524,17 @@
 function fwl_read_eeprom_op()
 {
 	[ ${read_eeprom_f:-0} -eq 0 ] && return 0
-	fwl_read_eeprom
+
+	host_run ls /lib/firmware/habanalabs/gaudi2/gaudi2-eeprom_read.itb > /dev/null 2>&1
+	[ $? -ne 0 ] && {
+		host_run ls /lib/firmware/habanalabs/gaudi2/gaudi2-sec-fit.tgz > /dev/null 2>&1
+		[ $? -ne 0 ] && {
+			host_cp_to ${imgs_src_dir}/gaudi2/gaudi2-sec-fit.tgz /lib/firmware/habanalabs/gaudi2/
+		} || {
+			host_run_sudo tar zxf /lib/firmware/habanalabs/gaudi2/gaudi2-sec-fit.tgz -C /lib/firmware/habanalabs/gaudi2/
+		}
+		fwl_read_eeprom
+	}
 }
 
 function host_depended_ops()
