From 2bc6c4ee545edc8bf11c92b8d8e09348faf77c90 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Sun, 30 Jan 2022 20:19:05 +0200
Subject: [PATCH] --lkd [status|load [<local file>]|remove

Change-Id: Ic7445665e913b700cd255b1ec902ec41e0bbb8a8
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/scripts/fwi/fw_installer.sh b/scripts/fwi/fw_installer.sh
index 566c185..3682c9e 100755
--- a/scripts/fwi/fw_installer.sh
+++ b/scripts/fwi/fw_installer.sh
@@ -1538,16 +1538,27 @@
 function lkd_op()
 {
 	local op
+	local file args
 
 	[ ${lkd_f:-0} -eq 0 ] && return 0
 	op=${lkd_args[0]}
 	[ ${#lkd_args[@]} -eq 0 ] && op=status || unset lkd_args[0]
-	case op in ${op}
+	file=${lkd_args[1]}
+	case ${op} in
 	status)
+		tmp=($(host_run lsmod \| grep habana))
+		[ $? -eq 0 ] && {
+			pinfo LKD up [$tmp]
+		} || {
+			pinfo LKD down
+		}
 		;;
 	l|ld|load)
+		[ -f ${file:-" "} ] && args=($(cat ${file}))
+		host_run build_and_insmod_habanalabs ${args[@]}
 		;;
 	r|rm|remove)
+		host_run build_and_insmod_habanalabs -r
 		;;
 	esac
 }
@@ -2060,6 +2071,7 @@
 	mods_var_register fwl
 	mods_var_register fwl_loc
 	mods_var_register read_eeprom
+	mods_var_register lkd
 
 	# User config files and platform selection was already
 	# handled in the early param handling
