From 641cf89881ca4a26e794fa76bdeaf20f1e42f73b Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Fri, 28 Jan 2022 19:50:45 +0200
Subject: [PATCH] [SW-65638] gaudi2: release: add error check

simple sanity check

Change-Id: I0cdbb7945b9faf5b4802e855e151225a0c596ba8
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/scripts/fwi/fw_installer.sh b/scripts/fwi/fw_installer.sh
index 444554a..056b6e9 100755
--- a/scripts/fwi/fw_installer.sh
+++ b/scripts/fwi/fw_installer.sh
@@ -1785,6 +1785,7 @@
 	finilize_env_before_execution
 	show_help_op
 	create_config_file_op
+	[ $? -ne 0 ] && return 1
 	api_sanity_check
 	[ $? -ne 0 ] && return 1
 	handle_ignore_op
