From 900faf8bd3477b34656af4002feb23f5ab9d63a7 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Sun, 30 Jan 2022 19:51:45 +0200
Subject: [PATCH] WIP - lkd_op

Change-Id: Ie629aee3b14a73885df029f5133d7c7f0eb20179
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/scripts/fwi/fw_installer.sh b/scripts/fwi/fw_installer.sh
index 1305d23..566c185 100755
--- a/scripts/fwi/fw_installer.sh
+++ b/scripts/fwi/fw_installer.sh
@@ -1535,6 +1535,23 @@
 	}
 }
 
+function lkd_op()
+{
+	local op
+
+	[ ${lkd_f:-0} -eq 0 ] && return 0
+	op=${lkd_args[0]}
+	[ ${#lkd_args[@]} -eq 0 ] && op=status || unset lkd_args[0]
+	case op in ${op}
+	status)
+		;;
+	l|ld|load)
+		;;
+	r|rm|remove)
+		;;
+	esac
+}
+
 function host_depended_ops()
 {
 	# The place to locate all the operation depend on the host
@@ -1544,6 +1561,7 @@
 		}
 		fwl_reset_devices_op
 		fwl_read_eeprom_op
+		lkd_op
 	} || true
 }
 
