From 62ff071e8cb5a079588db65bdea8b752a85504b4 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Sat, 29 Jan 2022 02:10:00 +0200
Subject: [PATCH] [SW-65638] gaudi2: release: WIP - read EEPROM

Change-Id: I8dcac47ce777b710bb4bb56e47dc903225343d62
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/scripts/fwi/fw_installer.sh b/scripts/fwi/fw_installer.sh
index 0f2bbfc..8f61f3c 100755
--- a/scripts/fwi/fw_installer.sh
+++ b/scripts/fwi/fw_installer.sh
@@ -655,6 +655,15 @@
 	return 0
 }
 
+function read_eeprom()
+{
+	local oam=$1
+	local pci_dev=$2
+	local local_itb_path=$3
+	local file_name=$(basename ${local_itb_path})
+
+	return 0
+}
 function burn_device()
 {
 	local pci_dev=$1
diff --git a/scripts/fwi/hls_platform.rc b/scripts/fwi/hls_platform.rc
index 5a4c65f..03a5557 100644
--- a/scripts/fwi/hls_platform.rc
+++ b/scripts/fwi/hls_platform.rc
@@ -1222,6 +1222,7 @@
 	pushd \${HOME}/fwl > /dev/null
 	echo ${rmt_passwd} | sudo -S cp hl-fw-loader /usr/sbin/
 	echo ${rmt_passwd} | sudo -S cp hl-smi /usr/bin/
+	tar xfz gaudi2-sec-fit.tgz
 	echo ${rmt_passwd} | sudo -S cp *.itb ${fw_imgs_dir}
 	echo ${rmt_passwd} | sudo -S cp eeprom/*.itb ${fw_imgs_dir}/eeprom
 	popd > /dev/null
