From fd3caab18fc03c23fb32f248bdda51e838f67c6a Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Sun, 30 Jan 2022 02:59:16 +0200
Subject: [PATCH] fix read eeprom

Change-Id: Ifd36a8e8b67cefbdbf806313fd996d7f6306a9da
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/scripts/fwi/fw_installer.sh b/scripts/fwi/fw_installer.sh
index f6ca681..8ebd0a2 100755
--- a/scripts/fwi/fw_installer.sh
+++ b/scripts/fwi/fw_installer.sh
@@ -1412,19 +1412,20 @@
 	return 0
 }
 
-function remote_fwl_devs_list()
+function remote_fwl()
 {
 	local args=()
 	for dev in ${devs_list_pci_ids[@]}; do
 		args+=(-d $dev)
+		pinfo args=${args[@]}
 	done
 
-	$dry_run host_run sudo hl-fw-loader $args *$
+	$dry_run host_run sudo hl-fw-loader ${args[@]} $*
 	[ $? -ne 0 ] && {
-		perr "[${pci_dev}] FWL failed"
+		perr "FWL failed"
 		return 1
 	} || {
-		pok "[${pci_dev}] FWL Succeed"
+		pok "FWL Succeed"
 	}
 	return 0
 }
@@ -1523,7 +1524,7 @@
 function fwl_read_eeprom_op()
 {
 	[ ${read_eeprom_f:-0} -eq 0 ] && return 0
-	read_eeprom
+	fwl_read_eeprom
 }
 
 function host_depended_ops()
@@ -2032,6 +2033,7 @@
 	mods_var_register copy_files
 	mods_var_register fwl
 	mods_var_register fwl_loc
+	mods_var_register read_eeprom
 
 	# User config files and platform selection was already
 	# handled in the early param handling
