From 406263d4c8995d23b3d3d4748207d82cc8458def Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Sat, 29 Jan 2022 23:43:15 +0200
Subject: [PATCH] alignment

Change-Id: I08165aa5f9107f1d8d7c5707b32d926ee7233b75
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/scripts/fwi/fw_installer.sh b/scripts/fwi/fw_installer.sh
index 00fe0e6..a369010 100755
--- a/scripts/fwi/fw_installer.sh
+++ b/scripts/fwi/fw_installer.sh
@@ -233,8 +233,9 @@
 	                          : explicitly ignores one or more entities in the setup. such
 	                            with no arguent ignores only the host.
 	                            EX, in case remote host is not relevant to the operation such
-                                    as toggling WP, ignores the absent of the rmt-host
+	                            as toggling WP, ignores the absent of the rmt-host
 	--pre-reboot              : do a reboot to the system before the update procedure
+
 	-l|--login <host|bmc>     : open an ssh on the host or the OAM's bmc
 	--run <host|bmc>[cmd]     : open an ssh on the host or the OAM's bmc and execute the supplied command
 	                            without command serve as a normal login shell
@@ -244,6 +245,7 @@
 	                            the file path can be part of the config file if used (rmt_virt_file/local_virt_file)
 	--show-pci-map            : shows a table of the OAM num, heperV real pci ID and VM's PCI ID if exist
 	--oam-con-out <oam_id>    : out a remote output console (temporary selution)
+
 	--create-cfg <file_name>  : create a configuration file out of the current running configuration
 	--edit-cfg                : combine with the -c option will open the config file for editing
 	--config-link [on|off]    : combine with the -c option link the basic config file as a temporary default
@@ -677,6 +679,7 @@
 
 	return 0
 }
+
 function burn_device()
 {
 	local pci_dev=$1
