From b1cc450f4c154a25c38d4565be7bb8eec57d29ac Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Sun, 30 Jan 2022 02:08:13 +0200
Subject: [PATCH] cuncurrency

Change-Id: Ibb75c7494e254cbf74696e2d486c2422e4bdff4e
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/scripts/fwi/fw_installer.sh b/scripts/fwi/fw_installer.sh
index 90b67c2..ed0a434 100755
--- a/scripts/fwi/fw_installer.sh
+++ b/scripts/fwi/fw_installer.sh
@@ -807,6 +807,42 @@
 	return $err
 }
 
+function concurrency()
+{
+	local err pci_dev
+	declare -A pids
+	declare -A ret_code
+	local err=0
+	local oam_id
+	local cmd=$1
+
+	shift
+
+	for pci_dev in ${devs_list_pci_ids[@]}; do
+		oam_id=$(hls2_pci2oam $elem)
+		pinfo [$pci_dev] Start update
+		echo "[$pci_dev] Open LOG file ${log_dir}/${pci_dev}.log"
+		${cmd} $oam_id $pci_dev $* > ${log_dir}/${pci_dev}.log 2>&1 &
+		pids[${pci_dev}]=$!
+	done
+
+	for pci_dev in ${devs_list_pci_ids[@]}; do
+		wait ${pids[$pci_dev]}
+		#err=$?
+		ret_code[$pci_dev]=$err
+	done
+
+	for pci_dev in ${devs_list_pci_ids[@]}; do
+		[ ${ret_code[${pci_dev}]} -eq 0 ] && {
+			pinfo "[${pci_dev}] Finished"
+		} || {
+			pinfo "[${pci_dev}] Failed"
+			err=1
+		}
+	done
+	return $err
+}
+
 function do_burn_single()
 {
 	local pci_dev
