From 23b4743ad266b4cd6c2b0aee4716a1b283db66bb Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Sun, 30 Jan 2022 02:37:05 +0200
Subject: [PATCH] typo

Change-Id: Ic5c3e67adc79db01213b672884ccb90aece8f11d
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/scripts/fwi/fw_installer.sh b/scripts/fwi/fw_installer.sh
index ed0a434..b7e666f 100755
--- a/scripts/fwi/fw_installer.sh
+++ b/scripts/fwi/fw_installer.sh
@@ -164,14 +164,14 @@
 	HLS: Update CPLD+FLASH, default PO revision
 	fw_installer.sh -p hls -rmt-host hls2-srvXX-csr -bmc-host bmc-hls2-srvXX-csr -bmc-oam hls2-bmc-srvXX-csr --flash --cpld
 
-	HLS: Update CPLD+FLASH of PO revision 05. update executed in concurency manner
+	HLS: Update CPLD+FLASH of PO revision 05. update executed in concurrency manner
 	     each device logs out to a separate log file. base on the log_file name with trailer of the pciaddr.
 	fw_installer.sh -p hls -rmt-host hls2-srvXX-csr -bmc-host bmc-hls2-srvXX-csr -bmc-oam hls2-bmc-srvXX-csr --flash --cpld --po-rev 05 -J --log ./fwi-$(date +%Y%m%d-%H%M).log
 
-	HLS: Update CPLD+FLASH of PO revision 05 with cpld version internal version 04. update executed in concurency manner, excluding 2 devices
+	HLS: Update CPLD+FLASH of PO revision 05 with cpld version internal version 04. update executed in concurrency manner, excluding 2 devices
 	fw_installer.sh -p hls -rmt-host hls2-srvXX-csr -bmc-host bmc-hls2-srvXX-csr -bmc-oam hls2-bmc-srvXX-csr --flash --cpld --po-rev 05 --cpld-rev 04 -J -r 9a:00.0 -r 9d:00.0
 
-	HLS: Update CPLD+FLASH of PO revision 05 with cpld version internal version 04. update executed in concurency manner, excluding 2 devices
+	HLS: Update CPLD+FLASH of PO revision 05 with cpld version internal version 04. update executed in concurrency manner, excluding 2 devices
 	fw_installer.sh -p hls -rmt-host hls2-srvXX-csr -bmc-host bmc-hls2-srvXX-csr -bmc-oam hls2-bmc-srvXX-csr --flash --cpld --po-rev 05 -J -d 9a:00.0 -d 9d:00.0
 
 	HLS: disable WP for pci 06:00.0 and enable pci 07:00.0 and OAM2 and read status
