From 34709a64c7a12bc3f0f24aa17934a1033b941236 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Sun, 30 Jan 2022 15:41:03 +0200
Subject: [PATCH] adj help

Change-Id: Iac9e358b07a3d04ef24ff9ee8e99aa631eac36ed
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/scripts/fwi/fw_installer.sh b/scripts/fwi/fw_installer.sh
index 8a1e01a..1305d23 100755
--- a/scripts/fwi/fw_installer.sh
+++ b/scripts/fwi/fw_installer.sh
@@ -226,8 +226,6 @@
                                     NOTE: rmt-host reboot is needed
 	--sys-pwr <on|off|cycle>  : system wide power control, it power off|on|cycle the host and devices
                                     NOTE: power cycle is 3 mins
-	--oam-pwr <on|off|cycle>  : power cycle only the OAMs, may need a host reboot, it may
-	                            cause a host reboot.
 	--rmt-host-reboot         : exlicitly send a request to reboot the remote host.
 	-I|--ignore [h|host][jc|bh|bmc-host]
 	                          : explicitly ignores one or more entities in the setup. such
@@ -2005,11 +2003,11 @@
 	hlp_redirect usage
 	is_function_exist usage_${platform_type} > /dev/null
 	[[ $? -eq 0  ]] && {
-		pinfo -l 2 ""
-		pinfo -l 2 "        Platform specific"
-		pinfo -l 2 "        ================="
-		pinfo -l 2 ""
-		hlp_redirect -o 4 usage_${platform_type}
+		pinfo ""
+		pinfo "        Platform specific (${platform_type})"
+		pinfo "        ==========================="
+		pinfo ""
+		hlp_redirect usage_${platform_type}
 	}
 	hlp_redirect -o 5 print_examples
 	plog -l 4 ${auto_created_vars_list[@]}
