From 41c1897358bcbbd8aa860973e173b8a4eec8a51d Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Mon, 31 Jan 2022 17:50:48 +0200
Subject: [PATCH] fix btl mode

Change-Id: Id561418d724439fa14d71bb6e41395dece48599c
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/scripts/fwi/hls_platform.rc b/scripts/fwi/hls_platform.rc
index 238309c..febe2ab 100644
--- a/scripts/fwi/hls_platform.rc
+++ b/scripts/fwi/hls_platform.rc
@@ -803,7 +803,7 @@
 		pinfo "[$pci_id] Enable BTL boot"
 		;;
 	*)
-		perr "Unknown disable btl option "
+		perr "Unknown disable btl option ${mode}"
 		;;
 	esac
 }
@@ -812,7 +812,7 @@
 {
 	[ ${btl_mode_f:-0} -eq 0 ] && return 0
 	[ ${#btl_mode_args[@]} -lt 1 ] && return 1
-	run_per_selected_devs  oam_set_btl_mode  ${btl_mode}
+	run_per_selected_devs  oam_set_btl_mode  ${btl_mode_args[@]}
 	[ $? -ne 0 ] && \
 		pwarn "[HLS] uknown BTL mode argument: $btl_mode. Ignoring"
 }
