From a2c9240112838633ab7a2718095f90fb71038225 Mon Sep 17 00:00:00 2001
From: Yotam Ben Dosa <ybendosa@habana.ai>
Date: Sun, 20 Feb 2022 12:07:45 +0200
Subject: [PATCH] test

Change-Id: Ic21caebb26be0515666e82df4bb15c452eed9175
Signed-off-by: Yotam Ben Dosa <ybendosa@habana.ai>
---

diff --git a/board/habanalabs/common/post-ami-build.sh b/board/habanalabs/common/post-ami-build.sh
index 8b6dd91..5556e4c 100755
--- a/board/habanalabs/common/post-ami-build.sh
+++ b/board/habanalabs/common/post-ami-build.sh
@@ -22,6 +22,7 @@
 LIB_SRC_DIR=${OUTPUT_DIR}/host/arm-buildroot-linux-gnueabi/sysroot/usr/lib
 LIB_INC_DIR=${OUTPUT_DIR}/host/arm-buildroot-linux-gnueabi/sysroot/usr/include
 HLS_INVENTORY_DIR=${EMBEDDED_DIR}/hls-utils/hls-inventory
+
 SPECS_SEARCH_DIR=(
 	${SPECS_ROOT}
 	${SPECS_EMBEDDED_ROOT}
