From a75da19b671c1ac56e4139a78edd4e373b5a4a17 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 09 Mar 2022 12:28:03 +0200
Subject: [PATCH] [SW-0] gaudi3: update register names

sync register names to latest specs

Change-Id: I813e9815525947229d1aa4c89823c800054b994a
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 692dd37..b502b3b 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -180,9 +180,9 @@
 
 	arcboot_rom_global.config_fuse = val;
 
-	strap_l = regmap_readl(mmD0_PSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L);
+	strap_l = regmap_readl(mmD0_PSOC_BOOT_CONF_BOOT_STRAP_PINS_L);
 	if (!(val & EFUSE_SYSCFG_EFUSE_LOCATION_MASK) &&
-		(strap_l & D0_PSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L_EFUSE_OVERRIDE_M)) {
+		(strap_l & D0_PSOC_BOOT_CONF_BOOT_STRAP_PINS_L_EFUSE_OVERRIDE_M)) {
 		spi_flash_fuse_init();
 	}
 
diff --git a/arch/gaudi3/arc/regmap.h b/arch/gaudi3/arc/regmap.h
index 89e4974..3cb8d50 100644
--- a/arch/gaudi3/arc/regmap.h
+++ b/arch/gaudi3/arc/regmap.h
@@ -24,7 +24,7 @@
 #define mmPRSTN				mmD0_PSOC_GLOBAL_CONF_PRSTN
 #define mmCOLD_RST_FLOPS_2		mmD0_PSOC_GLOBAL_CONF_COLD_RST_FLOPS_2
 #define mmPCIE_PHY_RST_N		mmD0_PSOC_GLOBAL_CONF_PCIE_PHY_RST_N
-#define mmBOOT_STRAP_PINS_L		mmD0_PSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L
+#define mmBOOT_STRAP_PINS_L		mmD0_PSOC_BOOT_CONF_BOOT_STRAP_PINS_L
 #define mmPRSTN_MASK			mmD0_PSOC_GLOBAL_CONF_PRSTN_MASK
 #define mmSPI_DMA_AWUSER		mmD0_PARC_GLOBAL_CONF_SPI_DMA_AWUSER
 #define mmSPI_DMA_CTRL			mmD0_PARC_GLOBAL_CONF_SPI_DMA_CTRL
