From 95cb78f6858a7e793d42b0e7429ea0b909080cb3 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Wed, 06 Nov 2019 20:57:37 +0200
Subject: [PATCH] Revert "[SW-0]: openbmc: add missing semicolon at end of SET_VERSION_INFO macro"

This reverts commit cf4db0996b1e2c868eaf4982e3a18214d7ff0bef.
---

diff --git a/package/openbmc/openbmc.mk b/package/openbmc/openbmc.mk
index a2d5812..cfa2814 100644
--- a/package/openbmc/openbmc.mk
+++ b/package/openbmc/openbmc.mk
@@ -17,7 +17,7 @@
 endef
 
 define SET_VERSION_INFO
-	export HL_VERSION=$(shell cut -c2- $(VERSION_FILE));
+	export HL_VERSION=$(shell cut -c2- $(VERSION_FILE))
 endef
 
 define OPENBMC_BUILD_CMDS
