From 4974841627f142646a1e462b7f4d34da973dc7a0 Mon Sep 17 00:00:00 2001
From: Varun Sharma <vsharma@habana.ai>
Date: Thu, 14 Jul 2022 13:55:20 +0300
Subject: [PATCH] clr btl sectors

Change-Id: Ia7a2a8a0295a4d62f442bbfec42c729e51fcd78a
Signed-off-by: Varun Sharma <vsharma@habana.ai>
---

diff --git a/board/habanalabs/greco/agent-fw_loader.its b/board/habanalabs/greco/agent-fw_loader.its
index 0218709..414abc9 100644
--- a/board/habanalabs/greco/agent-fw_loader.its
+++ b/board/habanalabs/greco/agent-fw_loader.its
@@ -15,6 +15,18 @@
 	#address-cells = <2>;
 
 	images {
+		btl-empty {
+			description = "Empty btl sectors";
+			data = /incbin/("128K-0x00.bin");
+			type = "standalone";
+			arch = "arc";
+			compression = "none";
+			offset = <0x00000000>;
+			loadaddr = <0x7F 0xF0000000>;
+			hash {
+				algo = "sha256";
+			};
+		};
 		ppboot {
 			description = "Habana PPboot";
 			data = /incbin/("@HL_DEV@-ppboot-arc.bin.le");
@@ -72,7 +84,7 @@
 		fw-loader-conf {
 			load-type = <@FIT_LOAD_TYPE_FLASH@>;
 			description = "Habanalabs device boot loader";
-			loadables = "ppboot", "img-ptr", "preboot";
+			loadables = "btl-empty", "ppboot", "img-ptr", "preboot";
 		};
 
 		preboot-conf {
diff --git a/board/habanalabs/greco/post-image-agents.sh b/board/habanalabs/greco/post-image-agents.sh
index 499eb8d..5b08783 100755
--- a/board/habanalabs/greco/post-image-agents.sh
+++ b/board/habanalabs/greco/post-image-agents.sh
@@ -11,7 +11,7 @@
 HL_BTL_VERSION=`cat ${BASE_DIR}/build/btl-custom/localversion-habanalabs`
 
 function handle_fw_copy {
-
+	dd status=none conv=sync bs=128K count=1 if=/dev/zero  > 128K-0x00.bin
 	# Copy common arc files required for agent fit creation
 	handle_fw_copy_arc
 }
