From 0fe51ea5453ac1aa6a54ea8e67ed585e17254d76 Mon Sep 17 00:00:00 2001
From: ran shalit <rshalit@habana.ai>
Date: Tue, 09 Aug 2022 10:34:37 +0300
Subject: [PATCH] add i2c-slave module

Change-Id: Ie5a16ac11c64d25e9218bfc518ec6a4622030145
Signed-off-by: ran shalit <rshalit@habana.ai>
---

diff --git a/package/zephyr/zephyr.mk b/package/zephyr/zephyr.mk
index 94422f8..23be285 100644
--- a/package/zephyr/zephyr.mk
+++ b/package/zephyr/zephyr.mk
@@ -119,7 +119,7 @@
 ZEPHYR_BUILD_CMDS += && $(TARGET_MAKE_ENV) $(MAKE) -C $(@D)/$(ZEPHYR_PREBOOT_DIR)
 ZEPHYR_INSTALL_TARGET_CMDS += && $(INSTALL) -D -m 0644 $(@D)/$(ZEPHYR_PREBOOT_DIR)/zephyr/zephyr.bin $(BINARIES_DIR)/preboot.bin
 ZEPHYR_INSTALL_TARGET_CMDS += && $(INSTALL) -D -m 0644 $(@D)/$(ZEPHYR_PREBOOT_DIR)/zephyr/zephyr.elf $(BINARIES_DIR)/zephyr_preboot.elf
-ZEPHYR_PREBOOT_MODULES = fdt boot_info mbedtls eeprom
+ZEPHYR_PREBOOT_MODULES = fdt boot_info mbedtls eeprom i2c-indirect
 $(foreach module,$(ZEPHYR_PREBOOT_MODULES),$(call DEFINE_MODULE_ENVVARS,$(module),$(ZEPHYR_PREBOOT_DIR),$(ZEPHYR_PREBOOT_NAME)))
 endif #BR2_PACKAGE_ZEPHYR_ARC_PREBOOT
 
