From 2f2a7b85892cc7daee9e494fe1fc94ce075da27e Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 24 Aug 2022 20:34:17 +0300
Subject: [PATCH] [SW-86869] common: rename arcboot_rom_in_recovery function

To enhance readibility, this patch renames arcboot_rom_in_recovery
to arcboot_rom_check_n_enter_recovery.

Change-Id: I586d24bb1b8ed7f5a2d5e173071794f1325fe6ba
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index daec631..71ee494 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -182,7 +182,7 @@
 	arcboot_rom_init_sp_sram();
 }
 
-__weak void arcboot_rom_in_recovery(void)
+__weak void arcboot_rom_check_n_enter_recovery(void)
 {
 	uint8_t override = 1;
 
@@ -475,7 +475,7 @@
 	arcboot_rom_security_init();
 	arcboot_rom_disable_spi_gw();
 	arcboot_rom_wait_for_signal();
-	arcboot_rom_in_recovery();
+	arcboot_rom_check_n_enter_recovery();
 	arcboot_rom_request_cpld_wdog();
 	if (arcboot_rom_verify(HL_FW_BOOT_PTR_SPI))
 		execute_recovery();
diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 0547cb5..f3e4309 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -989,7 +989,7 @@
 	return false;
 }
 
-void arcboot_rom_in_recovery(void)
+void arcboot_rom_check_n_enter_recovery(void)
 {
 	uint8_t override = 1;
 
