From 2e5ab6313a35f03896e93ac38a307742eed9342d Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Mon, 05 Dec 2022 17:38:59 +0200
Subject: [PATCH] Revert "[SW-112749] gaudi2: reduce linux mem to 80mb"

This reverts commit d57c9d6274756eeb2175baf020862df66ceae8d3.
---

diff --git a/board/habanalabs/gaudi2/linux_asic.fragment b/board/habanalabs/gaudi2/linux_asic.fragment
index d3b1754..c63886d 100644
--- a/board/habanalabs/gaudi2/linux_asic.fragment
+++ b/board/habanalabs/gaudi2/linux_asic.fragment
@@ -2,7 +2,7 @@
 CONFIG_SERIAL_MPS2_UART_CONSOLE=y
 CONFIG_SERIAL_MPS2_UART=y
 CONFIG_NR_CPUS=2
-CONFIG_CMDLINE="earlycon console=ttyMPS0,115200n8 root=/dev/ram mem=80M"
+CONFIG_CMDLINE="earlycon console=ttyMPS0,115200n8 root=/dev/ram mem=256M"
 CONFIG_EEPROM_AT24=n
 CONFIG_SENSORS_POWR1220=n
 CONFIG_SENSORS_LM75=n
diff --git a/board/habanalabs/gaudi2/post-image-fit.sh b/board/habanalabs/gaudi2/post-image-fit.sh
index c83d313..042fd4e 100644
--- a/board/habanalabs/gaudi2/post-image-fit.sh
+++ b/board/habanalabs/gaudi2/post-image-fit.sh
@@ -3,7 +3,7 @@
 # DDR/HBM Start address as needed by Linux
 KERNEL_START_ADDR_LO=0x00000000
 KERNEL_START_ADDR_HI=0x00000080
-KERNEL_LOAD_ADDR_LO=0x02080000
+KERNEL_LOAD_ADDR_LO=0x08080000
 
 NIC_FW_FILE=${BR2_EXTERNAL}/board/habanalabs/${HL_DEV}/fw/condor.fw.bin
 
