From ccab59f1b8d1a2387410d5e8fea679cb0b09e37c Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Thu, 12 Mar 2020 19:24:16 +0200
Subject: [PATCH] [SW-10412] goya2: i2c slave address compliance

To complie with melrose spec we changed the i2c slave address.

Change-Id: I859bbdbaf76f44fbe7ffb4b85418f819db2e5147
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/common/security.c b/common/security.c
index 5110338..9a604ba 100644
--- a/common/security.c
+++ b/common/security.c
@@ -472,7 +472,7 @@
 void secuirty_i2c_s_enable(void)
 {
 	/* Slave address */
-	regmap_writel(0x55, mmPSOC_I2C_S_IC_SAR);
+	regmap_writel(0x6A, mmPSOC_I2C_S_IC_SAR);
 	/* IC_RESTART_EN and speed: standard mode - 100 kbit/s */
 	regmap_writel(0x22, mmPSOC_I2C_S_IC_CON);
 	/* Enable all interrupt via mask */
