From 08e15fc1d993f45d81a7c0b03e74bfe8b1a69ff2 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Fri, 20 Mar 2020 11:48:20 +0200
Subject: [PATCH] [SW-10751] goya2: change offset in flash to sector 6

To better support future changes to reduce flash size we reduced offset
to sector number 6.

Change-Id: I6856b961c81a382b6f279b3ba3a44a9dc9b04bb4
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/Makefile b/Makefile
index 3197f4b..2759833 100644
--- a/Makefile
+++ b/Makefile
@@ -25,7 +25,7 @@
 ARCBOOT_CFLAGS += -I$(dir $(BR2_EXTERNAL))/embedded/specs
 ARCBOOT_CFLAGS += -DBTLVER=$(ARCBOOT_VER)
 ARCBOOT_CFLAGS += -DBTLID=0x30
-ARCBOOT_CFLAGS += -DFLASH_START_OFFSET=0x400000
+ARCBOOT_CFLAGS += -DFLASH_START_OFFSET=0x60000
 
 LDFLAGS = -flto
 LDFLAGS += -L $(LIBDIR) -lsb -lellipsys -L $(HL_DEV)
