From efbb33934275a808268fc06d838f7f39645df959 Mon Sep 17 00:00:00 2001
From: Nina Gertskin <ngertskin@habana.ai>
Date: Mon, 01 May 2023 11:09:37 +0300
Subject: [PATCH] [SW-0] emb: stam for test

Change-Id: I89a31708d2d573482510fbb66fc9af1111887e13
Signed-off-by: Nina Gertskin <ngertskin@habana.ai>
---

diff --git a/scripts/tests/local_run b/scripts/tests/local_run
index 30bf8cf..22c0b28 100755
--- a/scripts/tests/local_run
+++ b/scripts/tests/local_run
@@ -20,6 +20,7 @@
 fi
 
 print_msg "Starting: local_run $cmd_prepare $cmd_build $cmd_program $fw_type $version"
+print_msg "stam for test"
 
 dev_gaudi2=`lspci -d 1da3:1020`
 if [ -z "$dev_gaudi2" ]; then
