From cf3f82b679a0e72a8f84bdcacbe74a6367b55029 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 10 May 2023 14:09:00 +0300
Subject: [PATCH] [SW-0] gitignore: ignore vscode meta data

Ignore vscode meta data.

Change-Id: I06e0036a2adf31ea28e7e7d75aab8a0991f31c51
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/.gitignore b/.gitignore
index 2c985f1..4c5fe91 100644
--- a/.gitignore
+++ b/.gitignore
@@ -2,3 +2,4 @@
 .cproject
 .project
 .settings/
+.vscode/
