From bb8be762d08396fd23462ad7bd500ff05076147d Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Thu, 28 Sep 2023 18:57:16 +0300
Subject: [PATCH] remove openssl from hl-fw-loader dependencies

Change-Id: I5eded4ac222152f6426207c9ab6146d384bbf751
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/package/hl-fw-loader/hl-fw-loader.mk b/package/hl-fw-loader/hl-fw-loader.mk
index f5f97fb..3cdb485 100644
--- a/package/hl-fw-loader/hl-fw-loader.mk
+++ b/package/hl-fw-loader/hl-fw-loader.mk
@@ -1,7 +1,7 @@
 HL_FW_LOADER_SITE = hl-fw-loader
 HL_FW_LOADER_LICENSE = unclear
 
-HL_FW_LOADER_DEPENDENCIES = libopenssl pciutils zlib eeprom-util dtc hlmi lz4
+HL_FW_LOADER_DEPENDENCIES = pciutils zlib eeprom-util dtc hlmi lz4
 
 HL_FW_LOADER_INC += -I $(SPECS_ROOT) -I $(SPECS_EMBEDDED_ROOT) \
 	  -I ${EMBEDDED_ROOT} -I ${EMBEDDED_ROOT}/hlmi/include
@@ -20,7 +20,7 @@
 	$(INSTALL) -D -m 0755 $(@D)/build/bin/hl-fw-loader $(BINARIES_DIR)/hl-fw-loader
 endef
 
-HOST_HL_FW_LOADER_DEPENDENCIES = host-libopenssl host-pciutils host-zlib host-eeprom-util host-dtc host-hl-i2c-tools host-hlmi
+HOST_HL_FW_LOADER_DEPENDENCIES = host-pciutils host-zlib host-eeprom-util host-dtc host-hl-i2c-tools host-hlmi
 
 HOST_HL_FW_LOADER_CONF_OPTS = -DTESTS_ENABLED=OFF -DCMAKE_EXE_LINKER_FLAGS="-static" -DCMAKE_BUILD_TYPE="Release"
 
