From 1a89437c40b77b033a8a4dd965379985dca08c51 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Wed, 25 Oct 2023 16:29:07 +0300
Subject: [PATCH] move zephyr-fwl -> zephyr-agent-fwl

Change-Id: Ic953846ddf7ac10f5a88bcede96bb701270f0780
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/package/zephyr-fwl/zephyr-agent-fwl.mk b/package/zephyr-fwl/zephyr-agent-fwl.mk
new file mode 100644
index 0000000..b5e078b
--- /dev/null
+++ b/package/zephyr-fwl/zephyr-agent-fwl.mk
@@ -0,0 +1,149 @@
+ZEPHYR_FWL_APP_NAME = fw_loader
+ZEPHYR_FWL_DEPENDENCIES = host-cmake host-lzop host-lz4
+ZEPHYR_FWL_APP_MODULES = boot_info eeprom fdt i2c-indirect lz4 mbedtls hl_shell soc_init
+
+ZEPHYR_FWL_SRCDIR=$(EMBEDDED_DIR)/zephyr-plugins/fw-loader
+ZEPHYR_FWL_SRC_DIR=$(EMBEDDED_DIR)/zephyr-plugins/fw-loader
+ZEPHYR_FWL_ZEPHYR_SRCDIR=$(EMBEDDED_DIR)/zephyr
+ZEPHYR_FWL_APP_PROD_NAME = zephyr_$(HL_TARGET)_$(ZEPHYR_FWL_APP_NAME)
+ZEPHYR_FWL_DEVICE_NAME = $(firstword $(subst _, ,$(HL_DEV)))
+ZEPHYR_FWL_APP_VERSION = $(shell cut -d- -f2- $(@D)/localversion-habanalabs)
+ZEPHYR_FWL_LOCAL_VERSION = $$(sed "s/^.*fw-\\([0-9]\\+\\.[0-9]\\+\\.[0-9]\\+\\).*$$/\\1/" $(@D)/localversion-habanalabs)
+
+ZEPHYR_FWL_DIR = build-agent-fw-loader
+
+ZEPHYR_FWL_CONF_APP_EXTRA_CFLAGS = \
+		-include $(EMBEDDED_ROOT)/specs/$(ZEPHYR_FWL_DEVICE_NAME)/$(ZEPHYR_FWL_DEVICE_NAME)_boot_info.h \
+		$(ZEPHYR_CONF_DEV_EXTRA_CFLAGS)
+
+ZEPHYR_FWL_CONF_APP_CFLAGS = \
+		-I$(ZEPHYR_FWL_SRC_DIR)/../common/include \
+		-I$(ZEPHYR_FWL_SRC_DIR)/../common/include/$(HL_DEV) \
+		-I$(ZEPHYR_FWL_SRC_DIR)/include
+
+ZEPHYR_FWL_CONF_OPTS =
+ifdef BR2_arcle
+ZEPHYR_FWL_CONF_APP_EXTRA_CFLAGS =
+ZEPHYR_FWL_CONF_OPTS += -DBOARD=hl_$(ZEPHYR_FWL_DEVICE_NAME)_agent_$(HL_TARGET)
+endif
+ifdef BR2_aarch64
+ZEPHYR_FWL_CONF_APP_EXTRA_CFLAGS += -mstrict-align
+ZEPHYR_FWL_CONF_OPTS += -DBOARD=hl_$(ZEPHYR_FWL_DEVICE_NAME)_$(HL_TARGET)
+endif
+
+ifeq ($(HL_TARGET),pldm)
+ZEPHYR_FWL_CONF_APP_EXTRA_CFLAGS = -DTARGET_IS_PLDM
+endif
+
+ifeq ($(HL_TARGET),onix)
+ZEPHYR_FWL_CONF_APP_EXTRA_CFLAGS = -DTARGET_IS_ONIX
+endif
+
+ZEPHYR_FWL_BUILD_DIR = $(@D)
+ZEPHYR_FWL_BUILDDIR = $(@D)
+ZEPHYR_FWL_DIR = $(@D)
+ZEPHYR_FWL_ZEPHYR_BUILD_DIR ?= $(@D)/zephyr
+
+ZEPHYR_FWL_CONF_ENV = CROSS_COMPILE=$(TARGET_CROSS)
+ZEPHYR_FWL_CONF_ENV += ZEPHYR_TOOLCHAIN_VARIANT=cross-compile
+ZEPHYR_FWL_CONF_ENV += ZEPHYR_BASE=$(ZEPHYR_FWL_ZEPHYR_BUILD_DIR)
+
+ZEPHYR_FWL_CONF_OPTS += -UCMAKE_TOOLCHAIN_FILE
+ZEPHYR_FWL_CONF_OPTS += -DUSER_CACHE_DIR=$(@D)/.cache
+ZEPHYR_FWL_CONF_OPTS += -DCMAKE_EXE_LINKER_FLAGS_INIT=-Wl,-e__start
+ZEPHYR_FWL_CONF_OPTS += -DEXTRA_CFLAGS="\
+		-Werror \
+		-Wno-misleading-indentation \
+		$(ZEPHYR_FWL_CONF_APP_EXTRA_CFLAGS)"
+
+ZEPHYR_FWL_CONF_OPTS += -DFAKE_SECURITY=$(FAKE_SECURITY)
+ZEPHYR_FWL_CONF_OPTS += -DBUILD_VERSION="$(ZEPHYR_FWL_APP_VERSION)"
+ZEPHYR_FWL_CONF_OPTS += -DBUILD_MACHINE="$(shell hostname -s)"
+ZEPHYR_FWL_CONF_OPTS += -DCMAKE_C_FLAGS="\
+		-I$(SPECS_EMBEDDED_DIR) \
+		-I$(SPECS_EMBEDDED_DIR)/$(DEV_PREFIX) \
+		-I$(SPECS_ROOT) \
+		-I$(SPECS_EXT_ROOT) \
+		-I$(SPECS_ROOT)/$(DEV_PREFIX)/asic_reg \
+		$(ZEPHYR_FWL_CONF_APP_CFLAGS)"
+
+ZEPHYR_FWL_CONF_OPTS += -DDTS_EXTRA_CPPFLAGS="-I$(SPECS_EMBEDDED_DIR)"
+ZEPHYR_FWL_CONF_OPTS += -DDEVICE=$(ZEPHYR_FWL_DEVICE_NAME)
+ZEPHYR_FWL_CONF_OPTS += -DAPP_NAME=$(ZEPHYR_FWL_APP_PROD_NAME)
+ZEPHYR_FWL_CONF_OPTS += -DAPP_VERSION=$(ZEPHYR_FWL_APP_VERSION)
+ZEPHYR_FWL_WEST_SRC_DIR :=
+ZEPHYR_FWL_WEST_DEST_DIR :=
+
+define ZEPHYR_FWL_COPY_ZEPHYR_SRC
+	rsync -au --chmod=u=rwX,go=rX  --exclude .svn --exclude .git --exclude .hg --exclude .bzr --exclude CVS $(ZEPHYR_FWL_ZEPHYR_SRCDIR) $(ZEPHYR_FWL_BUILD_DIR)
+endef
+
+# Arguments:
+# - module name
+# - component dir
+# - component name
+#
+define ZEPHYR_FWL_DEFINE_MODULE_ENVVARS
+	$(eval MODULE_NAME=$(strip $1))
+	$(eval MODULE_NAME_UPPER_CASE=$(shell echo $(MODULE_NAME) | tr a-z A-Z))
+	$(eval MODULE_NAME_UPPER_CASE=$(strip $(MODULE_NAME_UPPER_CASE)))
+	$(eval $(MODULE_NAME_UPPER_CASE)_SRCDIR=$(EMBEDDED_DIR)/zephyr-modules/$(MODULE_NAME))
+	$(eval $(MODULE_NAME_UPPER_CASE)_DIR=$(@D)/build/modules/$(MODULE_NAME))
+	$(eval $(MODULE_NAME_UPPER_CASE)_NAME=$2_$(MODULE_NAME))
+endef
+
+$(foreach module,$(ZEPHYR_FWL_APP_MODULES),$(call ZEPHYR_FWL_DEFINE_MODULE_ENVVARS,$(module),$(ZEPHYR_FWL_APP_PROD_NAME)))
+
+# Arguments:
+# - module (mbedtls, eeprom, etc.)
+define ZEPHYR_FWL_MODULE_VERSION_HOOK
+	$(eval MODULE_NAME_UPPER_CASE=$(shell echo $1 | tr a-z A-Z))
+	$(eval MODULE_NAME_UPPER_CASE=$(strip $(MODULE_NAME_UPPER_CASE)))
+	$(call HL_VERSION_GEN, ${${MODULE_NAME_UPPER_CASE}_SRCDIR}, ${${MODULE_NAME_UPPER_CASE}_DIR}, ${${MODULE_NAME_UPPER_CASE}_NAME})
+endef
+
+define ZEPHYR_FWL_APP_VERSION_GEN
+	@$(call HL_VERSION_GEN, ${ZEPHYR_FWL_SRCDIR},.,${ZEPHYR_FWL_APP_PROD_NAME})
+	@$(call HL_VERSION_GEN, ${ZEPHYR_FWL_SRCDIR},build,${ZEPHYR_FWL_APP_PROD_NAME})
+	@$(foreach module,$(ZEPHYR_FWL_APP_MODULES),$(call ZEPHYR_FWL_MODULE_VERSION_HOOK,$(module)))
+endef
+
+define ZEPHYR_FWL_CREATE_HL_VERSION
+	$(call MESSAGE, " [$(ZEPHYR_FWL_BUILD_DIR)/include/_hl_version.h]")
+
+	@mkdir -p $(ZEPHYR_FWL_BUILD_DIR)/include
+	@echo "#ifndef _HL_VERSION_" >  $(ZEPHYR_FWL_BUILD_DIR)/include/_hl_version.h
+	@echo "#define _HL_VERSION_" >>  $(ZEPHYR_FWL_BUILD_DIR)/include/_hl_version.h
+	@echo "#define HL_VERSION_MAJOR $$(echo $(ZEPHYR_FWL_LOCAL_VERSION) | cut -d. -f1)" >> $(ZEPHYR_FWL_BUILD_DIR)/include/_hl_version.h
+	@echo "#define HL_VERSION_MINOR $$(echo $(ZEPHYR_FWL_LOCAL_VERSION) | cut -d. -f2)" >> $(ZEPHYR_FWL_BUILD_DIR)/include/_hl_version.h
+	@echo "#define HL_VERSION_PATCH $$(echo $(ZEPHYR_FWL_LOCAL_VERSION) | cut -d. -f3)" >> $(ZEPHYR_FWL_BUILD_DIR)/include/_hl_version.h
+	@echo "#endif" >>  $(ZEPHYR_FWL_BUILD_DIR)/include/_hl_version.h
+endef
+
+define ZEPHYR_FWL_CONFIGURE_CMDS
+	@$(call ZEPHYR_FWL_CREATE_HL_VERSION)
+	$(ZEPHYR_FWL_CONF_ENV) $(HOST_DIR)/bin/cmake $(ZEPHYR_FWL_CONF_OPTS) -S $(ZEPHYR_FWL_BUILD_DIR)/fw-loader -B $(ZEPHYR_FWL_BUILD_DIR)/build
+endef
+
+ZEPHYR_FWL_POST_RSYNC_HOOKS += HL_VERSION_GEN
+ZEPHYR_FWL_POST_RSYNC_HOOKS += ZEPHYR_FWL_COPY_ZEPHYR_SRC
+ZEPHYR_FWL_POST_RSYNC_HOOKS += ZEPHYR_FWL_APP_VERSION_GEN
+
+define ZEPHYR_FWL_BUILD_CMDS
+	$(TARGET_MAKE_ENV) $(MAKE) -C $(ZEPHYR_FWL_BUILD_DIR)/build
+endef
+
+define ZEPHYR_FWL_INSTALL_TARGET_CMDS
+
+	@$(INSTALL) -D -m 0644 $(@D)/build/zephyr/zephyr.bin $(BINARIES_DIR)/zephyr_agent_$(ZEPHYR_FWL_APP_NAME).bin
+	@$(INSTALL) -D -m 0644 $(@D)/build/zephyr/zephyr.elf $(BINARIES_DIR)/zephyr_agent_$(ZEPHYR_FWL_APP_NAME).elf
+	@$(INSTALL) -D -m 0644 $(@D)/build/zephyr/zephyr.stat $(BINARIES_DIR)/zephyr_agent_$(ZEPHYR_FWL_APP_NAME).stat
+
+endef
+
+zephyr-fwl-configure: $(ZEPHYR_FWL_DEPENDENCIES)
+zephyr-fwl-menuconfig zephyr-fwl-guiconfig: | zephyr-fwl-configure
+	$(TARGET_MAKE_ENV) $(MAKE) -C $(ZEPHYR_FWL_BUILD_DIR)/build
+	rm -f $(ZEPHYR_FWL_BUILD_DIR)/.stamp_{configured,built}
+
+$(eval $(cmake-package))
diff --git a/package/zephyr-fwl/zephyr-fwl.mk b/package/zephyr-fwl/zephyr-fwl.mk
deleted file mode 100644
index b5e078b..0000000
--- a/package/zephyr-fwl/zephyr-fwl.mk
+++ /dev/null
@@ -1,149 +0,0 @@
-ZEPHYR_FWL_APP_NAME = fw_loader
-ZEPHYR_FWL_DEPENDENCIES = host-cmake host-lzop host-lz4
-ZEPHYR_FWL_APP_MODULES = boot_info eeprom fdt i2c-indirect lz4 mbedtls hl_shell soc_init
-
-ZEPHYR_FWL_SRCDIR=$(EMBEDDED_DIR)/zephyr-plugins/fw-loader
-ZEPHYR_FWL_SRC_DIR=$(EMBEDDED_DIR)/zephyr-plugins/fw-loader
-ZEPHYR_FWL_ZEPHYR_SRCDIR=$(EMBEDDED_DIR)/zephyr
-ZEPHYR_FWL_APP_PROD_NAME = zephyr_$(HL_TARGET)_$(ZEPHYR_FWL_APP_NAME)
-ZEPHYR_FWL_DEVICE_NAME = $(firstword $(subst _, ,$(HL_DEV)))
-ZEPHYR_FWL_APP_VERSION = $(shell cut -d- -f2- $(@D)/localversion-habanalabs)
-ZEPHYR_FWL_LOCAL_VERSION = $$(sed "s/^.*fw-\\([0-9]\\+\\.[0-9]\\+\\.[0-9]\\+\\).*$$/\\1/" $(@D)/localversion-habanalabs)
-
-ZEPHYR_FWL_DIR = build-agent-fw-loader
-
-ZEPHYR_FWL_CONF_APP_EXTRA_CFLAGS = \
-		-include $(EMBEDDED_ROOT)/specs/$(ZEPHYR_FWL_DEVICE_NAME)/$(ZEPHYR_FWL_DEVICE_NAME)_boot_info.h \
-		$(ZEPHYR_CONF_DEV_EXTRA_CFLAGS)
-
-ZEPHYR_FWL_CONF_APP_CFLAGS = \
-		-I$(ZEPHYR_FWL_SRC_DIR)/../common/include \
-		-I$(ZEPHYR_FWL_SRC_DIR)/../common/include/$(HL_DEV) \
-		-I$(ZEPHYR_FWL_SRC_DIR)/include
-
-ZEPHYR_FWL_CONF_OPTS =
-ifdef BR2_arcle
-ZEPHYR_FWL_CONF_APP_EXTRA_CFLAGS =
-ZEPHYR_FWL_CONF_OPTS += -DBOARD=hl_$(ZEPHYR_FWL_DEVICE_NAME)_agent_$(HL_TARGET)
-endif
-ifdef BR2_aarch64
-ZEPHYR_FWL_CONF_APP_EXTRA_CFLAGS += -mstrict-align
-ZEPHYR_FWL_CONF_OPTS += -DBOARD=hl_$(ZEPHYR_FWL_DEVICE_NAME)_$(HL_TARGET)
-endif
-
-ifeq ($(HL_TARGET),pldm)
-ZEPHYR_FWL_CONF_APP_EXTRA_CFLAGS = -DTARGET_IS_PLDM
-endif
-
-ifeq ($(HL_TARGET),onix)
-ZEPHYR_FWL_CONF_APP_EXTRA_CFLAGS = -DTARGET_IS_ONIX
-endif
-
-ZEPHYR_FWL_BUILD_DIR = $(@D)
-ZEPHYR_FWL_BUILDDIR = $(@D)
-ZEPHYR_FWL_DIR = $(@D)
-ZEPHYR_FWL_ZEPHYR_BUILD_DIR ?= $(@D)/zephyr
-
-ZEPHYR_FWL_CONF_ENV = CROSS_COMPILE=$(TARGET_CROSS)
-ZEPHYR_FWL_CONF_ENV += ZEPHYR_TOOLCHAIN_VARIANT=cross-compile
-ZEPHYR_FWL_CONF_ENV += ZEPHYR_BASE=$(ZEPHYR_FWL_ZEPHYR_BUILD_DIR)
-
-ZEPHYR_FWL_CONF_OPTS += -UCMAKE_TOOLCHAIN_FILE
-ZEPHYR_FWL_CONF_OPTS += -DUSER_CACHE_DIR=$(@D)/.cache
-ZEPHYR_FWL_CONF_OPTS += -DCMAKE_EXE_LINKER_FLAGS_INIT=-Wl,-e__start
-ZEPHYR_FWL_CONF_OPTS += -DEXTRA_CFLAGS="\
-		-Werror \
-		-Wno-misleading-indentation \
-		$(ZEPHYR_FWL_CONF_APP_EXTRA_CFLAGS)"
-
-ZEPHYR_FWL_CONF_OPTS += -DFAKE_SECURITY=$(FAKE_SECURITY)
-ZEPHYR_FWL_CONF_OPTS += -DBUILD_VERSION="$(ZEPHYR_FWL_APP_VERSION)"
-ZEPHYR_FWL_CONF_OPTS += -DBUILD_MACHINE="$(shell hostname -s)"
-ZEPHYR_FWL_CONF_OPTS += -DCMAKE_C_FLAGS="\
-		-I$(SPECS_EMBEDDED_DIR) \
-		-I$(SPECS_EMBEDDED_DIR)/$(DEV_PREFIX) \
-		-I$(SPECS_ROOT) \
-		-I$(SPECS_EXT_ROOT) \
-		-I$(SPECS_ROOT)/$(DEV_PREFIX)/asic_reg \
-		$(ZEPHYR_FWL_CONF_APP_CFLAGS)"
-
-ZEPHYR_FWL_CONF_OPTS += -DDTS_EXTRA_CPPFLAGS="-I$(SPECS_EMBEDDED_DIR)"
-ZEPHYR_FWL_CONF_OPTS += -DDEVICE=$(ZEPHYR_FWL_DEVICE_NAME)
-ZEPHYR_FWL_CONF_OPTS += -DAPP_NAME=$(ZEPHYR_FWL_APP_PROD_NAME)
-ZEPHYR_FWL_CONF_OPTS += -DAPP_VERSION=$(ZEPHYR_FWL_APP_VERSION)
-ZEPHYR_FWL_WEST_SRC_DIR :=
-ZEPHYR_FWL_WEST_DEST_DIR :=
-
-define ZEPHYR_FWL_COPY_ZEPHYR_SRC
-	rsync -au --chmod=u=rwX,go=rX  --exclude .svn --exclude .git --exclude .hg --exclude .bzr --exclude CVS $(ZEPHYR_FWL_ZEPHYR_SRCDIR) $(ZEPHYR_FWL_BUILD_DIR)
-endef
-
-# Arguments:
-# - module name
-# - component dir
-# - component name
-#
-define ZEPHYR_FWL_DEFINE_MODULE_ENVVARS
-	$(eval MODULE_NAME=$(strip $1))
-	$(eval MODULE_NAME_UPPER_CASE=$(shell echo $(MODULE_NAME) | tr a-z A-Z))
-	$(eval MODULE_NAME_UPPER_CASE=$(strip $(MODULE_NAME_UPPER_CASE)))
-	$(eval $(MODULE_NAME_UPPER_CASE)_SRCDIR=$(EMBEDDED_DIR)/zephyr-modules/$(MODULE_NAME))
-	$(eval $(MODULE_NAME_UPPER_CASE)_DIR=$(@D)/build/modules/$(MODULE_NAME))
-	$(eval $(MODULE_NAME_UPPER_CASE)_NAME=$2_$(MODULE_NAME))
-endef
-
-$(foreach module,$(ZEPHYR_FWL_APP_MODULES),$(call ZEPHYR_FWL_DEFINE_MODULE_ENVVARS,$(module),$(ZEPHYR_FWL_APP_PROD_NAME)))
-
-# Arguments:
-# - module (mbedtls, eeprom, etc.)
-define ZEPHYR_FWL_MODULE_VERSION_HOOK
-	$(eval MODULE_NAME_UPPER_CASE=$(shell echo $1 | tr a-z A-Z))
-	$(eval MODULE_NAME_UPPER_CASE=$(strip $(MODULE_NAME_UPPER_CASE)))
-	$(call HL_VERSION_GEN, ${${MODULE_NAME_UPPER_CASE}_SRCDIR}, ${${MODULE_NAME_UPPER_CASE}_DIR}, ${${MODULE_NAME_UPPER_CASE}_NAME})
-endef
-
-define ZEPHYR_FWL_APP_VERSION_GEN
-	@$(call HL_VERSION_GEN, ${ZEPHYR_FWL_SRCDIR},.,${ZEPHYR_FWL_APP_PROD_NAME})
-	@$(call HL_VERSION_GEN, ${ZEPHYR_FWL_SRCDIR},build,${ZEPHYR_FWL_APP_PROD_NAME})
-	@$(foreach module,$(ZEPHYR_FWL_APP_MODULES),$(call ZEPHYR_FWL_MODULE_VERSION_HOOK,$(module)))
-endef
-
-define ZEPHYR_FWL_CREATE_HL_VERSION
-	$(call MESSAGE, " [$(ZEPHYR_FWL_BUILD_DIR)/include/_hl_version.h]")
-
-	@mkdir -p $(ZEPHYR_FWL_BUILD_DIR)/include
-	@echo "#ifndef _HL_VERSION_" >  $(ZEPHYR_FWL_BUILD_DIR)/include/_hl_version.h
-	@echo "#define _HL_VERSION_" >>  $(ZEPHYR_FWL_BUILD_DIR)/include/_hl_version.h
-	@echo "#define HL_VERSION_MAJOR $$(echo $(ZEPHYR_FWL_LOCAL_VERSION) | cut -d. -f1)" >> $(ZEPHYR_FWL_BUILD_DIR)/include/_hl_version.h
-	@echo "#define HL_VERSION_MINOR $$(echo $(ZEPHYR_FWL_LOCAL_VERSION) | cut -d. -f2)" >> $(ZEPHYR_FWL_BUILD_DIR)/include/_hl_version.h
-	@echo "#define HL_VERSION_PATCH $$(echo $(ZEPHYR_FWL_LOCAL_VERSION) | cut -d. -f3)" >> $(ZEPHYR_FWL_BUILD_DIR)/include/_hl_version.h
-	@echo "#endif" >>  $(ZEPHYR_FWL_BUILD_DIR)/include/_hl_version.h
-endef
-
-define ZEPHYR_FWL_CONFIGURE_CMDS
-	@$(call ZEPHYR_FWL_CREATE_HL_VERSION)
-	$(ZEPHYR_FWL_CONF_ENV) $(HOST_DIR)/bin/cmake $(ZEPHYR_FWL_CONF_OPTS) -S $(ZEPHYR_FWL_BUILD_DIR)/fw-loader -B $(ZEPHYR_FWL_BUILD_DIR)/build
-endef
-
-ZEPHYR_FWL_POST_RSYNC_HOOKS += HL_VERSION_GEN
-ZEPHYR_FWL_POST_RSYNC_HOOKS += ZEPHYR_FWL_COPY_ZEPHYR_SRC
-ZEPHYR_FWL_POST_RSYNC_HOOKS += ZEPHYR_FWL_APP_VERSION_GEN
-
-define ZEPHYR_FWL_BUILD_CMDS
-	$(TARGET_MAKE_ENV) $(MAKE) -C $(ZEPHYR_FWL_BUILD_DIR)/build
-endef
-
-define ZEPHYR_FWL_INSTALL_TARGET_CMDS
-
-	@$(INSTALL) -D -m 0644 $(@D)/build/zephyr/zephyr.bin $(BINARIES_DIR)/zephyr_agent_$(ZEPHYR_FWL_APP_NAME).bin
-	@$(INSTALL) -D -m 0644 $(@D)/build/zephyr/zephyr.elf $(BINARIES_DIR)/zephyr_agent_$(ZEPHYR_FWL_APP_NAME).elf
-	@$(INSTALL) -D -m 0644 $(@D)/build/zephyr/zephyr.stat $(BINARIES_DIR)/zephyr_agent_$(ZEPHYR_FWL_APP_NAME).stat
-
-endef
-
-zephyr-fwl-configure: $(ZEPHYR_FWL_DEPENDENCIES)
-zephyr-fwl-menuconfig zephyr-fwl-guiconfig: | zephyr-fwl-configure
-	$(TARGET_MAKE_ENV) $(MAKE) -C $(ZEPHYR_FWL_BUILD_DIR)/build
-	rm -f $(ZEPHYR_FWL_BUILD_DIR)/.stamp_{configured,built}
-
-$(eval $(cmake-package))
