From fc9adc47a4d9ec4ce4f185348ae65e07b87cc151 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Mon, 13 Nov 2023 18:58:09 +0200
Subject: [PATCH] [SW-15774] package: add preboot

Add the preboot standaalone rule

Change-Id: I139abeec515a94d84b9668d2adf25612644d684e
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/board/habanalabs/common/local.mk b/board/habanalabs/common/local.mk
index 6dc76b3..ce80d21 100644
--- a/board/habanalabs/common/local.mk
+++ b/board/habanalabs/common/local.mk
@@ -28,3 +28,4 @@
 AGENT_HBM_OVERRIDE_SRCDIR = $(TOPDIR)/../zephyr-plugins
 AGENT_PCIE_OVERRIDE_SRCDIR = $(TOPDIR)/../zephyr-plugins
 AGENT_SERDES_OVERRIDE_SRCDIR = $(TOPDIR)/../zephyr-plugins
+PREBOOT_OVERRIDE_SRCDIR = $(TOPDIR)/../zephyr-apps/preboot
diff --git a/board/habanalabs/gaudi2/post-image-agents.sh b/board/habanalabs/gaudi2/post-image-agents.sh
index b9a2580..5a41ff8 100755
--- a/board/habanalabs/gaudi2/post-image-agents.sh
+++ b/board/habanalabs/gaudi2/post-image-agents.sh
@@ -24,7 +24,7 @@
 AGENT_ARCH="arm64"
 LOADABLES="security\", \"ppboot\", \"preboot\", \"img-ptr"
 
-HL_PREBOOT_VERSION_SECUREBOOT=`cat ${BR2_EXTERNAL}/output/${HL_DEV}_arc/build/zephyr-custom/build-preboot/buildversion-habanalabs`
+HL_PREBOOT_VERSION_SECUREBOOT=`cat ${BR2_EXTERNAL}/output/${HL_DEV}_arc/build/preboot-custom/build/buildversion-habanalabs`
 HL_PPBOOT_VERSION_SECUREBOOT=`cat ${BR2_EXTERNAL}/output/${HL_DEV}_arc/build/baremetal-custom/buildversion-habanalabs | cut -d ' ' -f1`
 HL_FW_VERSION=`cut -d- -f2- ${BASE_DIR}/build/linux-custom/localversion-habanalabs`
 # Set hl-*-version = "hl-<device>-<HL_VERSION>[-rc]-fw-<fw ver>[-<git data>]"; inside FIT header
diff --git a/board/habanalabs/gaudi3_arc/post-image-agents.sh b/board/habanalabs/gaudi3_arc/post-image-agents.sh
index 2329a03..81e3ed1 100644
--- a/board/habanalabs/gaudi3_arc/post-image-agents.sh
+++ b/board/habanalabs/gaudi3_arc/post-image-agents.sh
@@ -8,7 +8,7 @@
 source ${BR2_EXTERNAL}/board/habanalabs/${HL_DEV}/create-fake-fuse-image.sh
 
 HL_COMPRESS=none
-HL_PREBOOT_VERSION_SECUREBOOT=`cat ${BR2_EXTERNAL}/output/${HL_DEV}/build/zephyr-custom/build-preboot/buildversion-habanalabs`
+HL_PREBOOT_VERSION_SECUREBOOT=`cat ${BR2_EXTERNAL}/output/${HL_DEV}/build/preboot-custom/build/buildversion-habanalabs`
 HL_PPBOOT_VERSION_SECUREBOOT=`cat ${BR2_EXTERNAL}/output/${HL_DEV}/build/baremetal-custom/buildversion-habanalabs | cut -d ' ' -f1`
 HL_UBOOT_VERSION=""
 HL_FW_VERSION=""
diff --git a/package/preboot/preboot.mk b/package/preboot/preboot.mk
new file mode 100644
index 0000000..d57d6b3
--- /dev/null
+++ b/package/preboot/preboot.mk
@@ -0,0 +1,152 @@
+PREBOOT_SITE = preboot
+PREBOOT_APP_NAME = preboot
+PREBOOT_DEPENDENCIES = host-cmake host-lzop host-lz4
+PREBOOT_APP_MODULES = fdt boot_info mbedtls eeprom soc_init hl_shell
+PREBOOT_SRC_DIR=$(EMBEDDED_DIR)/zephyr-apps/preboot
+PREBOOT_ZEPHYR_SRCDIR=$(EMBEDDED_DIR)/zephyr
+
+PREBOOT_CONF_APP_EXTRA_CFLAGS =
+
+ifeq ($(PREBOOT_RUN_FROM_ICCM),true)
+PREBOOT_CONF_APP_OPTS += -DCONFIG_PREBOOT_RUN_FROM_ICCM=y
+endif #PREBOOT_RUN_FROM_ICCM
+
+#TODO - TARGET_IS_ASIC: Need to check why on ZEPHYR_CONF_OPTS its not working as expected
+PREBOOT_CONF_APP_EXTRA_CFLAGS = \
+	-include $(EMBEDDED_ROOT)/specs/$(PREBOOT_DEVICE_NAME)/$(PREBOOT_DEVICE_NAME)_boot_info.h \
+	$($(call UPPERCASE,$(DEV_PREFIX))_CONF_DEV_EXTRA_CFLAGS) \
+	 $(ZEPHYR_CONF_DEV_EXTRA_CFLAGS)
+
+ifeq ($(HL_TARGET),asic)
+	PREBOOT_CONF_APP_EXTRA_CFLAGS += -DTARGET_IS_ASIC
+endif
+ifeq ($(HL_TARGET),onix)
+	PREBOOT_CONF_APP_EXTRA_CFLAGS += -DTARGET_IS_ONIX
+endif
+ifeq ($(HL_TARGET),pldm)
+	PREBOOT_CONF_APP_EXTRA_CFLAGS += -DTARGET_IS_PLDM
+endif
+
+PREBOOT_CONF_APP_CFLAGS = \
+		-I$(PREBOOT_ZEPHYR_BUILD_DIR) \
+		-I$(PREBOOT_ZEPHYR_BUILD_DIR)/include \
+		-I$(PREBOOT_ZEPHYR_BUILD_DIR)/include/drivers/misc/hl_npu \
+		-I$(PREBOOT_ZEPHYR_BUILD_DIR)/include/drivers/misc/hl_npu/$(HL_DEV) \
+		-I$(PREBOOT_ZEPHYR_BUILD_DIR)/include/drivers/misc/hl_npu/common \
+		-I$(PREBOOT_ZEPHYR_BUILD_DIR)/drivers/misc/hl_npu \
+		-I$(PREBOOT_ZEPHYR_BUILD_DIR)/drivers/misc/hl_npu/$(HL_DEV) \
+		-I$(PREBOOT_ZEPHYR_BUILD_DIR)/drivers/misc/hl_npu/common \
+		-I$(PREBOOT_ZEPHYR_BUILD_DIR)/include/generated \
+		-I$(PREBOOT_ZEPHYR_BUILD_DIR)/include/generated/syscalls \
+		-I$(EMBEDDED_DIR)/zephyr-modules/fdt/libfdt
+
+###################################################################
+
+PREBOOT_APP_PROD_NAME = zephyr_arc_$(PREBOOT_APP_NAME)
+PREBOOT_DEVICE_NAME = $(firstword $(subst _, ,$(HL_DEV)))
+PREBOOT_APP_VERSION = $(shell cut -d- -f2- $(@D)/localversion-habanalabs)
+PREBOOT_LOCAL_VERSION = $$(sed "s/^.*fw-\\([0-9]\\+\\.[0-9]\\+\\.[0-9]\\+\\).*$$/\\1/" $(@D)/localversion-habanalabs)
+
+PREBOOT_BUILD_DIR = $(@D)
+PREBOOT_ZEPHYR_BUILD_DIR ?= $(@D)/zephyr
+
+PREBOOT_CONF_ENV = CROSS_COMPILE=$(TARGET_CROSS)
+PREBOOT_CONF_ENV += ZEPHYR_TOOLCHAIN_VARIANT=cross-compile
+PREBOOT_CONF_ENV += ZEPHYR_BASE=$(PREBOOT_ZEPHYR_BUILD_DIR)
+
+PREBOOT_CONF_OPTS  = $(ZEPHYR_CONF_OPTS)
+PREBOOT_CONF_OPTS += -UCMAKE_TOOLCHAIN_FILE
+PREBOOT_CONF_OPTS += -DUSER_CACHE_DIR=$(@D)/.cache
+PREBOOT_CONF_OPTS += -DCMAKE_EXE_LINKER_FLAGS_INIT=-Wl,-e__start
+PREBOOT_CONF_OPTS += -DTARGET=$(HL_TARGET)
+PREBOOT_CONF_OPTS += -DEXTRA_CFLAGS="\
+		-Werror \
+		-Wno-misleading-indentation \
+		$(PREBOOT_CONF_APP_EXTRA_CFLAGS)"
+PREBOOT_CONF_OPTS += $(PREBOOT_CONF_APP_OPTS)
+PREBOOT_CONF_OPTS += -DBUILD_VERSION="$(PREBOOT_APP_VERSION)"
+PREBOOT_CONF_OPTS += -DBUILD_MACHINE="$(shell hostname -s)"
+PREBOOT_CONF_OPTS += -DCMAKE_C_FLAGS="\
+		$($(call UPPERCASE,$(DEV_PREFIX))_CONF_DEV_EXTRA_CFLAGS) \
+		$(PREBOOT_CONF_APP_CFLAGS)"
+
+PREBOOT_CONF_OPTS += -DDTS_EXTRA_CPPFLAGS="-I$(SPECS_EMBEDDED_DIR)"
+PREBOOT_CONF_OPTS += -DBOARD=hl_$(PREBOOT_DEVICE_NAME)_$(PREBOOT_APP_NAME)_$(HL_TARGET)
+PREBOOT_CONF_OPTS += -DDEVICE=$(PREBOOT_DEVICE_NAME)
+PREBOOT_CONF_OPTS += -DAPP_NAME=$(PREBOOT_APP_PROD_NAME)
+PREBOOT_CONF_OPTS += -DAPP_VERSION=$(PREBOOT_APP_VERSION)
+
+PREBOOT_WEST_SRC_DIR :=
+PREBOOT_WEST_DEST_DIR :=
+
+define PREBOOT_COPY_ZEPHYR_SRC
+	rsync -au --chmod=u=rwX,go=rX  --exclude .svn --exclude .git --exclude .hg --exclude .bzr --exclude CVS $(PREBOOT_ZEPHYR_SRCDIR) $(PREBOOT_BUILD_DIR)
+endef
+
+# Arguments:
+# - module name
+# - component dir
+# - component name
+#
+define PREBOOT_DEFINE_MODULE_ENVVARS
+	$(eval MODULE_NAME=$(strip $1))
+	$(eval MODULE_NAME_UPPER_CASE=$(shell echo $(MODULE_NAME) | tr a-z A-Z))
+	$(eval MODULE_NAME_UPPER_CASE=$(strip $(MODULE_NAME_UPPER_CASE)))
+	$(eval $(MODULE_NAME_UPPER_CASE)_SRCDIR=$(EMBEDDED_DIR)/zephyr-modules/$(MODULE_NAME))
+	$(eval $(MODULE_NAME_UPPER_CASE)_DIR=$(@D)/build/modules/$(MODULE_NAME))
+	$(eval $(MODULE_NAME_UPPER_CASE)_NAME=$2_$(MODULE_NAME))
+endef
+
+$(foreach module,$(PREBOOT_APP_MODULES),$(call PREBOOT_DEFINE_MODULE_ENVVARS,$(module),$(PREBOOT_APP_PROD_NAME)))
+
+# Arguments:
+# - module (mbedtls, eeprom, etc.)
+define PREBOOT_MODULE_VERSION_HOOK
+	$(eval MODULE_NAME_UPPER_CASE=$(shell echo $1 | tr a-z A-Z))
+	$(eval MODULE_NAME_UPPER_CASE=$(strip $(MODULE_NAME_UPPER_CASE)))
+	$(call HL_VERSION_GEN, ${${MODULE_NAME_UPPER_CASE}_SRCDIR}, ${${MODULE_NAME_UPPER_CASE}_DIR}, ${${MODULE_NAME_UPPER_CASE}_NAME})
+endef
+
+define PREBOOT_APP_VERSION_GEN
+	@$(call HL_VERSION_GEN, ${PREBOOT_SRCDIR},.,${PREBOOT_APP_PROD_NAME})
+	@$(call HL_VERSION_GEN, ${PREBOOT_SRCDIR},build,${PREBOOT_APP_PROD_NAME})
+	@$(foreach module,$(PREBOOT_APP_MODULES),$(call PREBOOT_MODULE_VERSION_HOOK,$(module)))
+endef
+
+define PREBOOT_CREATE_HL_VERSION
+	$(call MESSAGE, " [$(PREBOOT_BUILD_DIR)/include/hl_version.h]")
+
+	@mkdir -p $(PREBOOT_BUILD_DIR)/include
+	@echo "#ifndef _HL_VERSION_" >  $(PREBOOT_BUILD_DIR)/include/hl_version.h
+	@echo "#define _HL_VERSION_" >>  $(PREBOOT_BUILD_DIR)/include/hl_version.h
+	@echo "#define HL_VERSION_MAJOR $$(echo $(PREBOOT_LOCAL_VERSION) | cut -d. -f1)" >> $(PREBOOT_BUILD_DIR)/include/hl_version.h
+	@echo "#define HL_VERSION_MINOR $$(echo $(PREBOOT_LOCAL_VERSION) | cut -d. -f2)" >> $(PREBOOT_BUILD_DIR)/include/hl_version.h
+	@echo "#define HL_VERSION_PATCH $$(echo $(PREBOOT_LOCAL_VERSION) | cut -d. -f3)" >> $(PREBOOT_BUILD_DIR)/include/hl_version.h
+	@echo "#endif" >>  $(PREBOOT_BUILD_DIR)/include/hl_version.h
+endef
+
+define PREBOOT_CONFIGURE_CMDS
+#	$(INSTALL) -D -m 0755 $(PREBOOT_WEST_SRC_DIR)/west.config $(PREBOOT_WEST_DEST_DIR)/config
+	@$(call PREBOOT_CREATE_HL_VERSION)
+	@$(PREBOOT_CONF_ENV) $(HOST_DIR)/bin/cmake $(PREBOOT_CONF_OPTS) -S $(PREBOOT_BUILD_DIR) -B $(PREBOOT_BUILD_DIR)/build
+endef
+
+PREBOOT_POST_RSYNC_HOOKS += HL_VERSION_GEN
+PREBOOT_POST_RSYNC_HOOKS += PREBOOT_COPY_ZEPHYR_SRC
+PREBOOT_POST_RSYNC_HOOKS += PREBOOT_APP_VERSION_GEN
+
+define PREBOOT_BUILD_CMDS
+	$(TARGET_MAKE_ENV) $(MAKE) -C $(PREBOOT_BUILD_DIR)/build all
+endef
+
+define PREBOOT_INSTALL_TARGET_CMDS
+	@$(INSTALL) -D -m 0644 $(@D)/build/zephyr/zephyr.elf $(BINARIES_DIR)/zephyr_$(PREBOOT_APP_NAME).elf
+	@$(INSTALL) -D -m 0644 $(@D)/build/zephyr/zephyr.bin $(BINARIES_DIR)/$(PREBOOT_APP_NAME).bin
+endef
+
+preboot-configure: $(PREBOOT_DEPENDENCIES)
+preboot-menuconfig preboot-guiconfig: | preboot-configure
+	$(TARGET_MAKE_ENV) $(MAKE) -C $(PREBOOT_BUILD_DIR)/build all
+	rm -f $(PREBOOT_BUILD_DIR)/.stamp_{configured,built}
+
+$(eval $(cmake-package))
