From 62ba8779e7ff0fabd124b4db3ec51c4f01c682da Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Tue, 21 Nov 2023 14:20:59 +0200
Subject: [PATCH] only zephyr mgmt

Change-Id: I133c0a1043ce0c79e8d8b0182dabdd9e252f6153
---

diff --git a/board/habanalabs/common/zephyr.its b/board/habanalabs/common/zephyr.its
index 141021a..fb42798 100644
--- a/board/habanalabs/common/zephyr.its
+++ b/board/habanalabs/common/zephyr.its
@@ -107,8 +107,7 @@
 		preboot-conf {
 			load-type = <0>; /* FIT_LOAD_TYPE_RUN */
 			description = "Habanalabs device preboot";
-			loadables = "zephyr-pid-iccm", "zephyr-pid-dccm", "zephyr-mgmt-iccm",
-				    "zephyr-mgmt-dccm", "zephyr-event-iccm", "zephyr-event-dccm",
+			loadables = "zephyr-mgmt-iccm", "zephyr-mgmt-dccm",
 				    "external_fw";
 
 		};
diff --git a/configs/gaudi3_arc_defconfig b/configs/gaudi3_arc_defconfig
index d3e51ce..d197060 100644
--- a/configs/gaudi3_arc_defconfig
+++ b/configs/gaudi3_arc_defconfig
@@ -26,7 +26,7 @@
 BR2_PACKAGE_ZEPHYR_ARC_PID=y
 BR2_PACKAGE_ZEPHYR_ARC_EVENT=y
 BR2_PACKAGE_ZEPHYR_ARC_PREBOOT=y
-BR2_PACKAGE_ZEPHYR_ARC_PREBOOT_D1=y
+#BR2_PACKAGE_ZEPHYR_ARC_PREBOOT_D1=y
 BR2_PACKAGE_PREBOOT_ARC_EXT_ROM=y
 BR2_PACKAGE_ZEPHYR_AGENT_FW_LOADER=y
 BR2_POST_IMAGE_FW_LOADER_ITS_TEMPLATE="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/agent-fw_loader.its"
