From dd0e1dc65582a38cb5762b9a0116bb56585c1019 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Mon, 04 Dec 2023 17:55:17 +0200
Subject: [PATCH] WIP baremetal-boot <=> baremetal

Change-Id: I55c5805256449af23e9a9a761c70df7997b1a99c
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/Config.in b/Config.in
index e63f306..5775d98 100644
--- a/Config.in
+++ b/Config.in
@@ -3,7 +3,7 @@
 source "$BR2_EXTERNAL_HABANALABS_PATH/package/sve/Config.in"
 #source "$BR2_EXTERNAL_HABANALABS_PATH/package/sve_zephyr/Config.in"
 source "$BR2_EXTERNAL_HABANALABS_PATH/package/thermal/Config.in"
-source "$BR2_EXTERNAL_HABANALABS_PATH/package/baremetal-boot/Config.in"
+source "$BR2_EXTERNAL_HABANALABS_PATH/package/baremetal/Config.in"
 source "$BR2_EXTERNAL_HABANALABS_PATH/package/btl/Config.in"
 source "$BR2_EXTERNAL_HABANALABS_PATH/package/eeprom-util/Config.in"
 source "$BR2_EXTERNAL_HABANALABS_PATH/package/arcboot/Config.in"
diff --git a/board/habanalabs/common/local.mk b/board/habanalabs/common/local.mk
index 5d8cab1..05b7462 100644
--- a/board/habanalabs/common/local.mk
+++ b/board/habanalabs/common/local.mk
@@ -3,7 +3,7 @@
 ARMCP_OVERRIDE_SRCDIR = $(TOPDIR)/../armcp
 DSV_OVERRIDE_SRCDIR = $(TOPDIR)/../dsv
 THERMAL_OVERRIDE_SRCDIR = $(TOPDIR)/../thermal
-BAREMETAL_BOOT_OVERRIDE_SRCDIR = $(TOPDIR)/../preboot
+BAREMETAL_OVERRIDE_SRCDIR = $(TOPDIR)/../preboot
 HL_FW_LOADER_OVERRIDE_SRCDIR = $(TOPDIR)/../fw_loader
 BTL_OVERRIDE_SRCDIR = $(TOPDIR)/../btl
 HL_SMI_OVERRIDE_SRCDIR = $(TOPDIR)/../hl-smi
diff --git a/board/habanalabs/common/post-image-agents.sh b/board/habanalabs/common/post-image-agents.sh
index 4c1defe..547b586 100755
--- a/board/habanalabs/common/post-image-agents.sh
+++ b/board/habanalabs/common/post-image-agents.sh
@@ -20,7 +20,7 @@
 HL_VERSION=`cut -d- -f4 ${BASE_DIR}/build/zephyr-custom/localversion-habanalabs`
 # Set hl-*-version = "hl-<device>-<HL_VERSION>[-rc]-fw-<fw ver>[-<git data>]"; inside FIT header
 # Set hl-preboot-version = "hl-<device>-<HL_VERSION>[-rc]-fw-<fw ver>[-<git data>] (<date> - <time>)"
-HL_PREBOOT_VERSION=`cat ${BASE_DIR}/build/baremetal-boot-custom/buildversion-habanalabs`
+HL_PREBOOT_VERSION=`cat ${BASE_DIR}/build/baremetal-custom/buildversion-habanalabs`
 # Set enum number of FIT_LOAD_TYPE_FLASH in FIT
 FIT_LOAD_TYPE_FLASH=$(grep -E "LOAD_TYPE_FLASH[^_]" ${SPECS_EMBEDDED_DIR}/fit_load_type.h | sed 's/.* = \(.*\),/\1/')
 FIT_LOAD_TYPE_RUN=$(grep -E "FIT_LOAD_TYPE_RUN[^_]" ${SPECS_EMBEDDED_DIR}/fit_load_type.h | sed 's/.* = \(.*\),/\1/')
diff --git a/board/habanalabs/common/post-image-fit.sh b/board/habanalabs/common/post-image-fit.sh
index 5394555..732f047 100755
--- a/board/habanalabs/common/post-image-fit.sh
+++ b/board/habanalabs/common/post-image-fit.sh
@@ -28,7 +28,7 @@
 HL_FW_VERSION=`cut -d- -f2- ${BASE_DIR}/build/linux-custom/localversion-habanalabs`
 HL_UBOOT_VERSION=`cut -d- -f2- ${BASE_DIR}/build/uboot-custom/localversion-habanalabs`
 # Set hl-preboot-version = "hl-<device>-<HL_VERSION>[-rc]-fw-<fw ver>[-<git data>] (<date> - <time>)"
-HL_PREBOOT_VERSION=`cat ${BASE_DIR}/build/baremetal-boot-custom/buildversion-habanalabs`
+HL_PREBOOT_VERSION=`cat ${BASE_DIR}/build/baremetal-custom/buildversion-habanalabs`
 # Set enum number of FIT_LOAD_TYPE_FLASH in FIT
 FIT_LOAD_TYPE_FLASH=$(grep -E "FIT_LOAD_TYPE_FLASH[^_]" ${SPECS_EMBEDDED_DIR}/fit_load_type.h | sed 's/.* = \(.*\),/\1/')
 # Set enum number of FIT_LOAD_TYPE_RUN in FIT
diff --git a/board/habanalabs/common/post-image-spi-arc_global.sh b/board/habanalabs/common/post-image-spi-arc_global.sh
index cc8a039..f252ee1 100755
--- a/board/habanalabs/common/post-image-spi-arc_global.sh
+++ b/board/habanalabs/common/post-image-spi-arc_global.sh
@@ -109,7 +109,7 @@
 
 function hl_get_ver {
 	FILE=$1
-	HL_VERSION=$(cut -d'-' -f4 ${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/build/baremetal-boot-custom/localversion-habanalabs)
+	HL_VERSION=$(cut -d'-' -f4 ${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/build/baremetal-custom/localversion-habanalabs)
 	W=${SECURITY_FW_VERSION}
 	X=$(echo ${HL_VERSION} | cut -d'.' -f1)
 	Y=$(echo ${HL_VERSION} | cut -d'.' -f2)
diff --git a/board/habanalabs/gaudi2/post-image-agents.sh b/board/habanalabs/gaudi2/post-image-agents.sh
index ed806aa..995f964 100755
--- a/board/habanalabs/gaudi2/post-image-agents.sh
+++ b/board/habanalabs/gaudi2/post-image-agents.sh
@@ -24,8 +24,8 @@
 AGENT_ARCH="arm64"
 LOADABLES="security\", \"ppboot\", \"preboot\", \"img-ptr"
 
-HL_PREBOOT_VERSION_SECUREBOOT=`cat ${BR2_EXTERNAL}/output/${HL_DEV}_arc/build/zephyr-custom/build-preboot/buildversion-habanalabs`
-HL_PPBOOT_VERSION_SECUREBOOT=`cat ${BR2_EXTERNAL}/output/${HL_DEV}_arc/build/baremetal-boot-custom/buildversion-habanalabs | cut -d ' ' -f1`
+HL_PREBOOT_VERSION_SECUREBOOT=`cat ${BR2_EXTERNAL}/output/${HL_DEV}_arc/build/preboot-custom/build/buildversion-habanalabs`
+HL_PPBOOT_VERSION_SECUREBOOT=`cat ${BR2_EXTERNAL}/output/${HL_DEV}_arc/build/baremetal-custom/buildversion-habanalabs | cut -d ' ' -f1`
 HL_FW_VERSION=`cut -d- -f2- ${BASE_DIR}/build/linux-custom/localversion-habanalabs`
 # Set hl-*-version = "hl-<device>-<HL_VERSION>[-rc]-fw-<fw ver>[-<git data>]"; inside FIT header
 HL_UBOOT_VERSION=`cut -d- -f2- ${BASE_DIR}/build/uboot-custom/localversion-habanalabs`
diff --git a/board/habanalabs/gaudi2/post-image-fit.sh b/board/habanalabs/gaudi2/post-image-fit.sh
index ccc03bf..20eedfc 100644
--- a/board/habanalabs/gaudi2/post-image-fit.sh
+++ b/board/habanalabs/gaudi2/post-image-fit.sh
@@ -32,7 +32,7 @@
 
 DEV_NAME=`echo $HL_DEV | cut -d'_' -f1`
 # Set hl-preboot-version = "hl-<device>-<HL_VERSION>[-rc]-fw-<fw ver>[-<git data>] (<date> - <time>)"
-HL_PREBOOT_VERSION=`cat ${BR2_EXTERNAL}/output/${DEV_NAME}_${HL_ARCH}/build/baremetal-boot-custom/buildversion-habanalabs`
+HL_PREBOOT_VERSION=`cat ${BR2_EXTERNAL}/output/${DEV_NAME}_${HL_ARCH}/build/baremetal-custom/buildversion-habanalabs`
 
 # Set the first found public key file name in dtb format inside the its template
 KEYS_DIR=${BR2_EXTERNAL}/board/habanalabs/${HL_DEV}/key/
diff --git a/board/habanalabs/gaudi2_arc/post-image-spi.sh b/board/habanalabs/gaudi2_arc/post-image-spi.sh
index 151eb5d..2168158 100755
--- a/board/habanalabs/gaudi2_arc/post-image-spi.sh
+++ b/board/habanalabs/gaudi2_arc/post-image-spi.sh
@@ -21,7 +21,7 @@
 
 EROM_ITS_TEMPLATE=${BR2_EXTERNAL}/board/habanalabs/${HL_DEV}/erom_data.its
 MKIMAGE_FILE=${BR2_EXTERNAL}/output/${HL_DEV}/host/bin/mkimage
-EROM_VERSION_FILE=${BR2_EXTERNAL}/output/${HL_DEV}/build/baremetal-boot-custom/buildversion-habanalabs
+EROM_VERSION_FILE=${BR2_EXTERNAL}/output/${HL_DEV}/build/baremetal-custom/buildversion-habanalabs
 
 PREBOOT_PRIMARY_IMAGE_OFFSET=0x00080000
 PREBOOT_SECONDARY_IMAGE_OFFSET=0x000D0000
diff --git a/board/habanalabs/gaudi3_arc/post-image-agents.sh b/board/habanalabs/gaudi3_arc/post-image-agents.sh
index a4df2d3..81e3ed1 100644
--- a/board/habanalabs/gaudi3_arc/post-image-agents.sh
+++ b/board/habanalabs/gaudi3_arc/post-image-agents.sh
@@ -8,8 +8,8 @@
 source ${BR2_EXTERNAL}/board/habanalabs/${HL_DEV}/create-fake-fuse-image.sh
 
 HL_COMPRESS=none
-HL_PREBOOT_VERSION_SECUREBOOT=`cat ${BR2_EXTERNAL}/output/${HL_DEV}/build/zephyr-custom/build-preboot/buildversion-habanalabs`
-HL_PPBOOT_VERSION_SECUREBOOT=`cat ${BR2_EXTERNAL}/output/${HL_DEV}/build/baremetal-boot-custom/buildversion-habanalabs | cut -d ' ' -f1`
+HL_PREBOOT_VERSION_SECUREBOOT=`cat ${BR2_EXTERNAL}/output/${HL_DEV}/build/preboot-custom/build/buildversion-habanalabs`
+HL_PPBOOT_VERSION_SECUREBOOT=`cat ${BR2_EXTERNAL}/output/${HL_DEV}/build/baremetal-custom/buildversion-habanalabs | cut -d ' ' -f1`
 HL_UBOOT_VERSION=""
 HL_FW_VERSION=""
 GAUDI3_SRAM_ADDR_LO=0xF0000000
diff --git a/configs/gaudi2_arc_defconfig b/configs/gaudi2_arc_defconfig
index 64fbce2..67e1f57 100644
--- a/configs/gaudi2_arc_defconfig
+++ b/configs/gaudi2_arc_defconfig
@@ -15,8 +15,8 @@
 # BR2_PACKAGE_BUSYBOX is not set
 # BR2_PACKAGE_IFUPDOWN_SCRIPTS is not set
 # BR2_TARGET_ROOTFS_TAR is not set
-BR2_PACKAGE_BAREMETAL_BOOT=y
-BR2_PACKAGE_BAREMETAL_BOOT_ARC_EXT_ROM=y
+BR2_PACKAGE_BAREMETAL=y
+BR2_PACKAGE_BAREMETAL_ARC_EXT_ROM=y
 BR2_PACKAGE_ARCBOOT=y
 BR2_PACKAGE_ZEPHYR_ARC_PID=y
 BR2_PACKAGE_ZEPHYR_ARC_MGMT=y
diff --git a/configs/gaudi2_arm_defconfig b/configs/gaudi2_arm_defconfig
index a5ffa07..4c83e81 100644
--- a/configs/gaudi2_arm_defconfig
+++ b/configs/gaudi2_arm_defconfig
@@ -60,7 +60,7 @@
 BR2_PACKAGE_HOST_UBOOT_TOOLS_FIT_SUPPORT=y
 BR2_PACKAGE_HOST_UBOOT_TOOLS_FIT_SIGNATURE_SUPPORT=y
 BR2_PACKAGE_ARMCP=y
-BR2_PACKAGE_BAREMETAL_BOOT=y
+BR2_PACKAGE_BAREMETAL=y
 BR2_PACKAGE_BTL=y
 BR2_PACKAGE_EEPROM_UTIL=y
 BR2_POST_IMAGE_UBOOT_ITS_TEMPLATE="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/uboot_extended.its"
diff --git a/configs/gaudi3_arc_defconfig b/configs/gaudi3_arc_defconfig
index 89b4557..61a27d7 100644
--- a/configs/gaudi3_arc_defconfig
+++ b/configs/gaudi3_arc_defconfig
@@ -19,7 +19,7 @@
 BR2_PACKAGE_HOST_UBOOT_TOOLS=y
 BR2_PACKAGE_HOST_UBOOT_TOOLS_FIT_SUPPORT=y
 BR2_PACKAGE_HOST_UBOOT_TOOLS_FIT_SIGNATURE_SUPPORT=y
-BR2_PACKAGE_BAREMETAL_BOOT=y
+BR2_PACKAGE_BAREMETAL=y
 BR2_PACKAGE_PCI_PIF_ARC=y
 BR2_PACKAGE_ARCBOOT=y
 BR2_PACKAGE_ZEPHYR_ARC_MGMT=y
@@ -27,7 +27,7 @@
 BR2_PACKAGE_ZEPHYR_ARC_EVENT=y
 BR2_PACKAGE_ZEPHYR_ARC_PREBOOT=y
 BR2_PACKAGE_ZEPHYR_ARC_PREBOOT_D1=y
-BR2_PACKAGE_BAREMETAL_BOOT_ARC_EXT_ROM=y
+BR2_PACKAGE_BAREMETAL_ARC_EXT_ROM=y
 BR2_PACKAGE_ZEPHYR_AGENT_FW_LOADER=y
 BR2_POST_IMAGE_FW_LOADER_ITS_TEMPLATE="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/agent-fw_loader.its"
 BR2_POST_IMAGE_AGENTS_ITS_TEMPLATE="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/agent.its"
diff --git a/package/baremetal-boot/Config.in b/package/baremetal-boot/Config.in
deleted file mode 100644
index 1ef6346..0000000
--- a/package/baremetal-boot/Config.in
+++ /dev/null
@@ -1,25 +0,0 @@
-config BR2_PACKAGE_BAREMETAL_BOOT
-	bool "Bare Metal Boot"
-	help
-	  Pre boot loader
-	  Manage PLLs on SoC
-
-config BR2_PACKAGE_BAREMETAL_BOOT_ARC_RECOVERY
-	bool "ARC recovery"
-	help
-	  ARC recovery image loaded on ARC1 in recovery mode
-
-config BR2_PACKAGE_BAREMETAL_BOOT_ARC_EXT_ROM
-	bool "ARC Extended ROM"
-	help
-	  ARC Extended ROM image loaded on ARC1, during ROM execution
-
-config BR2_PACKAGE_PCI_PIF_ARC
-	bool "PIF ARC CPU firmware"
-	help
-	  dedicated PciE ARC CPU image for pci initialization and FLR handling"
-
-config BR2_PACKAGE_PLDM_BLOADER
-	bool "preboot rtos bootloader for pldm"
-	help
-	  bootloader application dedicated for loading preboot rtos on pldm
diff --git a/package/baremetal-boot/baremetal-boot.mk b/package/baremetal-boot/baremetal-boot.mk
deleted file mode 100644
index 991d769..0000000
--- a/package/baremetal-boot/baremetal-boot.mk
+++ /dev/null
@@ -1,75 +0,0 @@
-BAREMETAL_BOOT_SITE = baremetal-boot
-BAREMETAL_BOOT_SRC_DIR = $(EMBEDDED_DIR)/preboot
-
-ifneq ($(HL_DEV),greco_arc_elf32)
-BAREMETAL_BOOT_DEPENDENCIES = dtc
-endif
-
-ifeq ($(FAKE_SECURITY),ON)
-BAREMETAL_BOOT_CONF_OPTS = -DFAKE_SECURITY=ON
-endif
-
-define BAREMETAL_BOOT_BUILD_CMDS
-
-	$(TARGET_MAKE_ENV) $(MAKE) CROSS_COMPILE="$(TARGET_CROSS)" -C $(@D) all
-
-	if [ "$(BR2_PACKAGE_BAREMETAL_BOOT_ARC_EXT_ROM)" == "y" ]; then \
-		$(TARGET_MAKE_ENV) $(MAKE) CROSS_COMPILE="$(TARGET_CROSS)" -C $(@D)/app/ext_rom all; \
-	fi;
-
-	if [ "$(BR2_PACKAGE_PCI_PIF_ARC)" == "y" ]; then \
-		$(TARGET_MAKE_ENV) $(MAKE) CROSS_COMPILE="$(TARGET_CROSS)" -C $(@D)/app/pcie_pif all; \
-	fi;
-
-	if [ "$(HL_TARGET)" == "pldm" ]; then \
-		if [ "$(BR2_PACKAGE_PLDM_BLOADER)" == "y" ]; then \
-			$(TARGET_MAKE_ENV) $(MAKE) CROSS_COMPILE="$(TARGET_CROSS)" -C $(@D)/app/pldm_bloader all; \
-		fi; \
-	fi;
-
-endef
-
-BAREMETAL_BOOT_INSTALL_IMAGES = YES
-
-define BAREMETAL_BOOT_INSTALL_IMAGES_CMDS
-	if [ "$(HL_DEV)" == "greco_arc" ]; then \
-		$(INSTALL) -D -m 0644 $(@D)/bin/preboot.bin $(BINARIES_DIR)/preboot.bin; \
-	elif [  "$(HL_DEV)" == "greco_arc_elf32" ]; then \
-		$(INSTALL) -D -m 0644 $(@D)/bin/ppboot.bin $(BINARIES_DIR)/ppboot.bin; \
-		$(INSTALL) -D -m 0644 $(@D)/bin/pcie_phy_fw.bin $(BINARIES_DIR)/pcie_phy_fw.bin; \
-	elif [  "$(HL_DEV)" == "gaudi2_arc" ] || [ "$(HL_DEV)" == "gaudi3_arc" ]; then \
-		$(INSTALL) -D -m 0644 $(@D)/bin/ppboot.bin $(BINARIES_DIR)/ppboot.bin; \
-		$(INSTALL) -D -m 0644 $(@D)/bin/pcie_phy_fw.bin $(BINARIES_DIR)/pcie_phy_fw.bin; \
-	else \
-		$(INSTALL) -D -m 0644 $(@D)/bin/ppboot.bin $(BINARIES_DIR)/ppboot.bin; \
-		$(INSTALL) -D -m 0644 $(@D)/bin/preboot.bin $(BINARIES_DIR)/preboot.bin; \
-		$(INSTALL) -D -m 0644 $(@D)/bin/pcie_phy_fw.bin $(BINARIES_DIR)/pcie_phy_fw.bin; \
-	fi;
-
-	if [ "$(BR2_PACKAGE_BAREMETAL_BOOT_ARC_EXT_ROM)" == "y" ]; then \
-		$(INSTALL) -D -m 0644 $(@D)/app/ext_rom/erom_recovery.bin $(BINARIES_DIR)/erom_recovery.bin; \
-		$(INSTALL) -D -m 0644 $(@D)/app/ext_rom/erom_pcie.bin $(BINARIES_DIR)/erom_pcie.bin; \
-		$(INSTALL) -D -m 0644 $(@D)/app/ext_rom/erom_tpm.bin $(BINARIES_DIR)/erom_tpm.bin; \
-	fi;
-
-	if [ "$(BR2_PACKAGE_PCI_PIF_ARC)" == "y" ]; then \
-		$(INSTALL) -D -m 0644 $(@D)/app/pcie_pif/pcie_pif.elf $(BINARIES_DIR)/pcie_pif.elf; \
-	fi;
-
-	if [ "$(HL_TARGET)" == "pldm" ]; then \
-		if [ "$(BR2_PACKAGE_PLDM_BLOADER)" == "y" ]; then \
-			$(INSTALL) -D -m 0644 $(@D)/app/pldm_bloader/pldm_bloader.elf $(BINARIES_DIR)/pldm_bloader.elf; \
-			$(INSTALL) -D -m 0644 $(@D)/app/pldm_bloader/pldm_bloader.bin $(BINARIES_DIR)/pldm_bloader.bin; \
-		fi; \
-	fi;
-
-endef
-
-# force BTL rebuild
-define BAREMETAL_BOOT_CLEAN_BTL
-	$(MAKE) -C $(BR2_EXTERNAL) btl-dirclean
-endef
-
-BAREMETAL_BOOT_POST_BUILD_HOOKS += BAREMETAL_BOOT_CLEAN_BTL
-
-$(eval $(generic-package))
diff --git a/package/baremetal/Config.in b/package/baremetal/Config.in
new file mode 100644
index 0000000..7c470c2
--- /dev/null
+++ b/package/baremetal/Config.in
@@ -0,0 +1,25 @@
+config BR2_PACKAGE_BAREMETAL
+	bool "Bare Metal Boot"
+	help
+	  Pre boot loader
+	  Manage PLLs on SoC
+
+config BR2_PACKAGE_BAREMETAL_ARC_RECOVERY
+	bool "ARC recovery"
+	help
+	  ARC recovery image loaded on ARC1 in recovery mode
+
+config BR2_PACKAGE_BAREMETAL_ARC_EXT_ROM
+	bool "ARC Extended ROM"
+	help
+	  ARC Extended ROM image loaded on ARC1, during ROM execution
+
+config BR2_PACKAGE_PCI_PIF_ARC
+	bool "PIF ARC CPU firmware"
+	help
+	  dedicated PciE ARC CPU image for pci initialization and FLR handling"
+
+config BR2_PACKAGE_PLDM_BLOADER
+	bool "preboot rtos bootloader for pldm"
+	help
+	  bootloader application dedicated for loading preboot rtos on pldm
diff --git a/package/baremetal/baremetal.mk b/package/baremetal/baremetal.mk
new file mode 100644
index 0000000..c4b9726
--- /dev/null
+++ b/package/baremetal/baremetal.mk
@@ -0,0 +1,75 @@
+BAREMETAL_SITE = baremetal
+BAREMETAL_SRC_DIR = $(EMBEDDED_DIR)/preboot
+
+ifneq ($(HL_DEV),greco_arc_elf32)
+BAREMETAL_DEPENDENCIES = dtc
+endif
+
+ifeq ($(FAKE_SECURITY),ON)
+BAREMETAL_CONF_OPTS = -DFAKE_SECURITY=ON
+endif
+
+define BAREMETAL_BUILD_CMDS
+
+	$(TARGET_MAKE_ENV) $(MAKE) CROSS_COMPILE="$(TARGET_CROSS)" -C $(@D) all
+
+	if [ "$(BR2_PACKAGE_BAREMETAL_ARC_EXT_ROM)" == "y" ]; then \
+		$(TARGET_MAKE_ENV) $(MAKE) CROSS_COMPILE="$(TARGET_CROSS)" -C $(@D)/app/ext_rom all; \
+	fi;
+
+	if [ "$(BR2_PACKAGE_PCI_PIF_ARC)" == "y" ]; then \
+		$(TARGET_MAKE_ENV) $(MAKE) CROSS_COMPILE="$(TARGET_CROSS)" -C $(@D)/app/pcie_pif all; \
+	fi;
+
+	if [ "$(HL_TARGET)" == "pldm" ]; then \
+		if [ "$(BR2_PACKAGE_PLDM_BLOADER)" == "y" ]; then \
+			$(TARGET_MAKE_ENV) $(MAKE) CROSS_COMPILE="$(TARGET_CROSS)" -C $(@D)/app/pldm_bloader all; \
+		fi; \
+	fi;
+
+endef
+
+BAREMETAL_INSTALL_IMAGES = YES
+
+define BAREMETAL_INSTALL_IMAGES_CMDS
+	if [ "$(HL_DEV)" == "greco_arc" ]; then \
+		$(INSTALL) -D -m 0644 $(@D)/bin/preboot.bin $(BINARIES_DIR)/preboot.bin; \
+	elif [  "$(HL_DEV)" == "greco_arc_elf32" ]; then \
+		$(INSTALL) -D -m 0644 $(@D)/bin/ppboot.bin $(BINARIES_DIR)/ppboot.bin; \
+		$(INSTALL) -D -m 0644 $(@D)/bin/pcie_phy_fw.bin $(BINARIES_DIR)/pcie_phy_fw.bin; \
+	elif [  "$(HL_DEV)" == "gaudi2_arc" ] || [ "$(HL_DEV)" == "gaudi3_arc" ]; then \
+		$(INSTALL) -D -m 0644 $(@D)/bin/ppboot.bin $(BINARIES_DIR)/ppboot.bin; \
+		$(INSTALL) -D -m 0644 $(@D)/bin/pcie_phy_fw.bin $(BINARIES_DIR)/pcie_phy_fw.bin; \
+	else \
+		$(INSTALL) -D -m 0644 $(@D)/bin/ppboot.bin $(BINARIES_DIR)/ppboot.bin; \
+		$(INSTALL) -D -m 0644 $(@D)/bin/preboot.bin $(BINARIES_DIR)/preboot.bin; \
+		$(INSTALL) -D -m 0644 $(@D)/bin/pcie_phy_fw.bin $(BINARIES_DIR)/pcie_phy_fw.bin; \
+	fi;
+
+	if [ "$(BR2_PACKAGE_BAREMETAL_ARC_EXT_ROM)" == "y" ]; then \
+		$(INSTALL) -D -m 0644 $(@D)/app/ext_rom/erom_recovery.bin $(BINARIES_DIR)/erom_recovery.bin; \
+		$(INSTALL) -D -m 0644 $(@D)/app/ext_rom/erom_pcie.bin $(BINARIES_DIR)/erom_pcie.bin; \
+		$(INSTALL) -D -m 0644 $(@D)/app/ext_rom/erom_tpm.bin $(BINARIES_DIR)/erom_tpm.bin; \
+	fi;
+
+	if [ "$(BR2_PACKAGE_PCI_PIF_ARC)" == "y" ]; then \
+		$(INSTALL) -D -m 0644 $(@D)/app/pcie_pif/pcie_pif.elf $(BINARIES_DIR)/pcie_pif.elf; \
+	fi;
+
+	if [ "$(HL_TARGET)" == "pldm" ]; then \
+		if [ "$(BR2_PACKAGE_PLDM_BLOADER)" == "y" ]; then \
+			$(INSTALL) -D -m 0644 $(@D)/app/pldm_bloader/pldm_bloader.elf $(BINARIES_DIR)/pldm_bloader.elf; \
+			$(INSTALL) -D -m 0644 $(@D)/app/pldm_bloader/pldm_bloader.bin $(BINARIES_DIR)/pldm_bloader.bin; \
+		fi; \
+	fi;
+
+endef
+
+# force BTL rebuild
+define BAREMETAL_CLEAN_BTL
+	$(MAKE) -C $(BR2_EXTERNAL) btl-dirclean
+endef
+
+BAREMETAL_POST_BUILD_HOOKS += BAREMETAL_CLEAN_BTL
+
+$(eval $(generic-package))
diff --git a/package/btl/btl.mk b/package/btl/btl.mk
index 6615d28..d7db1c0 100644
--- a/package/btl/btl.mk
+++ b/package/btl/btl.mk
@@ -2,7 +2,7 @@
 BTL_SITE = btl
 BTL_LICENSE = unclear
 
-BTL_DEPENDENCIES = baremetal-boot
+BTL_DEPENDENCIES = baremetal
 
 define BTL_BUILD_CMDS
 	$(TARGET_MAKE_ENV) $(MAKE) -C $(@D) all
diff --git a/package/version-hooks/version-hooks.mk b/package/version-hooks/version-hooks.mk
index 18147a1..fcb627a 100644
--- a/package/version-hooks/version-hooks.mk
+++ b/package/version-hooks/version-hooks.mk
@@ -108,7 +108,7 @@
 LINUX_POST_RSYNC_HOOKS += HL_VERSION_GEN
 THERMAL_POST_RSYNC_HOOKS += HL_VERSION_GEN
 ARMCP_POST_RSYNC_HOOKS += HL_VERSION_GEN
-BAREMETAL_BOOT_POST_RSYNC_HOOKS += HL_VERSION_GEN
+BAREMETAL_POST_RSYNC_HOOKS += HL_VERSION_GEN
 ARCBOOT_POST_RSYNC_HOOKS += HL_VERSION_GEN
 BTL_POST_RSYNC_HOOKS += HL_BTL_VERSION_GEN
 ZEPHYR_POST_RSYNC_HOOKS += HL_VERSION_GEN
diff --git a/scripts/make/gaudi2_arc/Makefile b/scripts/make/gaudi2_arc/Makefile
index 5ec6bb7..5353ae7 100644
--- a/scripts/make/gaudi2_arc/Makefile
+++ b/scripts/make/gaudi2_arc/Makefile
@@ -1,5 +1,5 @@
 ARCBOOT_BASE := ${OUTPUT_DIR}/build/arcboot-custom
-PPBOOT_BASE := ${OUTPUT_DIR}/build/baremetal-boot-custom
+PPBOOT_BASE := ${OUTPUT_DIR}/build/baremetal-custom
 HL_INSTALL_ONIX_LAB_IP ?= emb-srv-host11
 .PHONY: gaudi2_arc-onix-install
 gaudi2_arc-onix-install: gaudi2_arc-onix-target
