From c877cfafe4a86f0e020ab910f5457a19e4aa8ce4 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Thu, 07 Dec 2023 10:21:20 +0200
Subject: [PATCH] test patch

Change-Id: I8ad87900e02dfafb0777cb20f048f6aaa6b8457d
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/package/preboot/preboot.mk b/package/preboot/preboot.mk
index 7327661..341cc64 100644
--- a/package/preboot/preboot.mk
+++ b/package/preboot/preboot.mk
@@ -1,7 +1,7 @@
 PREBOOT_VERSION = 1.0.0
 PREBOOT_SITE = preboot
 PREBOOT_LICENSE = unclear
-
+#this is a test comment
 ifneq ($(HL_DEV),greco_arc_elf32)
 PREBOOT_DEPENDENCIES = dtc
 endif
