From ca8373d812e6c481ca7e640154aeded4c671c0ad Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Thu, 11 Jan 2024 16:30:43 +0200
Subject: [PATCH] Revert "[SW-170522] spi: fixup endianess and sha"

This reverts commit 8853b956997d4c32ab5a7fb3bc11241652081307.
---

diff --git a/board/habanalabs/common/post-image-spi-arc_global.sh b/board/habanalabs/common/post-image-spi-arc_global.sh
index 569bb32..00bd867 100755
--- a/board/habanalabs/common/post-image-spi-arc_global.sh
+++ b/board/habanalabs/common/post-image-spi-arc_global.sh
@@ -5,7 +5,7 @@
 set -o pipefail # stop on any error in pipeline
 
 SHA256_SIZE=32
-SHA256_CHKSUM=sha256sum
+SHA_CHKSUM=sha256sum
 HL_BASE_PATH=${BR2_EXTERNAL}/board/habanalabs
 COMMON_BASE=${HL_BASE_PATH}/common
 BOARD_BASE=${HL_BASE_PATH}/${HL_DEV}
@@ -356,19 +356,17 @@
 			 ${DEV_PREFIX}-${PIF_ICCM_BINFILE}.${ENDIANNESS} ${DEV_PREFIX}-${PIF_DCCM_BINFILE}.${ENDIANNESS} \
 			 ${DEV_PREFIX}-preboot_d1.bin.${ENDIANNESS}
 
-	swap_file ${DEV_PREFIX}-spi-flash.bin.nocrc ${DEV_PREFIX}-spi-flash.bin.nocrc.le
-
 	# Prepare checksum
-	CKSUM=`${SHA256_CHKSUM} ${DEV_PREFIX}-spi-flash.bin.nocrc.le | cut -d' ' -f1`
+	CKSUM=`${SHA_CHKSUM} ${DEV_PREFIX}-spi-flash.bin.nocrc | cut -d' ' -f1`
 	echo ${CKSUM} | xxd -p -r								> ${DEV_PREFIX}-spi-flash.crc
-	dd status=none conv=sync bs=64K			if=${DEV_PREFIX}-spi-flash.bin.nocrc.le	> ${DEV_PREFIX}-spi-flash.bin
+	dd status=none conv=sync bs=64K			if=${DEV_PREFIX}-spi-flash.bin.nocrc	> ${DEV_PREFIX}-spi-flash.bin
 	dd status=none conv=sync bs=$SHA256_SIZE	if=${DEV_PREFIX}-spi-flash.crc		>> ${DEV_PREFIX}-spi-flash.bin
 
 	rm -rf ${PPBOOT_PCIE_PHY_BINFILE}
 
 	## Convert bin file into Palladium hex format ##
 	# for SPI flash
-	hl_pldm_flash_hex spi-flash ${DEV_PREFIX}-spi-flash.bin.nocrc
+	hl_pldm_flash_hex spi-flash ${DEV_PREFIX}-spi-flash.bin
 
 	# for otp flash
 	hl_pldm_flash_hex spi-otp-flash ${DEV_PREFIX}-spi-otp-flash.bin
