From 95af5370f5ab61220709e11710aadaf5d3f84d94 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Thu, 08 Feb 2024 09:40:06 +0200
Subject: [PATCH] [DO NOT MERGE] CI testing

commit msg body

Change-Id: Iba86deb3f150153149e9cd2d8a082b157ff12967
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi3/arc/asic.c b/arch/gaudi3/arc/asic.c
index 50cc976..e1d1ecf 100644
--- a/arch/gaudi3/arc/asic.c
+++ b/arch/gaudi3/arc/asic.c
@@ -33,6 +33,8 @@
 #include <psoc_reset_conf_regs.h>
 #include <psoc_reset_conf_masks.h>
 
+/* khhhhh testing one two */
+
 static void gaudi3_sram_self_repair(u64 sram_loadaddr, struct hl_ras_sram_bist_data *bist_data)
 {
 	u32 *hl_gaudi3_sram_mbist_data_start_address;
