From 65cfc2c813d7891fcd9c6737725881516531a60b Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Mon, 26 Feb 2024 13:08:01 +0200
Subject: [PATCH] [SW-0] build: dsm: fix typo

mctp instead of mcpt

Change-Id: I2442a7e2b6cd2f793dc1b8eb13a25c7607149e03
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/package/dsm/dsm.mk b/package/dsm/dsm.mk
index 4c187ca..bda430d 100644
--- a/package/dsm/dsm.mk
+++ b/package/dsm/dsm.mk
@@ -1,6 +1,6 @@
 DSM_APP_NAME = dsm
 DSM_DEPENDENCIES = host-cmake host-lzop host-lz4
-DSM_APP_MODULES = eeprom mbedtls boot_info i2c-indirect soc_init hl_shell mcpt
+DSM_APP_MODULES = eeprom mbedtls boot_info i2c-indirect soc_init hl_shell mctp
 DSM_SRCDIR=$(EMBEDDED_DIR)/zephyr-apps/pcie-ss/dsm
 DSM_ZEPHYR_SRCDIR=$(EMBEDDED_DIR)/zephyr
 
