From f3be8229628a20d91461ed1ea0c1358295124187 Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Thu, 28 Mar 2024 16:03:16 +0200
Subject: [PATCH] [SW-0] gaudi2: mask double perst

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: I3f88ebd210974dfb5a8fb71f45320105f086cf08
---

diff --git a/app/ppboot/ppboot.c b/app/ppboot/ppboot.c
index f2864fa..825efe2 100644
--- a/app/ppboot/ppboot.c
+++ b/app/ppboot/ppboot.c
@@ -314,6 +314,7 @@
 __weak void hl_pcie_enable_elbi_access(void) {}
 __weak void ppboot_tpm_extend(void) {}
 __weak void ppboot_boot_info_init(void) {}
+__weak void ppboot_mask_perst(void) {}
 
 void ppboot_main(void)
 {
@@ -324,6 +325,7 @@
 	/************time critical section**************/
 	ppboot_time_critical_section_start();
 	device_init();
+	ppboot_mask_perst();
 	/* Set ARCs to privileged mode */
 	ppboot_secure_arc();
 	sram_set_extension();
diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index 052267a..682e93e 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -1342,3 +1342,9 @@
 {
 	return hl_fuse_read(mmPSOC_EFUSE_BASE, word, val);
 }
+
+#define GAUDI2_PERST_MASK_GPIO	70
+void ppboot_mask_perst(void)
+{
+	gaudi2_gpio_write(GAUDI2_PERST_MASK_GPIO, 0);
+}
\ No newline at end of file
