From 36b237db6ade2be7f4f7436f7f6d0ed2f8262c7f Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Tue, 16 Apr 2024 14:36:23 +0300
Subject: [PATCH] sq bins 1

Change-Id: I225fb7384017660bea4fd6b3ad9c7ca281cc5360
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/board/habanalabs/common/post-image-agents.sh b/board/habanalabs/common/post-image-agents.sh
index 89d289e..c27189c 100755
--- a/board/habanalabs/common/post-image-agents.sh
+++ b/board/habanalabs/common/post-image-agents.sh
@@ -57,10 +57,10 @@
 
 	# copy img-pointer, ppboot, preboot, security binaries from other HL targets required for agent fit creation
 	# This will be called by gaudi2 agent scripts
-	cp ${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/images/${HL_DEV}-ppboot-pcie_phy_fw.bin.le ${BINARIES_DIR}/${HL_DEV}-ppboot-arc.bin.le
-	cp ${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/images/${HL_DEV}-preboot.bin.le ${BINARIES_DIR}/${HL_DEV}-preboot-arc.bin.le
-	cp ${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/images/${HL_DEV}-images-pointers.bin.le ${BINARIES_DIR}/${HL_DEV}-images-pointers-arc.bin.le
-	cp ${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/images/${HL_DEV}-security-arc.bin.le ${BINARIES_DIR}
+	cp ${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/images/${HL_DEV}-ppboot-pcie_phy_fw.bin.le ${BINS_DIR}/${HL_DEV}-ppboot-arc.bin.le
+	cp ${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/images/${HL_DEV}-preboot.bin.le ${BINS_DIR}/${HL_DEV}-preboot-arc.bin.le
+	cp ${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/images/${HL_DEV}-images-pointers.bin.le ${BINS_DIR}/${HL_DEV}-images-pointers-arc.bin.le
+	cp ${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/images/${HL_DEV}-security-arc.bin.le ${BINS_DIR}
 }
 
 function handle_agent_fit {
diff --git a/board/habanalabs/common/post-image-fit-arc.sh b/board/habanalabs/common/post-image-fit-arc.sh
index 00c9eb5..bc7c061 100755
--- a/board/habanalabs/common/post-image-fit-arc.sh
+++ b/board/habanalabs/common/post-image-fit-arc.sh
@@ -64,9 +64,9 @@
 
 pushd ${OUTPUT_DIR}
 
-	hl_copy_keys ${OUTPUT_DIR}
+	hl_copy_keys ${BINS_DIR}
 
-	hl_copy_binaries ${OUTPUT_DIR}
+	hl_copy_binaries ${BINS_DIR}
 
 	sed ${ITS_SED_PARAMS} ${ZEPHYR_ITS_TEMPLATE} > ${ZEPHYR_ITS_FILE}
 	${MKIMAGE} -D "-W no-unit_address_vs_reg" -f ${ZEPHYR_ITS_FILE} -k ${OUTPUT_DIR} ${ZEPHYR_ITB_FILE}
diff --git a/board/habanalabs/common/post-image-fit-global.sh b/board/habanalabs/common/post-image-fit-global.sh
index 830c91e..17215b3 100755
--- a/board/habanalabs/common/post-image-fit-global.sh
+++ b/board/habanalabs/common/post-image-fit-global.sh
@@ -18,7 +18,6 @@
 IFS=$'\n'
 
 DEV_NAME=`echo $HL_DEV | cut -d'_' -f1`
-OUTPUT_DIR=${BR2_EXTERNAL}/output/${DEV_NAME}/images
 BOARD_BASE=${BR2_EXTERNAL}/board/habanalabs/${DEV_NAME}
 MKIMAGE=${BR2_EXTERNAL}/output/${DEV_NAME}/host/usr/bin/mkimage
 LZOP=${BR2_EXTERNAL}/output/${DEV_NAME}/host/usr/bin/lzop
@@ -41,24 +40,24 @@
 HL_ZEPHYR_MGMT_VERSION=`cat ${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/build/zephyr-custom/build-mgmt/buildversion-habanalabs`
 HL_ZEPHYR_PID_VERSION=`cat ${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/build/zephyr-custom/build-pid/buildversion-habanalabs`
 
-UBOOT_ZEPHYR_ITS_FILE=uboot-zephyr-${DEV_NAME}_${HL_TARGET}.its
-UBOOT_ZEPHYR_ITB_FILE=uboot-zephyr-fit.itb
+UBOOT_ZEPHYR_ITS_FILE=${STAGE_DIR}/uboot-zephyr-${DEV_NAME}_${HL_TARGET}.its
+UBOOT_ZEPHYR_ITB_FILE=${STAGE_DIR}/uboot-zephyr-fit.itb
 NIC_FW_FILE=${BR2_EXTERNAL}/board/habanalabs/${HL_DEV}/fw/nic_fw.bin
 
-SPI_ITS_FILE=spi-${DEV_NAME}_${HL_TARGET}.its
-SPI_ITB_FILE=spi-fpga-fit.itb
+SPI_ITS_FILE=${STAGE_DIR}/spi-${DEV_NAME}_${HL_TARGET}.its
+SPI_ITB_FILE=${STAGE_DIR}/spi-fpga-fit.itb
 SPI_FPGA_ITS_TEMPLATE=${COMMON_BASE}/spi-fpga-secureboot.its
 
-LINUX_ITS_FILE=linux-${DEV_NAME}_${HL_TARGET}.its
-LINUX_ITB_FILE=linux-fit.itb
+LINUX_ITS_FILE=${STAGE_DIR}/linux-${DEV_NAME}_${HL_TARGET}.its
+LINUX_ITB_FILE=${STAGE_DIR}/linux-fit.itb
 LINUX_ITS_TEMPLATE=${COMMON_BASE}/linux.its
 
-DSV_BIN=dsv.bin
-DSV_ITS_FILE=dsv.its
-DSV_ITS_TEMPLATE=${COMMON_BASE}/${DSV_ITS_FILE}
-DSV_ITB_FILE=dsv-fit.itb
+DSV_BIN=${BINS_DIR}dsv.bin
+DSV_ITS_FILE=${STAGE_DIR}dsv.its
+DSV_ITS_TEMPLATE=${COMMON_BASE}/dsv.its
+DSV_ITB_FILE=${ITBS_DIR}dsv-fit.itb
 
-LINUX_PRODUCTS_DIR=${BASE_DIR}/build/linux-custom/arch/arm64/boot/
+LINUX_PRODUCTS_DIR=${OUTPUT_DIR}/build/linux-custom/arch/arm64/boot/
 LINUX_DTB_DIR=${LINUX_PRODUCTS_DIR}/dts/habanalabs/${HL_DEV}/
 LINUX_ROOTFS_DIR=${IMGS_DIR}
 
@@ -138,20 +137,20 @@
 ITS_SED_PARAMS+="s#@LINUX_ROOTFS_DIR@#${LINUX_ROOTFS_DIR}#g;"
 
 if [ ! -f ${NIC_FW_FILE} ]; then
-	dd bs=4 count=1 if=/dev/zero of=${OUTPUT_DIR}/empty_nic_fw.bin
-	NIC_FW_FILE=${OUTPUT_DIR}/empty_nic_fw.bin
+	dd bs=4 count=1 if=/dev/zero of=${BINS_DIR}/empty_nic_fw.bin
+	NIC_FW_FILE=${BINS_DIR}/empty_nic_fw.bin
 fi
 ITS_SED_PARAMS+="s#@NIC_FW@#${NIC_FW_FILE}#g;"
 
 function generate_dsv_fit
 {
-	if [ ! -f ${OUTPUT_DIR}/${DSV_BIN} ]; then
+	if [ ! -f ${DSV_BIN} ]; then
 		return
 	fi
 
 	# Compress binary
 	if [ ${HL_COMPRESS} == "lzo" ]; then
-		${LZOP} -f ${OUTPUT_DIR}/${DSV_BIN} -o ${OUTPUT_DIR}/${DSV_BIN}.lzop
+		${LZOP} -f ${DSV_BIN} -o ${DSV_BIN}.lzop
 		DSV_BIN+=.lzop
 	elif [ ${HL_COMPRESS} != "none" ]; then
 		echo "Unsupported compression ${HL_COMPRESS}" >&2
@@ -163,41 +162,41 @@
 	${MKIMAGE} -D "-W no-unit_address_vs_reg" -f ${DSV_ITS_FILE} ${DSV_ITB_FILE}
 
 	# LKD likes this to be 4B aligned in size
-	dd status=none conv=sync bs=4 if=${DSV_ITB_FILE} > ${ITBS_DIR}${DEV_PREFIX}-${DSV_ITB_FILE}
+	dd status=none conv=sync bs=4 if=${DSV_ITB_FILE} > ${ITBS_DIR}/${DEV_PREFIX}-${DSV_ITB_FILE}
 
 	rm -f ${DSV_ITB_FILE}
 }
 
-pushd ${OUTPUT_DIR}
+pushd ${STAGE_DIR}
 
-	hl_copy_keys ${OUTPUT_DIR}
+	hl_copy_keys ${BINS_DIR}
 
-	hl_copy_binaries ${OUTPUT_DIR}
+	hl_copy_binaries ${BINS_DIR}
 
 	# Extract the first 4 bytes which contains the jump address
-	dd if=${OUTPUT_DIR}/${HL_DEV}-zephyr-mgmt-iccm.bin.le of=${OUTPUT_DIR}/${HL_DEV}-zephyr-mgmt-iccm.bin.jmp_sram bs=1 count=4
+	dd if=${BINS_DIR}/${HL_DEV}-zephyr-mgmt-iccm.bin.le of=${BINS_DIR}/${HL_DEV}-zephyr-mgmt-iccm.bin.jmp_sram bs=1 count=4
 
 	if [ ${HL_TARGET} == "onix" ]; then
 		sed ${ITS_SED_PARAMS} ${SPI_FPGA_ITS_TEMPLATE} > ${SPI_ITS_FILE}
 		${MKIMAGE} -D "-W no-unit_address_vs_reg" -f ${SPI_ITS_FILE} ${SPI_ITB_FILE}
-		dd status=none conv=sync bs=4 if=${SPI_ITB_FILE} > ${ITBS_DIR}${DEV_NAME}-boot-spi-fpga-fit.itb
+		dd status=none conv=sync bs=4 if=${SPI_ITB_FILE} > ${ITBS_DIR}/${DEV_NAME}-boot-spi-fpga-fit.itb
 	fi
 
-	uboot_bin_sz=$(stat -Lc "%s" ${HL_UBOOT_BIN_FILE})
+	uboot_bin_sz=$(stat -Lc "%s" ${STAGE_DIR}i/${HL_UBOOT_BIN_FILE})
 	if [ ${uboot_bin_sz} -gt ${UBOOT_BIN_MAX_SZ} ]; then
-		echo "Error: ${HL_UBOOT_BIN_FILE} file is larger (${uboot_bin_sz}) than max (${UBOOT_BIN_MAX_SZ})" 1>&2
+		echo "Error: ${STAGE_DIR}/${HL_UBOOT_BIN_FILE} file is larger (${uboot_bin_sz}) than max (${UBOOT_BIN_MAX_SZ})" 1>&2
 		exit 1
 	fi
 	sed ${ITS_SED_PARAMS} ${UBOOT_ZEPHYR_ITS_TEMPLATE} > ${UBOOT_ZEPHYR_ITS_FILE}
-	${MKIMAGE} -D "-W no-unit_address_vs_reg" -f ${UBOOT_ZEPHYR_ITS_FILE} -k ${OUTPUT_DIR} ${UBOOT_ZEPHYR_ITB_FILE}
+	${MKIMAGE} -D "-W no-unit_address_vs_reg" -f ${UBOOT_ZEPHYR_ITS_FILE} -k ${BINS_DIR} ${UBOOT_ZEPHYR_ITB_FILE}
 
 	# TODO: Replace DEV_PREFIX to DEV_NAME
-	dd status=none conv=sync bs=4 if=${UBOOT_ZEPHYR_ITB_FILE} > ${ITBS_DIR}${DEV_PREFIX}-boot-fit.itb
+	dd status=none conv=sync bs=4 if=${UBOOT_ZEPHYR_ITB_FILE} > ${ITBS_DIR}/${DEV_PREFIX}-boot-fit.itb
 
 	sed ${ITS_SED_PARAMS} ${LINUX_ITS_TEMPLATE} > ${LINUX_ITS_FILE}
-	${MKIMAGE} -D "-W no-unit_address_vs_reg" -f ${LINUX_ITS_FILE} -k ${OUTPUT_DIR} ${LINUX_ITB_FILE}
+	${MKIMAGE} -D "-W no-unit_address_vs_reg" -f ${LINUX_ITS_FILE} -k ${BINS_DIR} ${LINUX_ITB_FILE}
 
-	dd status=none conv=sync bs=4 if=${LINUX_ITB_FILE} > ${ITBS_DIR}${DEV_PREFIX}-fit.itb
+	dd status=none conv=sync bs=4 if=${LINUX_ITB_FILE} > ${ITBS_DIR}/${DEV_PREFIX}-fit.itb
 
 	generate_dsv_fit
 popd
diff --git a/board/habanalabs/common/post-image-fit.sh b/board/habanalabs/common/post-image-fit.sh
index 82b61f0..d5e1bf3 100755
--- a/board/habanalabs/common/post-image-fit.sh
+++ b/board/habanalabs/common/post-image-fit.sh
@@ -26,12 +26,12 @@
 IFS=$'\n'
 
 # Set hl-version = "x.y.z"; inside FIT header to be used for KMD/FW compatability validation
-HL_VERSION=`cut -d- -f4 ${BASE_DIR}/build/linux-custom/localversion-habanalabs`
+HL_VERSION=`cut -d- -f4 ${OUTPUT_DIR}/build/linux-custom/localversion-habanalabs`
 # Set hl-*-version = "hl-<device>-<HL_VERSION>[-rc]-fw-<fw ver>[-<git data>]"; inside FIT header
-HL_FW_VERSION=`cut -d- -f2- ${BASE_DIR}/build/linux-custom/localversion-habanalabs`
-HL_UBOOT_VERSION=`cut -d- -f2- ${BASE_DIR}/build/uboot-custom/localversion-habanalabs`
+HL_FW_VERSION=`cut -d- -f2- ${OUTPUT_DIR}/build/linux-custom/localversion-habanalabs`
+HL_UBOOT_VERSION=`cut -d- -f2- ${OUTPUT_DIR}/build/uboot-custom/localversion-habanalabs`
 # Set hl-preboot-version = "hl-<device>-<HL_VERSION>[-rc]-fw-<fw ver>[-<git data>] (<date> - <time>)"
-HL_PREBOOT_VERSION=`cat ${BASE_DIR}/build/baremetal-custom/buildversion-habanalabs`
+HL_PREBOOT_VERSION=`cat ${OUTPUT_DIR}/build/baremetal-custom/buildversion-habanalabs`
 # Set enum number of FIT_LOAD_TYPE_FLASH in FIT
 FIT_LOAD_TYPE_FLASH=$(grep -E "FIT_LOAD_TYPE_FLASH[^_]" ${SPECS_EMBEDDED_DIR}/fit_load_type.h | sed 's/.* = \(.*\),/\1/')
 # Set enum number of FIT_LOAD_TYPE_RUN in FIT
@@ -42,8 +42,8 @@
 HL_SEC_FW_VERSION=`grep HL_SEC_FW_VERSION $HL_SEC_FILE | awk '{ print $NF }'`
 
 NIC_FW_FILE=${BR2_EXTERNAL_HABANALABS_PATH}/board/habanalabs/${HL_DEV}/fw/condor.fw.bin
-UBOOT_ITS_FILE=uboot-${HL_DEV}_${HL_TARGET}.its
-UBOOT_ITB_FILE=uboot-fit.itb
+UBOOT_ITS_FILE=${STAGE_DIR}uboot-${HL_DEV}_${HL_TARGET}.its
+UBOOT_ITB_FILE=${STAGE_DIR}uboot-fit.itb
 UBOOT_ITS_TEMPLATE=`sed -e 's/^"//' -e 's/"$//' <<< ${BR2_POST_IMAGE_UBOOT_ITS_TEMPLATE}`
 SPI_ITS_FILE=${STAGE_DIR}/spi-${HL_DEV}_${HL_TARGET}.its
 SPI_ITB_FILE=${STAGE_DIR}/spi-fpga-fit.itb
@@ -58,11 +58,11 @@
 	LINUX_ITS_TEMPLATE=${COMMON_BASE}/linux.its
 fi
 
-LINUX_PRODUCTS_DIR=${BASE_DIR}/build/linux-custom/arch/arm64/boot/
+LINUX_PRODUCTS_DIR=${OUTPUT_DIR}/build/linux-custom/arch/arm64/boot/
 LINUX_DTB_DIR=${LINUX_PRODUCTS_DIR}/dts/habanalabs/${HL_DEV}/
 LINUX_ROOTFS_DIR=${IMGS_DIR}
 
-DSV_BIN=dsv.bin
+DSV_BIN=${BINS_DIR}/dsv.bin
 DSV_ITS_FILE=${STAGE_DIR}/dsv.its
 DSV_ITS_TEMPLATE=${COMMON_BASE}/${DSV_ITS_FILE}
 DSV_ITB_FILE=${STAGE_DIR}/dsv-fit.itb
@@ -89,7 +89,7 @@
 
 function generate_dsv_fit
 {
-	if [ ! -f ${BINARIES_DIR}/${DSV_BIN} ]; then
+	if [ ! -f ${DSV_BIN} ]; then
 		return
 	fi
 
@@ -104,7 +104,7 @@
 		sed ${ITS_SED_PARAMS} ${SPI_FPGA_ITS_TEMPLATE} > ${SPI_ITS_FILE}
 		${HOST_DIR}/usr/bin/mkimage -D "-W no-unit_address_vs_reg" -f ${SPI_ITS_FILE} -k ${KEYS_DIR} ${SPI_ITB_FILE}
 		# LKD like this to be 4B aligned in size
-		dd status=none conv=sync bs=4 if=${SPI_ITB_FILE} > ${ITBS_DIR}${HL_DEV}-boot-spi-fpga-fit.itb
+		dd status=none conv=sync bs=4 if=${SPI_ITB_FILE} > ${ITBS_DIR}/${HL_DEV}-boot-spi-fpga-fit.itb
 	fi
 
 	if [ ! -f ${NIC_FW_FILE} ]; then
@@ -113,7 +113,7 @@
 	fi
 	ITS_SED_PARAMS+="s#@NIC_FW@#${NIC_FW_FILE}#g;"
 
-	uboot_bin_sz=$(stat -Lc "%s" ${HL_UBOOT_BIN_FILE})
+	uboot_bin_sz=$(stat -Lc "%s" ${BINS_DIR}/${HL_UBOOT_BIN_FILE})
 	if [ ${uboot_bin_sz} -gt ${UBOOT_BIN_MAX_SZ} ]; then
 		echo "Error: ${HL_UBOOT_BIN_FILE} file is larger (${uboot_bin_sz}) than max (${UBOOT_BIN_MAX_SZ})" 1>&2
 		exit 1
@@ -122,14 +122,14 @@
 	${HOST_DIR}/usr/bin/mkimage -D "-W no-unit_address_vs_reg" -f ${UBOOT_ITS_FILE} -k ${KEYS_DIR} ${UBOOT_ITB_FILE}
 
 	# LKD like this to be 4B aligned in size
-	dd status=none conv=sync bs=4 if=${UBOOT_ITB_FILE} > ${HL_DEV}-boot-fit.itb
+	dd status=none conv=sync bs=4 if=${UBOOT_ITB_FILE} > ${ITBS_DIR}/${HL_DEV}-boot-fit.itb
 
 	sed ${ITS_SED_PARAMS} ${LINUX_ITS_TEMPLATE} > ${LINUX_ITS_FILE}
-	cp ${BASE_DIR}/build/linux-custom/arch/arm64/boot/Image .
+	cp ${LINUX_PRODUCTS_DIR}/Image ${BINS_DIR}
 	${HOST_DIR}/usr/bin/mkimage -D "-W no-unit_address_vs_reg" -f ${LINUX_ITS_FILE} -k ${KEYS_DIR} ${LINUX_ITB_FILE}
 
 	# KMD like this to be 4B aligned in size
-	dd status=none conv=sync bs=4 if=${LINUX_ITB_FILE} > ${ITBS_DIR}${HL_DEV}-fit.itb
+	dd status=none conv=sync bs=4 if=${LINUX_ITB_FILE} > ${ITBS_DIR}/${HL_DEV}-fit.itb
 
 	generate_dsv_fit
 popd
diff --git a/board/habanalabs/common/post-image-spi-arc_global.sh b/board/habanalabs/common/post-image-spi-arc_global.sh
index b595ab4..3c375e7 100755
--- a/board/habanalabs/common/post-image-spi-arc_global.sh
+++ b/board/habanalabs/common/post-image-spi-arc_global.sh
@@ -409,7 +409,7 @@
 
 pushd ${OUTPUT_DIR}
 
-	hl_copy_binaries ${OUTPUT_DIR}
+	hl_copy_binaries ${BINS_DIR}
 
 	for i in $(seq 1 $NUM_PROD_KEYS); do
 		CUST=cust${i}
@@ -417,10 +417,10 @@
 	done
 
 	## create otp spi image
-	dd status=none conv=sync count=12 bs=16 if=/dev/zero | tr '\000' $'\xFF' > ${DEV_PREFIX}-spi-otp-flash.bin
+	dd status=none conv=sync count=12 bs=16 if=/dev/zero | tr '\000' $'\xFF' > ${BINS_DIR}${DEV_PREFIX}-spi-otp-flash.bin
 
 	if [ ${CREATE_DUMMY_FLASH} -eq 1 ]; then
-		dd status=none conv=sync count=${CREATE_DUMMY_FLASH_SIZE_64B_COUNT} bs=64 if=/dev/zero > ${DEV_PREFIX}-dummy.bin
+		dd status=none conv=sync count=${CREATE_DUMMY_FLASH_SIZE_64B_COUNT} bs=64 if=/dev/zero > ${BINS_DIR}${DEV_PREFIX}-dummy.bin
 
 		PREBOOT_BINFILE=${DEV_PREFIX}-dummy.bin
 		PPBOOT_BINFILE=${DEV_PREFIX}-dummy.bin
@@ -435,22 +435,20 @@
 	generate_public_keys
 
 	# Create pif arc binaries
-	prep_image_pcie_pif ${PIF_ARC_ELFFILE}
-
+	prep_image_pcie_pif ${BINS_DIR}${PIF_ARC_ELFFILE}
 	# Create erom binaries
-	prep_image_erom ${DEV_PREFIX}-erom.bin.le
-
+	prep_image_erom ${BINS_DIR}${DEV_PREFIX}-erom.bin.le
 	# Create fake fuse binary
-	prep_image_fake_fuse ${DEV_PREFIX}-fake-fuse.bin.le
+	prep_image_fake_fuse ${BINS_DIR}${DEV_PREFIX}-fake-fuse.bin.le
 
 	# Create sram mbist binary
-	prep_image_sram_mbist_data sram-mbist-data.bin
+	prep_image_sram_mbist_data ${BINS_DIR}/sram-mbist-data.bin
 
 	# Pad preboot binary and then pack
-	zephyr_preboot_pack ${PREBOOT_BINFILE} ${PRIME_IMAGE_RAM_KEY} ${PRIME_IMAGE_RAM_CRC_ERROR} ${PRIME_IMAGE_RAM_SIGN_ERROR}
+	zephyr_preboot_pack ${BINS_DIR}${PREBOOT_BINFILE} ${PRIME_IMAGE_RAM_KEY} ${PRIME_IMAGE_RAM_CRC_ERROR} ${PRIME_IMAGE_RAM_SIGN_ERROR}
 
 	# Pad die1 preboot binary and then pack
-	zephyr_preboot_d1_pack ${PREBOOT_D1_BINFILE} ${PRIME_IMAGE_RAM_KEY} ${PRIME_IMAGE_RAM_CRC_ERROR} ${PRIME_IMAGE_RAM_SIGN_ERROR}
+	zephyr_preboot_d1_pack ${BINS_DIR}${PREBOOT_D1_BINFILE} ${PRIME_IMAGE_RAM_KEY} ${PRIME_IMAGE_RAM_CRC_ERROR} ${PRIME_IMAGE_RAM_SIGN_ERROR}
 
 	## To be populated into FW SRAM
 
@@ -460,58 +458,58 @@
 	PPBOOT_PCIE_PHY_BINFILE=${PPBOOT}-${PCIE_PHY}.bin
 
 	# concatenate both FW
-	hl_join_bin ${PPBOOT} ${PCIE_PHY} ${PPBOOT_PCIE_PHY_BINFILE}
+	hl_join_bin ${BINS_DIR}${PPBOOT} ${BINS_DIR}${PCIE_PHY} ${BINS_DIR}${PPBOOT_PCIE_PHY_BINFILE}
 
 	# pack and sign image
-	hl_pack_bin ${PPBOOT_PCIE_PHY_BINFILE} 64K ${PRIME_IMAGE_PPBOOT_KEY} ${PRIME_IMAGE_PPBOOT_CRC_ERROR} ${PRIME_IMAGE_PPBOOT_SIGN_ERROR}
+	hl_pack_bin ${BINS_DIR}${PPBOOT_PCIE_PHY_BINFILE} 64K ${PRIME_IMAGE_PPBOOT_KEY} ${PRIME_IMAGE_PPBOOT_CRC_ERROR} ${PRIME_IMAGE_PPBOOT_SIGN_ERROR}
 
 	# Create image pointers
-	prep_image_ptrs ${DEV_PREFIX}-images-pointers.bin.le
+	prep_image_ptrs ${BINS_DIR}${DEV_PREFIX}-images-pointers.bin.le
 
 	if [ ${HL_TARGET} == "pldm" ]; then
 		hl_efuse_sys_config_generate
 	fi
 
-	hl_sec_fw ${DEV_PREFIX}-spi-otp-flash.bin ${SECURITY_FW_VERSION}
+	hl_sec_fw ${BINS_DIR}${DEV_PREFIX}-spi-otp-flash.bin ${SECURITY_FW_VERSION}
 
-	hl_sec_key ${DEV_PREFIX}-spi-otp-flash.bin hl 0
+	hl_sec_key ${BINS_DIR}${DEV_PREFIX}-spi-otp-flash.bin hl 0
 
 	# generate 4 customer keys and hash them to spi security
 	for i in $(seq 1 $NUM_PROD_KEYS); do
 		CUST=cust${i}
-		hl_sec_key ${DEV_PREFIX}-spi-otp-flash.bin ${CUST} ${i}
+		hl_sec_key ${BINS_DIR}${DEV_PREFIX}-spi-otp-flash.bin ${CUST} ${i}
 	done
 
 	# set revoke status as a 40bit (8bit * 5) binary value
-	hl_set_revoke_status ${DEV_PREFIX}-spi-otp-flash.bin
+	hl_set_revoke_status ${BINS_DIR}${DEV_PREFIX}-spi-otp-flash.bin
 
-	cat ${DEV_PREFIX}-spi-otp-flash.bin	> ${DEV_PREFIX}-security-arc.bin.le
+	cat ${BINS_DIR}${DEV_PREFIX}-spi-otp-flash.bin	> ${BINS_DIR}${DEV_PREFIX}-security-arc.bin.le
 
-	swap_file ${DEV_PREFIX}-security-arc.bin.le ${DEV_PREFIX}-security-arc.bin.be
-	swap_file ${DEV_PREFIX}-erom.bin.le ${DEV_PREFIX}-erom.bin.be
-	swap_file ${DEV_PREFIX}-preboot.bin.le ${DEV_PREFIX}-preboot.bin.be
-	swap_file ${DEV_PREFIX}-ppboot-pcie_phy_fw.bin.le ${DEV_PREFIX}-ppboot-pcie_phy_fw.bin.be
-	swap_file  ${DEV_PREFIX}-images-pointers.bin.le  ${DEV_PREFIX}-images-pointers.bin.be
-	swap_file ${DEV_PREFIX}-fake-fuse.bin.le ${DEV_PREFIX}-fake-fuse.bin.be
-	swap_file ${DEV_PREFIX}-preboot_d1.bin.le ${DEV_PREFIX}-preboot_d1.bin.be
-	swap_file ${DEV_PREFIX}-sram-mbist-data.bin.le ${DEV_PREFIX}-sram-mbist-data.bin.be
+	swap_file ${BINS_DIR}${DEV_PREFIX}-security-arc.bin.le ${BINS_DIR}${DEV_PREFIX}-security-arc.bin.be
+	swap_file ${BINS_DIR}${DEV_PREFIX}-erom.bin.le ${BINS_DIR}${DEV_PREFIX}-erom.bin.be
+	swap_file ${BINS_DIR}${DEV_PREFIX}-preboot.bin.le ${BINS_DIR}${DEV_PREFIX}-preboot.bin.be
+	swap_file ${BINS_DIR}${DEV_PREFIX}-ppboot-pcie_phy_fw.bin.le ${BINS_DIR}${DEV_PREFIX}-ppboot-pcie_phy_fw.bin.be
+	swap_file ${BINS_DIR}${DEV_PREFIX}-images-pointers.bin.le ${BINS_DIR}${DEV_PREFIX}-images-pointers.bin.be
+	swap_file ${BINS_DIR}${DEV_PREFIX}-fake-fuse.bin.le ${BINS_DIR}${DEV_PREFIX}-fake-fuse.bin.be
+	swap_file ${BINS_DIR}${DEV_PREFIX}-preboot_d1.bin.le ${BINS_DIR}${DEV_PREFIX}-preboot_d1.bin.be
+	swap_file ${BINS_DIR}${DEV_PREFIX}-sram-mbist-data.bin.le ${BINS_DIR}${DEV_PREFIX}-sram-mbist-data.bin.be
 
-	calc_image_ptrs_sha384 ${DEV_PREFIX}-images-pointers.bin.be
+	calc_image_ptrs_sha384 ${BINS_DIR}/${DEV_PREFIX}-images-pointers.bin.be
 
-	rm -rf ${PPBOOT_PCIE_PHY_BINFILE}
+	rm -rf ${BINS_DIR}${PPBOOT_PCIE_PHY_BINFILE}
 
 	## Convert bin file into Palladium hex format ##
 	# for otp flash
-	hl_pldm_flash_hex spi-otp-flash ${DEV_PREFIX}-spi-otp-flash.bin ${HEXS_DIR}
+	hl_pldm_flash_hex spi-otp-flash ${BINS_DIR}${DEV_PREFIX}-spi-otp-flash.bin ${HEXS_DIR}
 
 	# for preboot binary
-	swap_file ${PREBOOT_BINFILE} ${PREBOOT_BINFILE}.orig.be
-	hl_pldm_hex preboot ${PREBOOT_BINFILE}.orig.be 1 ${HEXS_DIR}
+	swap_file ${BINS_DIR}${PREBOOT_BINFILE} ${BINS_DIR}${PREBOOT_BINFILE}.orig.be
+	hl_pldm_hex preboot ${BINS_DIR}${PREBOOT_BINFILE}.orig.be 1 ${HEXS_DIR}
 
-	PREBOOT_BIN_SZ=$(stat ${PREBOOT_BINFILE}.orig.be --format=%s)
+	PREBOOT_BIN_SZ=$(stat ${BINS_DIR}${PREBOOT_BINFILE}.orig.be --format=%s)
 	PREBOOT_CHUNK_SZ=131072 #128kb
 	PREBOOT_CHUNKS=$((PREBOOT_BIN_SZ / PREBOOT_CHUNK_SZ + 1))
-	split -b$PREBOOT_CHUNK_SZ -d --additional-suffix=${PREBOOT_BINFILE}.split.be ${PREBOOT_BINFILE}.orig.be
+	split -b$PREBOOT_CHUNK_SZ -d --additional-suffix=${PREBOOT_BINFILE}.split.be ${BINS_DIR}${PREBOOT_BINFILE}.orig.be
 	for (( i=0; i<$PREBOOT_CHUNKS; i++ )); do
 		if test -f "x0${i}${PREBOOT_BINFILE}.split.be"; then
 			hl_pldm_hex preboot${i} x0${i}${PREBOOT_BINFILE}.split.be 1 ${HEXS_DIR}
@@ -522,6 +520,6 @@
 
 	# for ROM binary
 	hl_pad_arcboot_rom
-	hl_pldm_hex arcboot_rom ${ARCBOOT_BINFILE} 16 ${HEXS_DIR}
+	hl_pldm_hex arcboot_rom ${BINS_DIR}${ARCBOOT_BINFILE} 16 ${HEXS_DIR}
 
 popd
diff --git a/board/habanalabs/fs1_arc/fs1-image-utility.sh b/board/habanalabs/fs1_arc/fs1-image-utility.sh
index 4dd49a0..d0e44b6 100755
--- a/board/habanalabs/fs1_arc/fs1-image-utility.sh
+++ b/board/habanalabs/fs1_arc/fs1-image-utility.sh
@@ -74,7 +74,7 @@
 	local SECTIONS=${*:4}
 
 	${OBJCOPY} -S --gap-fill 0xff --output-target=binary --remove-section=.comment --remove-section=COMMON \
-	--remove-section=.eh_frame $(echo -n $SECTIONS | sed 's/[^ ]* */ --only-section=&/g') ${BINARIES_DIR}/zephyr_${PREFIX}.elf \
+	--remove-section=.eh_frame $(echo -n $SECTIONS | sed 's/[^ ]* */ --only-section=&/g') ${ELFS_DIR}/zephyr_${PREFIX}.elf \
 	--set-section-flags=bss=alloc,load,contents --set-section-flags=noinit=alloc,load,contents \
 	${STAGE_DIR}/zephyr-${PREFIX}-${MEM}.bin
 
diff --git a/board/habanalabs/fs1_arc/fs1-post-image-fit-arc.sh b/board/habanalabs/fs1_arc/fs1-post-image-fit-arc.sh
index c50da10..01cda26 100755
--- a/board/habanalabs/fs1_arc/fs1-post-image-fit-arc.sh
+++ b/board/habanalabs/fs1_arc/fs1-post-image-fit-arc.sh
@@ -52,14 +52,14 @@
 ITS_SED_PARAMS+="s/@FIT_CORE_ID_D0_PSOC_ARC0@/${FIT_CORE_ID_D0_PSOC_ARC0}/g;"
 ITS_SED_PARAMS+="s/@FIT_CORE_ID_D0_PSOC_ARC1@/${FIT_CORE_ID_D0_PSOC_ARC1}/g;"
 
-pushd ${OUTPUT_DIR}
+pushd ${STAGE_DIR}
 
-	hl_copy_keys ${OUTPUT_DIR}
+	hl_copy_keys ${STAGE_DIR}
 
-	hl_copy_binaries ${OUTPUT_DIR}
+	hl_copy_binaries ${STAGE_DIR}
 
 	sed ${ITS_SED_PARAMS} ${ZEPHYR_ITS_TEMPLATE} > ${ZEPHYR_ITS_FILE}
-	${MKIMAGE} -D "-W no-unit_address_vs_reg" -f ${ZEPHYR_ITS_FILE} -k ${OUTPUT_DIR} ${ZEPHYR_ITB_FILE}
+	${MKIMAGE} -D "-W no-unit_address_vs_reg" -f ${ZEPHYR_ITS_FILE} -k ${STAGE_DIR} ${ZEPHYR_ITB_FILE}
 
 	dd status=none conv=sync bs=4 if=${ZEPHYR_ITB_FILE} > ${DEV_PREFIX}-boot-fit.itb
 popd
diff --git a/board/habanalabs/fs1_arc/fs1-post-image-spi-arc_global.sh b/board/habanalabs/fs1_arc/fs1-post-image-spi-arc_global.sh
index 5f11dbb..2fb52c7 100755
--- a/board/habanalabs/fs1_arc/fs1-post-image-spi-arc_global.sh
+++ b/board/habanalabs/fs1_arc/fs1-post-image-spi-arc_global.sh
@@ -6,11 +6,11 @@
 
 SHA256_SIZE=32
 SHA256_CHKSUM=sha256sum
-HL_BASE_PATH=${BR2_EXTERNAL}/board/habanalabs
-BOARD_BASE=${HL_BASE_PATH}/${HL_DEV}
-DEVICE_BASE=${HL_BASE_PATH}/${HL_DEV}
-KEY_BASE=${BR2_EXTERNAL}/board/habanalabs/${DEV_PREFIX}_arc
-OUTPUT_DIR=${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/images
+HL_BASE_PATH=${BR2_EXTERNAL}/board/habanalabs/
+BOARD_BASE=${HL_BASE_PATH}/${HL_DEV}/
+DEVICE_BASE=${HL_BASE_PATH}/${HL_DEV}/
+KEY_BASE=${BR2_EXTERNAL}/board/habanalabs/${DEV_PREFIX}_arc/
+OUTPUT_DIR=${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/images/
 OBJCOPY=objcopy
 ARCBOOT_BINFILE=${ARCBOOT_BINFILE:-boot_rom.bin}
 PPBOOT_BINFILE=${PPBOOT_BINFILE:-ppboot.bin}
@@ -86,16 +86,16 @@
 	FILE=$2
 	BYTES_PER_LINE=$3
 	# pad file to 128bit (16Byte) as pladium emulates 128bit memory, so xxd wont trunc lines
-	dd status=none conv=sync bs=16 if=${FILE} > ${FILE}.tmp
-	xxd -e -g ${BYTES_PER_LINE} ${FILE}.tmp | xxd -r | xxd -ps -c${BYTES_PER_LINE} > ${DEV_PREFIX}-${HL_TARGET}-${NAME}.hex
-	rm ${FILE}.tmp
+	dd status=none conv=sync bs=16 if=${FILE} > ${STAGE_DIR}/${FILE}.tmp
+	xxd -e -g ${BYTES_PER_LINE} ${STAGE_DIR}/${FILE}.tmp | xxd -r | xxd -ps -c${BYTES_PER_LINE} > ${HEXS_DIR}/${DEV_PREFIX}-${HL_TARGET}-${NAME}.hex
+	rm ${STAGE_DIR}/${FILE}.tmp
 }
 
 function hl_pldm_flash_hex {
 	NAME=$(echo "$1" | cut -f 1 -d '.')
 	FILE=$2
 	# save hex as a 1 byte list of values
-	xxd -e -g 4 ${FILE} | xxd -r | xxd -ps -c1			> ${DEV_PREFIX}-${HL_TARGET}-${NAME}.hex
+	xxd -e -g 4 ${FILE} | xxd -r | xxd -ps -c1			> ${HEXS_DIR}/${DEV_PREFIX}-${HL_TARGET}-${NAME}.hex
 }
 
 function hl_sec_fw {
@@ -172,9 +172,9 @@
 	echo "force {h5_core.h5_psoc_top.psoc_macro.efuse.efuse_cfg_reg.u_efuse_cfg_reg_fields.u_cfg_reg_base_data_58_r.data_in_r[31:0]} 'h${DWORD}" > ${HL_TARGET}-force.hex
 }
 
-pushd ${OUTPUT_DIR}
+pushd ${STAGE_DIR}
 
-	hl_copy_binaries ${OUTPUT_DIR}
+	hl_copy_binaries ${STAGE_DIR}
 
 	export IMG_PTR_SZ="$(get_struct_size $BOOT_POINTER_HEADER hl_fw_boot_ptrs)"
 	export IMG_DESC_SZ="$(get_struct_size $BOOT_POINTER_HEADER hl_fw_img_desc)"
@@ -185,7 +185,7 @@
 	done
 
 	## create otp spi image
-	dd status=none conv=sync count=12 bs=16 if=/dev/zero | tr '\000' $'\xFF' > ${DEV_PREFIX}-spi-otp-flash.bin
+	dd status=none conv=sync count=12 bs=16 if=/dev/zero | tr '\000' $'\xFF' > ${BINS_DIR}/${DEV_PREFIX}-spi-otp-flash.bin
 
 	# get private and public keys
 	copy_all_keys
@@ -194,13 +194,13 @@
 	generate_public_keys
 
 	# Create erom binaries
-	prep_image_erom ${DEV_PREFIX}-erom.bin.le
+	prep_image_erom ${BINS_DIR}/${DEV_PREFIX}-erom.bin.le
 
 	# Create fake fuse binary
-	prep_image_fake_fuse ${DEV_PREFIX}-fake-fuse.bin.le
+	prep_image_fake_fuse ${BINS_DIR}/${DEV_PREFIX}-fake-fuse.bin.le
 
 	# Pad preboot binary and then pack
-	zephyr_preboot_pack ${PREBOOT_BINFILE} ${PRIME_IMAGE_RAM_KEY} ${PRIME_IMAGE_RAM_CRC_ERROR} ${PRIME_IMAGE_RAM_SIGN_ERROR}
+	zephyr_preboot_pack ${BINS_DIR}/${PREBOOT_BINFILE} ${PRIME_IMAGE_RAM_KEY} ${PRIME_IMAGE_RAM_CRC_ERROR} ${PRIME_IMAGE_RAM_SIGN_ERROR}
 
 	# create combined name of ppboot and pcie
 	PPBOOT=$(echo "$PPBOOT_BINFILE" | cut -f 1 -d '.')
@@ -208,47 +208,46 @@
 	PPBOOT_PCIE_PHY_BINFILE=${PPBOOT}-${PCIE_PHY}.bin
 
 	# concatenate both FW
-	hl_join_bin ${PPBOOT} ${PCIE_PHY} ${PPBOOT_PCIE_PHY_BINFILE}
+	hl_join_bin ${BINS_DIR}/${PPBOOT} ${PCIE_PHY} ${BINS_DIR}/${PPBOOT_PCIE_PHY_BINFILE}
 
 	# pack and sign image
-	hl_pack_bin ${PPBOOT_PCIE_PHY_BINFILE} 64K ${PRIME_IMAGE_PPBOOT_KEY} ${PRIME_IMAGE_PPBOOT_CRC_ERROR} ${PRIME_IMAGE_PPBOOT_SIGN_ERROR}
+	hl_pack_bin ${BINS_DIR}/${PPBOOT_PCIE_PHY_BINFILE} 64K ${PRIME_IMAGE_PPBOOT_KEY} ${PRIME_IMAGE_PPBOOT_CRC_ERROR} ${PRIME_IMAGE_PPBOOT_SIGN_ERROR}
 
 	# Create image pointers
-	prep_image_ptrs ${DEV_PREFIX}-images-pointers.bin.le
+	prep_image_ptrs ${BINS_DIR}/${DEV_PREFIX}-images-pointers.bin.le
 
 	if [ ${HL_TARGET} == "pldm" ]; then
 		hl_efuse_sys_config_generate
 	fi
+	swap_file ${BINS_DIR}/${DEV_PREFIX}-ppboot-pcie_phy_fw.bin.le ${BINS_DIR}/${DEV_PREFIX}-ppboot-pcie_phy_fw.bin.be
+	swap_file ${BINS_DIR}/${DEV_PREFIX}-images-pointers.bin.le  ${BINS_DIR}/${DEV_PREFIX}-images-pointers.bin.be
+	swap_file ${BINS_DIR}/${DEV_PREFIX}-fake-fuse.bin.le ${BINS_DIR}/${DEV_PREFIX}-fake-fuse.bin.be
+	swap_file ${BINS_DIR}/${DEV_PREFIX}-preboot.bin.le ${BINS_DIR}/${DEV_PREFIX}-preboot.bin.be
 
-	swap_file ${DEV_PREFIX}-ppboot-pcie_phy_fw.bin.le ${DEV_PREFIX}-ppboot-pcie_phy_fw.bin.be
-	swap_file ${DEV_PREFIX}-images-pointers.bin.le  ${DEV_PREFIX}-images-pointers.bin.be
-	swap_file ${DEV_PREFIX}-fake-fuse.bin.le ${DEV_PREFIX}-fake-fuse.bin.be
-	swap_file ${DEV_PREFIX}-preboot.bin.le ${DEV_PREFIX}-preboot.bin.be
-
-	create_spi_image ${DEV_PREFIX}-spi-flash.bin.nocrc \
-			 ${DEV_PREFIX}-ppboot-pcie_phy_fw.bin.${ENDIANNESS} ${DEV_PREFIX}-preboot.bin.${ENDIANNESS} \
-			 ${DEV_PREFIX}-images-pointers.bin.${ENDIANNESS} \
-			 ${DEV_PREFIX}-fake-fuse.bin.${ENDIANNESS}
+	create_spi_image ${BINS_DIR}/${DEV_PREFIX}-spi-flash.bin.nocrc \
+			 ${BINS_DIR}/${DEV_PREFIX}-ppboot-pcie_phy_fw.bin.${ENDIANNESS} \
+			 ${BINS_DIR}/${DEV_PREFIX}-preboot.bin.${ENDIANNESS} \
+			 ${BINS_DIR}/${DEV_PREFIX}-images-pointers.bin.${ENDIANNESS} \
+			 ${BINS_DIR}/${DEV_PREFIX}-fake-fuse.bin.${ENDIANNESS}
 
 	# SPI flash binary is provided as Big Endian, for manufacturing programmer.
-	swap_file ${DEV_PREFIX}-spi-flash.bin.nocrc ${DEV_PREFIX}-spi-flash.bin.nocrc.le
+	swap_file ${BINS_DIR}/${DEV_PREFIX}-spi-flash.bin.nocrc ${BINS_DIR}/${DEV_PREFIX}-spi-flash.bin.nocrc.le
 
 	# Prepare checksum
-	CKSUM=`${SHA256_CHKSUM} ${DEV_PREFIX}-spi-flash.bin.nocrc.le | cut -d' ' -f1`
-	echo ${CKSUM} | xxd -p -r								> ${DEV_PREFIX}-spi-flash.crc
-	dd status=none conv=sync bs=64K			if=${DEV_PREFIX}-spi-flash.bin.nocrc.le	> ${DEV_PREFIX}-spi-flash.bin
-	dd status=none conv=sync bs=$SHA256_SIZE	if=${DEV_PREFIX}-spi-flash.crc		>> ${DEV_PREFIX}-spi-flash.bin
+	CKSUM=`${SHA256_CHKSUM} ${BINS_DIR}/${DEV_PREFIX}-spi-flash.bin.nocrc.le | cut -d' ' -f1`
+	echo ${CKSUM} | xxd -p -r								> ${BINS_DIR}/${DEV_PREFIX}-spi-flash.crc
+	dd status=none conv=sync bs=64K			if=${BINS_DIR}/${DEV_PREFIX}-spi-flash.bin.nocrc.le	> ${BINS_DIR}/${DEV_PREFIX}-spi-flash.bin
+	dd status=none conv=sync bs=$SHA256_SIZE	if=${BINS_DIR}/${DEV_PREFIX}-spi-flash.crc		>> ${BINS_DIR}/${DEV_PREFIX}-spi-flash.bin
 
-	rm -rf ${PPBOOT_PCIE_PHY_BINFILE}
+	rm -rf ${BINS_DIR}/${PPBOOT_PCIE_PHY_BINFILE}
 
 	## Convert bin file into Palladium hex format ##
 	# for SPI flash
-	hl_pldm_flash_hex spi-flash ${DEV_PREFIX}-spi-flash.bin.nocrc ${HEXS_DIR}
+	hl_pldm_flash_hex spi-flash ${BINS_DIR}/${DEV_PREFIX}-spi-flash.bin.nocrc ${HEXS_DIR}
 
 	# for preboot binary
-	swap_file ${PREBOOT_BINFILE} ${PREBOOT_BINFILE}.orig.be
-	hl_pldm_hex preboot ${PREBOOT_BINFILE}.orig.be 1 ${HEXS_DIR}
-
+	swap_file ${BINS_DIR}/${PREBOOT_BINFILE} ${BINS_DIR}/${PREBOOT_BINFILE}.orig.be
+	hl_pldm_hex preboot ${BINS_DIR}/${PREBOOT_BINFILE}.orig.be 1 ${HEXS_DIR}
 	# for ROM binary
 	hl_pad_arcboot_rom
 popd
diff --git a/board/habanalabs/fs1_arc/post-image-spi.sh b/board/habanalabs/fs1_arc/post-image-spi.sh
index db4e4c4..2ad8e99 100755
--- a/board/habanalabs/fs1_arc/post-image-spi.sh
+++ b/board/habanalabs/fs1_arc/post-image-spi.sh
@@ -194,14 +194,16 @@
 MAX_SPI_DMA_DATA_TRANSFER=65520
 SPI_DMA_DEST_64KB_OFFS=65536
 function hl_pack_bin {
-	FILEPATH=$1
+	FILEFULLPATH=$1
+	FILENAME=$(basename ${FILEFULLPATH})
+	FILEPATH=$(dirname ${FILEFULLPATH})
 	BSIZE=$2
 	KEY_NUM=$3
 	CRC_ERROR=$4
 	SIGN_ERROR=$5
 
-	PADFULLBINFILE=${FILEPATH}.pad
-	dd status=none conv=sync bs=16 if=${FILEPATH} of=${PADFULLBINFILE}
+	PADFULLBINFILE=${FILEFULLPATH}.pad
+	dd status=none conv=sync bs=16 if=${FILEFULLPATH} of=${PADFULLBINFILE}
 
 	SIZE_WITHOUT_CRC=$(stat --printf="%s" ${PADFULLBINFILE})
 	NUM_SLICES=$((SIZE_WITHOUT_CRC/MAX_SPI_DMA_DATA_TRANSFER))
@@ -252,13 +254,13 @@
 		fi
 		flash_offset=$((SPI_DMA_DEST_64KB_OFFS/4))
 		flash_offset=$((flash_offset*slice_cnt))
-		dd status=none conv=sync if=${BIN_FILE} bs=4 seek=$flash_offset of=${DEV_PREFIX}-${FILEPATH}.le.int
+		dd status=none conv=sync if=${BIN_FILE} bs=4 seek=$flash_offset of=${STAGE_DIR}/${DEV_PREFIX}-${FILENAME}.le.int
 		leftover=$((leftover-size_bytes))
 		skip_offset=$((skip_offset+size_words))
 	done
 
-	hl_reverse_words ${DEV_PREFIX}-${FILEPATH}.le.int ${DEV_PREFIX}-${FILEPATH}.le 4
-	rm -f ${DEV_PREFIX}-${FILEPATH}.le.int
+	hl_reverse_words ${STAGE_DIR}/${DEV_PREFIX}-${FILENAME}.le.int ${FILEPATH}/${DEV_PREFIX}-${FILENAME}.le 4
+	rm -f ${STAGE_DIR}/${DEV_PREFIX}-${FILENAME}.le.int
 }
 
 function hl_sign_bin {
diff --git a/board/habanalabs/gaudi2_arc/post-image-spi.sh b/board/habanalabs/gaudi2_arc/post-image-spi.sh
index 63cc9a8..ba0c56c 100755
--- a/board/habanalabs/gaudi2_arc/post-image-spi.sh
+++ b/board/habanalabs/gaudi2_arc/post-image-spi.sh
@@ -159,7 +159,7 @@
 	VPKFILE=${TEMPNAME}.ver.payload.key
 
 	#No need to pack Public Key, since Key Props are enough
-	cat ${TEMPNAME}.ver ${TEMPNAME} ${STAGE_DIR}/${KEY_NAME}-key-props.bin > ${STAGE_DIR}/${VPKFILE}
+	cat ${TEMPFULLPATH}.ver ${TEMPFULLPATH} ${BINS_DIR}/${KEY_NAME}-key-props.bin > ${STAGE_DIR}/${VPKFILE}
 
 	$OPENSSL dgst -sha384 -binary ${STAGE_DIR}/${VPKFILE} > ${STAGE_DIR}/psk_${VPKFILE}.sha384
 
@@ -206,7 +206,7 @@
 		fi
 		flash_offset=$((SPI_DMA_DEST_64KB_OFFS/4))
 		flash_offset=$((flash_offset*slice_cnt))
-		dd status=none conv=sync if=${BIN_FILE} bs=4 seek=$flash_offset of=${BINS_DIR}/${DEV_PREFIX}-${FILEPATH}.le
+		dd status=none conv=sync if=${BIN_FILE} bs=4 seek=$flash_offset of=${BINS_DIR}/${DEV_PREFIX}-${FILENAME}.le
 		leftover=$((leftover-size_bytes))
 		skip_offset=$((skip_offset+size_words))
 	done
@@ -224,7 +224,7 @@
 function get_image_size {
 	local name=${1^^}
 
-	stat -Lc "%s" $(eval "echo ${DEV_PREFIX}-\${${name}_BINFILE}")
+	stat -Lc "%s" $(eval "echo ${BINS_DIR}/${DEV_PREFIX}-\${${name}_BINFILE}")
 }
 
 function hl_join_bin {
@@ -415,8 +415,8 @@
 	prep_erom_with_crc ${ITBS_DIR}/${EROM_FIT_BINFILE}
 
 	# Create erom image pointers binary
-	prep_image_ptrs_erom ${EROM_IMAGES_POINTERS_BINFILE}
-	prep_erom_with_crc ${EROM_IMAGES_POINTERS_BINFILE}
+	prep_image_ptrs_erom ${BINS_DIR}/${EROM_IMAGES_POINTERS_BINFILE}
+	prep_erom_with_crc ${BINS_DIR}/${EROM_IMAGES_POINTERS_BINFILE}
 
 	if [ ${HL_TARGET} == "pldm" ]; then
 		#Create erom hex files for paladium
@@ -433,7 +433,6 @@
 	cp -f ${BINS_DIR}/${DEV_PREFIX}-${EROM_RECOVERY_BINFILE} ${BINS_DIR}/${DEV_PREFIX}-${EROM_RECOVERY_BINFILE}.le
 	cp -f ${BINS_DIR}/${DEV_PREFIX}-${EROM_FIT_BINFILE} ${BINS_DIR}/${DEV_PREFIX}-${EROM_FIT_BINFILE}.le
 	cp -f ${BINS_DIR}/${DEV_PREFIX}-${EROM_IMAGES_POINTERS_BINFILE} ${BINS_DIR}/${DEV_PREFIX}-${EROM_IMAGES_POINTERS_BINFILE}.le
-
 	# Reserved space (80820 bytes) filled by 0xFF
 	HI_OFFSET=$(printf %d ${EROM_TPM_OFFSET})
 	dd status=none conv=sync bs=${HI_OFFSET} count=1 if=/dev/zero |  tr "\000" "\377" > ${OUTF}
diff --git a/board/habanalabs/gaudi3_arc/post-image-spi.sh b/board/habanalabs/gaudi3_arc/post-image-spi.sh
index f7773d8..a344615 100755
--- a/board/habanalabs/gaudi3_arc/post-image-spi.sh
+++ b/board/habanalabs/gaudi3_arc/post-image-spi.sh
@@ -325,7 +325,7 @@
 	cat ${BINFILE}>${BINFILE}.tmp.pad
 	dd status=none conv=sync bs=16 if=${BINFILE}.tmp.pad of=${BINFILE}.tmp.algn
 	hl_calc_crc32 ${BINFILE}.tmp.algn
-	cat ${BINFILE}.tmp.algn ${BINFILE}.tmp.algn.crc32 > ${BINS_DIR}${DEV_PREFIX}-${OUTFILENAME}
+	cat ${BINFILE}.tmp.algn ${BINFILE}.tmp.algn.crc32 > ${BINS_DIR}/${DEV_PREFIX}-${OUTFILENAME}
 	rm -rf ${BINFILE}.tmp*
 }
 
@@ -354,7 +354,7 @@
 	local BINFILE=$(eval "echo \${${NAME}_BINFILE}")
 
 	#align the source binaries to 16 bytes
-	dd status=none conv=sync bs=16 if=${BINS_DIR}${BINFILE} of=${STAGE_DIR}/${BINFILE}.tmp.pad
+	dd status=none conv=sync bs=16 if=${BINS_DIR}/${BINFILE} of=${STAGE_DIR}/${BINFILE}.tmp.pad
 	# Prepare checksum
 	local CKSUM=`sha384sum ${STAGE_DIR}/${BINFILE}.tmp.pad`
 	hl_chksum_add ${CKSUM}
@@ -624,12 +624,12 @@
 		hl_pldm_flash_hex erom-binary-img-ptr ${BINS_DIR}/${DEV_PREFIX}-erom-images-pointers.bin.be ${HEXS_DIR}
 	fi
 
-	dd status=none conv=sync if=${DEV_PREFIX}-erom-images-pointers.bin.${ENDIANNESS} \
+	dd status=none conv=sync if=${BINS_DIR}/${DEV_PREFIX}-erom-images-pointers.bin.${ENDIANNESS} \
 	bs=$(printf %d ${EROM_PCIE_OFFSET})	> ${STAGE_DIR}/${OUTF}.tmp
 	dd status=none conv=sync seek=$(printf %d ${EROM_PCIE_OFFSET}) \
-	if=${STAGE_DIR}/${DEV_PREFIX}-${EROM_PCIE_BINFILE} bs=1 of=${STAGE_DIR}/${OUTF}.tmp
+	if=${BINS_DIR}/${DEV_PREFIX}-${EROM_PCIE_BINFILE} bs=1 of=${STAGE_DIR}/${OUTF}.tmp
 	dd status=none conv=sync seek=$(printf %d ${EROM_RECOVERY_OFFSET}) \
-	if=${STAGE_DIR}/${DEV_PREFIX}-${EROM_RECOVERY_BINFILE} bs=1 of=${STAGE_DIR}/${OUTF}.tmp
+	if=${BINS_DIR}/${DEV_PREFIX}-${EROM_RECOVERY_BINFILE} bs=1 of=${STAGE_DIR}/${OUTF}.tmp
 
 	hl_reverse_words ${STAGE_DIR}/${OUTF}.tmp ${OUTFULLPATH} 4
 	rm -f ${STAGE_DIR}/${OUTF}.tmp
@@ -677,7 +677,7 @@
 }
 
 function hl_pad_arcboot_rom {
-	mv ${BINS_DIR}i/${ARCBOOT_BINFILE} ${BINS_DIR}/${ARCBOOT_BINFILE_ORIG}
+	mv ${BINS_DIR}/${ARCBOOT_BINFILE} ${BINS_DIR}/${ARCBOOT_BINFILE_ORIG}
 
 	# pad arcboot rom binary to size of ROM memory
 	dd status=none conv=sync bs=128K if=${BINS_DIR}/${ARCBOOT_BINFILE_ORIG} > ${BINS_DIR}/${ARCBOOT_BINFILE}
diff --git a/package/zephyr/zephyr.mk b/package/zephyr/zephyr.mk
index 607b507..6e4b7d6 100644
--- a/package/zephyr/zephyr.mk
+++ b/package/zephyr/zephyr.mk
@@ -104,6 +104,7 @@
 ZEPHYR_CONFIGURE_CMDS += && $(ZEPHYR_CONF_ENV) $(HOST_DIR)/bin/cmake $(ZEPHYR_CONF_OPTS) $(call CREATE_APP_VERSION,arcpid,$(ZEPHYR_PID_DIR)) $(ZEPHYR_CONF_OPTS_PID) -B $(ZEPHYR_PID_DIR)
 ZEPHYR_BUILD_CMDS += && $(TARGET_MAKE_ENV) $(MAKE) -C $(@D)/$(ZEPHYR_PID_DIR)
 ZEPHYR_INSTALL_TARGET_CMDS += && $(INSTALL) -D -m 0644 $(@D)/$(ZEPHYR_PID_DIR)/zephyr/zephyr.elf $(BINARIES_DIR)/zephyr_pid.elf
+ZEPHYR_INSTALL_TARGET_CMDS += && $(INSTALL) -D -m 0644 $(@D)/$(ZEPHYR_PID_DIR)/zephyr/zephyr.stat $(STATS_DIR)/zephyr_pid.stat
 ZEPHYR_PID_MODULES = mbedtls boot_info eeprom
 $(foreach module,$(ZEPHYR_PID_MODULES),$(call DEFINE_MODULE_ENVVARS,$(module),$(ZEPHYR_PID_DIR),$(ZEPHYR_PID_NAME)))
 endif
@@ -126,6 +127,7 @@
 ZEPHYR_CONFIGURE_CMDS += && $(ZEPHYR_CONF_ENV) $(HOST_DIR)/bin/cmake $(ZEPHYR_CONF_OPTS) $(call CREATE_APP_VERSION,arcmgmt,$(ZEPHYR_MGMT_DIR)) $(ZEPHYR_CONF_OPTS_MGMT) -B $(ZEPHYR_MGMT_DIR)
 ZEPHYR_BUILD_CMDS += && $(TARGET_MAKE_ENV) $(MAKE) -C $(@D)/$(ZEPHYR_MGMT_DIR)
 ZEPHYR_INSTALL_TARGET_CMDS += && $(INSTALL) -D -m 0644 $(@D)/$(ZEPHYR_MGMT_DIR)/zephyr/zephyr.elf $(BINARIES_DIR)/zephyr_mgmt.elf
+ZEPHYR_INSTALL_TARGET_CMDS += && $(INSTALL) -D -m 0644 $(@D)/$(ZEPHYR_MGMT_DIR)/zephyr/zephyr.stat $(STATS_DIR)/zephyr_mgmt.stat
 ZEPHYR_MGMT_MODULES = eeprom mbedtls boot_info i2c-indirect soc_init hl_shell mctp pldm
 $(foreach module,$(ZEPHYR_MGMT_MODULES),$(call DEFINE_MODULE_ENVVARS,$(module),$(ZEPHYR_MGMT_DIR),$(ZEPHYR_MGMT_NAME)))
 endif
@@ -147,6 +149,7 @@
 ZEPHYR_CONFIGURE_CMDS += && $(ZEPHYR_CONF_ENV) $(HOST_DIR)/bin/cmake $(ZEPHYR_CONF_OPTS) $(ZEPHYR_CONF_OPTS_EVENT) -B $(ZEPHYR_EVENT_DIR)
 ZEPHYR_BUILD_CMDS += && $(TARGET_MAKE_ENV) $(MAKE) -C $(@D)/$(ZEPHYR_EVENT_DIR)
 ZEPHYR_INSTALL_TARGET_CMDS += && $(INSTALL) -D -m 0644 $(@D)/$(ZEPHYR_EVENT_DIR)/zephyr/zephyr.elf $(BINARIES_DIR)/zephyr_event.elf
+ZEPHYR_INSTALL_TARGET_CMDS += && $(INSTALL) -D -m 0644 $(@D)/$(ZEPHYR_EVENT_DIR)/zephyr/zephyr.stat $(STATS_DIR)/zephyr_event.stat
 ZEPHYR_EVENT_MODULES = boot_info mbedtls
 $(foreach module,$(ZEPHYR_EVENT_MODULES),$(call DEFINE_MODULE_ENVVARS,$(module),$(ZEPHYR_EVENT_DIR),$(ZEPHYR_EVENT_NAME)))
 endif
@@ -175,7 +178,8 @@
 ZEPHYR_BUILD_CMDS += && $(TARGET_MAKE_ENV) $(MAKE) -C $(@D)/$(ZEPHYR_PREBOOT_DIR)
 ZEPHYR_INSTALL_TARGET_CMDS += && $(INSTALL) -D -m 0644 $(@D)/$(ZEPHYR_PREBOOT_DIR)/zephyr/zephyr.bin $(BINARIES_DIR)/preboot.bin
 ZEPHYR_INSTALL_TARGET_CMDS += && $(INSTALL) -D -m 0644 $(@D)/$(ZEPHYR_PREBOOT_DIR)/zephyr/zephyr.elf $(BINARIES_DIR)/zephyr_preboot.elf
-ZEPHYR_INSTALL_TARGET_CMDS += && $(INSTALL) -D -m 0644 $(@D)/$(ZEPHYR_PREBOOT_DIR)/buildversion-habanalabs $(BINARIES_DIR)/zephyr_preboot_buildversion
+ZEPHYR_INSTALL_TARGET_CMDS += && $(INSTALL) -D -m 0644 $(@D)/$(ZEPHYR_PREBOOT_DIR)/zephyr/zephyr.stat $(STATS_DIR)/zephyr_preboot.stat
+
 ZEPHYR_PREBOOT_MODULES = fdt boot_info mbedtls eeprom soc_init hl_shell mctp pldm
 $(foreach module,$(ZEPHYR_PREBOOT_MODULES),$(call DEFINE_MODULE_ENVVARS,$(module),$(ZEPHYR_PREBOOT_DIR),$(ZEPHYR_PREBOOT_NAME)))
 endif #BR2_PACKAGE_ZEPHYR_ARC_PREBOOT
@@ -195,6 +199,7 @@
 ZEPHYR_BUILD_CMDS += && $(TARGET_MAKE_ENV) $(MAKE) -C $(@D)/$(ZEPHYR_PREBOOT_D1_DIR)
 ZEPHYR_INSTALL_TARGET_CMDS += && $(INSTALL) -D -m 0644 $(@D)/$(ZEPHYR_PREBOOT_D1_DIR)/zephyr/zephyr.bin $(BINARIES_DIR)/preboot_d1.bin
 ZEPHYR_INSTALL_TARGET_CMDS += && $(INSTALL) -D -m 0644 $(@D)/$(ZEPHYR_PREBOOT_D1_DIR)/zephyr/zephyr.elf $(BINARIES_DIR)/zephyr_preboot_d1.elf
+ZEPHYR_INSTALL_TARGET_CMDS += && $(INSTALL) -D -m 0644 $(@D)/$(ZEPHYR_PREBOOT_D1_DIR)/zephyr/zephyr.stat $(STATS_DIR)/zephyr_preboot_d1.stat
 endif #BR2_PACKAGE_ZEPHYR_ARC_PREBOOT_D1
 
 ifdef BR2_PACKAGE_ZEPHYR_ARC_DSM
@@ -209,6 +214,7 @@
 ZEPHYR_CONFIGURE_CMDS += && $(ZEPHYR_CONF_ENV) $(HOST_DIR)/bin/cmake $(ZEPHYR_CONF_OPTS) $(ZEPHYR_CONF_OPTS_DSM) -B $(ZEPHYR_DSM_DIR)
 ZEPHYR_BUILD_CMDS += && $(TARGET_MAKE_ENV) $(MAKE) -C $(@D)/$(ZEPHYR_DSM_DIR)
 ZEPHYR_INSTALL_TARGET_CMDS += && $(INSTALL) -D -m 0644 $(@D)/$(ZEPHYR_DSM_DIR)/zephyr/zephyr.elf $(BINARIES_DIR)/zephyr_dsm.elf
+ZEPHYR_INSTALL_TARGET_CMDS += && $(INSTALL) -D -m 0644 $(@D)/$(ZEPHYR_DSM_DIR)/zephyr/zephyr.stat $(STATS_DIR)/zephyr_dsm.stat
 endif
 
 
@@ -233,7 +239,7 @@
 ZEPHYR_BUILD_CMDS += && $(TARGET_MAKE_ENV) $(MAKE) -C $(@D)/$(ZEPHYR_FW_LOADER_DIR)
 ZEPHYR_INSTALL_TARGET_CMDS += && $(INSTALL) -D -m 0644 $(@D)/$(ZEPHYR_FW_LOADER_DIR)/zephyr/zephyr.bin $(BINARIES_DIR)/zephyr_agent_fw_loader.bin
 # The stat file is used for extracting the binary's load address and start address
-ZEPHYR_INSTALL_TARGET_CMDS += && $(INSTALL) -D -m 0644 $(@D)/$(ZEPHYR_FW_LOADER_DIR)/zephyr/zephyr.stat $(BINARIES_DIR)/zephyr_agent_fw_loader.stat
+ZEPHYR_INSTALL_TARGET_CMDS += && $(INSTALL) -D -m 0644 $(@D)/$(ZEPHYR_FW_LOADER_DIR)/zephyr/zephyr.stat $(STATS_DIR)/zephyr_agent_fw_loader.stat
 ZEPHYR_FW_LOADER_MODULES = boot_info eeprom fdt i2c-indirect lz4 mbedtls hl_shell soc_init
 $(foreach module,$(ZEPHYR_FW_LOADER_MODULES),$(call DEFINE_MODULE_ENVVARS,$(module),$(ZEPHYR_FW_LOADER_DIR),$(ZEPHYR_FW_LOADER_NAME)))
 endif
@@ -262,7 +268,7 @@
 $(eval ZEPHYR_CONFIGURE_CMDS += && $$(ZEPHYR_CONF_ENV) $(HOST_DIR)/bin/cmake $$(ZEPHYR_CONF_OPTS) $(ZEPHYR_CONF_OPTS_AGENT) -B $(ZEPHYR_AGENT_DIR)) \
 $(eval ZEPHYR_BUILD_CMDS += && $(TARGET_MAKE_ENV) $(MAKE) -C $$(@D)/$(ZEPHYR_AGENT_DIR)) \
 $(eval ZEPHYR_INSTALL_TARGET_CMDS += && $(INSTALL) -D -m 0644 $$(@D)/$(ZEPHYR_AGENT_DIR)/zephyr/zephyr.bin $(BINARIES_DIR)/zephyr_agent_$1.bin) \
-$(eval ZEPHYR_INSTALL_TARGET_CMDS += && $(INSTALL) -D -m 0644 $$(@D)/$(ZEPHYR_AGENT_DIR)/zephyr/zephyr.stat $(BINARIES_DIR)/zephyr_agent_$1.stat) \
+$(eval ZEPHYR_INSTALL_TARGET_CMDS += && $(INSTALL) -D -m 0644 $$(@D)/$(ZEPHYR_AGENT_DIR)/zephyr/zephyr.stat $(STATS_DIR)/zephyr_agent_$1.stat) \
 $(foreach module,$(ZEPHYR_$(AGENT_UPPER_CASE)_MODULES),$(call DEFINE_MODULE_ENVVARS,$(module),$(ZEPHYR_$(AGENT_UPPER_CASE)_DIR), $(ZEPHYR_$(AGENT_UPPER_CASE)_NAME)))
 endef
 
