From 8c2c179966bd65679788d626524517425b4e746c Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Fri, 26 Apr 2024 00:19:40 +0300
Subject: [PATCH] sq files_path_1

Change-Id: Ida41b8d2d434188e8e76fba6f227768979a3b62f
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/board/habanalabs/common/post-image-exports.sh b/board/habanalabs/common/post-image-exports.sh
index 1c1e0d5..f5f92cd 100755
--- a/board/habanalabs/common/post-image-exports.sh
+++ b/board/habanalabs/common/post-image-exports.sh
@@ -17,3 +17,6 @@
 ITS_SED_PARAMS+="s#@HL_UBOOT_BIN_FILE@#${HL_UBOOT_BIN_FILE}#g;"
 ITS_SED_PARAMS+="s#@FILES_PATH_1@#${BINS_DIR}#g;"
 ITS_SED_PARAMS+="s#@FILES_PATH_2@#${BINS_PROD_DIR}#g;"
+
+ITS_SED_PARAMS+="s#@FILES_PATH_1@#${BINS_DIR}#g;"
+ITS_SED_PARAMS+="s#@FILES_PATH_2@#${BINS_PROD_DIR}#g;"
