From fc4c7e0786c16ff4e4f2aa65c35d2981aee25e94 Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Tue, 30 Apr 2024 14:45:38 +0300
Subject: [PATCH] [SW-0] gaudi2: increase pcie lbw drain timeout

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: Ie36e0e5eda2e1255615b1792fde2eef4cf6a368d
---

diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index f516f6d..c84529c 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -1108,7 +1108,7 @@
 }
 
 #define HBW_DRAIN_TIMEOUT	0x1000
-#define LBW_DRAIN_TIMEOUT	0x5000
+#define LBW_DRAIN_TIMEOUT	0xF000
 
 void ppboot_pcie_drain_init(void)
 {
