From 242846807783810de2c1e145f1707e4754a246d3 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Wed, 01 May 2024 09:31:39 +0300
Subject: [PATCH] [SW-15774] scripts: adjust path

Change-Id: I8e1cef7d2547ebc25c175dff9814f9595e6ddddc
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/board/habanalabs/gaudi2_arc/post-image-spi.sh b/board/habanalabs/gaudi2_arc/post-image-spi.sh
index 547097a..c580f09 100755
--- a/board/habanalabs/gaudi2_arc/post-image-spi.sh
+++ b/board/habanalabs/gaudi2_arc/post-image-spi.sh
@@ -53,8 +53,8 @@
 	# hl_copy_binaries function is called from post-image-spi-arc_global.sh
 	# common script. All the binaries required for creation of spi
 	# are compiled as part of gaudi2_arc HL_TARGET
-	touch ${DEV_PREFIX}-preboot_d1.bin.le
-	touch ${DEV_PREFIX}-sram-mbist-data.bin.le
+	touch ${BINS_DIR}/${DEV_PREFIX}-preboot_d1.bin.le
+	touch ${BINS_DIR}/${DEV_PREFIX}-sram-mbist-data.bin.le
 	return
 }
 
