From f63293afde02f8092f871f2989440c17f89a46c6 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Fri, 03 May 2024 19:24:27 +0300
Subject: [PATCH] sq: mgmt

Change-Id: I62f333821e4520c550beff469695624daf228433
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/board/habanalabs/fs1_arc/fs1-local.mk b/board/habanalabs/fs1_arc/fs1-local.mk
index f77aa4b..8e381b3 100644
--- a/board/habanalabs/fs1_arc/fs1-local.mk
+++ b/board/habanalabs/fs1_arc/fs1-local.mk
@@ -2,3 +2,4 @@
 BAREMETAL_OVERRIDE_SRCDIR = $(TOPDIR)/../preboot
 ZEPHYR_OVERRIDE_SRCDIR = $(TOPDIR)/../zephyr
 PID_OVERRIDE_SRCDIR = $(TOPDIR)/../zephyr-apps/arcpid
+MGMT_OVERRIDE_SRCDIR = $(TOPDIR)/../zephyr-apps/arcmgmt
diff --git a/package/mgmt/mgmt.mk b/package/mgmt/mgmt.mk
index 1c81055..8511c08 100644
--- a/package/mgmt/mgmt.mk
+++ b/package/mgmt/mgmt.mk
@@ -2,7 +2,7 @@
 MGMT_APP_NAME = mgmt
 MGMT_DEPENDENCIES = host-cmake host-lzop host-lz4
 MGMT_APP_MODULES = eeprom mbedtls boot_info i2c-indirect soc_init hl_shell mctp pldm
-MGMT_SRC_DIR=$(EMBEDDED_DIR)/zephyr-apps/arcmgmt
+MGMT_SRCDIR=$(EMBEDDED_DIR)/zephyr-apps/arcmgmt
 MGMT_ZEPHYR_SRCDIR=$(EMBEDDED_DIR)/zephyr
 
 MGMT_APP_PROD_NAME = zephyr_arc_$(MGMT_APP_NAME)
@@ -51,7 +51,7 @@
 		$(MGMT_CONF_CFLAGS)"
 
 MGMT_CONF_OPTS += -DDTS_EXTRA_CPPFLAGS="-I$(SPECS_EMBEDDED_DIR)"
-MGMT_CONF_OPTS += -DBOARD=hl_$(MGMT_DEVICE_NAME)_$(MGMT_APP_NAME)_$(HL_TARGET)
+MGMT_CONF_OPTS += -DBOARD=$(MGMT_DEVICE_NAME)_$(MGMT_APP_NAME)_$(HL_TARGET)
 MGMT_CONF_OPTS += -DDEVICE=$(MGMT_DEVICE_NAME)
 MGMT_CONF_OPTS += -DAPP_NAME=$(MGMT_APP_PROD_NAME)
 MGMT_CONF_OPTS += -DAPP_VERSION=$(MGMT_APP_VERSION)
@@ -59,7 +59,7 @@
 
 ifdef BR2_arcle
 ifneq ($(STACK_REDUCTION_SIZE),0)
-MGMT_CONF_OPTS += -DOVERLAY_CONFIG=$(BR2_EXTERNAL_HABANALABS_PATH)/configs/zephyr_arc_mgmt_stack_defconfig
+MGMT_CONF_OPTS += -DOVERLAY_CONFIG=$(BR2_EXTERNAL)/configs/zephyr_arc_mgmt_stack_defconfig
 endif
 endif
 
