From 3a53cefbff727d7f9e94c64b735a880e2eabd732 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Mon, 06 May 2024 16:35:43 +0300
Subject: [PATCH] sq: agent-pcie

Change-Id: Ib3f1dcadcabd419086444773f6d16224c0e91757
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/package/agent-pcie/agent-pcie.mk b/package/agent-pcie/agent-pcie.mk
index 46c54e4..2437f0a 100644
--- a/package/agent-pcie/agent-pcie.mk
+++ b/package/agent-pcie/agent-pcie.mk
@@ -22,7 +22,12 @@
 		-I$(AGENT_SRC_DIR)/common/include/$(HL_DEV)
 
 AGENT_PCIE_CONF_OPTS  = $(ZEPHYR_CONF_OPTS)
+ifdef BR2_arcle
+AGENT_PCIE_CONF_OPTS += -DBOARD=$(ZEPHYR_BOARD_PREFIX)_agent_$(HL_TARGET)
+endif
+ifdef BR2_aarch64
 AGENT_PCIE_CONF_OPTS += -DBOARD=$(ZEPHYR_BOARD_PREFIX)_$(HL_TARGET)
+endif
 
 AGENT_PCIE_BUILD_DIR = $(@D)
 AGENT_PCIE_ZEPHYR_BUILD_DIR ?= $(@D)/zephyr
