From 409dcbc0e7c31762d0eaffdc9beee57d41ba800a Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Mon, 06 May 2024 20:34:18 +0300
Subject: [PATCH] [SW-15774] board: scripts: set the elfs directory

Preparation to separate directories

Change-Id: Ib0f49bf6a365000c6c64ccf92a52b296b59562c9
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/board/habanalabs/common/dirs.sh b/board/habanalabs/common/dirs.sh
index 12cc9a0..76c6444 100644
--- a/board/habanalabs/common/dirs.sh
+++ b/board/habanalabs/common/dirs.sh
@@ -9,7 +9,7 @@
 export ITBS_DIR=${IMGS_DIR}
 export BINS_DIR=${IMGS_DIR}
 export STATS_DIR=${IMGS_DIR}
-export ELFS_DIR=${IMGS_DIR}
+export ELFS_DIR=${IMGS_DIR}/elfs/
 export BINS_PROD_DIR=${IMGS_PROD_DIR}
 
 # TODO: move the below to FS1 specific make file
diff --git a/scripts/make/common/hl_dirs.mk b/scripts/make/common/hl_dirs.mk
index b74a095..56ecdb1 100644
--- a/scripts/make/common/hl_dirs.mk
+++ b/scripts/make/common/hl_dirs.mk
@@ -5,8 +5,9 @@
 export HEXS_DIR=$(IMGS_DIR)
 export BINS_DIR=$(IMGS_DIR)
 export ITBS_DIR=$(IMGS_DIR)
-export ELFS_DIR=$(IMGS_DIR)
+export ELFS_DIR=$(IMGS_DIR)/elfs
 export STATS_DIR=$(IMGS_DIR)
 
 export DIRS_LIST = \
-	$(STAGE_DIR)
+	$(STAGE_DIR) \
+	$(ELFS_DIR)
