From e6035be7a69cf08d71bb5549ca2bfffac2bfef39 Mon Sep 17 00:00:00 2001
From: Tamal <tasaha@habana.ai>
Date: Thu, 09 May 2024 14:46:09 +0300
Subject: [PATCH] [DNR-NEW]: fs1: SPDM stack integration

Change-Id: I7977f5ac938f2e206fa3c6f3c9d2abaa83741438
Signed-off-by: Tamal <tasaha@habana.ai>
---

diff --git a/board/habanalabs/fs1_arc/fs1-post-image-spi-arc_global.sh b/board/habanalabs/fs1_arc/fs1-post-image-spi-arc_global.sh
index b45df1a..330773f 100755
--- a/board/habanalabs/fs1_arc/fs1-post-image-spi-arc_global.sh
+++ b/board/habanalabs/fs1_arc/fs1-post-image-spi-arc_global.sh
@@ -16,6 +16,7 @@
 PPBOOT_BINFILE=${PPBOOT_BINFILE:-ppboot.bin}
 PREBOOT_BINFILE=${PREBOOT_BINFILE:-preboot.bin}
 PCIE_PHY_BINFILE=${PCIE_PHY_BINFILE:-pcie_phy_fw.bin}
+ZEPHYR_DSM_BINFILE=${ZEPHYR_DSM_BINFILE:-zephyr_dsm.bin}
 
 
 #SECURITY_BOOT_MODE
@@ -232,6 +233,10 @@
 	swap_file ${PREBOOT_BINFILE} ${PREBOOT_BINFILE}.orig.be
 	hl_pldm_hex preboot ${PREBOOT_BINFILE}.orig.be 1
 
+	# for zephyr_dsm binary
+        swap_file ${ZEPHYR_DSM_BINFILE} ${ZEPHYR_DSM_BINFILE}.orig.be
+        hl_pldm_hex zephyr_dsm ${ZEPHYR_DSM_BINFILE}.orig.be 1
+
 	# for ROM binary
 	hl_pad_arcboot_rom
 popd
diff --git a/package/zephyr/zephyr.mk b/package/zephyr/zephyr.mk
index 5ec2378..56d60ce 100644
--- a/package/zephyr/zephyr.mk
+++ b/package/zephyr/zephyr.mk
@@ -209,6 +209,9 @@
 ZEPHYR_BUILD_CMDS += && $(TARGET_MAKE_ENV) $(MAKE) -C $(@D)/$(ZEPHYR_DSM_DIR)
 ZEPHYR_INSTALL_TARGET_CMDS += && $(INSTALL) -D -m 0644 $(@D)/$(ZEPHYR_DSM_DIR)/zephyr/zephyr.elf $(ELFS_DIR)/zephyr_dsm.elf
 ZEPHYR_INSTALL_TARGET_CMDS += && $(INSTALL) -D -m 0644 $(@D)/$(ZEPHYR_DSM_DIR)/zephyr/zephyr.stat $(STATS_DIR)/zephyr_dsm.stat
+ifeq ($(HL_TARGET),pldm)
+ZEPHYR_INSTALL_TARGET_CMDS += && $(INSTALL) -D -m 0644 $(@D)/$(ZEPHYR_DSM_DIR)/zephyr/zephyr.bin $(BINARIES_DIR)/zephyr_dsm.bin
+endif
 endif
 
 
