From 8973be61cc00b8d987cfdd7e364327d13de039b4 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Thu, 16 May 2024 10:37:46 +0300
Subject: [PATCH] sq event

Change-Id: Ibfd447ce67cd30e52a8229acafe36c580ee6586d
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/package/event/event.mk b/package/event/event.mk
index 12f01ac..893a757 100644
--- a/package/event/event.mk
+++ b/package/event/event.mk
@@ -73,7 +73,6 @@
 endif
 endif
 
-EVENT_CONF_OPTS += -DCMAKE_C_FLAGS="-I$(SPECS_EMB_PROJ_DIR) -I$(SPECS_EMB_PROJ_DIR)/$(DEV_PREFIX) -I$(SPECS_EMB_PROJ_DIR)/common -I$(SPECS_ROOT) -I$(SPECS_EXT_ROOT) -I$(SPECS_ROOT)/$(DEV_PREFIX)/asic_reg"
 EVENT_WEST_SRC_DIR :=
 EVENT_WEST_DEST_DIR :=
 
