From b402625e8297607b1133ff2d48068ad3d0e92c89 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Thu, 16 May 2024 10:36:38 +0300
Subject: [PATCH] mgmt cmake fix

Change-Id: I6a76b6f01353f57fa088ad52357a175dce166695
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/package/mgmt/mgmt.mk b/package/mgmt/mgmt.mk
index 528b75d..39cff23 100644
--- a/package/mgmt/mgmt.mk
+++ b/package/mgmt/mgmt.mk
@@ -66,7 +66,6 @@
 MGMT_CONF_OPTS += -DDEVICE=$(MGMT_DEVICE_NAME)
 MGMT_CONF_OPTS += -DAPP_NAME=$(MGMT_APP_PROD_NAME)
 MGMT_CONF_OPTS += -DAPP_VERSION=$(MGMT_APP_VERSION)
-MGMT_CONF_OPTS += -DCMAKE_C_FLAGS="-I$(SPECS_EMB_PROJ_DIR) -I$(SPECS_EMB_PROJ_DIR)/$(DEV_PREFIX) -I$(SPECS_EMB_PROJ_DIR)/common -I$(SPECS_ROOT) -I$(SPECS_EXT_ROOT) -I$(SPECS_ROOT)/$(DEV_PREFIX)/asic_reg"
 
 ifdef BR2_arcle
 ifneq ($(STACK_REDUCTION_SIZE),0)
