From 169edd1e20b12b8d897b2954834f5a7d1a56c3c8 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Wed, 22 May 2024 19:39:33 +0300
Subject: [PATCH] [SW-15774] board: change dirs

Fix missed out stage directory replacment.

Change-Id: I76054924f89f12806ef9104d4a7904193e9c65c7
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/board/habanalabs/common/create-psu-ctrl-fit.sh b/board/habanalabs/common/create-psu-ctrl-fit.sh
index fe968d7..f1ee0df 100755
--- a/board/habanalabs/common/create-psu-ctrl-fit.sh
+++ b/board/habanalabs/common/create-psu-ctrl-fit.sh
@@ -7,7 +7,6 @@
 source ${BUILDROOT_ROOT}/board/habanalabs/common/image-utility.sh
 
 MKIMAGE_FILE=${BR2_EXTERNAL}/output/${HL_DEV}/host/bin/mkimage
-IMAGE_DIRETORY=${BR2_EXTERNAL}/output/${HL_DEV}/images
 PSU_CTRL_ITS_TEMPLATE=${BR2_EXTERNAL}/board/habanalabs/common/psu-ctrl.its
 FIT_LOAD_TYPE_PSU_CTRL=$(grep -w FIT_LOAD_TYPE_PSU_CTRL ${SPECS_EMB_PROJ_DIR}/fit_load_type.h | sed 's/.* = \(.*\),/\1/')
 KEYS_DIR=${BR2_EXTERNAL}/board/habanalabs/${DEV_PREFIX}_arc/key/
@@ -69,7 +68,7 @@
 	${BR2_EXTERNAL}/board/habanalabs/common/add-image-descriptor.sh ${PSU_CTRL_ITB_FILE}
 }
 
-pushd ${IMAGE_DIRETORY}
+pushd ${STAGE_DIR}
 	PSU_CTRL_FW_IMAGE_FILES=$(ls ${PSU_CTRL_FW_IMAGE_NAME})
 	if [ ! -z "$PSU_CTRL_FW_IMAGE_FILES" ]; then
 		cp ${KEYS_DIR}/${KEY_FILE} ${KEY_FILE}
