From 75c10e4e297afd6619b248adcf3bbf11b28b2703 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Sat, 25 May 2024 23:30:10 +0300
Subject: [PATCH] [SW-15774] board: fake fuse

Adjust the fake fuse script.

Change-Id: Id7128f246173b2698ab28d21b12c84082c1f55f4
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/board/habanalabs/gaudi3/create-fake-fuse-image.sh b/board/habanalabs/gaudi3/create-fake-fuse-image.sh
index 5a1146c..5a24235 100755
--- a/board/habanalabs/gaudi3/create-fake-fuse-image.sh
+++ b/board/habanalabs/gaudi3/create-fake-fuse-image.sh
@@ -4,26 +4,26 @@
 set -u          # stop on any reference to undefine variable
 set -o pipefail # stop on any error in pipeline
 
+COMMON_BASE=${BR2_EXTERNAL}/board/habanalabs/common
 BOARD_BASE=${BR2_EXTERNAL}/board/habanalabs/${HL_DEV}
+source ${COMMON_BASE}/image-utility.sh
 MKIMAGE_FILE=${BR2_EXTERNAL}/output/${HL_DEV}/host/bin/mkimage
 IMAGE_DIRETORY=${BR2_EXTERNAL}/output/${HL_DEV}/images
 FAKE_FUSE_IMAGE_NAME=fake_fuse.bin
-FAKE_FUSE_IMAGE=${IMAGE_DIRETORY}/${FAKE_FUSE_IMAGE_NAME}
+FAKE_FUSE_IMAGE=${BINS_DIR}/${FAKE_FUSE_IMAGE_NAME}
 FAKE_FUSE_ITS_TEMPLATE=${BR2_EXTERNAL}/board/habanalabs/${HL_DEV}/fake_fuse.its
-FAKE_FUSE_ITS_FILE=${IMAGE_DIRETORY}/${DEV_PREFIX}-fake_fuse.its
-FAKE_FUSE_ITB_FILE=${IMAGE_DIRETORY}/${DEV_PREFIX}-fake_fuse.itb
-COMMON_BASE=${BR2_EXTERNAL}/board/habanalabs/common
+FAKE_FUSE_ITS_FILE=${STAGE_DIR}/${DEV_PREFIX}-fake_fuse.its
+FAKE_FUSE_ITB_FILE=${ITBS_DIR}/${DEV_PREFIX}-fake_fuse.itb
 source ${BOARD_BASE}/post-image-common.sh
-source ${COMMON_BASE}/image-utility.sh
 BR2_EXTERNAL_HABANALABS_PATH=${BR2_EXTERNAL}
 source ${BOARD_BASE}/post-image-spi.sh
 
 #fake fuse itb can also be created standalone by just changing
 #board/habanalabs/gaudi3_arc/d*_fake_fuse.txt contents, so
 #below functions update image to be use by itb
-pushd ${IMAGE_DIRETORY}
-	prep_image_fake_fuse ${DEV_PREFIX}-fake-fuse.bin.le
-	swap_file ${DEV_PREFIX}-fake-fuse.bin.le ${DEV_PREFIX}-fake-fuse.bin.be
+pushd ${STAGE_DIR}
+	prep_image_fake_fuse ${BINS_DIR}/${DEV_PREFIX}-fake-fuse.bin.le
+	swap_file ${BINS_DIR}/${DEV_PREFIX}-fake-fuse.bin.le ${BINS_DIR}/${DEV_PREFIX}-fake-fuse.bin.be
 popd
 
 hl_create_section_variables
