From d6adb5e700b6c582641ccdb3a1765d7a8922c5af Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Mon, 27 May 2024 14:16:03 +0300
Subject: [PATCH] [SW-0] gaudi2: add uart prints to arc0

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: I70c1090f6c8c90b2fcb458ca0ab38464e7565653
---

diff --git a/app/ppboot/ppboot.c b/app/ppboot/ppboot.c
index 1748ae6..9472b6b 100644
--- a/app/ppboot/ppboot.c
+++ b/app/ppboot/ppboot.c
@@ -287,7 +287,7 @@
 static void device_init(void)
 {
 	device_init_efuse();
-	device_init_uart();
+	//device_init_uart();
 	device_init_pci();
 	device_init_global_conf();
 	device_init_ca53_cfg();
diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index dccb198..8511bbc 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -727,17 +727,25 @@
 	device_init_efuse();
 	device_init_pci();
 	device_init_pll();
+	device_init_uart();
 	udelay_init();
 	preboot_device_id_init();
+	hl_uart_init();
 }
 
 void secondary_main(void)
 {
+	//preboot_uart_rbuf_init();
 	preboot_update_arc0_status(0);
 	secondary_device_init();
+	preboot_puts("uart print enabled\n");
+	preboot_uart_rbuf_flush();
 	preboot_update_arc0_status(1);
 	preboot_pcie_monitoring();
 	preboot_update_arc0_status(14);
+	preboot_puts("arc0:pcie link up\n");
+	preboot_uart_rbuf_flush();
+	preboot_puts("\n");
 	/* Put secondary CPU in sleep state until further notice */
 	wfe();
 }
diff --git a/arch/gaudi2/common/hl.h b/arch/gaudi2/common/hl.h
index deab740..f0f257f 100644
--- a/arch/gaudi2/common/hl.h
+++ b/arch/gaudi2/common/hl.h
@@ -16,7 +16,8 @@
 #define HL_PLL_DIV_SEL		0x280
 #define HL_PLL_DIV_EN		0x2a0
 #define preboot_update_arc0_status(bit_pos) \
-	regmap_writel(regmap_readl(mmPSOC_GLOBAL_CONF_SCRATCHPAD_13) | BIT(bit_pos), mmPSOC_GLOBAL_CONF_SCRATCHPAD_13)
+	regmap_writel(regmap_readl(mmPSOC_GLOBAL_CONF_SCRATCHPAD_13) | BIT(bit_pos), mmPSOC_GLOBAL_CONF_SCRATCHPAD_13); \
+	preboot_puts_hex(" ",bit_pos, false);
 void set_pci_msix_pba_offset_val(void);
 void set_slv_cfg_reg_val(void);
 void halt_n_reset_arc(u8 id);
