From 8b0cc8caa649bb496cb318cd3863632649e95671 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Thu, 23 May 2024 16:28:43 +0300
Subject: [PATCH] [SW-15774] board: fix hash calc

Using wrong directories.

Change-Id: I46f2b16c14bac53fb196272623154d285e478908
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/board/habanalabs/common/dirs.sh b/board/habanalabs/common/dirs.sh
index a7e24ba..1be1ad9 100644
--- a/board/habanalabs/common/dirs.sh
+++ b/board/habanalabs/common/dirs.sh
@@ -12,6 +12,8 @@
 export STATS_DIR=${IMGS_DIR}
 export ELFS_DIR=${IMGS_PROD_DIR}/elfs/
 export BINS_PROD_DIR=${IMGS_PROD_DIR}
+export BINS_ARC_DIR=${DEV_PREFIX_ARC_BASE_DIR}/images/
+
 # TODO: move the below to FS1 specific make file
 case ${DEV_PREFIX} in
 fs1)
diff --git a/board/habanalabs/gaudi2/post-image-common.sh b/board/habanalabs/gaudi2/post-image-common.sh
index 7b689c9..b0c1b2e 100755
--- a/board/habanalabs/gaudi2/post-image-common.sh
+++ b/board/habanalabs/gaudi2/post-image-common.sh
@@ -16,7 +16,7 @@
 	OUT_NAME=$3
 
 	TMPNAME=${KEY_NAME}-${PROP_NAME}.txt
-	fdtget ${STAGE_DIR}/${KEY_NAME}.dtb /signature/key-${KEY_BASE_NAME} rsa,${PROP_NAME} -t bx > ${STAGE_DIR}/${TMPNAME}
+	fdtget ${BINS_PROD_DIR}/${KEY_NAME}.dtb /signature/key-${KEY_BASE_NAME} rsa,${PROP_NAME} -t bx > ${STAGE_DIR}/${TMPNAME}
 	sed 's/\s\+/\n/g' ${STAGE_DIR}/${TMPNAME} > ${STAGE_DIR}/${TMPNAME}.tmp
 	sed -e 's/\b[0-9]\b/0&/' -e 's/\b[a-f]\b/0&/' ${STAGE_DIR}/${TMPNAME}.tmp  > ${STAGE_DIR}/${TMPNAME}.tmp.le
 	xxd -r -p ${STAGE_DIR}/${TMPNAME}.tmp.le > ${OUT_NAME}
diff --git a/board/habanalabs/gaudi3_arc/post-image-spi.sh b/board/habanalabs/gaudi3_arc/post-image-spi.sh
index e93380a..7b491b8 100755
--- a/board/habanalabs/gaudi3_arc/post-image-spi.sh
+++ b/board/habanalabs/gaudi3_arc/post-image-spi.sh
@@ -50,6 +50,12 @@
 
 function copy_all_keys {
 	#TODO: add functionality
+	cp ${KEY_BASE}/key/*.dtb ${1:-.}
+	cp ${KEY_BASE}/key/*.key ${1:-.}
+	cp ${KEY_BASE}/key/*.crt ${1:-.}
+}
+
+function hl_copy_binaries {
 	return
 }
 
@@ -99,7 +105,7 @@
 	OUT_NAME=$3
 
 	TMPNAME=${STAGE_DIR}/${KEY_NAME}-${PROP_NAME}.txt
-	fdtget ${KEY_NAME}.dtb /signature/key-${KEY_NAME} rsa,${PROP_NAME} -t bx > ${TMPNAME}
+	fdtget ${BINS_PROD_DIR}/${KEY_NAME}.dtb /signature/key-${KEY_NAME} rsa,${PROP_NAME} -t bx > ${TMPNAME}
 	sed 's/\s\+/\n/g' ${TMPNAME} > ${TMPNAME}.tmp
 	sed -e 's/\b[0-9]\b/0&/' -e 's/\b[a-f]\b/0&/' ${TMPNAME}.tmp  > ${TMPNAME}.tmp.le
 	xxd -r -p ${TMPNAME}.tmp.le > ${OUT_NAME}
@@ -129,8 +135,8 @@
 
 	#no separate public key will be part of the image, since these props serve the purpose.
 	# Hash will be computed on the Props itself
-	cat ${STAGE_DIR}/rr.prop ${STAGE_DIR}/mod.prop ${STAGE_DIR}/exp.prop ${STAGE_DIR}/inv.prop nb.prop	> ${STAGE_DIR}/${KEY_NAME}.pub
-	cat ${STAGE_DIR}/rr.prop ${STAGE_DIR}/mod.prop ${STAGE_DIR}/exp.prop ${STAGE_DIR}/inv.prop nb.prop	> ${STAGE_DIR}/${KEY_NAME}-key-props.bin
+	cat ${STAGE_DIR}/rr.prop ${STAGE_DIR}/mod.prop ${STAGE_DIR}/exp.prop ${STAGE_DIR}/inv.prop nb.prop	> ${BINS_PROD_DIR}/${KEY_NAME}.pub
+	cat ${STAGE_DIR}/rr.prop ${STAGE_DIR}/mod.prop ${STAGE_DIR}/exp.prop ${STAGE_DIR}/inv.prop nb.prop	> ${BINS_PROD_DIR}/${KEY_NAME}-key-props.bin
 
 	rm -f ${STAGE_DIR}/*.txt ${STAGE_DIR}/*.proptemp ${STAGE_DIR}/*.prop
 }
@@ -554,7 +560,7 @@
 	PKFILE=${TEMPNAME}.payload.key
 	PKFULLFILE=${TEMPFULLNAME}.payload.key
 
-	cat ${TEMPFULLNAME} ${STAGE_DIR}/${KEY_NAME}-key-props.bin > ${PKFULLFILE}
+	cat ${TEMPFULLNAME} ${BINS_PROD_DIR}/${KEY_NAME}-key-props.bin > ${PKFULLFILE}
 
 	#The size of the complete image(Payload + Key + Signature) needs to be calculated
 	#and placed in the manifest even before hashing and signing over it as the size
