From f3c8d9504c5bdc989ed2e847574449d8d6d906ac Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Mon, 03 Jun 2024 19:01:25 +0300
Subject: [PATCH] [SW-0] gaudi2: update arc0 alive status in while loop

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: I9451fa338b1954e768bd438f419fa7f6a87ea190
---

diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index c1c1713..c3b0ce5 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -639,6 +639,7 @@
 			preboot_update_arc0_status(12);
 			continue;
 		}
+		preboot_update_arc0_alive(0xAABB);
 	}
 
 	/*
diff --git a/arch/gaudi2/common/hl.h b/arch/gaudi2/common/hl.h
index deab740..8ecf892 100644
--- a/arch/gaudi2/common/hl.h
+++ b/arch/gaudi2/common/hl.h
@@ -17,6 +17,8 @@
 #define HL_PLL_DIV_EN		0x2a0
 #define preboot_update_arc0_status(bit_pos) \
 	regmap_writel(regmap_readl(mmPSOC_GLOBAL_CONF_SCRATCHPAD_13) | BIT(bit_pos), mmPSOC_GLOBAL_CONF_SCRATCHPAD_13)
+#define preboot_update_arc0_alive(magic) \
+	regmap_writel(magic, mmPSOC_GLOBAL_CONF_SCRATCHPAD_16)
 void set_pci_msix_pba_offset_val(void);
 void set_slv_cfg_reg_val(void);
 void halt_n_reset_arc(u8 id);
