From 46c1f5fdf6cffb4980e6f2b495aaab151d58dc4f Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Thu, 06 Jun 2024 14:36:00 +0300
Subject: [PATCH] [SW-0] gaudi2: set pci dbi sd_control1_reg

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: Ia321ed5e2cc595568931691e15d8ced7e8c04d17
---

diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index c3b0ce5..3101945 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -599,6 +599,9 @@
 
 void preboot_pcie_monitoring(void)
 {
+	bool magic = false;
+	u32 val;
+
 	preboot_update_arc0_status(2);
 	/* Wait for PRSTn */
 	while (!regmap_readl(mmPSOC_GLOBAL_CONF_PRSTN))
@@ -609,6 +612,11 @@
 	preboot_pcie_enable();
 
 	preboot_update_arc0_status(9);
+	udelay(100);
+	val = regmap_readl(mmPCIE_DBI_SD_CONTROL1_REG);
+	val |= 0x1FFFF;
+	regmap_writel(val, mmPCIE_DBI_SD_CONTROL1_REG);
+
 	while (1) {
 		unsigned int prstn, ltssm, smlh, rdlh;
 
@@ -639,7 +647,11 @@
 			preboot_update_arc0_status(12);
 			continue;
 		}
-		preboot_update_arc0_alive(0xAABB);
+		if (magic)
+			preboot_update_arc0_alive(0xAABB);
+		else
+			preboot_update_arc0_alive(0xCCDD);
+		magic = !magic;
 	}
 
 	/*
