From 67df056b30030069dac2ac82895acb20fab06f42 Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Thu, 13 Jun 2024 15:33:27 +0300
Subject: [PATCH] Revert "[SW-176859] gaudi3: ignore temperature sensors"

This reverts commit 952b91ab104517522c3bc4c9c83c0fab0d20a8c4.
---

diff --git a/package/sensors/hl_sensors.conf b/package/sensors/hl_sensors.conf
index 1e01f02..0d4f806 100644
--- a/package/sensors/hl_sensors.conf
+++ b/package/sensors/hl_sensors.conf
@@ -262,20 +262,20 @@
 label temp55 "On Chip TD 1"
 label temp56 "On Chip TD 2"
 label temp57 "On Chip TD 3"
-ignore temp58
-ignore temp59
-ignore temp60
-ignore temp61
-ignore temp62
-ignore temp63
-ignore temp64
-ignore temp65
-ignore temp66
-ignore temp67
-ignore temp68
-ignore temp69
-ignore temp70
-ignore temp71
+label temp58 "TMP PCB D0"
+label temp59 "On Chip TD 4"
+label temp60 "On Chip TD 5"
+label temp61 "On Chip TD 6"
+label temp62 "On Chip TD 7"
+label temp63 "TMP PCB D1"
+label temp64 "Onboard Temp D0"
+label temp65 "Onboard Temp D1"
+label temp66 "CPLD Temp"
+label temp67 "ASC10 Temp"
+label temp68 "VRM1 Temp"
+label temp69 "VRM2_1 Temp"
+label temp70 "VRM2_2 Temp"
+label temp71 "VRM2_3 Temp"
 
 
 ignore curr1
