From 852a0297b35afab2479cf767c5c0958fe66d98e5 Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Thu, 27 Jun 2024 21:36:50 +0300
Subject: [PATCH] [SW-0] cpld fit: add debug prints

Temporary add debug prints to script

Signed-off-by: Guy Nisan <gnisan@habana.ai>
Change-Id: Ifefafc8d76c57b2a7c1723827868e3384a3a4822
---

diff --git a/board/habanalabs/gaudi3/create-cpld-fit.sh b/board/habanalabs/gaudi3/create-cpld-fit.sh
index 8fe0f23..a4134a3 100755
--- a/board/habanalabs/gaudi3/create-cpld-fit.sh
+++ b/board/habanalabs/gaudi3/create-cpld-fit.sh
@@ -3,6 +3,7 @@
 set -e          # stop on any error in this script
 set -u          # stop on any reference to undefine variable
 set -o pipefail # stop on any error in pipeline
+set -x
 
 FW_ROOT_DIR=${BR2_EXTERNAL}/board/habanalabs/${DEV_PREFIX}/fw
 FIT_TARGET_DEV_ENUM=$(grep "ENTITY_IDS_GAUDI3\b" ${SPECS_EMBEDDED_DIR}/entity_ids.h | sed 's/.*[\t]\+= \(.*\),/\1/')
