From 2b74f306adea3a4e6632a4f7fd118364c09f1793 Mon Sep 17 00:00:00 2001
From: Peleg Vigodny <peleg.vigodny@tandemg.com>
Date: Thu, 16 Jul 2020 16:17:59 +0300
Subject: [PATCH] uboot-btl

Change-Id: I5839f8f0355d25d76fb911f1f698e4194d6ff340
Signed-off-by: Peleg Vigodny <peleg.vigodny@tandemg.com>
---

diff --git a/board/habanalabs/common/uboot.its b/board/habanalabs/common/uboot.its
index 6e3f39b..aed5c83 100644
--- a/board/habanalabs/common/uboot.its
+++ b/board/habanalabs/common/uboot.its
@@ -21,82 +21,7 @@
 			description = "Habana BTL primary";
 			data = /incbin/("s1.bin.le");
 			compression = "none";
-			offset = <0x00000000>;
-			hash {
-				algo = "sha256";
-			};
-		};
-
-		btl-s2 {
-			description = "Habana BTL secondary";
-			data = /incbin/("s2.bin.le");
-			compression = "none";
-			offset = <0x00010000>;
-			hash {
-				algo = "sha256";
-			};
-		};
-
-		btl-s3 {
-			description = "Habana BTL prstn";
-			data = /incbin/("s3.bin.le");
-			compression = "none";
-			offset = <0x00020000>;
-			hash {
-				algo = "sha256";
-			};
-		};
-
-		btl-s4 {
-			description = "Habana BTL pcie";
-			data = /incbin/("s4.bin.le");
-			compression = "none";
-			offset = <0x00030000>;
-			hash {
-				algo = "sha256";
-			};
-		};
-
-		img-ptr {
-			description = "Habana image pointers";
-			data = /incbin/("@HL_DEV@-images-pointers.bin.le");
-			compression = "none";
-			offset = <0x00060000>;
-			hash {
-				algo = "sha256";
-			};
-		};
-
-		uboot-env {
-			description = "Habana U-Boot environment";
-			data = /incbin/("@HL_DEV@-uboot-env.bin.le");
-			compression = "none";
-			offset = <0x00070000>;
-			hash {
-				algo = "sha256";
-			};
-		};
-
-		preboot {
-			description = "Habana Preboot";
-			data = /incbin/("@HL_DEV@-preboot.bin.le");
-			type = "standalone";
-			arch = "arm64";
-			compression = "none";
-			offset = <0x00040000>, <0x00050000>;
-			loadaddr = <0x7F 0xF0004000>;
-			hash {
-				algo = "sha256";
-			};
-		};
-
-		uboot {
-			description = "U-Boot (64-bit)";
-			data = /incbin/("@HL_DEV@-u-boot.bin.le");
-			type = "standalone";
-			arch = "arm64";
-			compression = "none";
-			loadaddr = <0x7F 0xF0100000>;
+			offset = <0x00200000>;
 			hash {
 				algo = "sha256";
 			};
@@ -118,22 +43,7 @@
 
 		fw-loader-conf {
 			description = "Habanalabs device boot loader";
-			loadables = "btl-s1", "btl-s2", "btl-s3", "btl-s4",
-				  "img-ptr", "uboot-env", "preboot";
-			btl-s1 = "btl-s1";
-			btl-s2 = "btl-s2";
-			btl-s3 = "btl-s3";
-			btl-s4 = "btl-s4";
-			img-ptr = "img-ptr";
-			uboot-env = "uboot-env";
-			preboot = "preboot";
-			signature {
-				algo = "sha256,rsa3072";
-				key-name-hint = "hl_rsa3072";
-				sign-images = "btl-s1", "btl-s2", "btl-s3",
-					"btl-s4", "img-ptr", "uboot-env",
-					"preboot";
-			};
+			loadables = "btl-s1";
 		};
 
 		preboot-conf {
