From 55ef8fd81f822060b9692369a449d26c0286d860 Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Mon, 05 Aug 2024 08:08:08 +0300
Subject: [PATCH] temp patch

Change-Id: Ia565b85fc9d9682ed770564ad4d1cf2fab1555fa
---

diff --git a/common/boot_flow.c b/common/boot_flow.c
index 6cf17d6..5a79655 100644
--- a/common/boot_flow.c
+++ b/common/boot_flow.c
@@ -195,6 +195,17 @@
 	return true;
 }
 
+int boot_rom_boot_flow_get_rec_img_count(enum boot_rom_manifests rom_manifest_type)
+{
+	enum hl_fw_manifest_type man_idx = primary_manifest_idx;
+
+	if (man_type == BOOT_ROM_SECONDARY) {
+		man_idx = (primary_manifest_idx + 1) % HL_FW_SPI_MANIFEST_MAX_NUM;
+	}
+
+	return boot_desc_get_recovery_image_count(man_idx);
+}
+
 bool boot_rom_boot_flow_is_recovery_imgs_valid(__unused enum boot_rom_manifests rom_manifest_type)
 {
 	/* Verify sha for recovery img0 and img1 */
diff --git a/common/boot_flow.h b/common/boot_flow.h
index 88e60fc..b74a504 100644
--- a/common/boot_flow.h
+++ b/common/boot_flow.h
@@ -15,5 +15,6 @@
 bool boot_rom_boot_flow_is_recovery_imgs_valid(__unused enum boot_rom_manifests rom_manifest_type);
 bool boot_rom_boot_flow_is_secondary_manifest_valid(void);
 void boot_rom_boot_flow_wait_for_dma_finish(void);
+int boot_rom_boot_flow_get_rec_img_count(enum boot_rom_manifests rom_manifest_type);
 
 #endif /* _BOOT_FLOW_H_ */
diff --git a/common/recovery.c b/common/recovery.c
index 07ff160..ae881ee 100644
--- a/common/recovery.c
+++ b/common/recovery.c
@@ -29,7 +29,7 @@
 	boot_rom_status_update(PROGRESS_BOOT_ROM_RECOVERY_END);
 }
 
-static int boot_rom_execute_recovery_img(__unused enum boot_rom_manifests recovery_manifest)
+static int boot_rom_execute_recovery_img(enum boot_rom_manifests recovery_manifest)
 {
 	/* Wait for any existing dma to finish */
 	boot_rom_boot_flow_wait_for_dma_finish();
