From 5e7d7a7dafbe23324e2dcb77e174cfc8b1d610f2 Mon Sep 17 00:00:00 2001
From: Jason Smith <jsmith@habana.ai>
Date: Mon, 12 Aug 2024 11:53:19 -0400
Subject: [PATCH] TEST: do not merge

Change-Id: Id5ec5b4587bbf024399dee1ee72203935ff28624
---

diff --git a/board/habanalabs/gaudi3/zephyr.its b/board/habanalabs/gaudi3/zephyr.its
index 4a38958..7f1fff2 100644
--- a/board/habanalabs/gaudi3/zephyr.its
+++ b/board/habanalabs/gaudi3/zephyr.its
@@ -1,5 +1,7 @@
 /dts-v1/;
 
+
+
 / {
 	hl-version = "@HL_VERSION@";
 	hl-device = "@DEV_PREFIX@";
