From 53ce2856ed7867077278a7a3a032137da59c2e94 Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Thu, 17 Oct 2024 14:35:56 +0300
Subject: [PATCH] [SW-0] gaudi2: btl rom d2bug early write rsp disabled

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: I6f21fe71d818655092ee14cb346d431299209c58
---

diff --git a/app/ppboot/ppboot.c b/app/ppboot/ppboot.c
index 9c7fe05..77ac7d6 100644
--- a/app/ppboot/ppboot.c
+++ b/app/ppboot/ppboot.c
@@ -373,6 +373,7 @@
 
 	ppboot_init_fw_info();
 	d2d_spi_clk_init();
+	hl_lbw_early_write_resp_enable(false);
 
 	ret = ppboot_load_preboot_img(&loadaddr);
 	ppboot_check_stack_monitor();
diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 72a8a68..4d84994 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -309,6 +309,10 @@
 
 	val = val | PARC_GLOBAL_CONF_ARC_LBU_AXI_SPLIT_CTRL2_FORCE_WR_BUF_M;
 	regmap_writel(val, D0_PARC_GLOBAL_CONF_REG(mmPARC_GLOBAL_CONF_ARC_LBU_AXI_SPLIT_CTRL2_1));
+	if (enable)
+		preboot_puts("early write response enabled\n");
+	else
+		preboot_puts("early write response disabled\n");
 }
 
 __weak void hl_dev_cpu_pll_init(void) {}
diff --git a/include/baremetal.h b/include/baremetal.h
index e2d949d..39bb2c3 100644
--- a/include/baremetal.h
+++ b/include/baremetal.h
@@ -319,6 +319,7 @@
 int hl_boot_info_finalize(void);
 void hl_check_pcie_link_up_by_erom(void);
 void d2d_spi_clk_init(void);
+void hl_lbw_early_write_resp_enable(bool enable);
 
 #endif /* __ASSEMBLY__ */
 
