From b0bd6fe0372c6ebd541d540ebab878db1c1a1c29 Mon Sep 17 00:00:00 2001
From: Jason Smith <jsmith@habana.ai>
Date: Tue, 18 Mar 2025 18:23:05 -0400
Subject: [PATCH] DO NOT MERGE: fix pldm package version

Change-Id: I8bd1e294edf4ef77cf32682cb5ae3bdb7b8e840f
---

diff --git a/board/habanalabs/gaudi3/create-pldm-fwup-package-main.sh b/board/habanalabs/gaudi3/create-pldm-fwup-package-main.sh
index 2cd4bff..b8e7334 100755
--- a/board/habanalabs/gaudi3/create-pldm-fwup-package-main.sh
+++ b/board/habanalabs/gaudi3/create-pldm-fwup-package-main.sh
@@ -2,6 +2,33 @@
 
 source ${BUILDROOT_ROOT}/board/habanalabs/${DEV_PREFIX}/create-pldm-fwup-package.sh
 
+VERSION_FILE="${BUILDROOT_ROOT}/output/${DEV_PREFIX}_arc/build/agent-fwl-custom/localversion-habanalabs"
+
+if [[ ! -f ${VERSION_FILE} ]]; then
+        echo "$0: FW loader agent not built - cannot create PLDM package"
+        exit 1
+fi
+
+VERSION_STR=$(cat ${VERSION_FILE})
+VERSION_STR=${VERSION_STR:1} # remove leading '-'
+VERSION_STAMP=
+
+if [[ "${VERSION_STR}" =~ hl-gaudi3-([0-9]+)\.([0-9]+)\.([0-9]+).*-fw-([0-9]+)\.([0-9]+)\.([0-9]+).* ]]
+then
+        sw_major=${BASH_REMATCH[1]}
+        sw_minor=${BASH_REMATCH[2]}
+        sw_patch=${BASH_REMATCH[3]}
+        fw_major=${BASH_REMATCH[4]}
+        fw_minor=${BASH_REMATCH[5]}
+        fw_patch=${BASH_REMATCH[6]}
+
+        PKG_VERSION="${sw_major}.${sw_minor}.${sw_patch}.0"
+        VERSION_STAMP=$((fw_major << 24 | fw_minor << 16 | fw_patch << 8))
+else
+        echo "$0: Failed to parse version string for comparison stamp: '${VERSION_STR}'"
+        exit 1
+fi
+
 PACKAGE_PREFIX="${ITBS_DIR}${DEV_PREFIX}-pldm-package-main"
 PACKAGE_JSON="${PACKAGE_PREFIX}.json"
 PACKAGE_BIN="${PACKAGE_PREFIX}.bin"
@@ -31,28 +58,56 @@
         exit 1
 fi
 
+# expecting 'gaudi3-cpld-LFMXO5-15D-ES3-03-TS677A4813.itb'
+if [[ "$(basename ${CPLD_FIT_2})" =~ gaudi3-cpld-LFMXO5-15D-ES3-([0-9]+)-TS([A-F0-9]+).itb ]]
+then
+        CPLD_FIT_2_MAJOR=${BASH_REMATCH[1]}
+        CPLD_FIT_2_TS=${BASH_REMATCH[2]}
+        CPLD_FIT_2_VERSION="main-${SOFTWARE_VERSION}-cp-${CPLD_FIT_2_MAJOR}.${CPLD_FIT_2_TS}"
+        CPLD_FIT_2_STAMP=$((16#${CPLD_FIT_2_TS}))
+else
+        echo "$0: Failed to parse version from CPLD 2 filename"
+        exit 1
+fi
+
+# expecting 'gaudi3-cpld-LFMXO5-15D-HBN-HL325-04-TS67714C1D.itb'
+if [[ "$(basename ${CPLD_FIT_3})" =~ gaudi3-cpld-LFMXO5-15D-HBN-HL325-([0-9]+)-TS([A-F0-9]+).itb ]]
+then
+        CPLD_FIT_3_MAJOR=${BASH_REMATCH[1]}
+        CPLD_FIT_3_TS=${BASH_REMATCH[2]}
+        CPLD_FIT_3_VERSION="main-${SOFTWARE_VERSION}-cp-${CPLD_FIT_3_MAJOR}.${CPLD_FIT_3_TS}"
+        CPLD_FIT_3_STAMP=$((16#${CPLD_FIT_3_TS}))
+else
+        echo "$0: Failed to parse version from CPLD 3 filename '${CPLD_FIT_3}"
+        exit 1
+fi
+
+# expecting 'gaudi3-cpld-LFMXO5-15D-HBN-HL325-04-TS67714C1D-production.itb'
+if [[ "$(basename ${CPLD_FIT_4})" =~ gaudi3-cpld-LFMXO5-15D-HBN-HL325-([0-9]+)-TS([A-F0-9]+)-production.itb ]]
+then
+        CPLD_FIT_4_MAJOR=${BASH_REMATCH[1]}
+        CPLD_FIT_4_TS=${BASH_REMATCH[2]}
+        CPLD_FIT_4_VERSION="main-${SOFTWARE_VERSION}-cp-${CPLD_FIT_4_MAJOR}.${CPLD_FIT_4_TS}"
+        CPLD_FIT_4_STAMP=$((16#${CPLD_FIT_4_TS}))
+else
+        echo "$0: Failed to parse version from CPLD 4 filename '${CPLD_FIT_4}'"
+        exit 1
+fi
+
 ${PLDM_FWUP_JSON_CTOR_SCRIPT} --output-file ${PACKAGE_JSON} \
                              --package-release-date-time "${PACKAGE_RELEASE_DATE_TIME}" \
                              --package-version "${MAIN_PACKAGE_VERSION}" \
                              --component-image-set-version "${MAIN_PACKAGE_VERSION}" \
                              --pci-device-id "${PCI_DEVICE_ID}" \
                              --pci-subsystem-id "${PCI_SUBSYSTEM_ID}" \
-                             --component $((COMPONENT_CLASSIFICATION_FIRMWARE)) $((COMPONENT_ID_FW_LOADER_1))   \
-                                "${AGENT_FW_LOADER_FIT_VERSION}" "${AGENT_FW_LOADER_FIT_VERSION_STAMP}"         \
-                             --component $((COMPONENT_CLASSIFICATION_FIRMWARE)) $((COMPONENT_ID_SPI))           \
-                                "${AGENT_FW_LOADER_FIT_VERSION}" "${AGENT_FW_LOADER_FIT_VERSION_STAMP}"         \
-                             --component $((COMPONENT_CLASSIFICATION_FIRMWARE)) $((COMPONENT_ID_FW_LOADER_2))   \
-                                "${AGENT_FW_LOADER_FIT_VERSION}" "${AGENT_FW_LOADER_FIT_VERSION_STAMP}"         \
-                             --component $((COMPONENT_CLASSIFICATION_FIRMWARE)) $((COMPONENT_ID_EROM))          \
-                                "${EROM_FW_LOADER_FIT_VERSION}"  "${EROM_FW_LOADER_FIT_VERSION_STAMP}"          \
-                             --component $((COMPONENT_CLASSIFICATION_FIRMWARE)) $((COMPONENT_ID_FW_LOADER_3))   \
-                                "${AGENT_FW_LOADER_FIT_VERSION}" "${AGENT_FW_LOADER_FIT_VERSION_STAMP}"         \
-                             --component $((COMPONENT_CLASSIFICATION_FIRMWARE)) $((COMPONENT_ID_CPLD_2))        \
-                                "${CPLD_2_FIT_VERSION}"          "${CPLD_2_FIT_VERSION_STAMP}"                  \
-                             --component $((COMPONENT_CLASSIFICATION_FIRMWARE)) $((COMPONENT_ID_CPLD_3))        \
-                                "${CPLD_3_FIT_VERSION}"          "${CPLD_3_FIT_VERSION_STAMP}"                  \
-                             --component $((COMPONENT_CLASSIFICATION_FIRMWARE)) $((COMPONENT_ID_CPLD_4))        \
-                                "${CPLD_4_FIT_VERSION}"          "${CPLD_4_FIT_VERSION_STAMP}"                  \
+                             --component $((COMPONENT_CLASSIFICATION_FIRMWARE)) $((COMPONENT_ID_FW_LOADER_1)) "${VERSION_STR}"        "${VERSION_STAMP}"    \
+                             --component $((COMPONENT_CLASSIFICATION_FIRMWARE)) $((COMPONENT_ID_SPI))         "${VERSION_STR}"        "${VERSION_STAMP}"    \
+                             --component $((COMPONENT_CLASSIFICATION_FIRMWARE)) $((COMPONENT_ID_FW_LOADER_2)) "${VERSION_STR}"        "${VERSION_STAMP}"    \
+                             --component $((COMPONENT_CLASSIFICATION_FIRMWARE)) $((COMPONENT_ID_EROM))        "${VERSION_STR}"        "${VERSION_STAMP}"    \
+                             --component $((COMPONENT_CLASSIFICATION_FIRMWARE)) $((COMPONENT_ID_FW_LOADER_3)) "${VERSION_STR}"        "${VERSION_STAMP}"    \
+                             --component $((COMPONENT_CLASSIFICATION_FIRMWARE)) $((COMPONENT_ID_CPLD_2))      "${CPLD_FIT_2_VERSION}" "${CPLD_FIT_2_STAMP}" \
+                             --component $((COMPONENT_CLASSIFICATION_FIRMWARE)) $((COMPONENT_ID_CPLD_3))      "${CPLD_FIT_3_VERSION}" "${CPLD_FIT_3_STAMP}" \
+                             --component $((COMPONENT_CLASSIFICATION_FIRMWARE)) $((COMPONENT_ID_CPLD_4))      "${CPLD_FIT_4_VERSION}" "${CPLD_FIT_4_STAMP}" \
 
 rv=$?
 if [[ ${rv} -ne 0 ]]; then
diff --git a/board/habanalabs/gaudi3/create-pldm-fwup-package-spi-flash.sh b/board/habanalabs/gaudi3/create-pldm-fwup-package-spi-flash.sh
index b34c425..2ad9c18 100755
--- a/board/habanalabs/gaudi3/create-pldm-fwup-package-spi-flash.sh
+++ b/board/habanalabs/gaudi3/create-pldm-fwup-package-spi-flash.sh
@@ -13,12 +13,17 @@
 VERSION_STR=${VERSION_STR:1} # remove leading '-'
 VERSION_STAMP=
 
-if [[ "${VERSION_STR}" =~ hl-gaudi3-.*-fw-([0-9]+)\.([0-9]+)\.([0-9]+).* ]]
+if [[ "${VERSION_STR}" =~ hl-gaudi3-([0-9]+)\.([0-9]+)\.([0-9]+).*-fw-([0-9]+)\.([0-9]+)\.([0-9]+).* ]]
 then
-        _major=${BASH_REMATCH[1]}
-        _minor=${BASH_REMATCH[2]}
-        _patch=${BASH_REMATCH[3]}
-        VERSION_STAMP=$((_major << 24 | _minor << 16 | _patch << 8))
+        sw_major=${BASH_REMATCH[1]}
+        sw_minor=${BASH_REMATCH[2]}
+        sw_patch=${BASH_REMATCH[3]}
+        fw_major=${BASH_REMATCH[4]}
+        fw_minor=${BASH_REMATCH[5]}
+        fw_patch=${BASH_REMATCH[6]}
+
+        PKG_VERSION="${sw_major}.${sw_minor}.${sw_patch}.0"
+        VERSION_STAMP=$((fw_major << 24 | fw_minor << 16 | fw_patch << 8))
 else
         echo "%0: Failed to parse version string for comparison stamp"
         exit 1
@@ -30,8 +35,8 @@
 
 ${PLDM_FWUP_JSON_CTOR_SCRIPT} --output-file ${PACKAGE_JSON} \
         --package-release-date-time "${PACKAGE_RELEASE_DATE_TIME}" \
-        --package-version "${VERSION_STR}" \
-        --component-image-set-version "${VERSION_STR}" \
+        --package-version "${PKG_VERSION}" \
+        --component-image-set-version "${PKG_VERSION}" \
         --pci-device-id "${PCI_DEVICE_ID}" \
         --pci-subsystem-id "${PCI_SUBSYSTEM_ID}" \
         --component $((COMPONENT_CLASSIFICATION_FIRMWARE)) $((COMPONENT_ID_LOADER)) \
diff --git a/build-main.sh b/build-main.sh
new file mode 100755
index 0000000..39253c1
--- /dev/null
+++ b/build-main.sh
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+FW_PATH=${PWD}/output/gaudi3/images/itbs
+CPLD_PATH=/software/users/jsmith/cpld
+
+make gaudi3-pldm-package-create \
+	SOFTWARE_VERSION=1.21.0 \
+	FIRMWARE_VERSION=59.2.4 \
+	MAIN_PACKAGE_VERSION=1.21.0.0 \
+ 	AGENT_FW_LOADER_FIT=${FW_PATH}/gaudi3-agent-fw_loader-fit.itb \
+ 	EROM_FW_LOADER_FIT=${FW_PATH}/gaudi3-agent-fw_loader-fit_erom.itb \
+	AGENT_PCIE_FIT=${FW_PATH}/gaudi3-agent-pcie-fit.itb \
+	AGENT_SERDES_FIT=${FW_PATH}/gaudi3-agent-serdes-fit.itb \
+	REVOKE_KEY_FIT=${FW_PATH}/gaudi3-revoke_key.itb \
+	UPDATE_SVN_FIT=${FW_PATH}/gaudi3-update_svn.itb \
+	EEPROM_FIT=${FW_PATH}/gaudi3-eeprom.itb \
+	CPLD_FIT_2=${CPLD_PATH}/gaudi3-cpld-LFMXO5-15D-ES3-03-TS677A4813.itb \
+	CPLD_FIT_3=${CPLD_PATH}/gaudi3-cpld-LFMXO5-15D-HBN-HL325-04-TS67714C1D.itb \
+	CPLD_FIT_4=${CPLD_PATH}/gaudi3-cpld-LFMXO5-15D-HBN-HL325-04-TS67714C1D-production.itb
+
