From b00c8aeb9abd1a8bc388c3e5005bb2ce0dbe2efa Mon Sep 17 00:00:00 2001
From: Akshay Pradip Bharad <apbharad@habana.ai>
Date: Fri, 16 May 2025 16:45:15 +0000
Subject: [PATCH] [SW-0] ci fix

fix for ci build failure

Change-Id: I0f9b16c6b672bf6471f446f8430a8d80c3049acf
Signed-off-by: Akshay Pradip Bharad <apbharad@habana.ai>
---

diff --git a/include/intr.h b/include/intr.h
index a0e0570..7ab9439 100644
--- a/include/intr.h
+++ b/include/intr.h
@@ -43,7 +43,7 @@
 	int (*arch_process_interrupt)(struct intr_config *config);
 };
 
-int dev_intr_prepare(struct hl_dev_config *dev_cfg)
+__weak int dev_intr_prepare(struct hl_dev_config *dev_cfg) { return 0; }
 int process_interrupt(struct intr_config *config, enum intr_action action);
 int intr_dispatch_component_handler(struct intr_config *config);
 
