From cbb04ca87d41745522c0e6d6641496b57b9f0622 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Tue, 03 Jul 2018 09:36:56 +0300
Subject: [PATCH] eeprog: update with new couple of fields


---

diff --git a/board/habanalabs/common/rootfs_overlay/usr/sbin/eeprog b/board/habanalabs/common/rootfs_overlay/usr/sbin/eeprog
index 17772bf..35484d8 100755
--- a/board/habanalabs/common/rootfs_overlay/usr/sbin/eeprog
+++ b/board/habanalabs/common/rootfs_overlay/usr/sbin/eeprog
@@ -38,13 +38,15 @@
 	print_field_num "Layout Version" 0 1
 	print_field_num "Production Serial Number" 1 5
 	print_field_num "PCB Assembly Version" 6 1
-	print_field_num "DDR Type and Speed" 7 1
-	print_field_num "Fans Vendor" 8 1
-	print_field_num "Mechanics Vendor" 9 1
-	print_field_num "Thermal Solution Vendor" 10 1
-	print_field_num "PCIe Gen3 or Gen4 Enabled" 11 1
-	print_field_num "MAC Address" 12 6
-	print_field_str "Board Configuration" 18 64
+	print_field_num "CPLD FW Version" 7 1
+	print_field_num "Infineon FW Version" 8 1
+	print_field_num "DDR Type and Speed" 9 1
+	print_field_num "Fans Vendor" 10 1
+	print_field_num "Mechanics Vendor" 11 1
+	print_field_num "Thermal Solution Vendor" 12 1
+	print_field_num "PCIe Gen3 or Gen4 Enabled" 13 1
+	print_field_num "MAC Address" 14 6
+	print_field_str "Board Configuration" 20 64
 else
 	echo "Unknown eeprom layout"
 fi
