From 880c22d6eee2d0214f6d55d370f84a6801b5786f Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Tue, 03 Jul 2018 10:04:52 +0300
Subject: [PATCH] install: Add ASIC_BIN_DIR for installing production binaries


---

diff --git a/Makefile b/Makefile
index d5449a9..3447bc3 100644
--- a/Makefile
+++ b/Makefile
@@ -17,6 +17,7 @@
 	@echo 'Install:'
 	@echo '  optional environment variables:'
 	@echo '    PLDM_HEX_DIR       - Palladium directory to install flash image'
+	@echo '    ASIC_BIN_DIR       - ASIC directory to install binaries'
 	@echo '  dali-<board>-install - Install to <board> e.g. onix, pldm, asic'
 	@echo '                           set DALI_INSTALL_LAB_IP=<lab IP> to'
 	@echo '                             override <board> default lab IP'
@@ -109,8 +110,14 @@
 	@mkdir -p $@
 dali-pldm-install: ${PLDM_HEX_DIR}
 	-cp $(IMAGES_DIR)/dali-spi-flash.hex $</
-dali-asic-install:
-	@
+ASIC_BIN_DIR ?= /tools/hl_shared_dir/production/embedded/latest
+${ASIC_BIN_DIR}:
+	@mkdir -p $@
+dali-asic-install: ${ASIC_BIN_DIR}
+	-cp $(IMAGES_DIR)/dali-fit.itb $</
+	-cp $(IMAGES_DIR)/dali-spi-flash.bin $</
+	-cp $(IMAGES_DIR)/dali-u-boot.bin $</
+	-cp $(IMAGES_DIR)/dali-u-boot-spl.bin $</
 
 .PHONY: dali-distclean
 dali-distclean: $(OUTPUT_DIR)/.config
