From a993ec632c789c3510f5d148152299e9fd883da3 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Sat, 07 Jul 2018 23:29:18 +0300
Subject: [PATCH] board: dali is a new ARM64 SoC fixup

update uboot ASIC sys clock rate for nominal rate
---

diff --git a/board/habanalabs/dali/uboot_asic.fragment b/board/habanalabs/dali/uboot_asic.fragment
index d859dbd..f40dd75 100644
--- a/board/habanalabs/dali/uboot_asic.fragment
+++ b/board/habanalabs/dali/uboot_asic.fragment
@@ -1,4 +1,4 @@
-CONFIG_SYS_CLK_FREQ=200000000
+CONFIG_SYS_CLK_FREQ=1200000000
 CONFIG_CONARGS="serial0:115200n8"
 CONFIG_BAUDRATE=115200
 CONFIG_BOOTDELAY=0
