From cdd0780fb1c183023df9752bf20dc6b12baa42f8 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Wed, 01 Aug 2018 08:02:34 +0300
Subject: [PATCH] BTL: mark for KMD that we started


---

diff --git a/board/habanalabs/dali/btl/primary.S b/board/habanalabs/dali/btl/primary.S
index cad6e64..8008a61 100644
--- a/board/habanalabs/dali/btl/primary.S
+++ b/board/habanalabs/dali/btl/primary.S
@@ -6,6 +6,9 @@
 
 nop				/* DUMMY */
 
+/* CPU_BOOT_STATUS_IN_BTL -> mmPSOC_GLOBAL_CONF_WARM_REBOOT */
+cfg 0xCC4B300, 0x4
+
 #include "i2c.s"
 #ifdef TARGET_IS_PLDM
 /* Disable PCIe PHY */
