From 99ac35ab9594e57139d01cb7e4c84e50be0da1c1 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Thu, 02 Aug 2018 06:43:36 +0300
Subject: [PATCH] install: update default host IP for ASIC

To avoid erronouse usage, and accessing wrong machine
---

diff --git a/Makefile b/Makefile
index 5174542..285a709 100644
--- a/Makefile
+++ b/Makefile
@@ -89,7 +89,7 @@
 dali-pldm-install: DALI_INSTALL_LAB_IP ?= 192.168.10.45
 dali-pldm-install: DALI_INSTALL_HOST_IP ?= 192.168.10.45
 dali-asic-install: DALI_INSTALL_LAB_IP ?= 192.168.10.109
-dali-asic-install: DALI_INSTALL_HOST_IP ?= 192.168.10.114
+dali-asic-install: DALI_INSTALL_HOST_IP ?= 192.168.10.134
 .PHONY: dali-install
 dali-install:
 	-sshpass -p123456        scp $(IMAGES_DIR)/dali-u-boot.bin	root@$(DALI_INSTALL_HOST_IP):/lib/firmware/habanalabs/dali/
