From 95920f9fbc75bdd0d1c6151477b09265cb1650b7 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Sat, 04 Aug 2018 21:48:15 +0300
Subject: [PATCH] Add BTL support fixup

remove leftover flags
---

diff --git a/board/habanalabs/dali/btl/Makefile b/board/habanalabs/dali/btl/Makefile
index 53fb13a..3090e41 100644
--- a/board/habanalabs/dali/btl/Makefile
+++ b/board/habanalabs/dali/btl/Makefile
@@ -59,7 +59,7 @@
 s2.bin: secondary.bin
 	@dd status=none if=$< of=$@
 s3.bin: prstn.bin
-	@dd status=none conv=sync bs=1K count=64 if=$< of=$@
+	@dd status=none if=$< of=$@
 
 primary.bin secondary.bin prstn.bin: pcie_phy_fw.bin
 
