From f0fa06325ff975f8658a7348bea1a81719f3465c Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Sun, 05 Aug 2018 09:52:29 +0300
Subject: [PATCH] Add dali packets support fixup

Update to new flash layout
---

diff --git a/board/habanalabs/dali/post-image.sh b/board/habanalabs/dali/post-image.sh
index 2788511..988a4e7 100755
--- a/board/habanalabs/dali/post-image.sh
+++ b/board/habanalabs/dali/post-image.sh
@@ -71,12 +71,14 @@
 	dd status=none conv=sync bs=64K		if=dali-ver.bin					>> dali-spi-flash.bin.nocrc
 	dd status=none conv=sync bs=64K		if=dali-next-image.bin				>> dali-spi-flash.bin.nocrc
 	dd status=none conv=sync bs=64K		if=dali-images-pointers.bin			>> dali-spi-flash.bin.nocrc
+	# U-Boot ENV
+	dd status=none conv=sync bs=64K		if=uboot-env.bin				>> dali-spi-flash.bin.nocrc
+	# U-Boot ENV Redundand
+	dd status=none conv=sync bs=64K		if=uboot-env.bin				>> dali-spi-flash.bin.nocrc
 	# SPL
 	dd status=none conv=sync bs=64K		if=${SPL_BINFILE}				>> dali-spi-flash.bin.nocrc
 	# U-Boot
 	dd status=none conv=sync bs=1M		if=${UBOOT_BINFILE}				>> dali-spi-flash.bin.nocrc
-	# U-Boot ENV
-	dd status=none conv=sync bs=64K		if=uboot-env.bin				>> dali-spi-flash.bin.nocrc
 	# PCIe FW
 	dd status=none conv=sync bs=64K		if=${PCIE_PHY_FW_BINFILE}			>> dali-spi-flash.bin.nocrc
 
