From b2ccdc700c31a4899194cfbde949c96cec5fa200 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Sun, 05 Aug 2018 11:30:31 +0300
Subject: [PATCH] Add dali packets support fixup

Fix bad bash syntax
---

diff --git a/board/habanalabs/common/rootfs_overlay/usr/sbin/flash_burn b/board/habanalabs/common/rootfs_overlay/usr/sbin/flash_burn
index 12f5481..fe05e59 100755
--- a/board/habanalabs/common/rootfs_overlay/usr/sbin/flash_burn
+++ b/board/habanalabs/common/rootfs_overlay/usr/sbin/flash_burn
@@ -65,7 +65,7 @@
 # 5 -- FW 1
 STABLE_IMAGE_CHAR=`xxd -p -l1 -c1 /dev/mtd2`
 # '1' / '2' equal (ascii) 0x31 / 0x32
-if ![ "${STABLE_IMAGE_CHAR}" -eq "31" ] && ![ "${STABLE_IMAGE_CHAR}" -eq "32" ]; then
+if [ "${STABLE_IMAGE_CHAR}" -ne "31" ] && [ "${STABLE_IMAGE_CHAR}" -ne "32" ]; then
 	STABLE_IMAGE_CHAR=31
 fi
 STABLE_IMAGE_NUM=$((0x${STABLE_IMAGE_CHAR} - 0x31))
