From 790250cbfdb423051fe7982b953242e6b0785f0f Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Mon, 06 Aug 2018 14:42:33 +0300
Subject: [PATCH] Add FW loader support fixup

Add missing install
---

diff --git a/Makefile b/Makefile
index 26e3c01..bdfbcea 100644
--- a/Makefile
+++ b/Makefile
@@ -95,6 +95,7 @@
 	-sshpass -p123456        scp $(IMAGES_DIR)/dali-u-boot.bin	root@$(DALI_INSTALL_HOST_IP):/lib/firmware/habanalabs/dali/
 	-sshpass -p123456        scp $(IMAGES_DIR)/dali-fit.itb		root@$(DALI_INSTALL_HOST_IP):/lib/firmware/habanalabs/dali/
 	-sshpass -p123456        scp $(IMAGES_DIR)/dali-spi-flash.bin	root@$(DALI_INSTALL_HOST_IP):/lib/firmware/habanalabs/dali/
+	-sshpass -p123456        scp $(IMAGES_DIR)/fw_loader		root@$(DALI_INSTALL_HOST_IP):/lib/firmware/habanalabs/dali/
 	-sshpass -p123456        scp $(IMAGES_DIR)/dali-u-boot.bin	root@$(DALI_INSTALL_LAB_IP):/lib/firmware/habanalabs/dali/
 	-sshpass -p123456        scp $(IMAGES_DIR)/dali-u-boot-spl.bin	root@$(DALI_INSTALL_LAB_IP):/lib/firmware/habanalabs/dali/
 	-sshpass -p123456        scp $(IMAGES_DIR)/dali-fit.itb		root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/
