From f9f32d15b5b687b90f474dd0a6a64556e6435514 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Wed, 08 Aug 2018 10:15:29 +0300
Subject: [PATCH] Add BTL support Fixup [use specs]

Now definitions from specs repository can be used
---

diff --git a/board/habanalabs/dali/btl/i2c.s b/board/habanalabs/dali/btl/i2c.s
index 8a1ae2f..8cca5ef 100644
--- a/board/habanalabs/dali/btl/i2c.s
+++ b/board/habanalabs/dali/btl/i2c.s
@@ -2,21 +2,23 @@
  * I2C Slave initialization
  */
 
+#include "psoc_i2c_s_regs.h"
+
 /* Program IC_SAR with address to which DW_apb_i2c slave responds */
-cfg 0xC42008, 0x47
+cfg mmPSOC_I2C_S_IC_SAR, 0x47
 
 /* Program IC_CON */
-cfg 0xC42000, 0x22
+cfg mmPSOC_I2C_S_IC_CON, 0x22
 
 /* Write to IC_INTR_MASK to enable all interrupts */
-cfg 0xC4202C, 0x7FFF
+cfg mmPSOC_I2C_S_IC_INTR_STAT, 0x7FFF
 
 /* Write to IC_RX_TL to set Rx FIFO threshold level */
-cfg 0xC42038, 0
+cfg mmPSOC_I2C_S_IC_RX_TL, 0
 
 /* Write to IC_TX_TL to set Tx FIFO threshold level */
-cfg 0xC4203C, 0x7
+cfg mmPSOC_I2C_S_IC_TX_TL, 0x7
 
 /* Write 1 to IC_ENABLE to enable DW_apb_i2c */
-cfg 0xC4206C, 0x1
+cfg mmPSOC_I2C_S_IC_ENABLE, 0x1
 
diff --git a/board/habanalabs/dali/btl/pcie.S b/board/habanalabs/dali/btl/pcie.S
index 678a634..f53fd46 100644
--- a/board/habanalabs/dali/btl/pcie.S
+++ b/board/habanalabs/dali/btl/pcie.S
@@ -2,8 +2,10 @@
  * This file is for PCIe controller initialization
  */
 
+#include "pcie_aux_regs.h"
+
 #ifdef TARGET_IS_PLDM
-cfg 0xC07428, 0x1		/* LTSSM_EN */
+cfg mmPCIE_AUX_LTSSM_EN, 0x1
 #endif
 
 end
diff --git a/board/habanalabs/dali/btl/pcie_phy.s b/board/habanalabs/dali/btl/pcie_phy.s
index a9f15ec..857aadf 100644
--- a/board/habanalabs/dali/btl/pcie_phy.s
+++ b/board/habanalabs/dali/btl/pcie_phy.s
@@ -2,54 +2,48 @@
  * This file is for PCIe PHY initialization
  */
 
+#include "pcie_phy_regs.h"
+#include "pcie_aux_regs.h"
+#include "pcie_wrap_regs.h"
+
 #ifndef TARGET_IS_PLDM
 /*
  * Set PCIe PHY CR Para I/F to CFG.
  * Select CR PARA I/F instead of JTAG
  */
-cfg 0xC10850, 0x00000001
-cfg 0xC10854, 0x00000001
-cfg 0xC10858, 0x00000001
-cfg 0xC1085C, 0x00000001
+cfg mmPCIE_PHY_CR_PARA_SEL_0, 0x00000001
+cfg mmPCIE_PHY_CR_PARA_SEL_1, 0x00000001
+cfg mmPCIE_PHY_CR_PARA_SEL_2, 0x00000001
+cfg mmPCIE_PHY_CR_PARA_SEL_3, 0x00000001
 
 /* Select indirect access to CR PARA I/F instead of PRBI */
-cfg 0xC10920, 0x00000001
+cfg mmPCIE_PHY_CR_PARA_CFG, 0x00000001
 
 /* Errata: disable unused design that cause PCIe phy SRAM DERR */
-cfg 0xC10840, 0x0
-#endif
+cfg mmPCIE_PHY_MEM_WR_MC, 0x0
 
 /* Adjust the incorrect PHY protocol0 defaults */
-/* writel(0xA067, mmPCIE_PHY_PROTOCOL0_EXT_MPLLA_BANDWIDTH); */
-cfg 0xC1010C, 0x0000A067
-/* writel(0x19, mmPCIE_PHY_PROTOCOL0_EXT_MPLLA_MULTIPLIER); */
-cfg 0xC10128, 0x00000019
-/* writel(0xA01E, mmPCIE_PHY_PROTOCOL0_EXT_MPLLB_BANDWIDTH); */
-cfg 0xC10140, 0x0000A01E
-/* writel(0x50, mmPCIE_PHY_PROTOCOL0_EXT_MPLLB_MULTIPLIER); */
-cfg 0xC10158, 0x00000050
-/* writel(0xE, mmPCIE_PHY_PROTOCOL0_EXT_RX_MISC_G1_SLICE0); */
-cfg 0xC1029C, 0x0000000E
-/* writel(0xC, mmPCIE_PHY_PROTOCOL0_EXT_RX_MISC_G2_SLICE0); */
-cfg 0xC102A4, 0x0000000C
-/* writel(0xA, mmPCIE_PHY_PROTOCOL0_EXT_RX_MISC_G3_SLICE0); */
-cfg 0xC102AC, 0x0000000A
-/* writel(0x9, mmPCIE_PHY_PROTOCOL0_EXT_RX_MISC_G4_SLICE0); */
-cfg 0xC102B4, 0x00000009
-/* writel(0x4, mmPCIE_PHY_PROTOCOL0_EXT_TX_VBOOST_LVL); */
-cfg 0xC10364, 0x00000004
-/* writel(0x1, mmPCIE_PHY_PHY_EXT_CTRL_SEL); */
-cfg 0xC10640, 0x00000001
+cfg mmPCIE_PHY_PROTOCOL0_EXT_MPLLA_BANDWIDTH, 0x0000A067
+cfg mmPCIE_PHY_PROTOCOL0_EXT_MPLLA_MULTIPLIER, 0x00000019
+cfg mmPCIE_PHY_PROTOCOL0_EXT_MPLLB_BANDWIDTH, 0x0000A01E
+cfg mmPCIE_PHY_PROTOCOL0_EXT_MPLLB_MULTIPLIER, 0x00000050
+cfg mmPCIE_PHY_PROTOCOL0_EXT_RX_MISC_G1_SLICE0, 0x0000000E
+cfg mmPCIE_PHY_PROTOCOL0_EXT_RX_MISC_G2_SLICE0, 0x0000000C
+cfg mmPCIE_PHY_PROTOCOL0_EXT_RX_MISC_G3_SLICE0, 0x0000000A
+cfg mmPCIE_PHY_PROTOCOL0_EXT_RX_MISC_G4_SLICE0, 0x00000009
+cfg mmPCIE_PHY_PROTOCOL0_EXT_TX_VBOOST_LVL, 0x00000004
+cfg mmPCIE_PHY_PHY_EXT_CTRL_SEL, 0x00000001
+#endif
 
 /* Clear phy_init (removes the workaround provided by SNPS) */
-cfg 0xC07100, 0x00000000
+cfg mmPCIE_AUX_PHY_INIT, 0x00000000
 
 /* Take PCIe PHY out of reset */
-cfg 0xC01300, 0x00000001
+cfg mmPCIE_WRAP_PHY_RST_N, 0x00000001
 
 #ifdef PRSTN
 /* Enable PCIe FW download FSM */
-cfg 0xC4B020, 0x00000001
+cfg mmPSOC_GLOBAL_CONF_PCI_FW_FSM, 0x00000001
 #endif
 
 #ifndef TARGET_IS_PLDM
diff --git a/board/habanalabs/dali/btl/pll_pci.s b/board/habanalabs/dali/btl/pll_pci.s
index 567e198..78b0013 100644
--- a/board/habanalabs/dali/btl/pll_pci.s
+++ b/board/habanalabs/dali/btl/pll_pci.s
@@ -1,30 +1,32 @@
 /* PLL PSOC PCIe setup */
 
-cfg 0xC72134, 0x1	/* WREG32(mmPSOC_PCI_PLL_RST, 1); */
-cfg 0xC72100, 0x4	/* WREG32(mmPSOC_PCI_PLL_NR, 4); */
-cfg 0xC72104, 0x157	/* WREG32(mmPSOC_PCI_PLL_NF, 343); */
-cfg 0xC72108, 0x3	/* WREG32(mmPSOC_PCI_PLL_OD, 3); */
-cfg 0xC7210C, 0xAB 	/* WREG32(mmPSOC_PCI_PLL_NB, 171); */
-cfg 0xC72130, 0x11	/* WREG32(mmPSOC_PCI_PLL_DATA_CHNG, 0x11); */
+#include "psoc_pci_pll_regs.h"
+
+cfg mmPSOC_PCI_PLL_RST, 0x1
+cfg mmPSOC_PCI_PLL_NR, 0x4
+cfg mmPSOC_PCI_PLL_NF, 0x157
+cfg mmPSOC_PCI_PLL_OD, 0x3
+cfg mmPSOC_PCI_PLL_NB, 0xAB
+cfg mmPSOC_PCI_PLL_DATA_CHNG, 0x11
 
 /* delay before taking PLL out of reset */
 #include "nop.s"	/* udelay(100); */
 
-cfg 0xC72134, 0	/* WREG32(mmPSOC_PCI_PLL_RST, 0); */
-cfg 0xC722A0, 0x1	/* WREG32(mmPSOC_PCI_PLL_DIV_EN_0, 0x1); */
-cfg 0xC72280, 0x1	/* WREG32(mmPSOC_PCI_PLL_DIV_SEL_0, 0x1); */
+cfg mmPSOC_PCI_PLL_RST, 0
+cfg mmPSOC_PCI_PLL_DIV_EN_0, 0x1
+cfg mmPSOC_PCI_PLL_DIV_SEL_0, 0x1
 
-cfg 0xC72204, 0x4	/* WREG32(mmPSOC_PCI_PLL_DIV_FACTOR_1, 0x4); */
-cfg 0xC72224, 0x1	/* WREG32(mmPSOC_PCI_PLL_DIV_FACTOR_CMD_1, 0x1); */
-cfg 0xC722A4, 0x1	/* WREG32(mmPSOC_PCI_PLL_DIV_EN_1, 0x1); */
-cfg 0xC72284, 0x3	/* WREG32(mmPSOC_PCI_PLL_DIV_SEL_1, 0x3); */
+cfg mmPSOC_PCI_PLL_DIV_FACTOR_1, 0x4
+cfg mmPSOC_PCI_PLL_DIV_FACTOR_CMD_1, 0x1
+cfg mmPSOC_PCI_PLL_DIV_EN_1, 0x1
+cfg mmPSOC_PCI_PLL_DIV_SEL_1, 0x3
 
-cfg 0xC72208, 0x8	/* WREG32(mmPSOC_PCI_PLL_DIV_FACTOR_2, 0x8); */
-cfg 0xC72228, 0x1	/* WREG32(mmPSOC_PCI_PLL_DIV_FACTOR_CMD_2, 0x1); */
-cfg 0xC722A8, 0x1	/* WREG32(mmPSOC_PCI_PLL_DIV_EN_2, 0x1); */
-cfg 0xC72288, 0x3	/* WREG32(mmPSOC_PCI_PLL_DIV_SEL_2, 0x3); */
+cfg mmPSOC_PCI_PLL_DIV_FACTOR_2, 0x8
+cfg mmPSOC_PCI_PLL_DIV_FACTOR_CMD_2, 0x1
+cfg mmPSOC_PCI_PLL_DIV_EN_2, 0x1
+cfg mmPSOC_PCI_PLL_DIV_SEL_2, 0x3
 
-cfg 0xC7220C, 0x55	/* WREG32(mmPSOC_PCI_PLL_DIV_FACTOR_3, 0x55); */
-cfg 0xC7222C, 0x1	/* WREG32(mmPSOC_PCI_PLL_DIV_FACTOR_CMD_3, 0x1); */
-cfg 0xC722AC, 0x1	/* WREG32(mmPSOC_PCI_PLL_DIV_EN_3, 0x1); */
-cfg 0xC7228C, 0x3	/* WREG32(mmPSOC_PCI_PLL_DIV_SEL_3, 0x3); */
+cfg mmPSOC_PCI_PLL_DIV_FACTOR_3, 0x55
+cfg mmPSOC_PCI_PLL_DIV_FACTOR_CMD_3, 0x1
+cfg mmPSOC_PCI_PLL_DIV_EN_3, 0x1
+cfg mmPSOC_PCI_PLL_DIV_SEL_3, 0x3
diff --git a/board/habanalabs/dali/btl/primary.S b/board/habanalabs/dali/btl/primary.S
index 05d8241..9a7e7ef 100644
--- a/board/habanalabs/dali/btl/primary.S
+++ b/board/habanalabs/dali/btl/primary.S
@@ -4,19 +4,23 @@
  * strict time regime of 10mS to establish link
  */
 
-nop				/* DUMMY */
+#include "psoc_global_conf_regs.h"
+#include "cpu_ca53_cfg_regs.h"
 
-/* CPU_BOOT_STATUS_IN_BTL -> mmPSOC_GLOBAL_CONF_WARM_REBOOT */
-cfg 0xC4B300, 0x4
+/* DUMMY */
+nop
 
-/* To avoid false prstn configurae pad as schmitt trigger */
-cfg 0xC4B668, 0x41
+/* Set CPU_BOOT_STATUS_IN_BTL */
+cfg mmPSOC_GLOBAL_CONF_WARM_REBOOT, 0x4
 
-/* Configure prstn to reset entire SoC */
-cfg 0xC4B470, 0x7FFFFF
+/* To avoid false PRSTN configure pad as Schmitt trigger */
+cfg mmPSOC_GLOBAL_CONF_PAD_3V3_CFG_10, 0x41
 
-/* Clear masking of prstn from SW reset */
-cfg 0xC4B484, 0x0
+/* Configure PRSTN to reset entire SoC */
+cfg mmPSOC_GLOBAL_CONF_PRSTN_RST_CFG, 0x7FFFFF
+
+/* Clear masking of PRSTN from SW reset */
+cfg mmPSOC_GLOBAL_CONF_PRSTN_MASK, 0x0
 
 #include "pcie_phy.s"
 
@@ -32,24 +36,19 @@
  * Using scratchpad registers for
  * passing flash layout to CPU SW
  * ------------------------------
- * 27 - BTL_ID
- * 28 - SPL_OFFSET
- * 29 - SPL_SIZE
- * 30 - UBOOT_OFFSET
- * 31 - UBOOT_SIZE
  */
-cfg 0xC4B16C, BTLID
-cfg 0xC4B170, SPL_OFFSET
-cfg 0xC4B174, SPL_SIZE
-cfg 0xC4B178, UBOOT_OFFSET
-cfg 0xC4B17C, UBOOT_SIZE
+cfg mmPSOC_GLOBAL_CONF_SCRATCHPAD_27, BTLID
+cfg mmPSOC_GLOBAL_CONF_SCRATCHPAD_28, SPL_OFFSET
+cfg mmPSOC_GLOBAL_CONF_SCRATCHPAD_29, SPL_SIZE
+cfg mmPSOC_GLOBAL_CONF_SCRATCHPAD_30, UBOOT_OFFSET
+cfg mmPSOC_GLOBAL_CONF_SCRATCHPAD_31, UBOOT_SIZE
 
 /* Set CPU0 Reset address to PREBOOT */
-cfg 0x441104, PREBOOT_ADDR_L
-cfg 0x441114, PREBOOT_ADDR_H
+cfg mmCPU_CA53_CFG_RST_ADDR_LSB_0, PREBOOT_ADDR_L
+cfg mmCPU_CA53_CFG_RST_ADDR_MSB_0, PREBOOT_ADDR_H
 
 /* Wake up CPU0, after we put all CPUs in reset */
-cfg 0x441124, 0x0
-cfg 0x441124, 0x10111
+cfg mmCPU_CA53_CFG_ARM_RST_CONTROL, 0x0
+cfg mmCPU_CA53_CFG_ARM_RST_CONTROL, 0x10111
 
 end
diff --git a/board/habanalabs/dali/btl/uart.s b/board/habanalabs/dali/btl/uart.s
index f5ecfb9..d418ea5 100644
--- a/board/habanalabs/dali/btl/uart.s
+++ b/board/habanalabs/dali/btl/uart.s
@@ -5,22 +5,25 @@
  * Using PLDM for creating VCDs to TESTER is an exception
  */
 
+#include "psoc_uart_0_regs.h"
+
 #if defined(TARGET_IS_PLDM) && !defined(TARGET_IS_TESTER)
 #include "pll_pci.s"
 #endif
 
-cfg 0xC4B9C0, 0x1	/* GPIO48 [UART0 RXD] */
-cfg 0xC4B9C4, 0x1	/* GPIO49 [UART0 TXD] */
-cfg 0xC45008, 0x0	/* UART0  [CTRL]      */
+cfg mmPSOC_GLOBAL_CONF_PAD_SEL_48, 0x1	/* UART0 RXD */
+cfg mmPSOC_GLOBAL_CONF_PAD_SEL_49, 0x1	/* UART0 TXD */
+cfg mmPSOC_UART_0_CTRL, 0x0
 #ifdef TARGET_IS_PLDM
-cfg 0xC45010, 0x17	/* UART0  [BAUDDIV]   */
+cfg mmPSOC_UART_0_BAUDDIV, 0x17
 #elif TARGET_IS_ASIC
-cfg 0xC45010, 0x1B2	/* UART0  [BAUDDIV]   */
+cfg mmPSOC_UART_0_BAUDDIV, 0x1B2
 #else
-cfg 0xC45010, 0x364	/* UART0  [BAUDDIV]   */
+cfg mmPSOC_UART_0_BAUDDIV, 0x364
 #endif
-cfg 0xC45008, 0x3	/* UART0  [CTRL]      */
+cfg mmPSOC_UART_0_CTRL, 0x3
 
-cfg 0xC45000, BTLID	/* UART0  [DATA]      */
-cfg 0xC45000, 0X0A	/* UART0  [DATA]      */
-cfg 0xC45000, 0X0D	/* UART0  [DATA]      */
+/* log BTLID\n\r */
+cfg mmPSOC_UART_0_DATA, BTLID
+cfg mmPSOC_UART_0_DATA, 0X0A
+cfg mmPSOC_UART_0_DATA, 0X0D
