From 7123e34ff14cfd997ddd4e86d671f474c103c2c4 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Thu, 09 Aug 2018 11:16:30 +0300
Subject: [PATCH] Revert "Add BTL support Fixup [PCIe PHY FW]"

This reverts commit a1f075334d6bbfb476cfaf5be0801b0072d01b8b.
---

diff --git a/board/habanalabs/dali/btl/Makefile b/board/habanalabs/dali/btl/Makefile
index 95f5fab..807429b 100644
--- a/board/habanalabs/dali/btl/Makefile
+++ b/board/habanalabs/dali/btl/Makefile
@@ -71,9 +71,7 @@
 	@gcc -E -P -nostdinc $(SECONDARY_CPP_FLAGS) -DBTLID=0x32 -o $@ $<
 
 pcie_phy_fw.bin: pcie_phy_fw.i
-	@cut -d' ' -f3- $< | xxd -r -p | dd status=none bs=16 conv=sync,swab of=$@.full
-	@# Run time optimization: Only first 12KB are used (rest is zeroes)
-	@dd status=none bs=16 count=$$((12 * 1024 / 16)) if=$@.full of=$@
+	@cut -d' ' -f3- $< | xxd -r -p | dd status=none bs=16 conv=sync,swab of=$@
 
 pcie_phy_fw.i: pcie_phy_fw.s
 	@gcc -x c -E -P -nostdinc -o $@ $<
