From 6ab1229e46b0eeb5cb3dd01665497e5878e5834e Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Thu, 09 Aug 2018 11:21:46 +0300
Subject: [PATCH] Add BTL support Fixup [PCIe PHY FW]

We revert to old CPLD
---

diff --git a/board/habanalabs/dali/btl/pcie_phy.s b/board/habanalabs/dali/btl/pcie_phy.s
index 857aadf..08000b4 100644
--- a/board/habanalabs/dali/btl/pcie_phy.s
+++ b/board/habanalabs/dali/btl/pcie_phy.s
@@ -41,10 +41,8 @@
 /* Take PCIe PHY out of reset */
 cfg mmPCIE_WRAP_PHY_RST_N, 0x00000001
 
-#ifdef PRSTN
 /* Enable PCIe FW download FSM */
 cfg mmPSOC_GLOBAL_CONF_PCI_FW_FSM, 0x00000001
-#endif
 
 #ifndef TARGET_IS_PLDM
 mem 0x07FF9000000, pcie_phy_fw.bin
