From d727802b29d1f004d2dba901e65b20d39fc5f497 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 07 Aug 2018 11:19:56 +0300
Subject: [PATCH] board: habana: finish rework the remote/local scripts

Actually add the scripts to the build.

Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/Makefile b/Makefile
index b7ceffd..19ed016 100644
--- a/Makefile
+++ b/Makefile
@@ -181,7 +181,7 @@
 .PHONY: dali-asic-tools-install
 dali-asic-tools-install:
 	-cd board/habanalabs/common/ &&					\
-	sshpass -p123456 scp -r remote_power remote_load_firmware	\
-	remote_cleanup local_install_packages ip_pdu t32 vivado_lab	\
+	sshpass -p123456 scp -r remote_power ip_pdu t32 vivado_lab	\
+	local_remote_common local_run remote_run			\
 	$(IMAGES_DIR)/fw_loader						\
 	root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
