From 6a3c066f5b6ddf8409c78090b9fa9a44648353aa Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Sat, 11 Aug 2018 23:00:21 +0300
Subject: [PATCH] hl: preperation to Gaudi step 3

rename R2_EXTERNAL_HL_DALI_PATH --> BR2_EXTERNAL_HABANALABS_PATH
---

diff --git a/Config.in b/Config.in
index d3e38b6..38e508f 100644
--- a/Config.in
+++ b/Config.in
@@ -1,14 +1,14 @@
-source "$BR2_EXTERNAL_HL_DALI_PATH/package/armcp/Config.in"
-source "$BR2_EXTERNAL_HL_DALI_PATH/package/specs/Config.in"
-source "$BR2_EXTERNAL_HL_DALI_PATH/package/spdlog/Config.in"
-source "$BR2_EXTERNAL_HL_DALI_PATH/package/specs_external/Config.in"
-source "$BR2_EXTERNAL_HL_DALI_PATH/package/uboot-test-hooks/Config.in"
-source "$BR2_EXTERNAL_HL_DALI_PATH/package/tbot/Config.in"
-source "$BR2_EXTERNAL_HL_DALI_PATH/package/dsv/Config.in"
-source "$BR2_EXTERNAL_HL_DALI_PATH/package/thermal/Config.in"
-source "$BR2_EXTERNAL_HL_DALI_PATH/package/preboot/Config.in"
+source "$BR2_EXTERNAL_HABANALABS_PATH/package/armcp/Config.in"
+source "$BR2_EXTERNAL_HABANALABS_PATH/package/specs/Config.in"
+source "$BR2_EXTERNAL_HABANALABS_PATH/package/spdlog/Config.in"
+source "$BR2_EXTERNAL_HABANALABS_PATH/package/specs_external/Config.in"
+source "$BR2_EXTERNAL_HABANALABS_PATH/package/uboot-test-hooks/Config.in"
+source "$BR2_EXTERNAL_HABANALABS_PATH/package/tbot/Config.in"
+source "$BR2_EXTERNAL_HABANALABS_PATH/package/dsv/Config.in"
+source "$BR2_EXTERNAL_HABANALABS_PATH/package/thermal/Config.in"
+source "$BR2_EXTERNAL_HABANALABS_PATH/package/preboot/Config.in"
 
 # Source the host tools
-source "$BR2_EXTERNAL_HL_DALI_PATH/Config.in.host"
+source "$BR2_EXTERNAL_HABANALABS_PATH/Config.in.host"
 
 # Below can add to make it appear in the top-level configuration menu
diff --git a/Config.in.host b/Config.in.host
index e3e57bc..dca3277 100644
--- a/Config.in.host
+++ b/Config.in.host
@@ -1,3 +1,3 @@
 menu "External Host utilities"
-	source "$BR2_EXTERNAL_HL_DALI_PATH/package/fw_loader/Config.in.host"
+	source "$BR2_EXTERNAL_HABANALABS_PATH/package/fw_loader/Config.in.host"
 endmenu
diff --git a/board/habanalabs/common/post-image.sh b/board/habanalabs/common/post-image.sh
index b4caf7a..0ded09b 100755
--- a/board/habanalabs/common/post-image.sh
+++ b/board/habanalabs/common/post-image.sh
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 
-BOARD_BASE=${BR2_EXTERNAL_HL_DALI_PATH}/board/habanalabs/${HL_DEV}
+BOARD_BASE=${BR2_EXTERNAL_HABANALABS_PATH}/board/habanalabs/${HL_DEV}
 SPL_BINFILE=${SPL_BINFILE:-u-boot-spl.bin}
 UBOOT_CMDFILE=${UBOOT_CMDFILE:-${BOARD_BASE}/uboot-eeprom.cmd}
 UBOOT_FITFILE=u-boot.itb
diff --git a/configs/dali_defconfig b/configs/dali_defconfig
index 7c41127..4e6caf4 100644
--- a/configs/dali_defconfig
+++ b/configs/dali_defconfig
@@ -1,7 +1,7 @@
 BR2_aarch64=y
 BR2_SVN="svn"
-BR2_DL_DIR="$(BR2_EXTERNAL_HL_DALI_PATH)/dl"
-BR2_PACKAGE_OVERRIDE_FILE="$(BR2_EXTERNAL_HL_DALI_PATH)/board/habanalabs/common/local.mk"
+BR2_DL_DIR="$(BR2_EXTERNAL_HABANALABS_PATH)/dl"
+BR2_PACKAGE_OVERRIDE_FILE="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/local.mk"
 BR2_TOOLCHAIN_EXTERNAL=y
 BR2_TARGET_GENERIC_HOSTNAME="$(HL_DEV)"
 BR2_TARGET_GENERIC_ISSUE="Welcome to Dali"
@@ -9,18 +9,18 @@
 BR2_TARGET_GENERIC_ROOT_PASSWD="root"
 BR2_SYSTEM_BIN_SH_BASH=y
 BR2_SYSTEM_DHCP="eth0"
-BR2_ROOTFS_OVERLAY="$(BR2_EXTERNAL_HL_DALI_PATH)/board/habanalabs/common/rootfs_overlay"
-BR2_ROOTFS_POST_BUILD_SCRIPT="$(BR2_EXTERNAL_HL_DALI_PATH)/board/habanalabs/common/post-build.sh"
-BR2_ROOTFS_POST_IMAGE_SCRIPT="$(BR2_EXTERNAL_HL_DALI_PATH)/board/habanalabs/common/post-image.sh"
+BR2_ROOTFS_OVERLAY="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/rootfs_overlay"
+BR2_ROOTFS_POST_BUILD_SCRIPT="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/post-build.sh"
+BR2_ROOTFS_POST_IMAGE_SCRIPT="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/post-image.sh"
 BR2_LINUX_KERNEL=y
 BR2_LINUX_KERNEL_USE_ARCH_DEFAULT_CONFIG=y
-BR2_LINUX_KERNEL_CONFIG_FRAGMENT_FILES="$(BR2_EXTERNAL_HL_DALI_PATH)/board/habanalabs/$(HL_DEV)/linux_$(HL_TARGET).fragment"
+BR2_LINUX_KERNEL_CONFIG_FRAGMENT_FILES="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/linux_$(HL_TARGET).fragment"
 BR2_LINUX_KERNEL_IMAGE_TARGET_CUSTOM=y
 BR2_LINUX_KERNEL_IMAGE_TARGET_NAME="Image.gz"
 BR2_LINUX_KERNEL_DTS_SUPPORT=y
 BR2_LINUX_KERNEL_INTREE_DTS_NAME="habanalabs/$(HL_DEV)_$(HL_TARGET)"
 BR2_PACKAGE_BUSYBOX_CONFIG="package/busybox/busybox-minimal.config"
-BR2_PACKAGE_BUSYBOX_CONFIG_FRAGMENT_FILES="$(BR2_EXTERNAL_HL_DALI_PATH)/board/habanalabs/common/busybox.fragment"
+BR2_PACKAGE_BUSYBOX_CONFIG_FRAGMENT_FILES="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/busybox.fragment"
 BR2_PACKAGE_BUSYBOX_SHOW_OTHERS=y
 BR2_PACKAGE_BUSYBOX_WATCHDOG=y
 BR2_PACKAGE_MTD=y
@@ -49,11 +49,11 @@
 BR2_TARGET_UBOOT=y
 BR2_TARGET_UBOOT_BUILD_SYSTEM_KCONFIG=y
 BR2_TARGET_UBOOT_BOARD_DEFCONFIG="$(HL_DEV)"
-BR2_TARGET_UBOOT_CONFIG_FRAGMENT_FILES="$(BR2_EXTERNAL_HL_DALI_PATH)/board/habanalabs/$(HL_DEV)/uboot_$(HL_TARGET).fragment"
+BR2_TARGET_UBOOT_CONFIG_FRAGMENT_FILES="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/uboot_$(HL_TARGET).fragment"
 BR2_TARGET_UBOOT_NEEDS_DTC=y
 BR2_TARGET_UBOOT_SPL=y
 BR2_TARGET_UBOOT_ENVIMAGE=y
-BR2_TARGET_UBOOT_ENVIMAGE_SOURCE="$(BR2_EXTERNAL_HL_DALI_PATH)/board/habanalabs/$(HL_DEV)/uboot-env-$(HL_TARGET).txt"
+BR2_TARGET_UBOOT_ENVIMAGE_SOURCE="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/uboot-env-$(HL_TARGET).txt"
 BR2_TARGET_UBOOT_ENVIMAGE_SIZE="0x1000"
 BR2_TARGET_UBOOT_ENVIMAGE_REDUNDANT=y
 BR2_PACKAGE_HOST_UBOOT_TOOLS=y
diff --git a/external.desc b/external.desc
index b122f98..4d6799c 100644
--- a/external.desc
+++ b/external.desc
@@ -1,2 +1,2 @@
-name: HL_DALI
-desc: br2-external tree for Habanalabs Dali
+name: HABANALABS
+desc: br2-external tree for Habanalabs product lines
diff --git a/external.mk b/external.mk
index eed6170..05c2e58 100644
--- a/external.mk
+++ b/external.mk
@@ -1,3 +1,3 @@
-include $(sort $(wildcard $(BR2_EXTERNAL_HL_DALI_PATH)/package/*/*.mk))
+include $(sort $(wildcard $(BR2_EXTERNAL_HABANALABS_PATH)/package/*/*.mk))
 
 # Below can add any target to be part of Makefile logic
