From c129c7cd3eac05568a5b088ff7334c04335e1baf Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Sun, 12 Aug 2018 16:03:49 +0300
Subject: [PATCH] board: gaudi is a new ARM64 SoC Fixup

Fix make dependency Goals
---

diff --git a/Makefile b/Makefile
index 95b40cf..b4e4d63 100644
--- a/Makefile
+++ b/Makefile
@@ -105,7 +105,8 @@
 dali-pldm-install: DALI_INSTALL_HOST_IP ?= 192.168.10.45
 dali-asic-install: DALI_INSTALL_LAB_IP ?= 192.168.10.109
 dali-asic-install: DALI_INSTALL_HOST_IP ?= 192.168.10.134
-dali-onix-install gaudi-onix-install: dali-onix-target
+gaudi-onix-install: gaudi-onix-target
+dali-onix-install: dali-onix-target
 dali-pldm-install: dali-pldm-target
 dali-asic-install: dali-asic-target
 
