From a32e0e54b786c4d77cd81a9872e96f46148e832d Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 13 Aug 2018 14:55:09 +0300
Subject: [PATCH] FW loader: rename to hl-fw-loader

This better defines/suits our naming conventions.

Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/Config.in.host b/Config.in.host
index dca3277..0e477d0 100644
--- a/Config.in.host
+++ b/Config.in.host
@@ -1,3 +1,3 @@
 menu "External Host utilities"
-	source "$BR2_EXTERNAL_HABANALABS_PATH/package/fw_loader/Config.in.host"
+	source "$BR2_EXTERNAL_HABANALABS_PATH/package/hl-fw-loader/Config.in.host"
 endmenu
diff --git a/Makefile b/Makefile
index b4e4d63..293b751 100644
--- a/Makefile
+++ b/Makefile
@@ -115,12 +115,12 @@
 	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-u-boot.bin		root@$(DALI_INSTALL_HOST_IP):/lib/firmware/habanalabs/$(HL_DEV)/
 	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-fit.itb		root@$(DALI_INSTALL_HOST_IP):/lib/firmware/habanalabs/$(HL_DEV)/
 	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-spi-flash.bin	root@$(DALI_INSTALL_HOST_IP):/lib/firmware/habanalabs/$(HL_DEV)/
-	-sshpass -p123456        scp $(IMAGES_DIR)/fw_loader			root@$(DALI_INSTALL_HOST_IP):/lib/firmware/habanalabs/$(HL_DEV)/
+	-sshpass -p123456        scp $(IMAGES_DIR)/hl-fw-loader			root@$(DALI_INSTALL_HOST_IP):/lib/firmware/habanalabs/$(HL_DEV)/
 	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-u-boot.bin		root@$(DALI_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
 	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-u-boot-spl.bin	root@$(DALI_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
 	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-fit.itb		root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/
 	-sshpass -p123456        scp $(IMAGES_DIR)/boot.scr			root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
-	-sshpass -p123456        scp $(IMAGES_DIR)/fw_loader			root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
+	-sshpass -p123456        scp $(IMAGES_DIR)/hl-fw-loader			root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
 	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-spi-flash.bin	root@$(DALI_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
 	-sshpass -p123456        scp $(UBOOT_BASE)/u-boot			root@$(DALI_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
 	-sshpass -p123456        scp $(LINUX_BASE)/vmlinux			root@$(DALI_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
@@ -150,7 +150,7 @@
 	-cp $(IMAGES_DIR)/$(HL_DEV)-spi-flash.bin $</
 	-cp $(IMAGES_DIR)/$(HL_DEV)-u-boot.bin $</
 	-cp $(IMAGES_DIR)/$(HL_DEV)-u-boot-spl.bin $</
-	-cp $(IMAGES_DIR)/fw_loader $</
+	-cp $(IMAGES_DIR)/hl-fw-loader $</
 ifneq ($(notdir $(realpath $(ASIC_BIN_DIR))), head)
 	ln -nfs $$(basename $(ASIC_BIN_DIR)) $$(dirname $(ASIC_BIN_DIR))/latest
 endif
@@ -211,5 +211,5 @@
 	-cd board/habanalabs/common/ &&					\
 	sshpass -p123456 scp -r remote_power ip_pdu t32 vivado_lab	\
 	local_remote_common local_run remote_run			\
-	$(IMAGES_DIR)/fw_loader						\
+	$(IMAGES_DIR)/hl-fw-loader					\
 	root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
diff --git a/board/habanalabs/common/local.mk b/board/habanalabs/common/local.mk
index 775db55..56d5244 100644
--- a/board/habanalabs/common/local.mk
+++ b/board/habanalabs/common/local.mk
@@ -9,4 +9,4 @@
 DSV_OVERRIDE_SRCDIR = $(TOPDIR)/../dsv
 THERMAL_OVERRIDE_SRCDIR = $(TOPDIR)/../thermal
 PREBOOT_OVERRIDE_SRCDIR = $(TOPDIR)/../preboot
-FW_LOADER_OVERRIDE_SRCDIR = $(TOPDIR)/../fw_loader
+HL_FW_LOADER_OVERRIDE_SRCDIR = $(TOPDIR)/../fw_loader
diff --git a/board/habanalabs/common/local_run b/board/habanalabs/common/local_run
index f813477..4d59093 100755
--- a/board/habanalabs/common/local_run
+++ b/board/habanalabs/common/local_run
@@ -7,7 +7,7 @@
 
 pre_pkg_lst="dkms"
 habana_pkg_lst="habanalabs"
-fw_ldr=$HOME/fw_loader
+fw_ldr=$HOME/hl-fw-loader
 
 source local_remote_common
 
diff --git a/board/habanalabs/common/remote_run b/board/habanalabs/common/remote_run
index 3821ae9..075230c 100755
--- a/board/habanalabs/common/remote_run
+++ b/board/habanalabs/common/remote_run
@@ -11,7 +11,7 @@
 fw_dir=/lib/firmware/habanalabs/dali
 local_scr_dir=/var/lib/tftpboot/tests
 scr_name=(local_run local_remote_common)
-fw_ldr=fw_loader
+fw_ldr=hl-fw-loader
 remote_scr_dir="."
 
 source $local_scr_dir/local_remote_common
diff --git a/configs/dali_defconfig b/configs/dali_defconfig
index 4e6caf4..25adc1a 100644
--- a/configs/dali_defconfig
+++ b/configs/dali_defconfig
@@ -64,4 +64,4 @@
 BR2_PACKAGE_DSV=y
 BR2_PACKAGE_THERMAL=y
 BR2_PACKAGE_PREBOOT=y
-BR2_PACKAGE_HOST_FW_LOADER=y
+BR2_PACKAGE_HOST_HL_FW_LOADER=y
diff --git a/configs/gaudi_defconfig b/configs/gaudi_defconfig
index b71ef9e..831893e 100644
--- a/configs/gaudi_defconfig
+++ b/configs/gaudi_defconfig
@@ -64,4 +64,4 @@
 BR2_PACKAGE_DSV=y
 BR2_PACKAGE_THERMAL=y
 BR2_PACKAGE_PREBOOT=y
-BR2_PACKAGE_HOST_FW_LOADER=y
+BR2_PACKAGE_HOST_HL_FW_LOADER=y
diff --git a/package/fw_loader/Config.in.host b/package/fw_loader/Config.in.host
deleted file mode 100644
index 78f2da9..0000000
--- a/package/fw_loader/Config.in.host
+++ /dev/null
@@ -1,7 +0,0 @@
-config BR2_PACKAGE_HOST_FW_LOADER
-	bool "FW Loader"
-	select BR2_PACKAGE_SPECS
-	select BR2_PACKAGE_SPDLOG
-	help
-	  FW loader is an SPI flash burning tool.
-	  Run on the host to update the SPI flash (U-Boot) of the Dali board.
diff --git a/package/fw_loader/fw_loader.mk b/package/fw_loader/fw_loader.mk
deleted file mode 100644
index fb3521b..0000000
--- a/package/fw_loader/fw_loader.mk
+++ /dev/null
@@ -1,13 +0,0 @@
-FW_LOADER_VERSION = 0.1.0
-FW_LOADER_SITE = fw_loader
-FW_LOADER_LICENSE = unclear
-
-HOST_FW_LOADER_DEPENDENCIES = specs spdlog
-
-HOST_FW_LOADER_CONF_OPTS = -DTESTS_ENABLED=OFF -DCMAKE_EXE_LINKER_FLAGS="-static" -DCMAKE_BUILD_TYPE="Release"
-
-define HOST_FW_LOADER_INSTALL_CMDS
-        $(INSTALL) -s -D -m 0755 $(@D)/bin/fw_loader $(BINARIES_DIR)/
-endef
-
-$(eval $(host-cmake-package))
diff --git a/package/hl-fw-loader/Config.in.host b/package/hl-fw-loader/Config.in.host
new file mode 100644
index 0000000..08ffcae
--- /dev/null
+++ b/package/hl-fw-loader/Config.in.host
@@ -0,0 +1,7 @@
+config BR2_PACKAGE_HOST_HL_FW_LOADER
+	bool "Habanalabs FW Loader"
+	select BR2_PACKAGE_SPECS
+	select BR2_PACKAGE_SPDLOG
+	help
+	  Habanalabs FW loader is an SPI flash burning tool.
+	  Run on the host to update the SPI flash (U-Boot) of the Habanalabs boards.
diff --git a/package/hl-fw-loader/hl-fw-loader.mk b/package/hl-fw-loader/hl-fw-loader.mk
new file mode 100644
index 0000000..638555b
--- /dev/null
+++ b/package/hl-fw-loader/hl-fw-loader.mk
@@ -0,0 +1,13 @@
+HL_FW_LOADER_VERSION = 0.1.0
+HL_FW_LOADER_SITE = hl-fw-loader
+HL_FW_LOADER_LICENSE = unclear
+
+HOST_HL_FW_LOADER_DEPENDENCIES = specs spdlog
+
+HOST_HL_FW_LOADER_CONF_OPTS = -DTESTS_ENABLED=OFF -DCMAKE_EXE_LINKER_FLAGS="-static" -DCMAKE_BUILD_TYPE="Release"
+
+define HOST_HL_FW_LOADER_INSTALL_CMDS
+        $(INSTALL) -s -D -m 0755 $(@D)/bin/hl-fw-loader $(BINARIES_DIR)/
+endef
+
+$(eval $(host-cmake-package))
