From 6b064293aec5cae4fa730d3d24259a2626dfbbc2 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 07 Aug 2018 11:19:56 +0300
Subject: [PATCH] board: habana: clean the local_run script

Remove the unused (commented out) code.

Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/board/habanalabs/common/local_run b/board/habanalabs/common/local_run
index 4d59093..3a70542 100755
--- a/board/habanalabs/common/local_run
+++ b/board/habanalabs/common/local_run
@@ -50,20 +50,13 @@
 }
 
 function local_program_flash() {
-#	local hl_vid=1da3
-#	local hl_did=0001
-#	local hl_slot=$(lspci -Dd ${hl_vid}:${hl_did} | cut -d\  -f1)
-#	local pci_rescan=/sys/bus/pci/rescan
-#	local hl_remove=/sys/bus/pci/devices/${hl_slot}/remove
 	local hl_syspath=/sys/class/habanalabs
 
 	print_msg "Programming SPI flash"
 	sudo $fw_ldr || return 1;
 	sleep 5;
-#	sudo bash -c "echo 1 > $hl_remove" || return 1;
 	sudo modprobe -r habanalabs
 	sleep 5;
-#	sudo bash -c "echo 1 > $pci_rescan" || return 1;
 	sudo modprobe habanalabs "heartbeat=0"
 	print_msg "New SPI flash FW versions"
 	sleep 5;
