From 3f7857b1fee37659ba6af3df04d51247d90fbe7c Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 14 Aug 2018 16:51:24 +0300
Subject: [PATCH] btl: give uart more time to send the '\n'

Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/board/habanalabs/common/btl/uart.s b/board/habanalabs/common/btl/uart.s
index 79dc4f9..f00b143 100644
--- a/board/habanalabs/common/btl/uart.s
+++ b/board/habanalabs/common/btl/uart.s
@@ -30,6 +30,11 @@
 nop
 nop
 nop
+nop
+nop
+nop
+nop
+nop
 /* #include "nop.s" */
 cfg mmPSOC_UART_0_DATA, BTLID
 cfg mmPSOC_UART_0_DATA, 0X0A
