From 7263197baa139ecea80ece4ed63d8abd95d82876 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Thu, 16 Aug 2018 22:08:28 +0300
Subject: [PATCH] Makefile [hl-dali/hl-gaudi]: update goal dependency

After distclean make sure we got $(OUTPUT_DIR)
---

diff --git a/Makefile b/Makefile
index 9d5d94b..127ace8 100644
--- a/Makefile
+++ b/Makefile
@@ -49,7 +49,7 @@
 endif
 
 .PHONY: hl-dali hl-gaudi
-hl-dali hl-gaudi:
+hl-dali hl-gaudi: $(OUTPUT_DIR)
 	@echo "export HL_DEV=$(patsubst hl-%,%,$@)"		 > $(HL_VARS_FILE)
 
 .PHONY: gaudi-onix-target
