From 38f8eaf25336b1198ffce0cd7c9a11098fc9bdb7 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Fri, 17 Aug 2018 08:41:12 +0300
Subject: [PATCH] btl: uart: remove commented out code

This comment has sneaked in by mistake. Remove it.

Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/board/habanalabs/common/btl/uart.s b/board/habanalabs/common/btl/uart.s
index f00b143..d06b846 100644
--- a/board/habanalabs/common/btl/uart.s
+++ b/board/habanalabs/common/btl/uart.s
@@ -35,7 +35,6 @@
 nop
 nop
 nop
-/* #include "nop.s" */
 cfg mmPSOC_UART_0_DATA, BTLID
 cfg mmPSOC_UART_0_DATA, 0X0A
 cfg mmPSOC_UART_0_DATA, 0X0D
