From 52ccb974437f6add0a2a4dd7aee6c4de8f912869 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Fri, 17 Aug 2018 13:23:55 +0300
Subject: [PATCH] board: habana: add empty line

This improves a bit the output readability.

Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/board/habanalabs/common/ip_pdu b/board/habanalabs/common/ip_pdu
index 1cf0479..ec9e369 100755
--- a/board/habanalabs/common/ip_pdu
+++ b/board/habanalabs/common/ip_pdu
@@ -44,6 +44,7 @@
 	;;
 on | off)
 	wget -q -O - $pdu_cred ${pdu_uri}/${action}s.cgi?led=$set_id
+	echo ""
 	status=3 # D_ON=2/D_OFF=3
 	while [ $status -gt 1 ]; do
 		sleep 1
