From 01ebf719e69eb2d6dad60e63ef644e2203558d2e Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Mon, 20 Aug 2018 09:45:29 +0300
Subject: [PATCH] Gaudi [tcl] update vivado script

To support latest dspg image for ONIX
---

diff --git a/board/habanalabs/common/vivado_lab/program-dspg.tcl b/board/habanalabs/common/vivado_lab/program-dspg.tcl
index 422a095..57a8858 100644
--- a/board/habanalabs/common/vivado_lab/program-dspg.tcl
+++ b/board/habanalabs/common/vivado_lab/program-dspg.tcl
@@ -4,13 +4,13 @@
 # DALI:
 #set_property PROGRAM.FILE {/opt/Xilinx/Vivado_Lab/h2_cpu_top_onix_dspg_070618.bit} [get_hw_devices xcvu440_0]
 # GAUDI:
-set_property PROGRAM.FILE {/opt/Xilinx/Vivado_Lab/h3_top_130818.bit} [get_hw_devices xcvu440_0]
+set_property PROGRAM.FILE {/opt/Xilinx/Vivado_Lab/h3_top_200818.bit} [get_hw_devices xcvu440_0]
 current_hw_device [get_hw_devices xcvu440_0]
 refresh_hw_device [lindex [get_hw_devices xcvu440_0] 0]
 # DALI:
 # set_property PROGRAM.FILE {/opt/Xilinx/Vivado_Lab/h2_cpu_top_onix_dspg_070618.bit} [get_hw_devices xcvu440_0]
 # GAUDI:
-set_property PROGRAM.FILE {/opt/Xilinx/Vivado_Lab/h3_top_130818.bit} [get_hw_devices xcvu440_0]
+set_property PROGRAM.FILE {/opt/Xilinx/Vivado_Lab/h3_top_200818.bit} [get_hw_devices xcvu440_0]
 program_hw_devices [get_hw_devices xcvu440_0]
 refresh_hw_device [lindex [get_hw_devices xcvu440_0] 0]
 exit
