From 2d1ca1ca09c95803c5643378459096d79bfd7845 Mon Sep 17 00:00:00 2001
From: Oded Gabbay <oded.gabbay@gmail.com>
Date: Wed, 22 Aug 2018 14:49:48 +0300
Subject: [PATCH] [SW-1525] final rename - dali folder to goya

Signed-off-by: Oded Gabbay <oded.gabbay@gmail.com>
---

diff --git a/board/habanalabs/common/btl/Makefile b/board/habanalabs/common/btl/Makefile
index 1a3e880..3c19fc1 100644
--- a/board/habanalabs/common/btl/Makefile
+++ b/board/habanalabs/common/btl/Makefile
@@ -8,11 +8,12 @@
 COMMON_CPP_FLAGS += -DPREBOOT_ADDR_H=0x7F
 COMMON_CPP_FLAGS += -DPREBOOT_ADDR_L=0xF0000000
 
-COMMON_CPP_FLAGS += -I$(dir $(BR2_EXTERNAL))/specs/$(HL_DEV)/asic_reg
 ifeq ($(HL_DEV),dali)
+	COMMON_CPP_FLAGS += -I$(dir $(BR2_EXTERNAL))/specs/goya/asic_reg
 	COMMON_CPP_FLAGS += -DDEV_IS_DALI
 	FW_SIZE=0x8000
 else
+	COMMON_CPP_FLAGS += -I$(dir $(BR2_EXTERNAL))/specs/gaudi/asic_reg
 	COMMON_CPP_FLAGS += -DDEV_IS_GAUDI
 	FW_SIZE=0x3000
 	COMMON_CPP_FLAGS += -DFW_SIZE=$(FW_SIZE)
