From 798a013dee5381a419a7085bf1d62b5c9c9b3520 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Thu, 23 Aug 2018 14:24:45 +0300
Subject: [PATCH] hl [linux fragments]: add CONFIG_PM_DEVFREQ_EVENT

Now Linux driver will be compiled and used.
---

diff --git a/board/habanalabs/dali/linux_asic.fragment b/board/habanalabs/dali/linux_asic.fragment
index d6c797f..ba2c3cc 100644
--- a/board/habanalabs/dali/linux_asic.fragment
+++ b/board/habanalabs/dali/linux_asic.fragment
@@ -13,6 +13,7 @@
 CONFIG_THERMAL=y
 CONFIG_DEVFREQ_THERMAL=y
 CONFIG_PM_DEVFREQ=y
+CONFIG_PM_DEVFREQ_EVENT=y
 CONFIG_DEVFREQ_GOV_SIMPLE_ONDEMAND=y
 CONFIG_SPI_DESIGNWARE=y
 CONFIG_SPI_DW_MMIO=y
diff --git a/board/habanalabs/dali/linux_onix.fragment b/board/habanalabs/dali/linux_onix.fragment
index 41b5e64..ec39967 100644
--- a/board/habanalabs/dali/linux_onix.fragment
+++ b/board/habanalabs/dali/linux_onix.fragment
@@ -13,6 +13,7 @@
 CONFIG_THERMAL=y
 CONFIG_DEVFREQ_THERMAL=y
 CONFIG_PM_DEVFREQ=y
+CONFIG_PM_DEVFREQ_EVENT=y
 CONFIG_DEVFREQ_GOV_SIMPLE_ONDEMAND=y
 CONFIG_SPI_DESIGNWARE=y
 CONFIG_SPI_DW_MMIO=y
diff --git a/board/habanalabs/gaudi/linux_asic.fragment b/board/habanalabs/gaudi/linux_asic.fragment
index fadde2a..cd5df62 100644
--- a/board/habanalabs/gaudi/linux_asic.fragment
+++ b/board/habanalabs/gaudi/linux_asic.fragment
@@ -13,6 +13,7 @@
 CONFIG_THERMAL=y
 CONFIG_DEVFREQ_THERMAL=y
 CONFIG_PM_DEVFREQ=y
+CONFIG_PM_DEVFREQ_EVENT=y
 CONFIG_DEVFREQ_GOV_SIMPLE_ONDEMAND=y
 CONFIG_SPI_DESIGNWARE=y
 CONFIG_SPI_DW_MMIO=y
diff --git a/board/habanalabs/gaudi/linux_onix.fragment b/board/habanalabs/gaudi/linux_onix.fragment
index b8327b7..c0adf52 100644
--- a/board/habanalabs/gaudi/linux_onix.fragment
+++ b/board/habanalabs/gaudi/linux_onix.fragment
@@ -13,6 +13,7 @@
 CONFIG_THERMAL=y
 CONFIG_DEVFREQ_THERMAL=y
 CONFIG_PM_DEVFREQ=y
+CONFIG_PM_DEVFREQ_EVENT=y
 CONFIG_DEVFREQ_GOV_SIMPLE_ONDEMAND=y
 CONFIG_SPI_DESIGNWARE=y
 CONFIG_SPI_DW_MMIO=y
