From ae8f72183e7a17f2c8b48f73de6e316de15482f3 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Fri, 24 Aug 2018 12:49:06 +0300
Subject: [PATCH] Gaudi [tcl] update vivado script

To support latest dspg image for ONIX
---

diff --git a/board/habanalabs/common/vivado_lab/program-avnet.tcl b/board/habanalabs/common/vivado_lab/program-avnet.tcl
index b40552e..5166c89 100644
--- a/board/habanalabs/common/vivado_lab/program-avnet.tcl
+++ b/board/habanalabs/common/vivado_lab/program-avnet.tcl
@@ -2,11 +2,11 @@
 connect_hw_server
 open_hw_target
 #set_property PROGRAM.FILE {/opt/Xilinx/Vivado_Lab/h2_cpu_top_onix_avnet_070618.bit} [get_hw_devices xcvu440_0]
-set_property PROGRAM.FILE {/opt/Xilinx/Vivado_Lab/h3_top_avnet_220818.bit} [get_hw_devices xcvu440_0]
+set_property PROGRAM.FILE {/opt/Xilinx/Vivado_Lab/h3_top_avnet_240818.bit} [get_hw_devices xcvu440_0]
 current_hw_device [get_hw_devices xcvu440_0]
 refresh_hw_device [lindex [get_hw_devices xcvu440_0] 0]
 #set_property PROGRAM.FILE {/opt/Xilinx/Vivado_Lab/h2_cpu_top_onix_avnet_070618.bit} [get_hw_devices xcvu440_0]
-set_property PROGRAM.FILE {/opt/Xilinx/Vivado_Lab/h3_top_avnet_220818.bit} [get_hw_devices xcvu440_0]
+set_property PROGRAM.FILE {/opt/Xilinx/Vivado_Lab/h3_top_avnet_240818.bit} [get_hw_devices xcvu440_0]
 program_hw_devices [get_hw_devices xcvu440_0]
 refresh_hw_device [lindex [get_hw_devices xcvu440_0] 0]
 exit
diff --git a/board/habanalabs/common/vivado_lab/program-dspg.tcl b/board/habanalabs/common/vivado_lab/program-dspg.tcl
index 4a8afd0..def1455 100644
--- a/board/habanalabs/common/vivado_lab/program-dspg.tcl
+++ b/board/habanalabs/common/vivado_lab/program-dspg.tcl
@@ -4,13 +4,13 @@
 # DALI:
 #set_property PROGRAM.FILE {/opt/Xilinx/Vivado_Lab/h2_cpu_top_onix_dspg_070618.bit} [get_hw_devices xcvu440_0]
 # GAUDI:
-set_property PROGRAM.FILE {/opt/Xilinx/Vivado_Lab/h3_top_dspg_220818.bit} [get_hw_devices xcvu440_0]
+set_property PROGRAM.FILE {/opt/Xilinx/Vivado_Lab/h3_top_dspg_240818.bit} [get_hw_devices xcvu440_0]
 current_hw_device [get_hw_devices xcvu440_0]
 refresh_hw_device [lindex [get_hw_devices xcvu440_0] 0]
 # DALI:
 # set_property PROGRAM.FILE {/opt/Xilinx/Vivado_Lab/h2_cpu_top_onix_dspg_070618.bit} [get_hw_devices xcvu440_0]
 # GAUDI:
-set_property PROGRAM.FILE {/opt/Xilinx/Vivado_Lab/h3_top_dspg_220818.bit} [get_hw_devices xcvu440_0]
+set_property PROGRAM.FILE {/opt/Xilinx/Vivado_Lab/h3_top_dspg_240818.bit} [get_hw_devices xcvu440_0]
 program_hw_devices [get_hw_devices xcvu440_0]
 refresh_hw_device [lindex [get_hw_devices xcvu440_0] 0]
 exit
