From ca6df6ed56731f55c4e3a7fd92b20fb93eeea5ca Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Wed, 29 Aug 2018 06:38:33 +0300
Subject: [PATCH] HL: Rename product name DALI --> GOYA


---

diff --git a/Makefile b/Makefile
index e0f54f5..3ee46d8 100644
--- a/Makefile
+++ b/Makefile
@@ -15,11 +15,11 @@
 .DEFAULT_GOAL := hl-help
 hl-help:
 	@echo 'Cleaning:'
-	@echo '  dali-distclean       - Delete all Dali files created by build'
+	@echo '  goya-distclean       - Delete all files created by build'
 	@echo
 	@echo 'Build:'
-	@echo '  dali-rpi-build       - Build for Raspberry Pi'
-	@echo '  dali-<borad>-build   - Build for <board> e.g. onix, pldm, asic'
+	@echo '  goya-rpi-build       - Build for Raspberry Pi'
+	@echo '  goya-<borad>-build   - Build for <board> e.g. onix, pldm, asic'
 	@echo '  optional environment variables:'
 	@echo '    SPL_BINFILE        - Alternative SPL binary <PATH>'
 	@echo '    UBOOT_ITSFILE      - Alternative ITS file <PATH>'
@@ -31,22 +31,22 @@
 	@echo 'Install:'
 	@echo '  optional environment variables:'
 	@echo '    PLDM_HEX_DIR       - Palladium directory to install flash image'
-	@echo '  dali-<board>-install - Install to <board> e.g. onix, pldm, asic'
-	@echo '                           set DALI_INSTALL_LAB_IP=<lab IP> to'
+	@echo '  goya-<board>-install - Install to <board> e.g. onix, pldm, asic'
+	@echo '                           set GOYA_INSTALL_LAB_IP=<lab IP> to'
 	@echo '                             override <board> default lab IP'
-	@echo '                           set DALI_INSTALL_HOST_IP=<host IP> to'
+	@echo '                           set GOYA_INSTALL_HOST_IP=<host IP> to'
 	@echo '                             override <board> default host IP'
 	@echo
 	@echo 'Release:'
 	@echo '  required environment variables:'
 	@echo '    HL_RELEASE_VER     - Habanalabs release version string'
-	@echo '  dali-release         - Prepare the release tag and push it upstream'
+	@echo '  goya-release         - Prepare the release tag and push it upstream'
 	@echo '                           usually, the version string would be something'
 	@echo '                             like: x.y.z'
 	@echo
 	@echo 'Testing:'
-	@echo '  dali-<board>-test    - Run uboot test on <board> e.g. onix, pldm, asic'
-	@echo '  dali-<board>-tbot    - Run tbot test on <board> e.g. onix, pldm, asic'
+	@echo '  goya-<board>-test    - Run uboot test on <board> e.g. onix, pldm, asic'
+	@echo '  goya-<board>-tbot    - Run tbot test on <board> e.g. onix, pldm, asic'
 	@echo '                           set HOST_TBOT_TC=<test name> to'
 	@echo '                             override dafault test tc_ub_help.py'
 	@echo '                           set HOST_TBOT_IP=<PC lab IP> to'
@@ -69,8 +69,8 @@
 RELEASE_REPOS := buildroot-external preboot u-boot linux armcp tbot fw_loader
 RELEASE_TAG_POINTER := latest-embedded-release
 
-.PHONY: gaudi-release dali-release
-gaudi-release dali-release: hl-check-release
+.PHONY: gaudi-release goya-release
+gaudi-release goya-release: hl-check-release
 	$(eval HL_FULL_VER=hl-$(patsubst %-release,%,$@)-$(HL_RELEASE_VER))
 	@repo forall $(RELEASE_REPOS) -c \
 		git tag -afm "Release Habanalabs $(HL_FULL_VER)" $(HL_FULL_VER)
@@ -82,40 +82,40 @@
 	@wget -q -O - http://192.168.0.4:8080/job/goya-release-test/build?token=goya-release
 	@echo "Started the \"goya-release-test\" jenkins job... check your email for results."
 
-.PHONY: hl-dali hl-gaudi
-hl-dali hl-gaudi: $(OUTPUT_DIR)
+.PHONY: hl-goya hl-gaudi
+hl-goya hl-gaudi: $(OUTPUT_DIR)
 	@echo "export HL_DEV=$(patsubst hl-%,%,$@)"		 > $(HL_VARS_FILE)
 
 .PHONY: gaudi-onix-target
 gaudi-onix-target: hl-gaudi
 	@echo "export HL_TARGET=$(patsubst gaudi-%-target,%,$@)"	>> $(HL_VARS_FILE)
 
-.PHONY: dali-onix-target dali-pldm-target dali-asic-target
-dali-onix-target dali-pldm-target dali-asic-target: hl-dali
-	@echo "export HL_TARGET=$(patsubst dali-%-target,%,$@)"	>> $(HL_VARS_FILE)
+.PHONY: goya-onix-target goya-pldm-target goya-asic-target
+goya-onix-target goya-pldm-target goya-asic-target: hl-goya
+	@echo "export HL_TARGET=$(patsubst goya-%-target,%,$@)"	>> $(HL_VARS_FILE)
 
-.PHONY: dali-prepare gaudi-prepare
+.PHONY: goya-prepare gaudi-prepare
 gaudi-prepare: $(BR2_EXTERNAL)/configs/gaudi_defconfig
 	@$(MAKE) $(<F)
-dali-prepare: $(BR2_EXTERNAL)/configs/dali_defconfig
+goya-prepare: $(BR2_EXTERNAL)/configs/goya_defconfig
 	@$(MAKE) $(<F)
 
-.PHONY: dali-onix-build dali-pldm-build dali-asic-build gaudi-onix-build
+.PHONY: goya-onix-build goya-pldm-build goya-asic-build gaudi-onix-build
 gaudi-onix-build: gaudi-onix-target
-dali-onix-build: dali-onix-target
-dali-pldm-build: dali-pldm-target
-dali-asic-build: dali-asic-target
+goya-onix-build: goya-onix-target
+goya-pldm-build: goya-pldm-target
+goya-asic-build: goya-asic-target
 gaudi-onix-build: $(OUTPUT_DIR) gaudi-prepare
-dali-onix-build dali-pldm-build dali-asic-build: $(OUTPUT_DIR) dali-prepare
-dali-onix-build dali-pldm-build dali-asic-build gaudi-onix-build:
+goya-onix-build goya-pldm-build goya-asic-build: $(OUTPUT_DIR) goya-prepare
+goya-onix-build goya-pldm-build goya-asic-build gaudi-onix-build:
 	@$(MAKE) all
 
-.PHONY: dali-rpi-build dali-rpi-armcp dali-rpi-gpio-mdio dali-rpi-regs-cli
-dali-rpi-gpio-mdio: SHELL=/bin/bash
-dali-rpi-gpio-mdio: .SHELLFLAGS=-i -c
-dali-rpi-regs-cli: SHELL=/bin/bash
-dali-rpi-regs-cli: .SHELLFLAGS=-i -c
-dali-rpi-armcp:
+.PHONY: goya-rpi-build goya-rpi-armcp goya-rpi-gpio-mdio goya-rpi-regs-cli
+goya-rpi-gpio-mdio: SHELL=/bin/bash
+goya-rpi-gpio-mdio: .SHELLFLAGS=-i -c
+goya-rpi-regs-cli: SHELL=/bin/bash
+goya-rpi-regs-cli: .SHELLFLAGS=-i -c
+goya-rpi-armcp:
 	@mkdir -p $(OUTPUT_DIR)/build
 	@rsync -au --exclude .git $(BR2_EXTERNAL)/../specs/ ${OUTPUT_DIR}/build/specs-custom
 	@rsync -au --exclude .git $(BR2_EXTERNAL)/../specs_external/ ${OUTPUT_DIR}/build/specs_external-custom
@@ -124,65 +124,65 @@
 	@mkdir -p ${OUTPUT_DIR}/build/armcp-custom/build
 	@cd ${OUTPUT_DIR}/build/armcp-custom/build && cmake -DTARGET_BUILD:STRING=fpga ../
 	@cd ${OUTPUT_DIR}/build/armcp-custom/build && make -j 4
-dali-rpi-gpio-mdio:
+goya-rpi-gpio-mdio:
 	@cd && build_and_insmod_gpio_mdio
-dali-rpi-regs-cli:
+goya-rpi-regs-cli:
 	@cd && build_armcp_regs_cli
-dali-rpi-build: dali-rpi-armcp dali-rpi-gpio-mdio dali-rpi-regs-cli
+goya-rpi-build: goya-rpi-armcp goya-rpi-gpio-mdio goya-rpi-regs-cli
 
 IMAGES_DIR := ${OUTPUT_DIR}/images
 UBOOT_BASE := ${OUTPUT_DIR}/build/uboot-custom
 LINUX_BASE := ${OUTPUT_DIR}/build/linux-custom
 UBOOT_XMPL := ${UBOOT_BASE}/examples/standalone
 DSV_BASE := ${OUTPUT_DIR}/build/dsv-custom
-.PHONY: dali-onix-install dali-pldm-install dali-asic-install gaudi-onix-install
-dali-onix-install gaudi-onix-install: DALI_INSTALL_LAB_IP ?= monster4-nuc
-dali-onix-install gaudi-onix-install: DALI_INSTALL_HOST_IP ?= monster4
-dali-pldm-install: DALI_INSTALL_LAB_IP ?= 192.168.10.45
-dali-pldm-install: DALI_INSTALL_HOST_IP ?= 192.168.10.45
-dali-asic-install: DALI_INSTALL_LAB_IP ?= 192.168.10.109
-dali-asic-install: DALI_INSTALL_HOST_IP ?= 192.168.10.134
+.PHONY: goya-onix-install goya-pldm-install goya-asic-install gaudi-onix-install
+goya-onix-install gaudi-onix-install: GOYA_INSTALL_LAB_IP ?= monster4-nuc
+goya-onix-install gaudi-onix-install: GOYA_INSTALL_HOST_IP ?= monster4
+goya-pldm-install: GOYA_INSTALL_LAB_IP ?= 192.168.10.45
+goya-pldm-install: GOYA_INSTALL_HOST_IP ?= 192.168.10.45
+goya-asic-install: GOYA_INSTALL_LAB_IP ?= 192.168.10.109
+goya-asic-install: GOYA_INSTALL_HOST_IP ?= 192.168.10.134
 gaudi-onix-install: gaudi-onix-target
-dali-onix-install: dali-onix-target
-dali-pldm-install: dali-pldm-target
-dali-asic-install: dali-asic-target
+goya-onix-install: goya-onix-target
+goya-pldm-install: goya-pldm-target
+goya-asic-install: goya-asic-target
 
 .PHONY: hl-install
 hl-install:
-	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-u-boot.bin		root@$(DALI_INSTALL_HOST_IP):/lib/firmware/habanalabs/$(HL_DEV)/
-	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-fit.itb		root@$(DALI_INSTALL_HOST_IP):/lib/firmware/habanalabs/$(HL_DEV)/
-	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-spi-flash.bin	root@$(DALI_INSTALL_HOST_IP):/lib/firmware/habanalabs/$(HL_DEV)/
-	-sshpass -p123456        scp $(IMAGES_DIR)/hl-fw-loader			root@$(DALI_INSTALL_HOST_IP):/lib/firmware/habanalabs/$(HL_DEV)/
-	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-u-boot.bin		root@$(DALI_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
-	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-u-boot-spl.bin	root@$(DALI_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
-	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-fit.itb		root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/
-	-sshpass -p123456        scp $(IMAGES_DIR)/boot.scr			root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
-	-sshpass -p123456        scp $(IMAGES_DIR)/hl-fw-loader			root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
-	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-spi-flash.bin	root@$(DALI_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
-	-sshpass -p123456        scp $(UBOOT_BASE)/u-boot			root@$(DALI_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
-	-sshpass -p123456        scp $(LINUX_BASE)/vmlinux			root@$(DALI_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
-	-sshpass -p123456        scp -r $(UBOOT_XMPL)/$(HL_DEV)*.bin        	root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
-	-sshpass -p123456        scp ${DSV_BASE}/dsv.bin			root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
+	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-u-boot.bin		root@$(GOYA_INSTALL_HOST_IP):/lib/firmware/habanalabs/$(HL_DEV)/
+	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-fit.itb		root@$(GOYA_INSTALL_HOST_IP):/lib/firmware/habanalabs/$(HL_DEV)/
+	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-spi-flash.bin	root@$(GOYA_INSTALL_HOST_IP):/lib/firmware/habanalabs/$(HL_DEV)/
+	-sshpass -p123456        scp $(IMAGES_DIR)/hl-fw-loader			root@$(GOYA_INSTALL_HOST_IP):/lib/firmware/habanalabs/$(HL_DEV)/
+	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-u-boot.bin		root@$(GOYA_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
+	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-u-boot-spl.bin	root@$(GOYA_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
+	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-fit.itb		root@$(GOYA_INSTALL_LAB_IP):/var/lib/tftpboot/
+	-sshpass -p123456        scp $(IMAGES_DIR)/boot.scr			root@$(GOYA_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
+	-sshpass -p123456        scp $(IMAGES_DIR)/hl-fw-loader			root@$(GOYA_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
+	-sshpass -p123456        scp $(IMAGES_DIR)/$(HL_DEV)-spi-flash.bin	root@$(GOYA_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
+	-sshpass -p123456        scp $(UBOOT_BASE)/u-boot			root@$(GOYA_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
+	-sshpass -p123456        scp $(LINUX_BASE)/vmlinux			root@$(GOYA_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
+	-sshpass -p123456        scp -r $(UBOOT_XMPL)/$(HL_DEV)*.bin        	root@$(GOYA_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
+	-sshpass -p123456        scp ${DSV_BASE}/dsv.bin			root@$(GOYA_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
 
 
-.PHONY: dali-install gaudi-install
-dali-install gaudi-install:
-	@$(MAKE) hl-install DALI_INSTALL_HOST_IP=$(DALI_INSTALL_HOST_IP) DALI_INSTALL_LAB_IP=$(DALI_INSTALL_LAB_IP)
+.PHONY: goya-install gaudi-install
+goya-install gaudi-install:
+	@$(MAKE) hl-install GOYA_INSTALL_HOST_IP=$(GOYA_INSTALL_HOST_IP) GOYA_INSTALL_LAB_IP=$(GOYA_INSTALL_LAB_IP)
 
 gaudi-onix-install: gaudi-install
-dali-onix-install dali-pldm-install dali-asic-install: dali-install
-dali-onix-install gaudi-onix-install:
+goya-onix-install goya-pldm-install goya-asic-install: goya-install
+goya-onix-install gaudi-onix-install:
 	@
 PLDM_HEX_DIR ?= $(IMAGES_DIR)/tools/hl_shared_dir/pldm
 ${PLDM_HEX_DIR}:
 	@mkdir -p $@
-dali-pldm-install: ${PLDM_HEX_DIR}
+goya-pldm-install: ${PLDM_HEX_DIR}
 	-cp $(IMAGES_DIR)/$(HL_DEV)-spi-flash.hex $</
-dali-asic-install:
+goya-asic-install:
 	@
 
-.PHONY: dali-distclean gaudi-distclean
-dali-distclean gaudi-distclean:
+.PHONY: goya-distclean gaudi-distclean
+goya-distclean gaudi-distclean:
 	@rm -f $(HL_VARS_FILE)
 	@if [ -d $(OUTPUT_DIR)/build ]; then \
 		cd $(OUTPUT_DIR)/build && \
@@ -200,32 +200,32 @@
 	@$(MAKE) $(HL_DEV)-distclean
 endif
 
-.PHONY: dali-onix-test dali-pldm-test dali-asic-test
-dali-onix-test: dali-onix-tools-install
-dali-onix-test dali-pldm-test dali-asic-test:
+.PHONY: goya-onix-test goya-pldm-test goya-asic-test
+goya-onix-test: goya-onix-tools-install
+goya-onix-test goya-pldm-test goya-asic-test:
 	@HOST_UBOOT_TEST_HOOKS_BRD=$(patsubst $(HL_DEV)-%-test,$(HL_DEV)_%,$@)	\
 		$(MAKE) host-uboot-test-hooks-rebuild
 
-dali-onix-tools-install: DALI_INSTALL_LAB_IP ?= monster4-nuc
-gaudi-onix-tools-install: DALI_INSTALL_LAB_IP ?= monster4-nuc
-.PHONY: dali-onix-tools-install gaudi-onix-tools-install
-dali-onix-tools-install gaudi-onix-tools-install:
+goya-onix-tools-install: GOYA_INSTALL_LAB_IP ?= monster4-nuc
+gaudi-onix-tools-install: GOYA_INSTALL_LAB_IP ?= monster4-nuc
+.PHONY: goya-onix-tools-install gaudi-onix-tools-install
+goya-onix-tools-install gaudi-onix-tools-install:
 	-cd board/habanalabs/common/ &&					\
 	sshpass -p123456 scp -r remote_power ip_pdu t32 vivado_lab	\
-	root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
+	root@$(GOYA_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
 
 HOST_TBOT_TC ?= tc_ub_help.py
-.PHONY: dali-onix-tbot dali-pldm-tbot dali-asic-tbot
-dali-onix-tbot: dali-onix-tools-install
-dali-onix-tbot: HOST_TBOT_IP ?= monster4-nuc
-dali-onix-tbot: PCIE_HOST_TBOT_IP ?= monster4
-dali-pldm-tbot: dali-pldm-tools-install
-dali-pldm-tbot: HOST_TBOT_IP ?= 192.168.10.45
-dali-pldm-tbot: PCIE_HOST_TBOT_IP ?= nohost
-dali-asic-tbot: dali-asic-tools-install
-dali-asic-tbot: HOST_TBOT_IP ?= 192.168.10.109
-dali-asic-tbot: PCIE_HOST_TBOT_IP ?= monster5
-dali-onix-tbot dali-pldm-tbot dali-asic-tbot:
+.PHONY: goya-onix-tbot goya-pldm-tbot goya-asic-tbot
+goya-onix-tbot: goya-onix-tools-install
+goya-onix-tbot: HOST_TBOT_IP ?= monster4-nuc
+goya-onix-tbot: PCIE_HOST_TBOT_IP ?= monster4
+goya-pldm-tbot: goya-pldm-tools-install
+goya-pldm-tbot: HOST_TBOT_IP ?= 192.168.10.45
+goya-pldm-tbot: PCIE_HOST_TBOT_IP ?= nohost
+goya-asic-tbot: goya-asic-tools-install
+goya-asic-tbot: HOST_TBOT_IP ?= 192.168.10.109
+goya-asic-tbot: PCIE_HOST_TBOT_IP ?= monster5
+goya-onix-tbot goya-pldm-tbot goya-asic-tbot:
 	@HOST_TBOT_LAB=lab_$(HL_DEV)					\
 	HOST_TBOT_CFG=$(HL_DEV)					\
 	HOST_TBOT_BRD=$(patsubst $(HL_DEV)-%-tbot,%,$@)		\
@@ -233,17 +233,17 @@
 	HOST_TBOT_LAB_IP=$(HOST_TBOT_IP)			\
 	PCIE_HOST_TBOT_LAB_IP=$(PCIE_HOST_TBOT_IP)		\
 		$(MAKE) host-tbot-rebuild
-dali-pldm-tools-install: DALI_INSTALL_LAB_IP ?= 192.168.10.45
-.PHONY: dali-pldm-tools-install
-dali-pldm-tools-install:
+goya-pldm-tools-install: GOYA_INSTALL_LAB_IP ?= 192.168.10.45
+.PHONY: goya-pldm-tools-install
+goya-pldm-tools-install:
 	-cd board/habanalabs/common/ &&				\
 	sshpass -p123456 scp -r remote_power ip_pdu		\
-	root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
-dali-asic-tools-install: DALI_INSTALL_LAB_IP ?= 192.168.10.109
-.PHONY: dali-asic-tools-install
-dali-asic-tools-install:
+	root@$(GOYA_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
+goya-asic-tools-install: GOYA_INSTALL_LAB_IP ?= 192.168.10.109
+.PHONY: goya-asic-tools-install
+goya-asic-tools-install:
 	-cd board/habanalabs/common/ &&					\
 	sshpass -p123456 scp -r remote_power ip_pdu t32 vivado_lab	\
 	local_remote_common local_run remote_run			\
 	$(IMAGES_DIR)/hl-fw-loader					\
-	root@$(DALI_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
+	root@$(GOYA_INSTALL_LAB_IP):/var/lib/tftpboot/tests/
diff --git a/board/habanalabs/common/hl_fit_spl.its b/board/habanalabs/common/hl_fit_spl.its
index 9ce365d..e602954 100644
--- a/board/habanalabs/common/hl_fit_spl.its
+++ b/board/habanalabs/common/hl_fit_spl.its
@@ -22,35 +22,35 @@
 		};
 
 		fdt {
-			description = "Dali 1000 flat device-tree";
+			description = "Goya 1000 flat device-tree";
 			data = /incbin/("u-boot.dtb");
 			type = "flat_dt";
 			compression = "none";
 		};
 /*
 		data0 {
-			description = "Dali data 0";
+			description = "Goya data 0";
 			data = /incbin/("uboot-data0.bin");
 			compression = "none";
 			load = <0x7f 0xf0000000>;
 		};
 
 		data1 {
-			description = "Dali data 1";
+			description = "Goya data 1";
 			data = /incbin/("uboot-data1.bin");
 			compression = "none";
 			load = <0x7f 0xf0010000>;
 		};
 
 		data2 {
-			description = "Dali data 2";
+			description = "Goya data 2";
 			data = /incbin/("uboot-data2.bin");
 			compression = "none";
 			load = <0x7f 0xf0020000>;
 		};
 
 		data3 {
-			description = "Dali data 3";
+			description = "Goya data 3";
 			data = /incbin/("uboot-data3.bin");
 			compression = "none";
 			load = <0x7f 0xf0030000>;
@@ -61,7 +61,7 @@
 	configurations {
 		default = "conf";
 		conf {
-			description = "Habanalabs Dali";
+			description = "Habanalabs Goya";
 			firmware = "uboot";
 			fdt = "fdt";
 /*
diff --git a/board/habanalabs/common/post-build.sh b/board/habanalabs/common/post-build.sh
index 4270fe4..76724e5 100755
--- a/board/habanalabs/common/post-build.sh
+++ b/board/habanalabs/common/post-build.sh
@@ -25,6 +25,6 @@
 # Add armcpd to hl device
 if [ -e ${TARGET_DIR}/etc/inittab ]; then
 	grep -qE "^${HL_DEV}::" ${TARGET_DIR}/etc/inittab || \
-	sed -i "/GENERIC_SERIAL/a# Put armcpd on Dali device/\n\
+	sed -i "/GENERIC_SERIAL/a# Put armcpd on Goya device/\n\
 ${HL_DEV}::respawn:/usr/bin/armcpd # ARM CP" ${TARGET_DIR}/etc/inittab
 fi
diff --git a/board/habanalabs/common/post-image.sh b/board/habanalabs/common/post-image.sh
index 1b1f588..f1a8f34 100755
--- a/board/habanalabs/common/post-image.sh
+++ b/board/habanalabs/common/post-image.sh
@@ -12,7 +12,7 @@
 	INF=$1
 	OUTF=$2
 
-	if [ "${HL_DEV}" == "dali" ]; then
+	if [ "${HL_DEV}" == "goya" ]; then
 		# SWAP needs this to be 4B aligned in size
 		dd status=none conv=sync bs=4 if=${INF} > ${OUTF}.tmp
 		# Sadly --endian=big is not swapping for -tx1 so I used some gawk magic
diff --git a/board/habanalabs/common/remote_run b/board/habanalabs/common/remote_run
index d333ac3..bcd73ef 100755
--- a/board/habanalabs/common/remote_run
+++ b/board/habanalabs/common/remote_run
@@ -8,7 +8,7 @@
 program_flash=$(echo $params | grep -ow program | uniq)
 
 result=~/$(basename $0).result
-fw_dir=/lib/firmware/habanalabs/dali
+fw_dir=/lib/firmware/habanalabs/goya
 local_scr_dir=/var/lib/tftpboot/tests
 scr_name=(local_run local_remote_common)
 fw_ldr=hl-fw-loader
diff --git a/board/habanalabs/common/t32/asic.cmm b/board/habanalabs/common/t32/asic.cmm
index 0205e8c..a3785cf 100644
--- a/board/habanalabs/common/t32/asic.cmm
+++ b/board/habanalabs/common/t32/asic.cmm
@@ -1,8 +1,8 @@
 ; --------------------------------------------------------------------------------------------------
-; @Title: Startup script for the DALI SoC
+; @Title: Startup script for the GOYA SoC
 ; @Description: This script configures the base addresses for debug and attaches to the core
 ; @Author: Oded Gabbay
-; @SoC: DALI
+; @SoC: GOYA
 ; @Chip: Cortex-A53
 ; @Copyright: (C) 2016-2017 HabanaLabs, Ltd.
 ; --------------------------------------------------------------------------------------------------
diff --git a/board/habanalabs/common/t32/onix.cmm b/board/habanalabs/common/t32/onix.cmm
index 8e1fd9b..4d40ece 100644
--- a/board/habanalabs/common/t32/onix.cmm
+++ b/board/habanalabs/common/t32/onix.cmm
@@ -1,8 +1,8 @@
 ; --------------------------------------------------------------------------------------------------
-; @Title: Startup script for the DALI SoC
+; @Title: Startup script for the GOYA SoC
 ; @Description: This script configures the base addresses for debug and attaches to the core
 ; @Author: Oded Gabbay
-; @SoC: DALI
+; @SoC: GOYA
 ; @Chip: Cortex-A53
 ; @Copyright: (C) 2016-2017 HabanaLabs, Ltd.
 ; --------------------------------------------------------------------------------------------------
diff --git a/board/habanalabs/common/t32/pldm.cmm b/board/habanalabs/common/t32/pldm.cmm
index 1474ec8..3b1d6f4 100644
--- a/board/habanalabs/common/t32/pldm.cmm
+++ b/board/habanalabs/common/t32/pldm.cmm
@@ -1,8 +1,8 @@
 ; --------------------------------------------------------------------------------------------------
-; @Title: Startup script for the DALI SoC
+; @Title: Startup script for the GOYA SoC
 ; @Description: This script configures the base addresses for debug and attaches to the core
 ; @Author: Oded Gabbay
-; @SoC: DALI
+; @SoC: GOYA
 ; @Chip: Cortex-A53
 ; @Copyright: (C) 2016-2017 HabanaLabs, Ltd.
 ; --------------------------------------------------------------------------------------------------
diff --git a/board/habanalabs/common/t32/prog_spi.cmm b/board/habanalabs/common/t32/prog_spi.cmm
index 8993f5e..85da93c 100644
--- a/board/habanalabs/common/t32/prog_spi.cmm
+++ b/board/habanalabs/common/t32/prog_spi.cmm
@@ -63,8 +63,8 @@
 
   FLASHFILE.unlock 0x0--0x29FFFF
   FLASHFILE.erase 0x0--0x29FFFF
-  FLASHFILE.load /lib/firmware/habanalabs/dali/dali-spi-flash.bin 0x0++0x29FFFF
-  FLASHFILE.load /lib/firmware/habanalabs/dali/dali-spi-flash.bin 0x0++0x29FFFF  /ComPare
+  FLASHFILE.load /lib/firmware/habanalabs/goya/goya-spi-flash.bin 0x0++0x29FFFF
+  FLASHFILE.load /lib/firmware/habanalabs/goya/goya-spi-flash.bin 0x0++0x29FFFF  /ComPare
 
 ENDDO
 
diff --git a/board/habanalabs/common/t32/state.cmm b/board/habanalabs/common/t32/state.cmm
index 4340b0c..966a58e 100644
--- a/board/habanalabs/common/t32/state.cmm
+++ b/board/habanalabs/common/t32/state.cmm
@@ -1,8 +1,8 @@
 ; --------------------------------------------------------------------------------------------------
-; @Title: Startup script for the DALI SoC
+; @Title: Startup script for the GOYA SoC
 ; @Description: This script configures the base addresses for debug and attaches to the core
 ; @Author: Oded Gabbay
-; @SoC: DALI
+; @SoC: GOYA
 ; @Chip: Cortex-A53
 ; @Copyright: (C) 2016-2017 HabanaLabs, Ltd.
 ; --------------------------------------------------------------------------------------------------
diff --git a/board/habanalabs/common/t32/uboot.cmm b/board/habanalabs/common/t32/uboot.cmm
index ff730bf..0ee16e6 100644
--- a/board/habanalabs/common/t32/uboot.cmm
+++ b/board/habanalabs/common/t32/uboot.cmm
@@ -8,8 +8,8 @@
 D.S axi:0x7FFCC4B474 %LE %Long 0x800
 
 ; Load u-boot
-B::data.LOAD.Binary /lib/firmware/habanalabs/dali/dali-u-boot.bin axi:0x7ff0100000
-B::data.LOAD.Binary /lib/firmware/habanalabs/dali/dali-u-boot-spl.bin axi:0x7ff0200000
-B::data.LOAD.elf /lib/firmware/habanalabs/dali/u-boot /NoCODE /NoClear
-B::data.LOAD.elf /lib/firmware/habanalabs/dali/u-boot 0x164000 /NoCODE /NoClear
-B::data.LOAD.elf /lib/firmware/habanalabs/dali/vmlinux /NoCODE /NoClear
+B::data.LOAD.Binary /lib/firmware/habanalabs/goya/goya-u-boot.bin axi:0x7ff0100000
+B::data.LOAD.Binary /lib/firmware/habanalabs/goya/goya-u-boot-spl.bin axi:0x7ff0200000
+B::data.LOAD.elf /lib/firmware/habanalabs/goya/u-boot /NoCODE /NoClear
+B::data.LOAD.elf /lib/firmware/habanalabs/goya/u-boot 0x164000 /NoCODE /NoClear
+B::data.LOAD.elf /lib/firmware/habanalabs/goya/vmlinux /NoCODE /NoClear
diff --git a/board/habanalabs/common/vivado_lab/erase.tcl b/board/habanalabs/common/vivado_lab/erase.tcl
index e8c7584..b6eca54 100644
--- a/board/habanalabs/common/vivado_lab/erase.tcl
+++ b/board/habanalabs/common/vivado_lab/erase.tcl
@@ -2,7 +2,7 @@
 connect_hw_server
 open_hw_target
 create_hw_cfgmem -hw_device [lindex [get_hw_devices] 0] -mem_dev [lindex [get_cfgmem_parts {mt25qu512-spi-x1_x2_x4_x8}] 0]
-set_property PROGRAM.FILES [list "/lib/firmware/habanalabs/dali/dali-spi-flash.bin" "/lib/firmware/habanalabs/dali/dali-spi-flash.bin" ] [ get_property PROGRAM.HW_CFGMEM [lindex [get_hw_devices xcvu440_0] 0]]
+set_property PROGRAM.FILES [list "/lib/firmware/habanalabs/goya/goya-spi-flash.bin" "/lib/firmware/habanalabs/goya/goya-spi-flash.bin" ] [ get_property PROGRAM.HW_CFGMEM [lindex [get_hw_devices xcvu440_0] 0]]
 set_property PROGRAM.ERASE  1 [ get_property PROGRAM.HW_CFGMEM [lindex [get_hw_devices xcvu440_0] 0]]
 program_hw_cfgmem -hw_cfgmem [ get_property PROGRAM.HW_CFGMEM [lindex [get_hw_devices xcvu440_0] 0]]
 exit
diff --git a/board/habanalabs/common/vivado_lab/program-dspg.tcl b/board/habanalabs/common/vivado_lab/program-dspg.tcl
index def1455..f091443 100644
--- a/board/habanalabs/common/vivado_lab/program-dspg.tcl
+++ b/board/habanalabs/common/vivado_lab/program-dspg.tcl
@@ -1,13 +1,13 @@
 open_hw
 connect_hw_server
 open_hw_target
-# DALI:
+# GOYA:
 #set_property PROGRAM.FILE {/opt/Xilinx/Vivado_Lab/h2_cpu_top_onix_dspg_070618.bit} [get_hw_devices xcvu440_0]
 # GAUDI:
 set_property PROGRAM.FILE {/opt/Xilinx/Vivado_Lab/h3_top_dspg_240818.bit} [get_hw_devices xcvu440_0]
 current_hw_device [get_hw_devices xcvu440_0]
 refresh_hw_device [lindex [get_hw_devices xcvu440_0] 0]
-# DALI:
+# GOYA:
 # set_property PROGRAM.FILE {/opt/Xilinx/Vivado_Lab/h2_cpu_top_onix_dspg_070618.bit} [get_hw_devices xcvu440_0]
 # GAUDI:
 set_property PROGRAM.FILE {/opt/Xilinx/Vivado_Lab/h3_top_dspg_240818.bit} [get_hw_devices xcvu440_0]
diff --git a/board/habanalabs/dali/dali_asic.its b/board/habanalabs/dali/dali_asic.its
index 4c61068..5bcba8b 100644
--- a/board/habanalabs/dali/dali_asic.its
+++ b/board/habanalabs/dali/dali_asic.its
@@ -11,7 +11,7 @@
 
 	images {
 		kernel@1 {
-			description = "Dali Linux kernel";
+			description = "Goya Linux kernel";
 			data = /incbin/("Image.gz");
 			type = "kernel";
 			arch = "arm64";
@@ -28,7 +28,7 @@
 		};
 		fdt@1 {
 			description = "Flattened Device Tree blob";
-			data = /incbin/("dali_asic.dtb");
+			data = /incbin/("goya_asic.dtb");
 			type = "flat_dt";
 			arch = "arm64";
 			compression = "none";
@@ -38,7 +38,7 @@
 			};
 		};
 		ramdisk@1 {
-			description = "Dali ramdisk";
+			description = "Goya ramdisk";
 			data = /incbin/("rootfs.cpio.gz");
 			type = "ramdisk";
 			arch = "arm64";
diff --git a/board/habanalabs/dali/dali_onix.its b/board/habanalabs/dali/dali_onix.its
index c0865d7..35145b0 100644
--- a/board/habanalabs/dali/dali_onix.its
+++ b/board/habanalabs/dali/dali_onix.its
@@ -11,7 +11,7 @@
 
 	images {
 		kernel@1 {
-			description = "Dali Linux kernel";
+			description = "Goya Linux kernel";
 			data = /incbin/("Image.gz");
 			type = "kernel";
 			arch = "arm64";
@@ -28,7 +28,7 @@
 		};
 		fdt@1 {
 			description = "Flattened Device Tree blob";
-			data = /incbin/("dali_onix.dtb");
+			data = /incbin/("goya_onix.dtb");
 			type = "flat_dt";
 			arch = "arm64";
 			compression = "none";
@@ -38,7 +38,7 @@
 			};
 		};
 		ramdisk@1 {
-			description = "Dali ramdisk";
+			description = "Goya ramdisk";
 			data = /incbin/("rootfs.cpio.gz");
 			type = "ramdisk";
 			arch = "arm64";
diff --git a/board/habanalabs/dali/dali_pldm.its b/board/habanalabs/dali/dali_pldm.its
index 6b112ea..786c11c 100644
--- a/board/habanalabs/dali/dali_pldm.its
+++ b/board/habanalabs/dali/dali_pldm.its
@@ -11,7 +11,7 @@
 
 	images {
 		kernel@1 {
-			description = "Dali Linux kernel";
+			description = "Goya Linux kernel";
 			data = /incbin/("Image");
 			type = "kernel";
 			arch = "arm64";
@@ -22,14 +22,14 @@
 		};
 		fdt@1 {
 			description = "Flattened Device Tree blob";
-			data = /incbin/("dali_pldm.dtb");
+			data = /incbin/("goya_pldm.dtb");
 			type = "flat_dt";
 			arch = "arm64";
 			compression = "none";
 			load = <0x00000000 0x00000000>;
 		};
 		ramdisk@1 {
-			description = "Dali ramdisk";
+			description = "Goya ramdisk";
 			data = /incbin/("rootfs.cpio");
 			type = "ramdisk";
 			arch = "arm64";
diff --git a/board/habanalabs/dali/linux_asic.fragment b/board/habanalabs/dali/linux_asic.fragment
index ba2c3cc..03b97cc 100644
--- a/board/habanalabs/dali/linux_asic.fragment
+++ b/board/habanalabs/dali/linux_asic.fragment
@@ -1,4 +1,4 @@
-CONFIG_ARCH_DALI=y
+CONFIG_ARCH_GOYA=y
 CONFIG_SERIAL_MPS2_UART_CONSOLE=y
 CONFIG_SERIAL_MPS2_UART=y
 CONFIG_STMMAC_ETH=y
diff --git a/board/habanalabs/dali/linux_onix.fragment b/board/habanalabs/dali/linux_onix.fragment
index ec39967..092f9b8 100644
--- a/board/habanalabs/dali/linux_onix.fragment
+++ b/board/habanalabs/dali/linux_onix.fragment
@@ -1,4 +1,4 @@
-CONFIG_ARCH_DALI=y
+CONFIG_ARCH_GOYA=y
 CONFIG_SERIAL_MPS2_UART_CONSOLE=y
 CONFIG_SERIAL_MPS2_UART=y
 CONFIG_STMMAC_ETH=y
diff --git a/board/habanalabs/dali/linux_pldm.fragment b/board/habanalabs/dali/linux_pldm.fragment
index fbf10d5..a70e9cd 100644
--- a/board/habanalabs/dali/linux_pldm.fragment
+++ b/board/habanalabs/dali/linux_pldm.fragment
@@ -1,4 +1,4 @@
-CONFIG_ARCH_DALI=y
+CONFIG_ARCH_GOYA=y
 CONFIG_SERIAL_MPS2_UART_CONSOLE=y
 CONFIG_SERIAL_MPS2_UART=y
 CONFIG_STMMAC_ETH=y
diff --git a/board/habanalabs/dali/uboot-eeprom.cmd b/board/habanalabs/dali/uboot-eeprom.cmd
index 4f10e0d..d7744f9 100644
--- a/board/habanalabs/dali/uboot-eeprom.cmd
+++ b/board/habanalabs/dali/uboot-eeprom.cmd
@@ -12,7 +12,7 @@
 eeprom update $LAYOUT "Thermal Solution Vendor" "5"
 eeprom update $LAYOUT "PCIe Gen3 or Gen4 Enabled" "6"
 eeprom update $LAYOUT "MAC Address" "00:11:22:33:44:55"
-eeprom update $LAYOUT "Board Configuration" "Dali HL1000"
+eeprom update $LAYOUT "Board Configuration" "Goya HL1000"
 echo "eeprom content is:"
 eeprom print  $LAYOUT
 setenv LAYOUT
diff --git a/board/habanalabs/dali/uboot-env-asic.txt b/board/habanalabs/dali/uboot-env-asic.txt
index d064803..d0d5134 100644
--- a/board/habanalabs/dali/uboot-env-asic.txt
+++ b/board/habanalabs/dali/uboot-env-asic.txt
@@ -1,4 +1,4 @@
 bootm_low=0
 ethaddr=00:33:66:99:cc:ff
-netboot=dhcp 192.168.10.109:dali-fit.itb && bootm $fileaddr#conf@1
+netboot=dhcp 192.168.10.109:goya-fit.itb && bootm $fileaddr#conf@1
 bootcmd=getfit && bootm $fileaddr#conf@1 || getfit_failure
diff --git a/board/habanalabs/dali/uboot-env-onix.txt b/board/habanalabs/dali/uboot-env-onix.txt
index d9a4f04..f0d4e91 100644
--- a/board/habanalabs/dali/uboot-env-onix.txt
+++ b/board/habanalabs/dali/uboot-env-onix.txt
@@ -1,4 +1,4 @@
 bootm_low=0
 ethaddr=00:11:22:33:44:55
-netboot=dhcp 192.168.10.86:dali-fit.itb && bootm $fileaddr#conf@1
+netboot=dhcp 192.168.10.86:goya-fit.itb && bootm $fileaddr#conf@1
 bootcmd=getfit && bootm $fileaddr#conf@1 || getfit_failure
diff --git a/board/habanalabs/dali/uboot_asic.fragment b/board/habanalabs/dali/uboot_asic.fragment
index c5c804f..0fcf240 100644
--- a/board/habanalabs/dali/uboot_asic.fragment
+++ b/board/habanalabs/dali/uboot_asic.fragment
@@ -4,7 +4,7 @@
 CONFIG_BOOTDELAY=0
 CONFIG_DEBUG_UART_CLOCK=107500000
 CONFIG_BOOTCOMMAND="getfit && bootm $fileaddr#conf@1"
-CONFIG_DEFAULT_DEVICE_TREE="dali_asic"
+CONFIG_DEFAULT_DEVICE_TREE="goya_asic"
 CONFIG_SPI_FLASH_MACRONIX=y
 CONFIG_SYS_I2C_EEPROM_BUS=1
 CONFIG_SYS_EEPROM_SIZE=256
diff --git a/board/habanalabs/dali/uboot_onix.fragment b/board/habanalabs/dali/uboot_onix.fragment
index e4a98d0..8dc7273 100644
--- a/board/habanalabs/dali/uboot_onix.fragment
+++ b/board/habanalabs/dali/uboot_onix.fragment
@@ -4,7 +4,7 @@
 CONFIG_BOOTDELAY=10
 CONFIG_DEBUG_UART_CLOCK=50000000
 CONFIG_BOOTCOMMAND="getfit && bootm $fileaddr#conf@1"
-CONFIG_DEFAULT_DEVICE_TREE="dali_onix"
+CONFIG_DEFAULT_DEVICE_TREE="goya_onix"
 CONFIG_SPI_FLASH_STMICRO=y
 CONFIG_SPI_FLASH_MACRONIX=y
 CONFIG_SYS_I2C_EEPROM_BUS=1
diff --git a/board/habanalabs/dali/uboot_pldm.fragment b/board/habanalabs/dali/uboot_pldm.fragment
index 655e20e..9c28c66 100644
--- a/board/habanalabs/dali/uboot_pldm.fragment
+++ b/board/habanalabs/dali/uboot_pldm.fragment
@@ -4,6 +4,6 @@
 CONFIG_BOOTDELAY=10
 CONFIG_DEBUG_UART_CLOCK=27000
 CONFIG_BOOTCOMMAND="getfit && bootm $fileaddr#conf@1"
-CONFIG_DEFAULT_DEVICE_TREE="dali_pldm"
+CONFIG_DEFAULT_DEVICE_TREE="goya_pldm"
 CONFIG_SPI_FLASH_MACRONIX=y
 CONFIG_PHY_FIXED=y
diff --git a/configs/dali_defconfig b/configs/dali_defconfig
index f7d88eb..f9690d0 100644
--- a/configs/dali_defconfig
+++ b/configs/dali_defconfig
@@ -4,7 +4,7 @@
 BR2_PACKAGE_OVERRIDE_FILE="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/local.mk"
 BR2_TOOLCHAIN_EXTERNAL=y
 BR2_TARGET_GENERIC_HOSTNAME="$(HL_DEV)"
-BR2_TARGET_GENERIC_ISSUE="Welcome to Dali"
+BR2_TARGET_GENERIC_ISSUE="Welcome to Goya"
 BR2_ROOTFS_DEVICE_CREATION_DYNAMIC_MDEV=y
 BR2_TARGET_GENERIC_ROOT_PASSWD="root"
 BR2_SYSTEM_BIN_SH_BASH=y
diff --git a/package/armcp/Config.in b/package/armcp/Config.in
index 7fc3453..1aafd4a 100644
--- a/package/armcp/Config.in
+++ b/package/armcp/Config.in
@@ -3,4 +3,4 @@
 	select BR2_PACKAGE_SPECS
 	select BR2_PACKAGE_SPDLOG
 	help
-	  armcp daemon is an agent to host application in monitoring Dali.
+	  armcp daemon is an agent to host application in monitoring HL devices.
diff --git a/package/specs/Config.in b/package/specs/Config.in
index 5f417d4..dc00242 100644
--- a/package/specs/Config.in
+++ b/package/specs/Config.in
@@ -1,4 +1,4 @@
 config BR2_PACKAGE_SPECS
 	bool "SPECS"
 	help
-	  Dali specifications used by armcp.
+	  Habanalabs specifications used by armcp.
diff --git a/package/specs_external/Config.in b/package/specs_external/Config.in
index b73bc60..c7cab59 100644
--- a/package/specs_external/Config.in
+++ b/package/specs_external/Config.in
@@ -1,4 +1,4 @@
 config BR2_PACKAGE_SPECS_EXTERNAL
 	bool "SPECS_EXTERNAL"
 	help
-	  Dali external specifications used by spdlog.
+	  Habanalabs external specifications used by spdlog.
