From e81e7d12f604f96737a645e3ef621d199a80cc81 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Wed, 29 Aug 2018 09:00:46 +0300
Subject: [PATCH] HL: Rename product name DALI --> GOYA [Part 2]


---

diff --git a/board/habanalabs/dali/dali_asic.its b/board/habanalabs/dali/dali_asic.its
deleted file mode 100644
index 5bcba8b..0000000
--- a/board/habanalabs/dali/dali_asic.its
+++ /dev/null
@@ -1,63 +0,0 @@
-/*
- * Simple U-Boot uImage source file containing a
- * single kernel and FDT blob
- */
-
-/dts-v1/;
-
-/ {
-	description = "Simple image with single Linux kernel and FDT blob";
-	#address-cells = <2>;
-
-	images {
-		kernel@1 {
-			description = "Goya Linux kernel";
-			data = /incbin/("Image.gz");
-			type = "kernel";
-			arch = "arm64";
-			os = "linux";
-			compression = "gzip";
-			load = <0x00000000 0x08080000>;
-			entry = <0x00000000 0x08080000>;
-			hash@1 {
-				algo = "crc32";
-			};
-			hash@2 {
-				algo = "sha1";
-			};
-		};
-		fdt@1 {
-			description = "Flattened Device Tree blob";
-			data = /incbin/("goya_asic.dtb");
-			type = "flat_dt";
-			arch = "arm64";
-			compression = "none";
-			load = <0x00000000 0x00000000>;
-			hash@1 {
-				algo = "sha1";
-			};
-		};
-		ramdisk@1 {
-			description = "Goya ramdisk";
-			data = /incbin/("rootfs.cpio.gz");
-			type = "ramdisk";
-			arch = "arm64";
-			os = "linux";
-			compression = "gzip";
-			load = <0x00000000 0x00000000>;
-			entry = <0x00000000 0x00000000>;
-			hash@1 {
-				algo = "sha1";
-			};
-		};
-	};
-	configurations {
-		default = "conf@1";
-		conf@1 {
-			description = "Boot Linux kernel with FDT blob";
-			kernel = "kernel@1";
-			ramdisk = "ramdisk@1";
-			fdt = "fdt@1";
-		};
-	};
-};
diff --git a/board/habanalabs/dali/dali_onix.its b/board/habanalabs/dali/dali_onix.its
deleted file mode 100644
index 35145b0..0000000
--- a/board/habanalabs/dali/dali_onix.its
+++ /dev/null
@@ -1,63 +0,0 @@
-/*
- * Simple U-Boot uImage source file containing a
- * single kernel and FDT blob
- */
-
-/dts-v1/;
-
-/ {
-	description = "Simple image with single Linux kernel and FDT blob";
-	#address-cells = <2>;
-
-	images {
-		kernel@1 {
-			description = "Goya Linux kernel";
-			data = /incbin/("Image.gz");
-			type = "kernel";
-			arch = "arm64";
-			os = "linux";
-			compression = "gzip";
-			load = <0x00000000 0x08080000>;
-			entry = <0x00000000 0x08080000>;
-			hash@1 {
-				algo = "crc32";
-			};
-			hash@2 {
-				algo = "sha1";
-			};
-		};
-		fdt@1 {
-			description = "Flattened Device Tree blob";
-			data = /incbin/("goya_onix.dtb");
-			type = "flat_dt";
-			arch = "arm64";
-			compression = "none";
-			load = <0x00000000 0x00000000>;
-			hash@1 {
-				algo = "sha1";
-			};
-		};
-		ramdisk@1 {
-			description = "Goya ramdisk";
-			data = /incbin/("rootfs.cpio.gz");
-			type = "ramdisk";
-			arch = "arm64";
-			os = "linux";
-			compression = "gzip";
-			load = <0x00000000 0x00000000>;
-			entry = <0x00000000 0x00000000>;
-			hash@1 {
-				algo = "sha1";
-			};
-		};
-	};
-	configurations {
-		default = "conf@1";
-		conf@1 {
-			description = "Boot Linux kernel with FDT blob";
-			kernel = "kernel@1";
-			ramdisk = "ramdisk@1";
-			fdt = "fdt@1";
-		};
-	};
-};
diff --git a/board/habanalabs/dali/dali_pldm.its b/board/habanalabs/dali/dali_pldm.its
deleted file mode 100644
index 786c11c..0000000
--- a/board/habanalabs/dali/dali_pldm.its
+++ /dev/null
@@ -1,51 +0,0 @@
-/*
- * Simple U-Boot uImage source file containing a
- * single kernel and FDT blob
- */
-
-/dts-v1/;
-
-/ {
-	description = "Simple image with single Linux kernel and FDT blob";
-	#address-cells = <2>;
-
-	images {
-		kernel@1 {
-			description = "Goya Linux kernel";
-			data = /incbin/("Image");
-			type = "kernel";
-			arch = "arm64";
-			os = "linux";
-			compression = "none";
-			load = <0x00000000 0x08080000>;
-			entry = <0x00000000 0x08080000>;
-		};
-		fdt@1 {
-			description = "Flattened Device Tree blob";
-			data = /incbin/("goya_pldm.dtb");
-			type = "flat_dt";
-			arch = "arm64";
-			compression = "none";
-			load = <0x00000000 0x00000000>;
-		};
-		ramdisk@1 {
-			description = "Goya ramdisk";
-			data = /incbin/("rootfs.cpio");
-			type = "ramdisk";
-			arch = "arm64";
-			os = "linux";
-			compression = "none";
-			load = <0x00000000 0x00000000>;
-			entry = <0x00000000 0x00000000>;
-		};
-	};
-	configurations {
-		default = "conf@1";
-		conf@1 {
-			description = "Boot Linux kernel with FDT blob";
-			kernel = "kernel@1";
-			ramdisk = "ramdisk@1";
-			fdt = "fdt@1";
-		};
-	};
-};
diff --git a/board/habanalabs/dali/linux_asic.fragment b/board/habanalabs/dali/linux_asic.fragment
deleted file mode 100644
index 03b97cc..0000000
--- a/board/habanalabs/dali/linux_asic.fragment
+++ /dev/null
@@ -1,20 +0,0 @@
-CONFIG_ARCH_GOYA=y
-CONFIG_SERIAL_MPS2_UART_CONSOLE=y
-CONFIG_SERIAL_MPS2_UART=y
-CONFIG_STMMAC_ETH=y
-CONFIG_NR_CPUS=2
-CONFIG_CMDLINE="earlycon console=ttyMPS0,115200n8 root=/dev/ram mem=256M"
-CONFIG_EEPROM_AT24=y
-CONFIG_SENSORS_POWR1220=y
-CONFIG_SENSORS_LM75=y
-CONFIG_PMBUS=y
-# CONFIG_SENSORS_PMBUS is not set
-CONFIG_SENSORS_IR35221=y
-CONFIG_THERMAL=y
-CONFIG_DEVFREQ_THERMAL=y
-CONFIG_PM_DEVFREQ=y
-CONFIG_PM_DEVFREQ_EVENT=y
-CONFIG_DEVFREQ_GOV_SIMPLE_ONDEMAND=y
-CONFIG_SPI_DESIGNWARE=y
-CONFIG_SPI_DW_MMIO=y
-CONFIG_MTD_PARTITIONED_MASTER=y
diff --git a/board/habanalabs/dali/linux_onix.fragment b/board/habanalabs/dali/linux_onix.fragment
deleted file mode 100644
index 092f9b8..0000000
--- a/board/habanalabs/dali/linux_onix.fragment
+++ /dev/null
@@ -1,19 +0,0 @@
-CONFIG_ARCH_GOYA=y
-CONFIG_SERIAL_MPS2_UART_CONSOLE=y
-CONFIG_SERIAL_MPS2_UART=y
-CONFIG_STMMAC_ETH=y
-CONFIG_NR_CPUS=2
-CONFIG_CMDLINE="earlycon console=ttyMPS0,57600n8 root=/dev/ram mem=256M"
-CONFIG_EEPROM_AT24=y
-CONFIG_SENSORS_POWR1220=y
-CONFIG_SENSORS_LM75=y
-CONFIG_PMBUS=y
-# CONFIG_SENSORS_PMBUS is not set
-CONFIG_SENSORS_IR35221=y
-CONFIG_THERMAL=y
-CONFIG_DEVFREQ_THERMAL=y
-CONFIG_PM_DEVFREQ=y
-CONFIG_PM_DEVFREQ_EVENT=y
-CONFIG_DEVFREQ_GOV_SIMPLE_ONDEMAND=y
-CONFIG_SPI_DESIGNWARE=y
-CONFIG_SPI_DW_MMIO=y
diff --git a/board/habanalabs/dali/linux_pldm.fragment b/board/habanalabs/dali/linux_pldm.fragment
deleted file mode 100644
index a70e9cd..0000000
--- a/board/habanalabs/dali/linux_pldm.fragment
+++ /dev/null
@@ -1,7 +0,0 @@
-CONFIG_ARCH_GOYA=y
-CONFIG_SERIAL_MPS2_UART_CONSOLE=y
-CONFIG_SERIAL_MPS2_UART=y
-CONFIG_STMMAC_ETH=y
-CONFIG_NR_CPUS=2
-CONFIG_HZ_12=y
-CONFIG_CMDLINE="earlycon console=ttyMPS0,1200n8 root=/dev/ram initcall_debug debug mem=160M"
diff --git a/board/habanalabs/dali/uboot-eeprom.cmd b/board/habanalabs/dali/uboot-eeprom.cmd
deleted file mode 100644
index d7744f9..0000000
--- a/board/habanalabs/dali/uboot-eeprom.cmd
+++ /dev/null
@@ -1,18 +0,0 @@
-setenv LAYOUT "-l legacy"
-mw.b 7ff0000000 ff 100
-eeprom write  $LAYOUT 7ff0000000 0 100
-eeprom update $LAYOUT "Layout Version" "0"
-eeprom update $LAYOUT "Production Serial Number" "a011223344"
-eeprom update $LAYOUT "PCB Assembly Version" "1a"
-eeprom update $LAYOUT "CPLD FW Version" "2b"
-eeprom update $LAYOUT "Infineon FW Version" "3c"
-eeprom update $LAYOUT "DDR Type and Speed" "12"
-eeprom update $LAYOUT "Fans Vendor" "3"
-eeprom update $LAYOUT "Mechanics Vendor" "4"
-eeprom update $LAYOUT "Thermal Solution Vendor" "5"
-eeprom update $LAYOUT "PCIe Gen3 or Gen4 Enabled" "6"
-eeprom update $LAYOUT "MAC Address" "00:11:22:33:44:55"
-eeprom update $LAYOUT "Board Configuration" "Goya HL1000"
-echo "eeprom content is:"
-eeprom print  $LAYOUT
-setenv LAYOUT
diff --git a/board/habanalabs/dali/uboot-env-asic.txt b/board/habanalabs/dali/uboot-env-asic.txt
deleted file mode 100644
index d0d5134..0000000
--- a/board/habanalabs/dali/uboot-env-asic.txt
+++ /dev/null
@@ -1,4 +0,0 @@
-bootm_low=0
-ethaddr=00:33:66:99:cc:ff
-netboot=dhcp 192.168.10.109:goya-fit.itb && bootm $fileaddr#conf@1
-bootcmd=getfit && bootm $fileaddr#conf@1 || getfit_failure
diff --git a/board/habanalabs/dali/uboot-env-onix.txt b/board/habanalabs/dali/uboot-env-onix.txt
deleted file mode 100644
index f0d4e91..0000000
--- a/board/habanalabs/dali/uboot-env-onix.txt
+++ /dev/null
@@ -1,4 +0,0 @@
-bootm_low=0
-ethaddr=00:11:22:33:44:55
-netboot=dhcp 192.168.10.86:goya-fit.itb && bootm $fileaddr#conf@1
-bootcmd=getfit && bootm $fileaddr#conf@1 || getfit_failure
diff --git a/board/habanalabs/dali/uboot-env-pldm.txt b/board/habanalabs/dali/uboot-env-pldm.txt
deleted file mode 100644
index 9f5255b..0000000
--- a/board/habanalabs/dali/uboot-env-pldm.txt
+++ /dev/null
@@ -1,3 +0,0 @@
-bootm_low=0
-ethaddr=00:22:44:66:88:aa
-bootcmd=getfit && bootm $fileaddr#conf@1
diff --git a/board/habanalabs/dali/uboot_asic.fragment b/board/habanalabs/dali/uboot_asic.fragment
deleted file mode 100644
index 0fcf240..0000000
--- a/board/habanalabs/dali/uboot_asic.fragment
+++ /dev/null
@@ -1,14 +0,0 @@
-CONFIG_SYS_CLK_FREQ=200000000
-CONFIG_CONARGS="serial0:115200n8"
-CONFIG_BAUDRATE=115200
-CONFIG_BOOTDELAY=0
-CONFIG_DEBUG_UART_CLOCK=107500000
-CONFIG_BOOTCOMMAND="getfit && bootm $fileaddr#conf@1"
-CONFIG_DEFAULT_DEVICE_TREE="goya_asic"
-CONFIG_SPI_FLASH_MACRONIX=y
-CONFIG_SYS_I2C_EEPROM_BUS=1
-CONFIG_SYS_EEPROM_SIZE=256
-CONFIG_SYS_EEPROM_PAGE_WRITE_BITS=4
-CONFIG_SYS_EEPROM_PAGE_WRITE_DELAY_MS=5
-CONFIG_SYS_I2C_EEPROM_ADDR=0x54
-CONFIG_SYS_I2C_EEPROM_ADDR_LEN=1
diff --git a/board/habanalabs/dali/uboot_onix.fragment b/board/habanalabs/dali/uboot_onix.fragment
deleted file mode 100644
index 8dc7273..0000000
--- a/board/habanalabs/dali/uboot_onix.fragment
+++ /dev/null
@@ -1,15 +0,0 @@
-CONFIG_SYS_CLK_FREQ=50000000
-CONFIG_CONARGS="serial0:57600n8"
-CONFIG_BAUDRATE=57600
-CONFIG_BOOTDELAY=10
-CONFIG_DEBUG_UART_CLOCK=50000000
-CONFIG_BOOTCOMMAND="getfit && bootm $fileaddr#conf@1"
-CONFIG_DEFAULT_DEVICE_TREE="goya_onix"
-CONFIG_SPI_FLASH_STMICRO=y
-CONFIG_SPI_FLASH_MACRONIX=y
-CONFIG_SYS_I2C_EEPROM_BUS=1
-CONFIG_SYS_EEPROM_SIZE=256
-CONFIG_SYS_EEPROM_PAGE_WRITE_BITS=4
-CONFIG_SYS_EEPROM_PAGE_WRITE_DELAY_MS=5
-CONFIG_SYS_I2C_EEPROM_ADDR=0x54
-CONFIG_SYS_I2C_EEPROM_ADDR_LEN=1
diff --git a/board/habanalabs/dali/uboot_pldm.fragment b/board/habanalabs/dali/uboot_pldm.fragment
deleted file mode 100644
index 9c28c66..0000000
--- a/board/habanalabs/dali/uboot_pldm.fragment
+++ /dev/null
@@ -1,9 +0,0 @@
-CONFIG_SYS_CLK_FREQ=164000
-CONFIG_CONARGS="serial0:1200n8"
-CONFIG_BAUDRATE=1200
-CONFIG_BOOTDELAY=10
-CONFIG_DEBUG_UART_CLOCK=27000
-CONFIG_BOOTCOMMAND="getfit && bootm $fileaddr#conf@1"
-CONFIG_DEFAULT_DEVICE_TREE="goya_pldm"
-CONFIG_SPI_FLASH_MACRONIX=y
-CONFIG_PHY_FIXED=y
diff --git a/board/habanalabs/goya/goya_asic.its b/board/habanalabs/goya/goya_asic.its
new file mode 100644
index 0000000..5bcba8b
--- /dev/null
+++ b/board/habanalabs/goya/goya_asic.its
@@ -0,0 +1,63 @@
+/*
+ * Simple U-Boot uImage source file containing a
+ * single kernel and FDT blob
+ */
+
+/dts-v1/;
+
+/ {
+	description = "Simple image with single Linux kernel and FDT blob";
+	#address-cells = <2>;
+
+	images {
+		kernel@1 {
+			description = "Goya Linux kernel";
+			data = /incbin/("Image.gz");
+			type = "kernel";
+			arch = "arm64";
+			os = "linux";
+			compression = "gzip";
+			load = <0x00000000 0x08080000>;
+			entry = <0x00000000 0x08080000>;
+			hash@1 {
+				algo = "crc32";
+			};
+			hash@2 {
+				algo = "sha1";
+			};
+		};
+		fdt@1 {
+			description = "Flattened Device Tree blob";
+			data = /incbin/("goya_asic.dtb");
+			type = "flat_dt";
+			arch = "arm64";
+			compression = "none";
+			load = <0x00000000 0x00000000>;
+			hash@1 {
+				algo = "sha1";
+			};
+		};
+		ramdisk@1 {
+			description = "Goya ramdisk";
+			data = /incbin/("rootfs.cpio.gz");
+			type = "ramdisk";
+			arch = "arm64";
+			os = "linux";
+			compression = "gzip";
+			load = <0x00000000 0x00000000>;
+			entry = <0x00000000 0x00000000>;
+			hash@1 {
+				algo = "sha1";
+			};
+		};
+	};
+	configurations {
+		default = "conf@1";
+		conf@1 {
+			description = "Boot Linux kernel with FDT blob";
+			kernel = "kernel@1";
+			ramdisk = "ramdisk@1";
+			fdt = "fdt@1";
+		};
+	};
+};
diff --git a/board/habanalabs/goya/goya_onix.its b/board/habanalabs/goya/goya_onix.its
new file mode 100644
index 0000000..35145b0
--- /dev/null
+++ b/board/habanalabs/goya/goya_onix.its
@@ -0,0 +1,63 @@
+/*
+ * Simple U-Boot uImage source file containing a
+ * single kernel and FDT blob
+ */
+
+/dts-v1/;
+
+/ {
+	description = "Simple image with single Linux kernel and FDT blob";
+	#address-cells = <2>;
+
+	images {
+		kernel@1 {
+			description = "Goya Linux kernel";
+			data = /incbin/("Image.gz");
+			type = "kernel";
+			arch = "arm64";
+			os = "linux";
+			compression = "gzip";
+			load = <0x00000000 0x08080000>;
+			entry = <0x00000000 0x08080000>;
+			hash@1 {
+				algo = "crc32";
+			};
+			hash@2 {
+				algo = "sha1";
+			};
+		};
+		fdt@1 {
+			description = "Flattened Device Tree blob";
+			data = /incbin/("goya_onix.dtb");
+			type = "flat_dt";
+			arch = "arm64";
+			compression = "none";
+			load = <0x00000000 0x00000000>;
+			hash@1 {
+				algo = "sha1";
+			};
+		};
+		ramdisk@1 {
+			description = "Goya ramdisk";
+			data = /incbin/("rootfs.cpio.gz");
+			type = "ramdisk";
+			arch = "arm64";
+			os = "linux";
+			compression = "gzip";
+			load = <0x00000000 0x00000000>;
+			entry = <0x00000000 0x00000000>;
+			hash@1 {
+				algo = "sha1";
+			};
+		};
+	};
+	configurations {
+		default = "conf@1";
+		conf@1 {
+			description = "Boot Linux kernel with FDT blob";
+			kernel = "kernel@1";
+			ramdisk = "ramdisk@1";
+			fdt = "fdt@1";
+		};
+	};
+};
diff --git a/board/habanalabs/goya/goya_pldm.its b/board/habanalabs/goya/goya_pldm.its
new file mode 100644
index 0000000..786c11c
--- /dev/null
+++ b/board/habanalabs/goya/goya_pldm.its
@@ -0,0 +1,51 @@
+/*
+ * Simple U-Boot uImage source file containing a
+ * single kernel and FDT blob
+ */
+
+/dts-v1/;
+
+/ {
+	description = "Simple image with single Linux kernel and FDT blob";
+	#address-cells = <2>;
+
+	images {
+		kernel@1 {
+			description = "Goya Linux kernel";
+			data = /incbin/("Image");
+			type = "kernel";
+			arch = "arm64";
+			os = "linux";
+			compression = "none";
+			load = <0x00000000 0x08080000>;
+			entry = <0x00000000 0x08080000>;
+		};
+		fdt@1 {
+			description = "Flattened Device Tree blob";
+			data = /incbin/("goya_pldm.dtb");
+			type = "flat_dt";
+			arch = "arm64";
+			compression = "none";
+			load = <0x00000000 0x00000000>;
+		};
+		ramdisk@1 {
+			description = "Goya ramdisk";
+			data = /incbin/("rootfs.cpio");
+			type = "ramdisk";
+			arch = "arm64";
+			os = "linux";
+			compression = "none";
+			load = <0x00000000 0x00000000>;
+			entry = <0x00000000 0x00000000>;
+		};
+	};
+	configurations {
+		default = "conf@1";
+		conf@1 {
+			description = "Boot Linux kernel with FDT blob";
+			kernel = "kernel@1";
+			ramdisk = "ramdisk@1";
+			fdt = "fdt@1";
+		};
+	};
+};
diff --git a/board/habanalabs/goya/linux_asic.fragment b/board/habanalabs/goya/linux_asic.fragment
new file mode 100644
index 0000000..03b97cc
--- /dev/null
+++ b/board/habanalabs/goya/linux_asic.fragment
@@ -0,0 +1,20 @@
+CONFIG_ARCH_GOYA=y
+CONFIG_SERIAL_MPS2_UART_CONSOLE=y
+CONFIG_SERIAL_MPS2_UART=y
+CONFIG_STMMAC_ETH=y
+CONFIG_NR_CPUS=2
+CONFIG_CMDLINE="earlycon console=ttyMPS0,115200n8 root=/dev/ram mem=256M"
+CONFIG_EEPROM_AT24=y
+CONFIG_SENSORS_POWR1220=y
+CONFIG_SENSORS_LM75=y
+CONFIG_PMBUS=y
+# CONFIG_SENSORS_PMBUS is not set
+CONFIG_SENSORS_IR35221=y
+CONFIG_THERMAL=y
+CONFIG_DEVFREQ_THERMAL=y
+CONFIG_PM_DEVFREQ=y
+CONFIG_PM_DEVFREQ_EVENT=y
+CONFIG_DEVFREQ_GOV_SIMPLE_ONDEMAND=y
+CONFIG_SPI_DESIGNWARE=y
+CONFIG_SPI_DW_MMIO=y
+CONFIG_MTD_PARTITIONED_MASTER=y
diff --git a/board/habanalabs/goya/linux_onix.fragment b/board/habanalabs/goya/linux_onix.fragment
new file mode 100644
index 0000000..092f9b8
--- /dev/null
+++ b/board/habanalabs/goya/linux_onix.fragment
@@ -0,0 +1,19 @@
+CONFIG_ARCH_GOYA=y
+CONFIG_SERIAL_MPS2_UART_CONSOLE=y
+CONFIG_SERIAL_MPS2_UART=y
+CONFIG_STMMAC_ETH=y
+CONFIG_NR_CPUS=2
+CONFIG_CMDLINE="earlycon console=ttyMPS0,57600n8 root=/dev/ram mem=256M"
+CONFIG_EEPROM_AT24=y
+CONFIG_SENSORS_POWR1220=y
+CONFIG_SENSORS_LM75=y
+CONFIG_PMBUS=y
+# CONFIG_SENSORS_PMBUS is not set
+CONFIG_SENSORS_IR35221=y
+CONFIG_THERMAL=y
+CONFIG_DEVFREQ_THERMAL=y
+CONFIG_PM_DEVFREQ=y
+CONFIG_PM_DEVFREQ_EVENT=y
+CONFIG_DEVFREQ_GOV_SIMPLE_ONDEMAND=y
+CONFIG_SPI_DESIGNWARE=y
+CONFIG_SPI_DW_MMIO=y
diff --git a/board/habanalabs/goya/linux_pldm.fragment b/board/habanalabs/goya/linux_pldm.fragment
new file mode 100644
index 0000000..a70e9cd
--- /dev/null
+++ b/board/habanalabs/goya/linux_pldm.fragment
@@ -0,0 +1,7 @@
+CONFIG_ARCH_GOYA=y
+CONFIG_SERIAL_MPS2_UART_CONSOLE=y
+CONFIG_SERIAL_MPS2_UART=y
+CONFIG_STMMAC_ETH=y
+CONFIG_NR_CPUS=2
+CONFIG_HZ_12=y
+CONFIG_CMDLINE="earlycon console=ttyMPS0,1200n8 root=/dev/ram initcall_debug debug mem=160M"
diff --git a/board/habanalabs/goya/uboot-eeprom.cmd b/board/habanalabs/goya/uboot-eeprom.cmd
new file mode 100644
index 0000000..d7744f9
--- /dev/null
+++ b/board/habanalabs/goya/uboot-eeprom.cmd
@@ -0,0 +1,18 @@
+setenv LAYOUT "-l legacy"
+mw.b 7ff0000000 ff 100
+eeprom write  $LAYOUT 7ff0000000 0 100
+eeprom update $LAYOUT "Layout Version" "0"
+eeprom update $LAYOUT "Production Serial Number" "a011223344"
+eeprom update $LAYOUT "PCB Assembly Version" "1a"
+eeprom update $LAYOUT "CPLD FW Version" "2b"
+eeprom update $LAYOUT "Infineon FW Version" "3c"
+eeprom update $LAYOUT "DDR Type and Speed" "12"
+eeprom update $LAYOUT "Fans Vendor" "3"
+eeprom update $LAYOUT "Mechanics Vendor" "4"
+eeprom update $LAYOUT "Thermal Solution Vendor" "5"
+eeprom update $LAYOUT "PCIe Gen3 or Gen4 Enabled" "6"
+eeprom update $LAYOUT "MAC Address" "00:11:22:33:44:55"
+eeprom update $LAYOUT "Board Configuration" "Goya HL1000"
+echo "eeprom content is:"
+eeprom print  $LAYOUT
+setenv LAYOUT
diff --git a/board/habanalabs/goya/uboot-env-asic.txt b/board/habanalabs/goya/uboot-env-asic.txt
new file mode 100644
index 0000000..d0d5134
--- /dev/null
+++ b/board/habanalabs/goya/uboot-env-asic.txt
@@ -0,0 +1,4 @@
+bootm_low=0
+ethaddr=00:33:66:99:cc:ff
+netboot=dhcp 192.168.10.109:goya-fit.itb && bootm $fileaddr#conf@1
+bootcmd=getfit && bootm $fileaddr#conf@1 || getfit_failure
diff --git a/board/habanalabs/goya/uboot-env-onix.txt b/board/habanalabs/goya/uboot-env-onix.txt
new file mode 100644
index 0000000..f0d4e91
--- /dev/null
+++ b/board/habanalabs/goya/uboot-env-onix.txt
@@ -0,0 +1,4 @@
+bootm_low=0
+ethaddr=00:11:22:33:44:55
+netboot=dhcp 192.168.10.86:goya-fit.itb && bootm $fileaddr#conf@1
+bootcmd=getfit && bootm $fileaddr#conf@1 || getfit_failure
diff --git a/board/habanalabs/goya/uboot-env-pldm.txt b/board/habanalabs/goya/uboot-env-pldm.txt
new file mode 100644
index 0000000..9f5255b
--- /dev/null
+++ b/board/habanalabs/goya/uboot-env-pldm.txt
@@ -0,0 +1,3 @@
+bootm_low=0
+ethaddr=00:22:44:66:88:aa
+bootcmd=getfit && bootm $fileaddr#conf@1
diff --git a/board/habanalabs/goya/uboot_asic.fragment b/board/habanalabs/goya/uboot_asic.fragment
new file mode 100644
index 0000000..0fcf240
--- /dev/null
+++ b/board/habanalabs/goya/uboot_asic.fragment
@@ -0,0 +1,14 @@
+CONFIG_SYS_CLK_FREQ=200000000
+CONFIG_CONARGS="serial0:115200n8"
+CONFIG_BAUDRATE=115200
+CONFIG_BOOTDELAY=0
+CONFIG_DEBUG_UART_CLOCK=107500000
+CONFIG_BOOTCOMMAND="getfit && bootm $fileaddr#conf@1"
+CONFIG_DEFAULT_DEVICE_TREE="goya_asic"
+CONFIG_SPI_FLASH_MACRONIX=y
+CONFIG_SYS_I2C_EEPROM_BUS=1
+CONFIG_SYS_EEPROM_SIZE=256
+CONFIG_SYS_EEPROM_PAGE_WRITE_BITS=4
+CONFIG_SYS_EEPROM_PAGE_WRITE_DELAY_MS=5
+CONFIG_SYS_I2C_EEPROM_ADDR=0x54
+CONFIG_SYS_I2C_EEPROM_ADDR_LEN=1
diff --git a/board/habanalabs/goya/uboot_onix.fragment b/board/habanalabs/goya/uboot_onix.fragment
new file mode 100644
index 0000000..8dc7273
--- /dev/null
+++ b/board/habanalabs/goya/uboot_onix.fragment
@@ -0,0 +1,15 @@
+CONFIG_SYS_CLK_FREQ=50000000
+CONFIG_CONARGS="serial0:57600n8"
+CONFIG_BAUDRATE=57600
+CONFIG_BOOTDELAY=10
+CONFIG_DEBUG_UART_CLOCK=50000000
+CONFIG_BOOTCOMMAND="getfit && bootm $fileaddr#conf@1"
+CONFIG_DEFAULT_DEVICE_TREE="goya_onix"
+CONFIG_SPI_FLASH_STMICRO=y
+CONFIG_SPI_FLASH_MACRONIX=y
+CONFIG_SYS_I2C_EEPROM_BUS=1
+CONFIG_SYS_EEPROM_SIZE=256
+CONFIG_SYS_EEPROM_PAGE_WRITE_BITS=4
+CONFIG_SYS_EEPROM_PAGE_WRITE_DELAY_MS=5
+CONFIG_SYS_I2C_EEPROM_ADDR=0x54
+CONFIG_SYS_I2C_EEPROM_ADDR_LEN=1
diff --git a/board/habanalabs/goya/uboot_pldm.fragment b/board/habanalabs/goya/uboot_pldm.fragment
new file mode 100644
index 0000000..9c28c66
--- /dev/null
+++ b/board/habanalabs/goya/uboot_pldm.fragment
@@ -0,0 +1,9 @@
+CONFIG_SYS_CLK_FREQ=164000
+CONFIG_CONARGS="serial0:1200n8"
+CONFIG_BAUDRATE=1200
+CONFIG_BOOTDELAY=10
+CONFIG_DEBUG_UART_CLOCK=27000
+CONFIG_BOOTCOMMAND="getfit && bootm $fileaddr#conf@1"
+CONFIG_DEFAULT_DEVICE_TREE="goya_pldm"
+CONFIG_SPI_FLASH_MACRONIX=y
+CONFIG_PHY_FIXED=y
diff --git a/configs/dali_defconfig b/configs/dali_defconfig
deleted file mode 100644
index f9690d0..0000000
--- a/configs/dali_defconfig
+++ /dev/null
@@ -1,68 +0,0 @@
-BR2_aarch64=y
-BR2_SVN="svn"
-BR2_DL_DIR="$(BR2_EXTERNAL_HABANALABS_PATH)/dl"
-BR2_PACKAGE_OVERRIDE_FILE="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/local.mk"
-BR2_TOOLCHAIN_EXTERNAL=y
-BR2_TARGET_GENERIC_HOSTNAME="$(HL_DEV)"
-BR2_TARGET_GENERIC_ISSUE="Welcome to Goya"
-BR2_ROOTFS_DEVICE_CREATION_DYNAMIC_MDEV=y
-BR2_TARGET_GENERIC_ROOT_PASSWD="root"
-BR2_SYSTEM_BIN_SH_BASH=y
-BR2_SYSTEM_DHCP="eth0"
-BR2_ROOTFS_OVERLAY="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/rootfs_overlay"
-BR2_ROOTFS_POST_BUILD_SCRIPT="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/post-build.sh"
-BR2_ROOTFS_POST_IMAGE_SCRIPT="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/post-image.sh"
-BR2_LINUX_KERNEL=y
-BR2_LINUX_KERNEL_USE_ARCH_DEFAULT_CONFIG=y
-BR2_LINUX_KERNEL_CONFIG_FRAGMENT_FILES="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/linux_$(HL_TARGET).fragment"
-BR2_LINUX_KERNEL_IMAGE_TARGET_CUSTOM=y
-BR2_LINUX_KERNEL_IMAGE_TARGET_NAME="Image.gz"
-BR2_LINUX_KERNEL_DTS_SUPPORT=y
-BR2_LINUX_KERNEL_INTREE_DTS_NAME="habanalabs/$(HL_DEV)_$(HL_TARGET)"
-BR2_PACKAGE_BUSYBOX_CONFIG="package/busybox/busybox-minimal.config"
-BR2_PACKAGE_BUSYBOX_CONFIG_FRAGMENT_FILES="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/busybox.fragment"
-BR2_PACKAGE_BUSYBOX_SHOW_OTHERS=y
-BR2_PACKAGE_BUSYBOX_WATCHDOG=y
-BR2_PACKAGE_MTD=y
-# BR2_PACKAGE_MTD_MTD_DEBUG is not set
-# BR2_PACKAGE_MTD_NANDDUMP is not set
-# BR2_PACKAGE_MTD_NANDTEST is not set
-# BR2_PACKAGE_MTD_NANDWRITE is not set
-# BR2_PACKAGE_MTD_UBIATTACH is not set
-# BR2_PACKAGE_MTD_UBICRC32 is not set
-# BR2_PACKAGE_MTD_UBIDETACH is not set
-# BR2_PACKAGE_MTD_UBIFORMAT is not set
-# BR2_PACKAGE_MTD_UBIMKVOL is not set
-# BR2_PACKAGE_MTD_UBINFO is not set
-# BR2_PACKAGE_MTD_UBINIZE is not set
-# BR2_PACKAGE_MTD_UBIRENAME is not set
-# BR2_PACKAGE_MTD_UBIRMVOL is not set
-# BR2_PACKAGE_MTD_UBIRSVOL is not set
-# BR2_PACKAGE_MTD_UBIUPDATEVOL is not set
-# BR2_PACKAGE_MTD_UBIBLOCK is not set
-BR2_PACKAGE_I2C_TOOLS=y
-BR2_PACKAGE_LM_SENSORS=y
-BR2_PACKAGE_KMOD=y
-BR2_TARGET_ROOTFS_CPIO=y
-BR2_TARGET_ROOTFS_CPIO_GZIP=y
-# BR2_TARGET_ROOTFS_TAR is not set
-BR2_TARGET_UBOOT=y
-BR2_TARGET_UBOOT_BUILD_SYSTEM_KCONFIG=y
-BR2_TARGET_UBOOT_BOARD_DEFCONFIG="$(HL_DEV)"
-BR2_TARGET_UBOOT_CONFIG_FRAGMENT_FILES="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/uboot_$(HL_TARGET).fragment"
-BR2_TARGET_UBOOT_NEEDS_DTC=y
-BR2_TARGET_UBOOT_SPL=y
-BR2_TARGET_UBOOT_ENVIMAGE=y
-BR2_TARGET_UBOOT_ENVIMAGE_SOURCE="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/uboot-env-$(HL_TARGET).txt"
-BR2_TARGET_UBOOT_ENVIMAGE_SIZE="0x1000"
-BR2_TARGET_UBOOT_ENVIMAGE_REDUNDANT=y
-BR2_PACKAGE_HOST_UBOOT_TOOLS=y
-BR2_PACKAGE_HOST_UBOOT_TOOLS_FIT_SUPPORT=y
-BR2_PACKAGE_ARMCP=y
-BR2_PACKAGE_UBOOT_TEST_HOOKS=y
-BR2_PACKAGE_TBOT=y
-BR2_PACKAGE_DSV=y
-BR2_PACKAGE_THERMAL=y
-BR2_PACKAGE_PREBOOT=y
-BR2_PACKAGE_HOST_HL_FW_LOADER=y
-BR2_PACKAGE_BTL=y
diff --git a/configs/goya_defconfig b/configs/goya_defconfig
new file mode 100644
index 0000000..f9690d0
--- /dev/null
+++ b/configs/goya_defconfig
@@ -0,0 +1,68 @@
+BR2_aarch64=y
+BR2_SVN="svn"
+BR2_DL_DIR="$(BR2_EXTERNAL_HABANALABS_PATH)/dl"
+BR2_PACKAGE_OVERRIDE_FILE="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/local.mk"
+BR2_TOOLCHAIN_EXTERNAL=y
+BR2_TARGET_GENERIC_HOSTNAME="$(HL_DEV)"
+BR2_TARGET_GENERIC_ISSUE="Welcome to Goya"
+BR2_ROOTFS_DEVICE_CREATION_DYNAMIC_MDEV=y
+BR2_TARGET_GENERIC_ROOT_PASSWD="root"
+BR2_SYSTEM_BIN_SH_BASH=y
+BR2_SYSTEM_DHCP="eth0"
+BR2_ROOTFS_OVERLAY="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/rootfs_overlay"
+BR2_ROOTFS_POST_BUILD_SCRIPT="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/post-build.sh"
+BR2_ROOTFS_POST_IMAGE_SCRIPT="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/post-image.sh"
+BR2_LINUX_KERNEL=y
+BR2_LINUX_KERNEL_USE_ARCH_DEFAULT_CONFIG=y
+BR2_LINUX_KERNEL_CONFIG_FRAGMENT_FILES="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/linux_$(HL_TARGET).fragment"
+BR2_LINUX_KERNEL_IMAGE_TARGET_CUSTOM=y
+BR2_LINUX_KERNEL_IMAGE_TARGET_NAME="Image.gz"
+BR2_LINUX_KERNEL_DTS_SUPPORT=y
+BR2_LINUX_KERNEL_INTREE_DTS_NAME="habanalabs/$(HL_DEV)_$(HL_TARGET)"
+BR2_PACKAGE_BUSYBOX_CONFIG="package/busybox/busybox-minimal.config"
+BR2_PACKAGE_BUSYBOX_CONFIG_FRAGMENT_FILES="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/busybox.fragment"
+BR2_PACKAGE_BUSYBOX_SHOW_OTHERS=y
+BR2_PACKAGE_BUSYBOX_WATCHDOG=y
+BR2_PACKAGE_MTD=y
+# BR2_PACKAGE_MTD_MTD_DEBUG is not set
+# BR2_PACKAGE_MTD_NANDDUMP is not set
+# BR2_PACKAGE_MTD_NANDTEST is not set
+# BR2_PACKAGE_MTD_NANDWRITE is not set
+# BR2_PACKAGE_MTD_UBIATTACH is not set
+# BR2_PACKAGE_MTD_UBICRC32 is not set
+# BR2_PACKAGE_MTD_UBIDETACH is not set
+# BR2_PACKAGE_MTD_UBIFORMAT is not set
+# BR2_PACKAGE_MTD_UBIMKVOL is not set
+# BR2_PACKAGE_MTD_UBINFO is not set
+# BR2_PACKAGE_MTD_UBINIZE is not set
+# BR2_PACKAGE_MTD_UBIRENAME is not set
+# BR2_PACKAGE_MTD_UBIRMVOL is not set
+# BR2_PACKAGE_MTD_UBIRSVOL is not set
+# BR2_PACKAGE_MTD_UBIUPDATEVOL is not set
+# BR2_PACKAGE_MTD_UBIBLOCK is not set
+BR2_PACKAGE_I2C_TOOLS=y
+BR2_PACKAGE_LM_SENSORS=y
+BR2_PACKAGE_KMOD=y
+BR2_TARGET_ROOTFS_CPIO=y
+BR2_TARGET_ROOTFS_CPIO_GZIP=y
+# BR2_TARGET_ROOTFS_TAR is not set
+BR2_TARGET_UBOOT=y
+BR2_TARGET_UBOOT_BUILD_SYSTEM_KCONFIG=y
+BR2_TARGET_UBOOT_BOARD_DEFCONFIG="$(HL_DEV)"
+BR2_TARGET_UBOOT_CONFIG_FRAGMENT_FILES="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/uboot_$(HL_TARGET).fragment"
+BR2_TARGET_UBOOT_NEEDS_DTC=y
+BR2_TARGET_UBOOT_SPL=y
+BR2_TARGET_UBOOT_ENVIMAGE=y
+BR2_TARGET_UBOOT_ENVIMAGE_SOURCE="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/uboot-env-$(HL_TARGET).txt"
+BR2_TARGET_UBOOT_ENVIMAGE_SIZE="0x1000"
+BR2_TARGET_UBOOT_ENVIMAGE_REDUNDANT=y
+BR2_PACKAGE_HOST_UBOOT_TOOLS=y
+BR2_PACKAGE_HOST_UBOOT_TOOLS_FIT_SUPPORT=y
+BR2_PACKAGE_ARMCP=y
+BR2_PACKAGE_UBOOT_TEST_HOOKS=y
+BR2_PACKAGE_TBOT=y
+BR2_PACKAGE_DSV=y
+BR2_PACKAGE_THERMAL=y
+BR2_PACKAGE_PREBOOT=y
+BR2_PACKAGE_HOST_HL_FW_LOADER=y
+BR2_PACKAGE_BTL=y
