From 3d94d59c2f06e8dfbec00cc0b6c6bdd2c3f96599 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Mon, 03 Sep 2018 14:31:53 +0300
Subject: [PATCH] Rootfs overlay: remove i2c binding to IR device

Device driver have OF support
---

diff --git a/board/habanalabs/common/rootfs_overlay/etc/init.d/S99misc b/board/habanalabs/common/rootfs_overlay/etc/init.d/S99misc
index 90b4093..3641d87 100755
--- a/board/habanalabs/common/rootfs_overlay/etc/init.d/S99misc
+++ b/board/habanalabs/common/rootfs_overlay/etc/init.d/S99misc
@@ -7,7 +7,7 @@
   start)
 	echo "Starting misc..."
 	echo powr1220 0x25 > /sys/bus/i2c/devices/i2c-0/new_device
-	echo ir35221 0x40 > /sys/bus/i2c/devices/i2c-0/new_device
+	#echo ir35221 0x40 > /sys/bus/i2c/devices/i2c-0/new_device
 	echo 0 > /sys/module/rcupdate/parameters/rcu_cpu_stall_suppress
 	for tz in /sys/class/thermal/thermal_zone*;
 	do
