From 4697886eac5d240588b8003cecbb36e7c3275aca Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Mon, 03 Sep 2018 15:17:12 +0300
Subject: [PATCH] Linux fragment: add CONFIG_I2C_DESIGNWARE_SLAVE

Needed by Future connection with BMC
---

diff --git a/board/habanalabs/goya/linux_asic.fragment b/board/habanalabs/goya/linux_asic.fragment
index 03b97cc..6a726ed 100644
--- a/board/habanalabs/goya/linux_asic.fragment
+++ b/board/habanalabs/goya/linux_asic.fragment
@@ -18,3 +18,4 @@
 CONFIG_SPI_DESIGNWARE=y
 CONFIG_SPI_DW_MMIO=y
 CONFIG_MTD_PARTITIONED_MASTER=y
+CONFIG_I2C_DESIGNWARE_SLAVE=y
