From a8f27be44a7bfea722682915389be5197b92170c Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Thu, 27 Sep 2018 10:20:50 +0300
Subject: [PATCH] Linux: remove simple_ondemand governor

Will use new hl_pid governor instead
---

diff --git a/board/habanalabs/gaudi/linux_asic.fragment b/board/habanalabs/gaudi/linux_asic.fragment
index cd5df62..ef3bc3e 100644
--- a/board/habanalabs/gaudi/linux_asic.fragment
+++ b/board/habanalabs/gaudi/linux_asic.fragment
@@ -14,7 +14,6 @@
 CONFIG_DEVFREQ_THERMAL=y
 CONFIG_PM_DEVFREQ=y
 CONFIG_PM_DEVFREQ_EVENT=y
-CONFIG_DEVFREQ_GOV_SIMPLE_ONDEMAND=y
 CONFIG_SPI_DESIGNWARE=y
 CONFIG_SPI_DW_MMIO=y
 CONFIG_MTD_PARTITIONED_MASTER=y
diff --git a/board/habanalabs/gaudi/linux_onix.fragment b/board/habanalabs/gaudi/linux_onix.fragment
index c0adf52..38f5e7e 100644
--- a/board/habanalabs/gaudi/linux_onix.fragment
+++ b/board/habanalabs/gaudi/linux_onix.fragment
@@ -14,6 +14,5 @@
 CONFIG_DEVFREQ_THERMAL=y
 CONFIG_PM_DEVFREQ=y
 CONFIG_PM_DEVFREQ_EVENT=y
-CONFIG_DEVFREQ_GOV_SIMPLE_ONDEMAND=y
 CONFIG_SPI_DESIGNWARE=y
 CONFIG_SPI_DW_MMIO=y
diff --git a/board/habanalabs/goya/linux_asic.fragment b/board/habanalabs/goya/linux_asic.fragment
index 6a726ed..f9af0f8 100644
--- a/board/habanalabs/goya/linux_asic.fragment
+++ b/board/habanalabs/goya/linux_asic.fragment
@@ -14,7 +14,6 @@
 CONFIG_DEVFREQ_THERMAL=y
 CONFIG_PM_DEVFREQ=y
 CONFIG_PM_DEVFREQ_EVENT=y
-CONFIG_DEVFREQ_GOV_SIMPLE_ONDEMAND=y
 CONFIG_SPI_DESIGNWARE=y
 CONFIG_SPI_DW_MMIO=y
 CONFIG_MTD_PARTITIONED_MASTER=y
diff --git a/board/habanalabs/goya/linux_onix.fragment b/board/habanalabs/goya/linux_onix.fragment
index 092f9b8..40930aa 100644
--- a/board/habanalabs/goya/linux_onix.fragment
+++ b/board/habanalabs/goya/linux_onix.fragment
@@ -14,6 +14,5 @@
 CONFIG_DEVFREQ_THERMAL=y
 CONFIG_PM_DEVFREQ=y
 CONFIG_PM_DEVFREQ_EVENT=y
-CONFIG_DEVFREQ_GOV_SIMPLE_ONDEMAND=y
 CONFIG_SPI_DESIGNWARE=y
 CONFIG_SPI_DW_MMIO=y
