From 175b963da58c1bb8d886cb10ff797d2d62fbf1ab Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Thu, 04 Oct 2018 13:13:13 +0300
Subject: [PATCH] uboot [eeprom script]: Fix syntax


---

diff --git a/board/habanalabs/gaudi/uboot-eeprom.cmd b/board/habanalabs/gaudi/uboot-eeprom.cmd
index f234531..97e5049 100644
--- a/board/habanalabs/gaudi/uboot-eeprom.cmd
+++ b/board/habanalabs/gaudi/uboot-eeprom.cmd
@@ -1,6 +1,6 @@
 setenv LAYOUT "-l legacy"
-mw.b 7ff0000000 ff 100
-eeprom write  $LAYOUT 7ff0000000 0 100
+mw.b 0x7ff0000000 ff 100
+eeprom write 0x7ff0000000 0 100
 eeprom update $LAYOUT "Layout Version" "0"
 eeprom update $LAYOUT "Production Serial Number" "a011223344"
 eeprom update $LAYOUT "PCB Assembly Version" "1a"
diff --git a/board/habanalabs/goya/uboot-eeprom.cmd b/board/habanalabs/goya/uboot-eeprom.cmd
index fa23316..177fecc 100644
--- a/board/habanalabs/goya/uboot-eeprom.cmd
+++ b/board/habanalabs/goya/uboot-eeprom.cmd
@@ -1,6 +1,6 @@
 setenv LAYOUT "-l v1"
-mw.b 7ff0000000 ff 100
-eeprom write  $LAYOUT 7ff0000000 0 100
+mw.b 0x7ff0000000 ff 100
+eeprom write 0x7ff0000000 0 100
 eeprom update $LAYOUT "Layout Version" "1"
 eeprom update $LAYOUT "DDR Type and Speed" "0314"
 eeprom update $LAYOUT "PCIe Gen3 or Gen4 Enabled" "3"
