From 7e3792f72ab93cae75afdbd683cf0d629c026fce Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Fri, 05 Oct 2018 21:26:10 +0300
Subject: [PATCH] gaudi-pldm: add support

Add proper targets and see that build target pass
---

diff --git a/Makefile b/Makefile
index 19dccc2..b5f10f8 100644
--- a/Makefile
+++ b/Makefile
@@ -101,9 +101,9 @@
 hl-goya hl-gaudi: $(OUTPUT_DIR)
 	@echo "export HL_DEV=$(patsubst hl-%,%,$@)"		 > $(HL_VARS_FILE)
 
-.PHONY: gaudi-onix-target
-gaudi-onix-target: hl-gaudi
-	@echo "export HL_TARGET=$(patsubst gaudi-%-target,%,$@)"	>> $(HL_VARS_FILE)
+.PHONY: gaudi-onix-target gaudi-pldm-target
+gaudi-onix-target gaudi-pldm-target: hl-gaudi
+	@echo "export HL_TARGET=$(patsubst gaudi-%-target,%,$@)" >> $(HL_VARS_FILE)
 
 .PHONY: goya-onix-target goya-pldm-target goya-asic-target
 goya-onix-target goya-pldm-target goya-asic-target: hl-goya
@@ -115,14 +115,15 @@
 goya-prepare: $(BR2_EXTERNAL)/configs/goya_defconfig
 	@$(MAKE) $(<F)
 
-.PHONY: goya-onix-build goya-pldm-build goya-asic-build gaudi-onix-build
+.PHONY: goya-onix-build goya-pldm-build goya-asic-build gaudi-onix-build gaudi-pldm-build
 gaudi-onix-build: gaudi-onix-target
+gaudi-pldm-build: gaudi-pldm-target
 goya-onix-build: goya-onix-target
 goya-pldm-build: goya-pldm-target
 goya-asic-build: goya-asic-target
-gaudi-onix-build: $(OUTPUT_DIR) gaudi-prepare
+gaudi-onix-build gaudi-pldm-build: $(OUTPUT_DIR) gaudi-prepare
 goya-onix-build goya-pldm-build goya-asic-build: $(OUTPUT_DIR) goya-prepare
-goya-onix-build goya-pldm-build goya-asic-build gaudi-onix-build:
+goya-onix-build goya-pldm-build goya-asic-build gaudi-onix-build gaudi-pldm-build:
 	@$(MAKE) all
 
 .PHONY: goya-rpi-build goya-rpi-armcp goya-rpi-gpio-mdio goya-rpi-regs-cli
@@ -150,14 +151,15 @@
 LINUX_BASE := ${OUTPUT_DIR}/build/linux-custom
 UBOOT_XMPL := ${UBOOT_BASE}/examples/standalone
 DSV_BASE := ${OUTPUT_DIR}/build/dsv-custom
-.PHONY: goya-onix-install goya-pldm-install goya-asic-install gaudi-onix-install
+.PHONY: goya-onix-install goya-pldm-install goya-asic-install gaudi-onix-install gaudi-pldm-install
 goya-onix-install gaudi-onix-install: GOYA_INSTALL_LAB_IP ?= monster4-nuc
 goya-onix-install gaudi-onix-install: GOYA_INSTALL_HOST_IP ?= monster4
-goya-pldm-install: GOYA_INSTALL_LAB_IP ?= 192.168.10.45
-goya-pldm-install: GOYA_INSTALL_HOST_IP ?= 192.168.10.45
+goya-pldm-install gaudi-pldm-install: GOYA_INSTALL_LAB_IP ?= 192.168.10.45
+goya-pldm-install gaudi-pldm-install: GOYA_INSTALL_HOST_IP ?= 192.168.10.45
 goya-asic-install: GOYA_INSTALL_LAB_IP ?= 192.168.10.109
 goya-asic-install: GOYA_INSTALL_HOST_IP ?= 192.168.10.134
 gaudi-onix-install: gaudi-onix-target
+gaudi-pldm-install: gaudi-pldm-target
 goya-onix-install: goya-onix-target
 goya-pldm-install: goya-pldm-target
 goya-asic-install: goya-asic-target
@@ -184,14 +186,14 @@
 goya-install gaudi-install:
 	@$(MAKE) hl-install GOYA_INSTALL_HOST_IP=$(GOYA_INSTALL_HOST_IP) GOYA_INSTALL_LAB_IP=$(GOYA_INSTALL_LAB_IP)
 
-gaudi-onix-install: gaudi-install
+gaudi-onix-install gaudi-pldm-install: gaudi-install
 goya-onix-install goya-pldm-install goya-asic-install: goya-install
 goya-onix-install gaudi-onix-install:
 	@
 PLDM_HEX_DIR ?= $(IMAGES_DIR)/tools/hl_shared_dir/pldm
 ${PLDM_HEX_DIR}:
 	@mkdir -p $@
-goya-pldm-install: ${PLDM_HEX_DIR}
+goya-pldm-install gaudi-pldm-install: ${PLDM_HEX_DIR}
 	-cp $(IMAGES_DIR)/$(HL_DEV)-spi-flash.hex $</
 goya-asic-install:
 	@
diff --git a/board/habanalabs/gaudi/linux_pldm.fragment b/board/habanalabs/gaudi/linux_pldm.fragment
index 2e1189d..3125f2e 100644
--- a/board/habanalabs/gaudi/linux_pldm.fragment
+++ b/board/habanalabs/gaudi/linux_pldm.fragment
@@ -1,7 +1,6 @@
 CONFIG_ARCH_GAUDI=y
 CONFIG_SERIAL_MPS2_UART_CONSOLE=y
 CONFIG_SERIAL_MPS2_UART=y
-CONFIG_STMMAC_ETH=y
 CONFIG_NR_CPUS=2
 CONFIG_HZ_12=y
 CONFIG_CMDLINE="earlycon console=ttyMPS0,1200n8 root=/dev/ram initcall_debug debug mem=160M"
diff --git a/board/habanalabs/gaudi/uboot-env-pldm.txt b/board/habanalabs/gaudi/uboot-env-pldm.txt
index 9f5255b..3fc2c10 100644
--- a/board/habanalabs/gaudi/uboot-env-pldm.txt
+++ b/board/habanalabs/gaudi/uboot-env-pldm.txt
@@ -1,3 +1,2 @@
 bootm_low=0
-ethaddr=00:22:44:66:88:aa
 bootcmd=getfit && bootm $fileaddr#conf@1
