From 945a18179bcd042b04d48c18760401fe0b76de03 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 07 Oct 2018 13:35:16 +0300
Subject: [PATCH] Makefile: fix type in help message

s/borad/board/

Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/Makefile b/Makefile
index 5a58c2f..7752fa2 100644
--- a/Makefile
+++ b/Makefile
@@ -19,7 +19,7 @@
 	@echo
 	@echo 'Build:'
 	@echo '  goya-rpi-build       - Build for Raspberry Pi'
-	@echo '  goya-<borad>-build   - Build for <board> e.g. onix, pldm, asic'
+	@echo '  goya-<board>-build   - Build for <board> e.g. onix, pldm, asic'
 	@echo '  optional environment variables:'
 	@echo '    SPL_BINFILE        - Alternative SPL binary <PATH>'
 	@echo '    UBOOT_ITSFILE      - Alternative ITS file <PATH>'
