From 8d6b42599fba901baa99cdc83fb147fa2cbbe98c Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 14 Oct 2018 18:27:17 +0300
Subject: [PATCH] Makefile: release: fixup the jenkins job token name

To generalize the release job even more, change the auth. token.

Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/Makefile b/Makefile
index 15c1474..aabe227 100644
--- a/Makefile
+++ b/Makefile
@@ -80,7 +80,7 @@
 RELEASE_JOB_VER_COMPAT := "release_version=$(RELEASE_VER_COMPAT)"
 RELEASE_JOB_URL := http://192.168.0.4:8080/job/goya-release-test/buildWithParameters
 RELEASE_JOB_CAUSE := "cause=Started_by_$(shell whoami)"
-RELEASE_JOB_TOKEN := "token=goya-release"
+RELEASE_JOB_TOKEN := "token=hl-release"
 RELEASE_JOB_PARAMS := $(RELEASE_JOB_TOKEN)&$(RELEASE_JOB_CAUSE)&$(RELEASE_JOB_BRANCH)&$(RELEASE_JOB_VER_COMPAT)
 
 define NO_RELEASE_BRANCH_ERROR =
