From 37d3c27f17ecbd542b5f3e1478dd69f99408c1c9 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Mon, 22 Oct 2018 13:17:58 +0300
Subject: [PATCH] goya-asic-install: update GOYA_INSTALL_HOST_IP

Like swhat provided by DHCP instead of static IP
---

diff --git a/Makefile b/Makefile
index aabe227..d39c60c 100644
--- a/Makefile
+++ b/Makefile
@@ -185,7 +185,7 @@
 goya-pldm-install gaudi-pldm-install: GOYA_INSTALL_LAB_IP ?= 192.168.10.45
 goya-pldm-install gaudi-pldm-install: GOYA_INSTALL_HOST_IP ?= 192.168.10.45
 goya-asic-install: GOYA_INSTALL_LAB_IP ?= 192.168.10.109
-goya-asic-install: GOYA_INSTALL_HOST_IP ?= 192.168.10.134
+goya-asic-install: GOYA_INSTALL_HOST_IP ?= 192.168.10.133
 gaudi-onix-install: gaudi-onix-target
 gaudi-pldm-install: gaudi-pldm-target
 goya-onix-install: goya-onix-target
