From a449e588ca7340493cb4bb738e6663f28865c3dc Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Tue, 08 Dec 2020 20:15:45 +0200
Subject: [PATCH] [DO NOT MERGE] goya2: disable replace signature for fit images

Change-Id: I15d4794b16f13b230b5f011cb3f2cdee857a8e64
---

diff --git a/board/habanalabs/common/post-image-fit-global.sh b/board/habanalabs/common/post-image-fit-global.sh
index 04aac1d..9de587f 100755
--- a/board/habanalabs/common/post-image-fit-global.sh
+++ b/board/habanalabs/common/post-image-fit-global.sh
@@ -159,14 +159,14 @@
 	sed ${ITS_SED_PARAMS} ${UBOOT_ZEPHYR_ITS_TEMPLATE} > ${UBOOT_ZEPHYR_ITS_FILE}
 	${MKIMAGE} -D "-W no-unit_address_vs_reg" -f ${UBOOT_ZEPHYR_ITS_FILE} -k ${KEYS_DIR} ${UBOOT_ZEPHYR_ITB_FILE}
 
-	replace_signature_in_fit ${UBOOT_ZEPHYR_ITB_FILE} ${FDT_PREBOOT_SIG_NODE}
+	#replace_signature_in_fit ${UBOOT_ZEPHYR_ITB_FILE} ${FDT_PREBOOT_SIG_NODE}
 
 	dd status=none conv=sync bs=4 if=${UBOOT_ZEPHYR_ITB_FILE} > ${DEV_NAME}-boot-fit.itb
 
 	sed ${ITS_SED_PARAMS} ${LINUX_ITS_TEMPLATE} > ${LINUX_ITS_FILE}
 	${MKIMAGE} -D "-W no-unit_address_vs_reg" -f ${LINUX_ITS_FILE} -k ${KEYS_DIR} ${LINUX_ITB_FILE}
 
-	replace_signature_in_fit ${LINUX_ITB_FILE} ${FDT_LINUX_SIG_NODE}
+	#replace_signature_in_fit ${LINUX_ITB_FILE} ${FDT_LINUX_SIG_NODE}
 
 	dd status=none conv=sync bs=4 if=${LINUX_ITB_FILE} > ${DEV_NAME}-fit.itb
 
