From 0559b5c88824a1e86f06ed346951962e0e275f84 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Fri, 12 Mar 2021 06:21:00 +0200
Subject: [PATCH] [SW-31828] gaudi: update bar4 size add comment

No fuctional change

Change-Id: I04bc0e2501fe7d3c69437c57969da53cf335886e
---

diff --git a/gaudi/common.c b/gaudi/common.c
index 90bf522..957a49f 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -160,6 +160,7 @@
 	regmap_writel(0x0004A07D, mmPCIE_PHY_CR_PARA_CMD);
 	udelay(20);
 
+	/* Set BAR4 size to 32GB */
 	regmap_writel(0x00100000, mmPCIE_AUX_DBI);
 	regmap_writel(0xFFFFFFFF, mmPCIE_DBI_BAR4_REG);
 	regmap_writel(0x00000007, mmPCIE_DBI_BAR5_REG);
