From c45ea74b840b724ca8788bd21e7eed7cf39ebf76 Mon Sep 17 00:00:00 2001
From: Ran Shalit <rshalit@habana.ai>
Date: Wed, 24 Mar 2021 17:03:43 +0200
Subject: [PATCH] add nic-en-create

Change-Id: I60f5579d78262cc20b6a75d5d8126d88635589cc
Signed-off-by: Ran Shalit <rshalit@habana.ai>
---

diff --git a/scripts/make/gaudi/Makefile b/scripts/make/gaudi/Makefile
index 613be62..38e0aae 100644
--- a/scripts/make/gaudi/Makefile
+++ b/scripts/make/gaudi/Makefile
@@ -115,6 +115,10 @@
 gaudi-otp-create:
 	$(HOST_MAKE_ENV) ${BR2_EXTERNAL}/board/habanalabs/$(HL_DEV)/create-otp-image.sh
 
+.PHONY: gaudi-nic-en-create
+gaudi-nic-en-create:
+	${BR2_EXTERNAL}/board/habanalabs/$(HL_DEV)/create-simple.sh nic_en
+
 .PHONY: gaudi-otp-install
 gaudi-otp-install:
 	-sshpass -p$(HL_PASSWD) $(SCP) $(IMAGES_DIR)/${HL_DEV}-otp.itb	root@$(HL_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
