From 831952f37a04561cf67c412bc6c1474288cefc0c Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Thu, 25 Mar 2021 09:57:18 +0200
Subject: [PATCH] [SW-0] fit: turn fit_pub_key_value_node_get() static

Keep name spacing clean

Change-Id: I3d062cf690687b5704c81814b488549c58f1396e
---

diff --git a/preboot/fit.c b/preboot/fit.c
index 7c7109b..560f4ed 100644
--- a/preboot/fit.c
+++ b/preboot/fit.c
@@ -391,7 +391,7 @@
 #ifdef FAKE_SECURITY
 __weak void preboot_security_sp_sram_write_key_hash(u8 *key, size_t len) {}
 
-int fit_pub_key_value_node_get(const void *fit)
+static int fit_pub_key_value_node_get(const void *fit)
 {
 	int conf = 0;
 	int hash_len = 0;
