From dec1191eb6b3e2d5823ecbe1bcd290bc4c1cdfed Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 05 May 2021 19:32:27 +0300
Subject: [PATCH] print armcp logs

Change-Id: I51aae14d7cba6b92556cb0467be686f4a04a1ffb
---

diff --git a/board/habanalabs/common/post-image-spi.sh b/board/habanalabs/common/post-image-spi.sh
index 579e039..f68e397 100755
--- a/board/habanalabs/common/post-image-spi.sh
+++ b/board/habanalabs/common/post-image-spi.sh
@@ -126,7 +126,7 @@
 	# Swap Preboot
 	swap_file ${HL_DEV}-preboot.bin.le ${HL_DEV}-preboot.bin.be
 	PREBOOT_BINFILE=${HL_DEV}-preboot.bin.be
-	hl_compress_bin ${HL_DEV}-preboot.bin.le
+	hl_compress_bin ${PREBOOT_BINFILE}
 
 	dd status=none conv=sync bs=8 if=u-boot.bin					 > ${HL_DEV}-u-boot.bin.le
 	# Swap U-boot
diff --git a/board/habanalabs/common/rootfs_overlay/etc/init.d/S99last b/board/habanalabs/common/rootfs_overlay/etc/init.d/S99last
index 230c3e4..6895956 100755
--- a/board/habanalabs/common/rootfs_overlay/etc/init.d/S99last
+++ b/board/habanalabs/common/rootfs_overlay/etc/init.d/S99last
@@ -7,6 +7,8 @@
 {
 	echo "Starting last..."
 	/usr/sbin/abt
+	echo "printing armcp content..."
+	cat /var/log/armcpd.log
 }
 
 function stop()
